commit 2285efe5dc2c83eda4578527b65f759f6382d094
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1325323706 -0600

    Fix FTBFS

diff --git a/src/modules/options/mkcreateinstanceproc.sh b/src/modules/options/mkcreateinstanceproc.sh
index c426230..5e504fa 100755
--- a/src/modules/options/mkcreateinstanceproc.sh
+++ b/src/modules/options/mkcreateinstanceproc.sh
@@ -56,7 +56,7 @@ typedef struct _KviOptionsWidgetInstanceEntry KviOptionsWidgetInstanceEntry;
 
 typedef struct _KviOptionsWidgetInstanceEntry
 {
-	KviOptionsWidget                          * (*createProc)(QWidget *);
+	KviOptionsWidget                          * (*createProc)(TQWidget *);
 	KviOptionsWidget                          * pWidget;   // singleton
 	int                                         iIcon;
 	QString                                     szName;
@@ -73,9 +73,10 @@ typedef struct _KviOptionsWidgetInstanceEntry
 } KviOptionsWidgetInstanceEntry;
 
 
-class KviOptionsInstanceManager : public QObject
+class KviOptionsInstanceManager : public TQObject
 {
 	Q_OBJECT
+	TQ_OBJECT
 public:
 	KviOptionsInstanceManager();
 	virtual ~KviOptionsInstanceManager();
@@ -83,14 +84,14 @@ protected:
 	KviPointerList<KviOptionsWidgetInstanceEntry> *  m_pInstanceTree;
 public:
 	KviPointerList<KviOptionsWidgetInstanceEntry> * instanceEntryTree(){ return m_pInstanceTree; };
-	KviOptionsWidget * getInstance(KviOptionsWidgetInstanceEntry * e,QWidget * par);
+	KviOptionsWidget * getInstance(KviOptionsWidgetInstanceEntry * e,TQWidget * par);
 	KviOptionsWidgetInstanceEntry * findInstanceEntry(const char * clName);
 	void cleanup(KviModule * m);
 protected:
 	KviOptionsWidgetInstanceEntry * findInstanceEntry(const char * clName,KviPointerList<KviOptionsWidgetInstanceEntry> * l);
-	KviOptionsWidgetInstanceEntry * findInstanceEntry(const QObject * ptr,KviPointerList<KviOptionsWidgetInstanceEntry> * l);
+	KviOptionsWidgetInstanceEntry * findInstanceEntry(const TQObject * ptr,KviPointerList<KviOptionsWidgetInstanceEntry> * l);
 	void deleteInstanceTree(KviPointerList<KviOptionsWidgetInstanceEntry> * l);
-protected Q_SLOTS:
+protected slots:
 	void widgetDestroyed();
 };
 
