--- kvirc/admin/acinclude.m4.in.ORI	2012-08-06 12:20:40.350261853 +0200
+++ kvirc/admin/acinclude.m4.in	2012-08-06 12:21:34.061121107 +0200
@@ -2149,7 +2149,7 @@
 	if test -n "$KDEDIR"; then
 		ss_kde_general_test_dirs="$KDEDIR $ss_kde_general_test_dirs"
 	fi
-	ss_kde_general_test_path_suffix="kde kde3.0 kde3 KDE Kde KDE3 kde3 ."
+	ss_kde_general_test_path_suffix="trinity kde kde3.0 kde3 KDE Kde KDE3 kde3 ."
 
 	AC_SS_FIND_FILE_PATH_EXT($SS_KDE_GENERAL_FILE_TO_SEARCH,$ss_kde_general_test_dirs,$ss_kde_general_test_path_suffix,SS_KDE_GENERAL_DIR)
 
@@ -2524,7 +2524,7 @@
 
 	if test -n "$SS_OTHER_LIBDIRS"; then
 		SS_LIBDIRS="$SS_LIBDIRS $SS_OTHER_LIBDIRS"
-		$SS_OTHER_LIBDIRS=`echo "$SS_OTHER_LIBDIRS" | sed -e s/-L//g`
+		SS_OTHER_LIBDIRS=`echo "$SS_OTHER_LIBDIRS" | sed -e s/-L//g`
 		SS_RPATH="$SS_RPATH -rpath $SS_OTHER_LIBDIRS"
 	fi
 
