--- ./kicker/kicker/ui/k_mnu.cpp~	2007-05-14 09:55:40.000000000 +0200
+++ ./kicker/kicker/ui/k_mnu.cpp	2007-09-28 07:41:27.000000000 +0200
@@ -361,7 +361,7 @@
     if (kapp->authorize("start_new_session") && (p = dm.numReserve()) >= 0)
     {
         if (kapp->authorize("lock_screen"))
-	  sessionsMenu->insertItem(/*SmallIconSet("lockfork"),*/ i18n("Lock Current && Start New Session"), 100 );
+	  sessionsMenu->insertItem(SmallIconSet("switchuser"), i18n("Lock Current && Start New Session"), 100 );
         sessionsMenu->insertItem(SmallIconSet("fork"), i18n("Start New Session"), 101 );
         if (!p) {
             sessionsMenu->setItemEnabled( 100, false );
