--- arts-1.5.9/mcop/mcoputils.cc.orig	2008-06-02 09:28:23.000000000 -0300
+++ arts-1.5.9/mcop/mcoputils.cc	2008-06-02 09:28:42.000000000 -0300
@@ -28,6 +28,7 @@
 #include <sys/types.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <limits.h>
 #include <netdb.h>
 #include <errno.h>
 #include <config.h>
--- arts-1.5.9/artsc/artsdsp.c.orig	2008-06-02 09:27:24.000000000 -0300
+++ arts-1.5.9/artsc/artsdsp.c	2008-06-02 09:27:52.000000000 -0300
@@ -21,6 +21,7 @@
  */
 
 #define _GNU_SOURCE 1
+#undef _FORTIFY_SOURCE
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
