--- a/mkspecs/linux-g++-64/qmake.conf
+++ b/mkspecs/linux-g++-64/qmake.conf
@@ -43,11 +43,11 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_T
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
 QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib64
+QMAKE_LIBDIR_X11	= /usr/X11R6/lib
 QMAKE_INCDIR_QT		= $(QTDIR)/include
 QMAKE_LIBDIR_QT		= $(QTDIR)/lib
 QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib64
+QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
 
 QMAKE_LINK		= g++
 QMAKE_LINK_SHLIB	= g++
