--- libcaldav-0.6.5/ChangeLog
+++ libcaldav-0.6.5/ChangeLog	2011-05-08 00:56:11.000000000 -0500
@@ -1,3 +1,9 @@
+libcaldav (0.6.5)
+  * Fix bug to be able to integrate the debian libcaldav
+    source with Arch PKGBUILD Patch provided by Timothy Pearson
+
+-- David Rankin <drankinatty@gmail.com>  Sun, 8 May 2011 22:51:03 +0100
+
 libcaldav (0.6.2)
   * Fix bug to be able to handle HTTP HEADER lines split
     over multiple lines. Patch provided by Timothy Pearson
