--- trinity-tdeio-apt-14.0.0~pre105+c7bc8b76/src/apt.cpp.ORI	2013-07-29 11:26:26.740982531 +0200
+++ trinity-tdeio-apt-14.0.0~pre105+c7bc8b76/src/apt.cpp	2013-07-29 11:26:53.970250335 +0200
@@ -143,10 +143,10 @@
  */
 
 static const TQString
-  html_form_begin("\n<form action=\"apt:/\" method=\"GET\">\n"
+  html_form_begin_apt("\n<form action=\"apt:/\" method=\"GET\">\n"
                   "<table class=\"query\">\n");
 static const TQString
-  html_form_end("<tr>\n"
+  html_form_end_apt("<tr>\n"
                 "\t<td class=\"button\" colspan=\"2\"><input type=\"submit\" value=\"%1\"></td>\n"
                 "</tr>\n"
                 "</table>\n"
@@ -197,7 +197,7 @@
   if (online)
     stream << "<table class=\"queryform\"><tr><td>\n";
 
-  stream << html_form_begin;
+  stream << html_form_begin_apt;
   stream << "<tr><td colspan=\"2\" class=\"title\">" + i18n("Offline search") + "</td></tr>" << endl;
   stream << make_html_form_line("search", i18n("Package search"));
   if (ext_form)
@@ -206,7 +206,7 @@
       stream << make_html_form_line("fsearch", i18n("File search"));
     stream << make_html_form_line("show", i18n("Package info"));
   }
-  stream << html_form_end.arg( i18n("Search") );
+  stream << html_form_end_apt.arg( i18n("Search") );
 
   if (online)
   {
