--- trinity-tdeutils-3.5.13.2/ConfigureChecks.cmake.rhel4	2013-01-09 02:43:16.000000000 +0100
+++ trinity-tdeutils-3.5.13.2/ConfigureChecks.cmake	2013-03-10 16:25:27.108634832 +0100
@@ -109,8 +107,6 @@
       set( CMAKE_REQUIRED_LIBRARIES )
       set( HAVE_DPMS 1 )
     endif( WITH_DPMS )
-  else( XEXT_FOUND )
-    tde_message_fatal( "libXext is requested, but was not found on your system" )
   endif( XEXT_FOUND )
 endif( )
 
