--- tqca.ORI/qca.pro	2012-04-24 17:24:02.760426575 -0400
+++ tqca.P/qca.pro	2012-04-24 17:39:36.323825994 -0400
@@ -21,6 +21,7 @@
 QCA_CPP = src
 INCLUDEPATH += $$QCA_CPP
 INCLUDEPATH += /usr/include/tqt
+INCLUDEPATH += /opt/trinity/include/tqt
 
 HEADERS += \
 	$$QCA_CPP/qca.h \
--- tqca-master/configure.qt	2012-06-20 03:30:19.000000000 +0200
+++ tqca-master/configure	2013-05-28 19:44:10.821100155 +0200
@@ -8,7 +8,7 @@
 
 Main options:
   --prefix=[path]   Base path for build/install.  Default: /usr/local
-  --qtdir=[path]    Directory where Qt is installed.
+  --qtdir=[path]    Directory where [T]Qt is installed.
   --help            This help text.
 
 EOT
@@ -37,7 +37,7 @@
 
 PREFIX=${PREFIX:-/usr/local}
 
-echo "Configuring TQt Cryptographic Architecture (QCA) ..."
+echo "Configuring TQt Cryptographic Architecture (TQCA) ..."
 
 if [ "$QC_DEBUG" = "Y" ]; then
 echo
@@ -52,7 +52,7 @@
 	if [ "$QC_DEBUG" = "Y" ]; then
 		echo \$QTDIR not set... trying to find Qt manually
 	fi
-	for p in /usr/lib/tqt /usr/share/tqt /usr/share/tqt3 /usr/local/lib/tqt /usr/local/share/tqt /usr/lib/tqt3 /usr/local/lib/tqt3 /usr/lib/qt /usr/share/qt /usr/share/qt3 /usr/local/lib/qt /usr/local/share/qt /usr/lib/qt3 /usr/local/lib/qt3 /usr/lib64/tqt /usr/local/lib64/tqt /usr/lib64/tqt3 /usr/local/lib64/tqt3 /usr/lib64/qt /usr/local/lib64/qt /usr/lib64/qt3 /usr/local/lib64/qt3 /usr/X11R6/share/qt /usr/qt/3 ; do
+	for p in /usr/lib/qt /usr/share/qt /usr/share/qt3 /usr/local/lib/qt /usr/local/share/qt /usr/lib/qt3 /usr/local/lib/qt3 /usr/lib/qt /usr/share/qt /usr/share/qt3 /usr/local/lib/qt /usr/local/share/qt /usr/lib/qt3 /usr/local/lib/qt3 /usr/lib64/qt /usr/local/lib64/qt /usr/lib64/qt3 /usr/local/lib64/qt3 /usr/lib64/qt /usr/local/lib64/qt /usr/lib64/qt3 /usr/local/lib64/qt3 /usr/X11R6/share/qt /usr/qt/3 ; do
 		if [ -d "$p/mkspecs" ]; then
 			QTDIR=$p
 			break;
@@ -62,7 +62,7 @@
 		echo fail
 		echo
 		echo Unable to find Qt 'mkspecs'. Please set QTDIR
-		echo manually. Perhaps you need to install Qt 3
+		echo manually. Perhaps you need to install [T]Qt 3
 		echo development utilities. You may download them either
 		echo from the vendor of your operating system or from
 		echo www.trolltech.com
@@ -71,15 +71,15 @@
 	fi
 fi
 
-if [ ! -x "$QTDIR/bin/tqmake" ]; then
+if [ ! -x "$QTDIR/bin/qmake" ]; then
 	if [ "$QC_DEBUG" = "Y" ]; then
-		echo Warning: tqmake not in \$QTDIR/bin/tqmake
+		echo Warning: qmake not in \$QTDIR/bin/qmake
 		echo trying to find it in \$PATH
 	fi
-	qm=`type -p tqmake`
+	qm=`type -p qmake`
 	if [ -x "$qm" ]; then
 		if [ "$QC_DEBUG" = "Y" ]; then
-			echo tqmake found in $qm
+			echo qmake found in $qm
 		fi
 	else
 		echo fail
@@ -88,12 +88,12 @@
 		echo or I missdetected \$QTDIR=$QTDIR
 		echo
 		echo Please set \$QTDIR manually and make sure that
-		echo \$QTDIR/bin/tqmake exists.
+		echo \$QTDIR/bin/qmake exists.
 		echo
 		exit 1;
 	fi
 else
-	qm=$QTDIR/bin/tqmake
+	qm=$QTDIR/bin/qmake
 fi
 
 gen_files() {
@@ -382,7 +382,7 @@
 
 		TQString str;
 		str += "# qconf\n";
-    str += "QT_PATH_PLUGINS = " + TQString(tqInstallPathPlugins()) + '\n';
+    str += "QT_PATH_PLUGINS = " + TQString(qInstallPathPlugins()) + '\n';
 		if(!conf->DEFINES.isEmpty())
 			str += "DEFINES += " + conf->DEFINES + '\n';
 		if(!conf->INCLUDEPATH.isEmpty())
@@ -417,7 +417,8 @@
 TEMPLATE = app
 CONFIG += qt x11 thread console
 TARGET = conf
-INCLUDEPATH += '/usr/include/tqt'
+INCLUDEPATH += '${QTINC:-${QTDIR}/include}'
+INCLUDEPATH += '/opt/trinity/include/tqt'
 
 DEFINES += X11_INC='"\$\$QMAKE_INCDIR_X11"'
 DEFINES += X11_LIBDIR='"\$\$QMAKE_LIBDIR_X11"'
@@ -442,16 +443,16 @@
 )
 
 if [ "$?" != "0" ]; then
-	rm -rf .qconftemp
+#	rm -rf .qconftemp
 	echo fail
 	echo
 	echo "There was an error compiling 'conf'.  Be sure you have a proper"
 	echo "TQt 3.x Multithreaded (MT) build environment set up."
-	if [ ! -f "$QTDIR/lib/libtqt-mt.so.3" ]; then
-		if [ ! -f "$QTDIR/lib64/libtqt-mt.so.3" ]; then
+	if [ ! -f "$QTDIR/lib/libqt-mt.so.3" ]; then
+		if [ ! -f "$QTDIR/lib64/libqt-mt.so.3" ]; then
 			echo
 			echo "One possible reason is that you don't have"
-			echo "libtqt-mt.so.3 installed in $QTDIR/lib/"
+			echo "libqt-mt.so.3 installed in $QTDIR/lib/"
 			echo "or $QTDIR/lib64/."
 		fi
 	fi
@@ -480,7 +481,7 @@
 if [ -x "./qcextra" ]; then
 	./qcextra
 fi
-# run tqmake
+# run qmake
 $qm qca.pro
 if [ "$?" != "0" ]; then
 	echo
