diff -Nuar tqca.ORI/configure tqca.P/configure
--- tqca.ORI/configure	2012-04-24 17:24:02.762426535 -0400
+++ tqca.P/configure	2012-04-24 17:41:57.164913623 -0400
@@ -8,7 +8,7 @@
 
 Main options:
   --prefix=[path]   Base path for build/install.  Default: /usr/local
-  --qtdir=[path]    Directory where Qt is installed.
+  --qtdir=[path]    Directory where [T]Qt is installed.
   --help            This help text.
 
 EOT
@@ -37,7 +37,7 @@
 
 PREFIX=${PREFIX:-/usr/local}
 
-echo "Configuring TQt Cryptographic Architecture (QCA) ..."
+echo "Configuring TQt Cryptographic Architecture (TQCA) ..."
 
 if [ "$QC_DEBUG" = "Y" ]; then
 echo
@@ -52,7 +52,7 @@
 	if [ "$QC_DEBUG" = "Y" ]; then
 		echo \$QTDIR not set... trying to find Qt manually
 	fi
-	for p in /usr/lib/tqt /usr/share/tqt /usr/share/tqt3 /usr/local/lib/tqt /usr/local/share/tqt /usr/lib/tqt3 /usr/local/lib/tqt3 /usr/lib/qt /usr/share/qt /usr/share/qt3 /usr/local/lib/qt /usr/local/share/qt /usr/lib/qt3 /usr/local/lib/qt3 /usr/X11R6/share/qt /usr/qt/3 ; do
+	for p in /usr/lib64/tqt /usr/lib/tqt /usr/share/tqt /usr/share/tqt3 /usr/local/lib64/tqt /usr/local/lib/tqt /usr/local/share/tqt /usr/lib64/tqt3 /usr/lib/tqt3 /usr/local/lib64/tqt3 /usr/local/lib/tqt3 /usr/lib64/qt /usr/lib/qt /usr/share/qt /usr/share/qt3 /usr/local/lib64/qt /usr/local/lib/qt /usr/local/share/qt /usr/lib64/qt3 /usr/lib/qt3 /usr/local/lib64/qt3 /usr/local/lib/qt3 /usr/X11R6/share/qt /usr/qt/3 ; do
 		if [ -d "$p/mkspecs" ]; then
 			QTDIR=$p
 			break;
@@ -62,7 +62,7 @@
 		echo fail
 		echo
 		echo Unable to find Qt 'mkspecs'. Please set QTDIR
-		echo manually. Perhaps you need to install Qt 3
+		echo manually. Perhaps you need to install [T]Qt 3
 		echo development utilities. You may download them either
 		echo from the vendor of your operating system or from
 		echo www.trolltech.com
@@ -71,12 +71,12 @@
 	fi
 fi
 
-if [ ! -x "$QTDIR/bin/qmake" ]; then
+if [ ! -x "$QTDIR/bin/qmake-tqt3" ] && [ ! -x "$QTDIR/bin/qmake" ]; then
 	if [ "$QC_DEBUG" = "Y" ]; then
 		echo Warning: qmake not in \$QTDIR/bin/qmake
 		echo trying to find it in \$PATH
 	fi
-	qm=`type -p qmake`
+	qm=`type -p qmake-tqt3 || type -p qmake`
 	if [ -x "$qm" ]; then
 		if [ "$QC_DEBUG" = "Y" ]; then
 			echo qmake found in $qm
@@ -93,7 +93,11 @@
 		exit 1;
 	fi
 else
-	qm=$QTDIR/bin/qmake
+	if [ -x $QTDIR/bin/qmake-tqt3 ]; then
+		qm=$QTDIR/bin/qmake-tqt3
+	else
+		qm=$QTDIR/bin/qmake
+	fi
 fi
 
 gen_files() {
@@ -414,6 +418,7 @@
 CONFIG += qt x11 thread console
 TARGET = conf
 INCLUDEPATH += '/usr/include/tqt'
+INCLUDEPATH += '/usr/include/tqt3'
 
 DEFINES += X11_INC='"\$\$QMAKE_INCDIR_X11"'
 DEFINES += X11_LIBDIR='"\$\$QMAKE_LIBDIR_X11"'
@@ -438,7 +443,7 @@
 )
 
 if [ "$?" != "0" ]; then
-	rm -rf .qconftemp
+#	rm -rf .qconftemp
 	echo fail
 	echo
 	echo "There was an error compiling 'conf'.  Be sure you have a proper"
diff -Nuar tqca.ORI/qca.pro tqca.P/qca.pro
--- tqca.ORI/qca.pro	2012-04-24 17:24:02.760426575 -0400
+++ tqca.P/qca.pro	2012-04-24 17:39:36.323825994 -0400
@@ -21,6 +21,7 @@
 QCA_CPP = src
 INCLUDEPATH += $$QCA_CPP
 INCLUDEPATH += /usr/include/tqt
+INCLUDEPATH += /usr/include/tqt3
 
 HEADERS += \
 	$$QCA_CPP/qca.h \
