commit d97e403f9f70425a59f0d3bdbcfd4dd6360fa6e0
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1331655316 -0500

    Remove inadvertent renaming. This resolves bug report 863.

diff --git a/kuickshow/src/imagewindow.cpp b/kuickshow/src/imagewindow.cpp
index 39af7f5..51367f2 100644
--- a/kuickshow/src/imagewindow.cpp
+++ b/kuickshow/src/imagewindow.cpp
@@ -182,10 +182,10 @@ void ImageWindow::setupActions()
                  TQT_TQOBJECT(this), TQT_SLOT( rotate270() ),
                  m_actions, "rotate270" );
 
-    new KAction( i18n("FlipQt::Horizontally"), Key_Asterisk,
+    new KAction( i18n("Flip Horizontally"), Key_Asterisk,
                  TQT_TQOBJECT(this), TQT_SLOT( flipHoriz() ),
                  m_actions, "flip_horicontally" );
-    new KAction( i18n("FlipQt::Vertically"), Key_Slash,
+    new KAction( i18n("Flip Vertically"), Key_Slash,
                  TQT_TQOBJECT(this), TQT_SLOT( flipVert() ),
                  m_actions, "flip_vertically" );
 
