--- kuickshow/kuickshow/src/Makefile.am.ORI	2012-08-06 11:35:57.271341272 +0200
+++ kuickshow/kuickshow/src/Makefile.am	2012-08-06 11:36:20.916830749 +0200
@@ -10,7 +10,7 @@
 METASOURCES = AUTO
 
 kuickshow_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kuickshow_la_LIBADD  = $(LIB_KDEPRINT) $(LIB_IMLIB)
+kuickshow_la_LIBADD  = $(LIB_KDEPRINT) $(LIB_IMLIB) $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KIO)
 kuickshow_la_SOURCES =	kuickshow.cpp \
 	aboutwidget.cpp     generalwidget.cpp  kuickconfigdlg.cpp  main.cpp \
 	defaultswidget.cpp  imagewindow.cpp    kuickdata.cpp  \
