--- kdelibs/kdecore/tdeaboutdata.cpp	2006-03-17 10:19:05.000000000 +0000
+++ kdelibs/kdecore/tdeaboutdata.cpp	2006-12-05 19:46:12.000000000 +0000
@@ -389,7 +389,7 @@
             "of the translation teams all over the world.</p>"
             "<p>For more information on KDE internationalization "
             "visit <a href=\"http://l10n.kde.org\">http://l10n.kde.org</a></p>"
-            );
+		) + i18n("<p>Kubuntu includes additions and alterations to the KDE translation from <a href=\"https://translations.launchpad.net/\">Launchpad</a>.</p>");
 }
 
 TQString
