--- a/src/main.cpp
+++ b/src/main.cpp
@@ -24,6 +24,7 @@
 #include <kmessagebox.h>
 #include <kmainwindow.h>
 #include <kapp.h>
+#include <kuser.h>
 
 #include <klocale.h>
 #include <dcopclient.h>
@@ -129,7 +130,12 @@
 // 			KMessageBox::error( 0, I18N_NOOP( "Kvpnc must be started as root!" ) );
 // 			exit(0);
 // 		}
-		
+        if (!KUser().isSuperUser())
+        {
+ 			KMessageBox::error( 0L, i18n( "Kvpnc must be started as root!" ) );
+ 			exit(0);
+        }
+	
 		KCmdLineArgs *args = KCmdLineArgs::parsedArgs();
 		
 		a.setMainWidget( kvpnc );
