diff --git a/src/main.cpp b/src/main.cpp
index 7c45fcf..b933832 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -24,6 +24,7 @@
 #include <kmessagebox.h>
 #include <kmainwindow.h>
 #include <kapp.h>
+#include <kuser.h>
 
 #include <klocale.h>
 #include <dcopclient.h>
@@ -134,11 +135,11 @@ int main ( int argc, char *argv[] )
 	}
 	else
 	{
-// 		if (getenv("USER") != "root")
-// 		{
-// 			KMessageBox::error( 0, I18N_NOOP( "Kvpnc must be started as root!" ) );
-// 			exit(0);
-// 		}
+ 		if (getenv("USER") != "root")
+ 		{
+ 			KMessageBox::error( 0, I18N_NOOP( "Kvpnc must be started as root!" ) );
+ 			exit(0);
+ 		}
 
 		KCmdLineArgs *args = KCmdLineArgs::parsedArgs();
 
