--- kdevelop/languages/cpp/cppcodecompletion.cpp.gcc47	2011-08-13 07:59:50.000000000 +0200
+++ kdevelop/languages/cpp/cppcodecompletion.cpp	2012-04-26 20:33:55.645135959 +0200
@@ -3892,15 +3892,15 @@
         ArgumentList fArgs = ( *it ) ->argumentList();
         if ( fArgs.count() != args.count() )
           continue;
-        ArgumentList::iterator it = args.begin();
+        ArgumentList::iterator it3 = args.begin();
         ArgumentList::iterator it2 = fArgs.begin();
         bool hit = true;
-        while ( it != args.end() ) {
-          if ( ( *it ) ->type() != ( *it2 ) ->type() ) {
+        while ( it3 != args.end() ) {
+          if ( ( *it3 ) ->type() != ( *it2 ) ->type() ) {
             hit = false;
             break;
           }
-          ++it;
+          ++it3;
           ++it2;
         }
         if ( hit ) {
