diff -Nur tdebase-3.5.4/kcontrol/componentchooser/componentchooser.desktop tdebase-3.5.4.new/kcontrol/componentchooser/componentchooser.desktop
--- tdebase/kcontrol/componentchooser/componentchooser.desktop
+++ tdebase/kcontrol/componentchooser/componentchooser.desktop
@@ -6,3 +6,4 @@
 X-TDE-ParentApp=kcontrol
 DocPath=kcontrol/componentchooser/index.html 
 Categories=Qt;TDE;X-TDE-settings-components;
+NoDisplay=true
