diff --git a/tdeprint/printmgr/printers.desktop b/tdeprint/printmgr/printers.desktop
index f4a363d..f990394 100644
--- a/tdeprint/printmgr/printers.desktop
+++ b/tdeprint/printmgr/printers.desktop
@@ -233,4 +233,4 @@ Keywords[zh_CN]=print;printer;printing management;打印;打印机;打印管理
 Keywords[zh_TW]=print;printer;printing management;列印;印表機;列印管理
 Keywords[zu]=shicilela;umshicileli;abaphathi boshicilelo
 
-Categories=Qt;TDE;X-TDE-settings-hardware;Settings;
+Categories=Qt;TDE;X-TDE-settings-hardware;
