--- tdm/kfrontend/tdmshutdown.cpp
+++ tdm/kfrontend/tdmshutdown.cpp
@@ -467,7 +467,7 @@
 	, targetList( 0 )
 {
 
-	bool doUbuntuLogout = KConfigGroup(KGlobal::config(), "Shutdown").readBoolEntry("doUbuntuLogout", false);
+	bool doUbuntuLogout = KConfigGroup(KGlobal::config(), "Shutdown").readBoolEntry("doUbuntuLogout", true);
 
 	TQVBoxLayout* vbox = new TQVBoxLayout( this );
 	TQHBoxLayout *hbox = new TQHBoxLayout( this, KDmh, KDsh );

--- ksmserver/shutdowndlg.cpp
+++ ksmserver/shutdowndlg.cpp
@@ -668,7 +668,7 @@
 							2 * KDialog::spacingHint() );
 
 	// default factor
-	bool doUbuntuLogout = KConfigGroup(KGlobal::config(), "Logout").readBoolEntry("doUbuntuLogout", false);
+	bool doUbuntuLogout = KConfigGroup(KGlobal::config(), "Logout").readBoolEntry("doUbuntuLogout", true);
 
 	// slighty more space for the new logout
 	int factor = 2;
