diff --git a/kcontrol/componentchooser/componentchooser.desktop b/kcontrol/componentchooser/componentchooser.desktop
index fdd1d41..e6913a7 100644
--- a/kcontrol/componentchooser/componentchooser.desktop
+++ b/kcontrol/componentchooser/componentchooser.desktop
@@ -6,6 +6,7 @@ X-TDE-Library=componentchooser
 X-TDE-ParentApp=kcontrol
 X-DocPath=kcontrol/componentchooser/index.html
 Categories=Qt;TDE;X-TDE-settings-components;
+NoDisplay=true
 
 Name=Default Applications
 Name[af]=Verstektoepassings
