--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h.ORI	2011-11-27 05:38:43.500726484 -0500
+++ kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h	2011-11-27 05:39:19.901925857 -0500
@@ -80,6 +80,7 @@
 
 #include <set>
 #include <list>
+#include <cstddef>
 
 // On our copy of sigslot.h, we force single threading
 #define SIGSLOT_PURE_ISO
--- kdenetwork/kopete/protocols/jabber/CMakeLists.txt.ORI	2011-11-27 05:55:08.369780670 -0500
+++ kdenetwork/kopete/protocols/jabber/CMakeLists.txt	2011-11-27 05:55:12.350662883 -0500
@@ -80,6 +80,6 @@
     jabberclient-static kopetejabberui-static
     iris_jabber-static iris_xmpp_core-static iris_xmpp_im-static iris-static
     qca-static cutestuff_network-static cutestuff_util-static ${JINGLE_LIBRARIES}
-    kopete-shared
+    kopete-shared idn
   DESTINATION ${PLUGIN_INSTALL_DIR}
 )
--- kdenetwork/kopete/protocols/jabber/kioslave/CMakeLists.txt.ORI	2011-11-27 06:01:44.412062605 -0500
+++ kdenetwork/kopete/protocols/jabber/kioslave/CMakeLists.txt	2011-11-27 06:01:56.658700254 -0500
@@ -37,6 +37,6 @@
   LINK
     jabberclient-static
     iris_xmpp_core-static iris_xmpp_im-static iris_jabber-static iris-static
-    qca-static cutestuff_network-static cutestuff_util-static kio-shared
+    qca-static cutestuff_network-static cutestuff_util-static kio-shared idn
   DESTINATION ${PLUGIN_INSTALL_DIR}
 )
