--- kdenetwork/kopete/plugins/motionautoaway/motionawayplugin.cpp.ORI	2011-11-27 13:27:13.236739020 -0500
+++ kdenetwork/kopete/plugins/motionautoaway/motionawayplugin.cpp	2011-11-27 13:28:48.217257573 -0500
@@ -75,7 +75,12 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,50)
 #define _LINUX_TIME_H
 #endif
+
+#ifdef HAVE_LIBV4L1_VIDEODEV_H
+#include <libv4l1-videodev.h>
+#else
 #include <linux/videodev.h>
+#endif
 
 #define DEF_WIDTH			352
 #define DEF_HEIGHT		288
