diff --git a/wifi/ConfigureChecks.cmake b/wifi/ConfigureChecks.cmake
index f3a7ddb..86ca98b 100644
--- a/wifi/ConfigureChecks.cmake
+++ b/wifi/ConfigureChecks.cmake
@@ -21,7 +21,9 @@ endif( )
 
 
 # check for wireless tools
+tde_save_and_set( CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} "-lm" )
 check_library_exists( iw iw_get_stats "" HAVE_IW )
+tde_restore( CMAKE_REQUIRED_FLAGS )
 if( HAVE_IW )
   check_include_file( iwlib.h HAVE_IWLIB_H )
   # we need at least wireless tools >= 27
