Index: b/kerry/src/Makefile.am
===================================================================
--- a/kerry/src/Makefile.am
+++ b/kerry/src/Makefile.am
@@ -31,7 +31,7 @@
 kerry_la_LIBADD = -ltdeio -lkonq $(LIB_TDEUI) $(LIBBEAGLE_LIBADD) $(GLIB_LIBADD) $(LIB_TDEABC) -lDCOP $(LIB_QT) $(LIB_TDECORE)
 
 autostartdir = $(datadir)/autostart
-autostart_DATA = kerry.autostart.desktop beagled.desktop
+autostart_DATA = kerry.autostart.desktop
 
 shutdowndir = $(prefix)/bin
 shutdown_SCRIPTS = beagled-shutdown
