commit fd6afacf70587de329d9b2647b96469bae5e74da
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1327530995 -0600

    Fix linear alphabet string errors

diff --git a/mpg123_artsplugin/mpg123/httpget.c b/mpg123_artsplugin/mpg123/httpget.c
index 12c6941..7175011 100644
--- a/mpg123_artsplugin/mpg123/httpget.c
+++ b/mpg123_artsplugin/mpg123/httpget.c
@@ -83,7 +83,7 @@ void readstring (char *string, int maxlen, FILE *f)
 void encode64 (char *source,char *destination)
 {
   static char *Base64Digits =
-    "ABCDEFGHIJKLMNOPTQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
   int n = 0;
   int ssiz=strlen(source);
   int i;
