commit 2f6e9c735fe8486ee3484c47c0e8ff999b699abc
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1347987185 -0500

    Fall back to ruby-1.9 pkgconfig file if needed

diff --git a/korundum/rubylib/korundum/configure.in.in b/korundum/rubylib/korundum/configure.in.in
index e4ab2fd..924f42f 100644
--- a/korundum/rubylib/korundum/configure.in.in
+++ b/korundum/rubylib/korundum/configure.in.in
@@ -10,6 +10,9 @@ else
   AC_MSG_CHECKING(for ruby dirs)
   if test -n "$PKGCONFIG"; then
     RUBY_VERSION=`$PKGCONFIG ruby --modversion 2>/dev/null`
+    if test -z "$RUBY_VERSION"; then
+      RUBY_VERSION=`$PKGCONFIG ruby-1.9 --modversion 2>/dev/null`
+    fi
     if test -n "$RUBY_VERSION"; then
       RUBY_ARCHDIR=`ruby -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"archdir"@:>@)'`
       RUBY_SITEARCHDIR=`ruby -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"sitearchdir"@:>@)'`
diff --git a/qtruby/rubylib/qtruby/configure.in.in b/qtruby/rubylib/qtruby/configure.in.in
index 7ad0e4d..e7f4f2c 100644
--- a/qtruby/rubylib/qtruby/configure.in.in
+++ b/qtruby/rubylib/qtruby/configure.in.in
@@ -10,6 +10,9 @@ else
   AC_MSG_CHECKING(for ruby dirs)
   if test -n "$PKGCONFIG"; then
     RUBY_VERSION=`$PKGCONFIG ruby --modversion 2>/dev/null`
+    if test -z "$RUBY_VERSION"; then
+      RUBY_VERSION=`$PKGCONFIG ruby-1.9 --modversion 2>/dev/null`
+    fi
     if test -n "$RUBY_VERSION"; then
       RUBY_ARCHDIR=`$RUBY -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"archdir"@:>@)'`
       RUBY_SITEARCHDIR=`$RUBY -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"sitearchdir"@:>@)'`
