--- kdebindings/dcopc/Makefile.am.ORI	2012-08-02 19:39:33.981833237 +0000
+++ kdebindings/dcopc/Makefile.am	2012-08-02 19:39:45.085467620 +0000
@@ -10,7 +10,7 @@
 libdcopc_la_LDFLAGS = -version-info 1:0 $(X_LDFLAGS) -lICE -no-undefined
 
-include_HEADERS = dcopc.h marshal.h dcopobject.h util.h
-includedir = $(prefix)/include/dcopc
+dcopcinclude_HEADERS = dcopc.h marshal.h dcopobject.h util.h
+dcopcincludedir = $(includedir)/dcopc
 
 noinst_HEADERS = global.h
 
