diff -urN tdebindings/configure.in.in tdebindings.new/configure.in.in
--- tdebindings/configure.in.in	2011-12-25 00:54:22.000000000 -0600
+++ tdebindings.new/configure.in.in	2012-02-25 17:56:52.000000000 -0600
@@ -302,4 +302,4 @@
 KDE_CHECK_KNS
 
 dnl remove when fixed
-DO_NOT_COMPILE="$DO_NOT_COMPILE dcopjava qtsharp"
+DO_NOT_COMPILE="$DO_NOT_COMPILE qtsharp"
diff -urN tdebindings/dcopc/configure.in.in tdebindings.new/dcopc/configure.in.in
--- tdebindings/dcopc/configure.in.in	2011-12-25 00:54:22.000000000 -0600
+++ tdebindings.new/dcopc/configure.in.in	2012-02-23 15:44:16.000000000 -0600
@@ -1,7 +1,3 @@
-dnl if test "$GTK_CONFIG" = "no"; then 
-    DO_NOT_COMPILE="$DO_NOT_COMPILE dcopc"
-dnl fi
-dnl 
 dnl AC_CHECK_GETHOSTNAME
 
 AC_GTK_2
diff -urN tdebindings/dcopperl/DCOP.xs tdebindings.new/dcopperl/DCOP.xs
--- tdebindings/dcopperl/DCOP.xs	2011-12-25 00:54:22.000000000 -0600
+++ tdebindings.new/dcopperl/DCOP.xs	2012-02-23 16:29:28.000000000 -0600
@@ -26,11 +26,11 @@
 #undef vform
 #endif
 
-#include <qpoint.h>
-#include <qrect.h>
-#include <qregexp.h>
-#include <qsize.h>
-#include <qstringlist.h>
+#include <tqpoint.h>
+#include <tqrect.h>
+#include <tqregexp.h>
+#include <tqsize.h>
+#include <tqstringlist.h>
 
 #include <dcopclient.h>
 #include <dcopref.h>
diff -urN tdebindings/qtsharp/src/generator/Printer.cs tdebindings.new/qtsharp/src/generator/Printer.cs
--- tdebindings/qtsharp/src/generator/Printer.cs	2011-12-25 00:54:23.000000000 -0600
+++ tdebindings.new/qtsharp/src/generator/Printer.cs	2012-02-25 16:22:12.000000000 -0600
@@ -226,7 +226,7 @@
 		{
 			if (check)
 				return "\n\n\t\t\tif ((qparent = parent) != null)\n" +
-					   "\t\t\t\parent.AddChild (this);\n";
+					   "\t\t\tparent.AddChild (this);\n";
 			else
 				return "";
 		}
diff -urN tdebindings/xparts/configure.in.in tdebindings.new/xparts/configure.in.in
--- tdebindings/xparts/configure.in.in	2011-12-25 00:54:23.000000000 -0600
+++ tdebindings.new/xparts/configure.in.in	2012-02-25 18:15:48.000000000 -0600
@@ -1 +1 @@
-DO_NOT_COMPILE="$DO_NOT_COMPILE xparts"
+# DO_NOT_COMPILE="$DO_NOT_COMPILE xparts"
--- kdebindings/dcopjava/binding/Makefile.am.ORI	2012-04-25 20:47:03.520286136 +0200
+++ kdebindings/dcopjava/binding/Makefile.am	2012-04-25 20:51:13.379853911 +0200
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = libjavadcop.la
 
 libjavadcop_la_SOURCES = client.cpp 
-libjavadcop_la_LDFLAGS = $(KDE_LDFLAGS) $(KDE_PLUGIN) -lDCOP -lkdecore -lkdeui $(LIB_QT) -lkjs -lkdefx
+libjavadcop_la_LDFLAGS = $(KDE_LDFLAGS) $(KDE_PLUGIN) -lDCOP -lkdecore -lkdeui $(LIB_QT) -lkjs -lkdefx -I$(QTINC) -L$(QTLIB)
 libjavadcop_la_LIBADD = $(LIB_KDECORE)
 
 INCLUDES = $(jni_includes) $(all_includes)
--- kdebindings/xparts/xpart_notepad/Makefile.am.ORI	2012-04-25 21:16:17.476461045 +0200
+++ kdebindings/xparts/xpart_notepad/Makefile.am	2012-04-25 21:16:30.908149806 +0200
@@ -10,7 +10,7 @@
 xp_notepad_SOURCES = xp_notepad.cpp xpart.skel xparthost.stub
 xp_notepad_LDADD = ../src/kde/libkdexparts.la
 #xp_notepad_LDFLAGS = -L$(prefix)/lib -lxpcom $(all_libraries)
-xp_notepad_LDFLAGS = -L$(prefix)/lib $(all_libraries)
+xp_notepad_LDFLAGS = -L$(prefix)/lib -lDCOP -lkdecore $(all_libraries)
 
 shell_xparthost_SOURCES = shell_xparthost.cpp
 shell_xparthost_LDADD = $(top_builddir)/xparts/src/kde/libkdexparts.la
