commit 14594d81891b0bdf9aa03f72b88d6939eee59f02
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1347389903 -0500

    Fix FTBFS on ruby 1.8.x

diff --git a/korundum/bin/krubyinit.cpp b/korundum/bin/krubyinit.cpp
index f53f830..12b0f6e 100644
--- a/korundum/bin/krubyinit.cpp
+++ b/korundum/bin/krubyinit.cpp
@@ -25,13 +25,18 @@
  *                                                                         *
  ***************************************************************************/
 
-// this name can be used to allow apps 
+// this name can be used to allow apps
 // to detect what they were started with
 static const char* script_name = "krubyinit_app";
 
 int main(int argc, char **argv) {
      ruby_init();
      ruby_script((char*)script_name);
-     void* node = ruby_options(argc, argv); 
+#ifdef HAVE_RUBY_1_9
+     void* node = ruby_options(argc, argv);
      ruby_run_node(node);
+#else // HAVE_RUBY_1_9
+     ruby_options(argc, argv);
+     ruby_run();
+#endif // HAVE_RUBY_1_9
 }
diff --git a/qtruby/bin/qtrubyinit.cpp b/qtruby/bin/qtrubyinit.cpp
index 65c00c4..c9569d9 100644
--- a/qtruby/bin/qtrubyinit.cpp
+++ b/qtruby/bin/qtrubyinit.cpp
@@ -25,13 +25,18 @@ qrubyinit  -  makes use of tdeinit_wrapper possible for ruby programs
  *                                                                         *
  ***************************************************************************/
 
-// this name can be used to allow apps 
+// this name can be used to allow apps
 // to detect what they were started with
 static const char* script_name = "qrubyinit_app";
 
 int main(int argc, char **argv) {
      ruby_init();
      ruby_script((char*)script_name);
-     void* node = ruby_options(argc, argv); 
+#ifdef HAVE_RUBY_1_9
+     void* node = ruby_options(argc, argv);
      ruby_run_node(node);
+#else // HAVE_RUBY_1_9
+     ruby_options(argc, argv);
+     ruby_run();
+#endif // HAVE_RUBY_1_9
 }
