diff -Nuar amarok.ORI/amarok/src/engine/akode/CMakeLists.txt amarok.P/amarok/src/engine/akode/CMakeLists.txt
--- amarok.ORI/amarok/src/engine/akode/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ amarok.P/amarok/src/engine/akode/CMakeLists.txt	2012-01-16 21:42:52.701799667 +0100
@@ -0,0 +1,37 @@
+#################################################
+#
+#  Improvements and feedback are welcome
+#
+#  This file is released under GPL >= 2
+#
+#################################################
+
+include_directories(
+  ${CMAKE_CURRENT_BINARY_DIR}
+  ${CMAKE_BINARY_DIR}
+  ${CMAKE_BINARY_DIR}/amarok/src/amarokcore
+  ${CMAKE_SOURCE_DIR}/amarok/src
+  ${CMAKE_SOURCE_DIR}/amarok/src/engine/akode
+  ${TDE_INCLUDE_DIR}
+  ${TQT_INCLUDE_DIRS}
+  ${AKODE_INCLUDE_DIRS}
+)
+
+link_directories(
+  ${TQT_LIBRARY_DIRS}
+)
+
+
+##### other data ################################
+
+install( FILES amarok_aKode-engine.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
+
+
+##### libamarok_aKode-engine (module) ############
+
+tde_add_kpart( libamarok_aKode-engine AUTOMOC
+  SOURCES
+    akode-scope.cpp akode-engine.cpp
+  LINK amarok-shared ${AKODE_LIBRARIES}
+  DESTINATION ${PLUGIN_INSTALL_DIR}
+)
diff -Nuar amarok.ORI/amarok/src/engine/CMakeLists.txt amarok.P/amarok/src/engine/CMakeLists.txt
--- amarok.ORI/amarok/src/engine/CMakeLists.txt	2012-01-16 21:26:50.433584399 +0100
+++ amarok.P/amarok/src/engine/CMakeLists.txt	2012-01-16 21:27:36.702218362 +0100
@@ -14,3 +14,4 @@
 add_subdirectory( void )
 tde_conditional_add_subdirectory( WITH_XINE xine )
 tde_conditional_add_subdirectory( WITH_YAUAP yauap )
+tde_conditional_add_subdirectory( WITH_AKODE akode )
diff -Nuar amarok.ORI/CMakeLists.txt amarok.P/CMakeLists.txt
--- amarok.ORI/CMakeLists.txt	2012-01-16 21:26:50.418583977 +0100
+++ amarok.P/CMakeLists.txt	2012-01-16 21:28:23.382857595 +0100
@@ -47,6 +47,7 @@
 option( WITH_KONQSIDEBAR "Enable konqsidebar plugin" OFF )
 option( WITH_XINE "Enable xine-engine support" OFF )
 option( WITH_YAUAP "Enable yauap-engine support" OFF )
+option( WITH_AKODE "Enable akode-engine support" OFF )
 option( WITH_IPOD "Enable iPod support from libgpod" OFF )
 option( WITH_IFP "Enable ifp support" OFF )
 option( WITH_NJB "Enable njb support" OFF )
diff -Nuar amarok.ORI/ConfigureChecks.cmake amarok.P/ConfigureChecks.cmake
--- amarok.ORI/ConfigureChecks.cmake	2012-01-16 21:26:50.491585184 +0100
+++ amarok.P/ConfigureChecks.cmake	2012-01-16 21:40:55.883193781 +0100
@@ -121,6 +121,18 @@
   endif( )
 endif( )
 
+# akode-engine
+if( WITH_AKODE )
+  pkg_search_module( AKODE akode )
+  if( AKODE_FOUND )
+    if( ${AKODE_VERSION} VERSION_LESS "2.0.0" )
+      tde_message_fatal( "your akode version is too old; at least 2.0.0 is required" )
+    endif( )
+  else( )
+    tde_message_fatal( "akode is requested, but was not found on your system" )
+  endif( )
+endif( )
+
 
 # iPod
 if( WITH_IPOD )
