commit 2a1d4a67d148c750266837db16a3e7b34cf13258
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1326326096 -0600

    Allow ark embedding
    This closes Bug 670

diff --git a/ark/ark_part.desktop b/ark/ark_part.desktop
index f09a772..e62db26 100644
--- a/ark/ark_part.desktop
+++ b/ark/ark_part.desktop
@@ -133,4 +133,3 @@ Icon=ark
 Type=Service
 ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=libarkpart
-Hidden=true
