--- src/tools/qcstring.h		(revision 658213)
+++ src/tools/qcstring.h		(working copy)
@@ -161,7 +161,11 @@ public:
 
     QCString	copy()	const;
 
-    QCString    &sprintf( const char *format, ... );
+    QCString    &sprintf( const char *format, ... )
+#if defined(Q_CC_GNU) && !defined(__INSURE__)
+    __attribute__ ((format (printf, 2, 3)))
+#endif
+        ;
 
     int		find( char c, int index=0, bool cs=TRUE ) const;
     int		find( const char *str, int index=0, bool cs=TRUE ) const;
