commit 445a515241430cef8e8458d782649fa0e8593a4c
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1332214882 -0500

    Fix inadvertent "TQ" changes.

diff --git a/apps/ktorrent/groups/queueddownloadsgroup.h b/apps/ktorrent/groups/queueddownloadsgroup.h
index 3e436a5..019a69c 100644
--- a/apps/ktorrent/groups/queueddownloadsgroup.h
+++ b/apps/ktorrent/groups/queueddownloadsgroup.h
@@ -17,8 +17,8 @@
  *   Free Software Foundation, Inc.,                                       *
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.          *
  ***************************************************************************/
-#ifndef KTQUEUEDDOWNLOADSGROUP_H
-#define KTQUEUEDDOWNLOADSGROUP_H
+#ifndef KQUEUEDDOWNLOADSGROUP_H
+#define KQUEUEDDOWNLOADSGROUP_H
 
 #include <group.h>
 
diff --git a/apps/ktorrent/groups/queueduploadsgroup.h b/apps/ktorrent/groups/queueduploadsgroup.h
index 675da62..601e02f 100644
--- a/apps/ktorrent/groups/queueduploadsgroup.h
+++ b/apps/ktorrent/groups/queueduploadsgroup.h
@@ -17,8 +17,8 @@
  *   Free Software Foundation, Inc.,                                       *
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.          *
  ***************************************************************************/
-#ifndef KTQUEUEDUPLOADSGROUP_H
-#define KTQUEUEDUPLOADSGROUP_H
+#ifndef KQUEUEDUPLOADSGROUP_H
+#define KQUEUEDUPLOADSGROUP_H
 
 #include <group.h>
 
diff --git a/apps/ktorrent/ktorrentviewitem.cpp b/apps/ktorrent/ktorrentviewitem.cpp
index a9392bd..ae37665 100644
--- a/apps/ktorrent/ktorrentviewitem.cpp
+++ b/apps/ktorrent/ktorrentviewitem.cpp
@@ -72,7 +72,7 @@ static TQColor StatusToColor(TorrentStatus s,const TQColorGroup & cg)
 			return TQt::red;
 		case kt::NOT_STARTED :
 		case kt::STOPPED:
-		case kt::TQUEUED:
+		case kt::QUEUED:
 		case kt::DOWNLOAD_COMPLETE :
 		case kt::SEEDING_COMPLETE :
 		default:
diff --git a/apps/ktorrent/queuedialog.h b/apps/ktorrent/queuedialog.h
index d025fbf..a6900e4 100644
--- a/apps/ktorrent/queuedialog.h
+++ b/apps/ktorrent/queuedialog.h
@@ -17,8 +17,8 @@
  *   Free Software Foundation, Inc.,                                       *
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.          *
  ***************************************************************************/
-#ifndef TQUEUEDIALOG_H
-#define TQUEUEDIALOG_H
+#ifndef QUEUEDIALOG_H
+#define QUEUEDIALOG_H
 
 #include "queuedlg.h"
 #include <interfaces/torrentinterface.h>
diff --git a/libktorrent/interfaces/torrentinterface.h b/libktorrent/interfaces/torrentinterface.h
index 8b21854..e76e00e 100644
--- a/libktorrent/interfaces/torrentinterface.h
+++ b/libktorrent/interfaces/torrentinterface.h
@@ -55,7 +55,7 @@ namespace kt
 		STOPPED,
 		ALLOCATING_DISKSPACE,
 		ERROR,
-		TQUEUED,
+		QUEUED,
 		CHECKING_DATA,
   		NO_SPACE_LEFT
 	};
diff --git a/libktorrent/torrent/queuemanager.h b/libktorrent/torrent/queuemanager.h
index 26b1012..e49ad44 100644
--- a/libktorrent/torrent/queuemanager.h
+++ b/libktorrent/torrent/queuemanager.h
@@ -18,8 +18,8 @@
  *   Free Software Foundation, Inc.,                                       *
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.             *
  ***************************************************************************/
-#ifndef TQUEUEMANAGER_H
-#define TQUEUEMANAGER_H
+#ifndef QUEUEMANAGER_H
+#define QUEUEMANAGER_H
 
 #include <set>
 #include <tqobject.h>
diff --git a/libktorrent/torrent/torrentcontrol.cpp b/libktorrent/torrent/torrentcontrol.cpp
index 3e22ec1..3cedfe9 100644
--- a/libktorrent/torrent/torrentcontrol.cpp
+++ b/libktorrent/torrent/torrentcontrol.cpp
@@ -932,7 +932,7 @@ namespace bt
 		else if (!stats.started)
 			stats.status = kt::NOT_STARTED;
 		else if(!stats.running && !stats.user_controlled)
-			stats.status = kt::TQUEUED;
+			stats.status = kt::QUEUED;
 		else if (!stats.running && stats.completed && (overMaxRatio() || overMaxSeedTime()))
 			stats.status = kt::SEEDING_COMPLETE;
 		else if (!stats.running && stats.completed)
@@ -1301,7 +1301,7 @@ namespace bt
 		istats.priority = p;
 		stats.user_controlled = p == 0 ? true : false;
 		if(p)
-			stats.status = kt::TQUEUED;
+			stats.status = kt::QUEUED;
 		else
 			updateStatusMsg();
 		
@@ -1378,7 +1378,7 @@ namespace bt
 				return i18n("Error: ") + getShortErrorMessage(); 
 			case kt::ALLOCATING_DISKSPACE:
 				return i18n("Allocating diskspace");
-			case kt::TQUEUED:
+			case kt::QUEUED:
 				return i18n("Queued");
 			case kt::CHECKING_DATA:
 				return i18n("Checking data");
