commit c3480dfed56038a7e8c11e2afbc454f59ab44843
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1324748239 -0600

    Rename a few stragglers

diff --git a/apps/ktorrent/newui/button.cpp b/apps/ktorrent/newui/button.cpp
index 100ab62..845df4b 100644
--- a/apps/ktorrent/newui/button.cpp
+++ b/apps/ktorrent/newui/button.cpp
@@ -131,7 +131,7 @@ void Button::drawButton(TQPainter *p)
     pm.fill(eraseColor());
     TQPainter p2(&pm);
 
-    tqstyle().drawControl(TQStyle::CE_PushButton,&p2,this, TQRect(0,0,pm.width(),pm.height()), colorGroup(),flags);
+    style().drawControl(TQStyle::CE_PushButton,&p2,this, TQRect(0,0,pm.width(),pm.height()), colorGroup(),flags);
 
     style().drawControl(TQStyle::CE_PushButtonLabel, &p2, this,
                         TQRect(0,0,pm.width(),pm.height()),
