--- trinity-k9copy/k9author/k9avidecode.cpp.k9author	2013-07-21 09:56:44.000000000 +0200
+++ trinity-k9copy/k9author/k9avidecode.cpp	2013-08-02 12:16:55.398266088 +0200
@@ -119,6 +119,13 @@
     m_opened=false;
 
     glibref++;
+    
+    m_FormatCtx = NULL;
+    m_CodecCtx = NULL;
+    m_Codec = NULL;
+    m_Frame = NULL;
+    m_FrameRGB = NULL;
+    m_buffer = NULL;
 }
 
 
@@ -148,7 +155,7 @@
     // Open video file
     if (
 #       if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(53, 2, 0)
-        avformat_open_input(&m_FormatCtx, _fileName.utf8(), 0, 0)!=0
+        avformat_open_input(&m_FormatCtx, _fileName.utf8(), NULL, NULL)!=0
 #       else
         av_open_input_file(&m_FormatCtx, _fileName.utf8(), NULL, 0, NULL)!=0
 #       endif
