commit ca864edeed0fe99b91c63616d4ec6d31ecdee217
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1336121912 -0500

    Fix FTBFS

diff --git a/libk9copy/k9saveimage.cpp b/libk9copy/k9saveimage.cpp
index 23613d8..758f8de 100644
--- a/libk9copy/k9saveimage.cpp
+++ b/libk9copy/k9saveimage.cpp
@@ -76,7 +76,7 @@ void k9SaveImage::run() {
     TQString sFileName=m_tempFile->name();
     m_image.save(sFileName,"PNG");
     sFileName="\rINFOIMAGE:"+sFileName;
-    fprintf(stderr,"%s",sFileName.utf8());
+    fprintf(stderr,"%s",sFileName.data());
 
     m_mutex.unlock();
 }
