commit d9ed8b3291e596eafed777956a062dc8816d0d3c
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1336083386 -0500

    Fix 'format not a string literal' error

diff --git a/libk9copy/k9cellcopylist.cpp b/libk9copy/k9cellcopylist.cpp
index be4e406..31e85a3 100644
--- a/libk9copy/k9cellcopylist.cpp
+++ b/libk9copy/k9cellcopylist.cpp
@@ -294,10 +294,10 @@ double k9CellCopyList::getMinFactor(bool _withMenus) {
    double totalSize=gettotalSize()+menuSize - (fforced) -m_inbytes;
    totalSize/=2.50;
 
-qDebug(TQString("totalSize(%1)=gettotalSize()(%2)+menuSize(%3) -(fforced(%4))-m_inbytes(%6)").tqarg(totalSize).tqarg(gettotalSize()).tqarg(menuSize).tqarg(fforced).tqarg(m_frcinbytes).tqarg(m_inbytes));
+   qDebug("%s", (TQString("totalSize(%1)=gettotalSize()(%2)+menuSize(%3) -(fforced(%4))-m_inbytes(%6)").tqarg(totalSize).tqarg(gettotalSize()).tqarg(menuSize).tqarg(fforced).tqarg(m_frcinbytes).tqarg(m_inbytes)).ascii());
 
    double minFactor=(fforced-m_frcinbytes) /(MaxSize- (totalSize +m_outbytes) - m_frcoutbytes) ;
-   qDebug(TQString("minfactor(%1)=(fforced(%2) -m_frinbytes(%3))/(MacSize(%4)-totalSize(%5)-m_outbytes(%6) - m_frcoutbytes(%7))").tqarg(minFactor).tqarg(fforced).tqarg(m_frcinbytes).tqarg(MaxSize).tqarg(totalSize).tqarg(m_outbytes).tqarg(m_frcoutbytes));
+   qDebug("%s", (TQString("minfactor(%1)=(fforced(%2) -m_frinbytes(%3))/(MacSize(%4)-totalSize(%5)-m_outbytes(%6) - m_frcoutbytes(%7))").tqarg(minFactor).tqarg(fforced).tqarg(m_frcinbytes).tqarg(MaxSize).tqarg(totalSize).tqarg(m_outbytes).tqarg(m_frcoutbytes)).ascii());
    
    if (minFactor<1)
 	minFactor=1;
