commit 62acebb74cc50671eebf21a314912b12b082be1d
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1336073612 -0500

    Fix FTBFS

diff --git a/libk9copy/k9burndvd.cpp b/libk9copy/k9burndvd.cpp
index 9f95352..3860b79 100644
--- a/libk9copy/k9burndvd.cpp
+++ b/libk9copy/k9burndvd.cpp
@@ -215,7 +215,7 @@ void k9BurnDVD::burnWithGenisoimage() {
                 if ( KMessageBox::warningContinueCancel ( 0,i18n("Insert a recordable DVD"), i18n("DVD burning"))!=KMessageBox::Continue)
                     cancelled=true;
             }
-            qDebug("%s", proc->debug());
+            qDebug("%s", proc->debug().ascii());
         }
 
 
