commit a8e98ad9ce209af0aba35ce26ec9200761b397df
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1336091322 -0500

    Fix remaining string format errors

diff --git a/k9author/k9menu.cpp b/k9author/k9menu.cpp
index 29f2b02..ea9c622 100644
--- a/k9author/k9menu.cpp
+++ b/k9author/k9menu.cpp
@@ -143,7 +143,7 @@ void k9Menu::convertJpegToMpeg(const TQString &_imageJpg,const TQString &_imageM
     if (!process->normalExit()) {
         qDebug("error converting jpeg to mpeg");
     }
-    qDebug(process->debug().latin1());
+    qDebug("%s", process->debug().latin1());
 }
 
 void k9Menu::createAudio(const TQString & _audioFile) {
diff --git a/k9author/k9newdvd.cpp b/k9author/k9newdvd.cpp
index a2c02e7..454099e 100644
--- a/k9author/k9newdvd.cpp
+++ b/k9author/k9newdvd.cpp
@@ -257,7 +257,7 @@ void k9NewDVD::createMencoderCmd(TQString &_cmd,TQString &_chapters, k9AviFile *
     *process << "-mpegopts" << "format=dvd" << "-vf" << "scale="+scale+",harddup" << "-srate" << "48000" << "-af" << "lavcresample=48000";
     *process << "-lavcopts" << TQString("vcodec=mpeg2video:vrc_buf_size=1835:vrc_maxrate=9800:vbitrate=%1:keyint=15:acodec=%3:abitrate=%2:aspect=16/9").tqarg(m_videoBitrate).tqarg(m_config->getPrefAudioBitrate()).tqarg(m_config->getPrefAudioFormat().lower());
     *process << "-ofps" << fps << "-o" << fileName << "-ss" << t1 << "-endpos" << t2 << _aviFile->getFileName();
-    qDebug(process->debug()); 
+    qDebug("%s", process->debug().ascii());
 /*
     if (!m_progress->execute()) {
         m_cancel=true;
diff --git a/k9devices/k9halconnection.cpp b/k9devices/k9halconnection.cpp
index 76c6882..e55c0b3 100644
--- a/k9devices/k9halconnection.cpp
+++ b/k9devices/k9halconnection.cpp
@@ -41,7 +41,7 @@ k9HalConnection::k9HalConnection(TQObject *parent, const char *name)
   dbus_error_init( &error );
   m_dbusConnect = dbus_bus_get( DBUS_BUS_SYSTEM, &error );
   if( dbus_error_is_set(&error) ) {
-    qDebug(TQString("Error connecting to DBUS : %1").tqarg(error.message));
+    qDebug("%s", (TQString("Error connecting to DBUS : %1").tqarg(error.message)).ascii());
     return;
   }
 
diff --git a/k9devices/k9haldevice.cpp b/k9devices/k9haldevice.cpp
index 9ddff76..0129efe 100644
--- a/k9devices/k9haldevice.cpp
+++ b/k9devices/k9haldevice.cpp
@@ -95,7 +95,7 @@ void k9HalDevice::getDriveProperties() {
     m_canReadDvd=(caps & LIBHAL_DRIVE_CDROM_CAPS_DVDROM)==LIBHAL_DRIVE_CDROM_CAPS_DVDROM;
     m_canBurnDvd=(caps & LIBHAL_DRIVE_CDROM_CAPS_DVDR)==LIBHAL_DRIVE_CDROM_CAPS_DVDR;
     m_model=TQString(libhal_drive_get_model(drive));
-    //    qDebug(TQString("canReadDvd:%1 canBurnDvd:%2 model:%3").tqarg(m_canReadDvd).tqarg(m_canBurnDvd).tqarg(m_model));
+    //    qDebug("%s", (TQString("canReadDvd:%1 canBurnDvd:%2 model:%3").tqarg(m_canReadDvd).tqarg(m_canBurnDvd).tqarg(m_model)).ascii());
 
 
     libhal_drive_free(drive);
diff --git a/k9vamps/k9vamps.cpp b/k9vamps/k9vamps.cpp
index f910979..d7bd43d 100755
--- a/k9vamps/k9vamps.cpp
+++ b/k9vamps/k9vamps.cpp
@@ -368,7 +368,7 @@ int k9vamps::check_video_packet (uchar *ptr) {
 
     if (ptr [7]) {
         if ((ptr [7] & 0xc0) != 0xc0)
-           qDebug (TQString("First video packet in sequence starting at %1 misses PTS or DTS, flags=%2").tqarg(rtell (ptr)).tqarg(ptr [7]));
+           qDebug ("%s", (TQString("First video packet in sequence starting at %1 misses PTS or DTS, flags=%2").tqarg(rtell (ptr)).tqarg(ptr [7])).ascii());
         else {
             sequence_header_code  = (uint32_t) (ptr [6 + 3 + ptr [8] + 0]) << 24;
             sequence_header_code |= (uint32_t) (ptr [6 + 3 + ptr [8] + 1]) << 16;
diff --git a/libk9copy/k9cellcopylist.cpp b/libk9copy/k9cellcopylist.cpp
index 31e85a3..fe880ac 100644
--- a/libk9copy/k9cellcopylist.cpp
+++ b/libk9copy/k9cellcopylist.cpp
@@ -298,7 +298,7 @@ double k9CellCopyList::getMinFactor(bool _withMenus) {
 
    double minFactor=(fforced-m_frcinbytes) /(MaxSize- (totalSize +m_outbytes) - m_frcoutbytes) ;
    qDebug("%s", (TQString("minfactor(%1)=(fforced(%2) -m_frinbytes(%3))/(MacSize(%4)-totalSize(%5)-m_outbytes(%6) - m_frcoutbytes(%7))").tqarg(minFactor).tqarg(fforced).tqarg(m_frcinbytes).tqarg(MaxSize).tqarg(totalSize).tqarg(m_outbytes).tqarg(m_frcoutbytes)).ascii());
-   
+
    if (minFactor<1)
 	minFactor=1;
    return minFactor;
@@ -347,7 +347,7 @@ double k9CellCopyList::getfactor(bool _withMenus,bool _streams) {
  
     if (dvdSize2 <0) {
     	factor=2.5;
-    	//qDebug(TQString("dvdSize (%1)- _outbytes(%2) - fforcedsh(%3)=%4").tqarg(dvdSize).tqarg(m_outbytes).tqarg(fforcedsh).tqarg(dvdSize2));	
+    	//qDebug("%s", (TQString("dvdSize (%1)- _outbytes(%2) - fforcedsh(%3)=%4").tqarg(dvdSize).tqarg(m_outbytes).tqarg(fforcedsh).tqarg(dvdSize2).ascii()));
     }
     else {
 	factor=(totalSize +menuSize -  fforced -m_inbytes)/ dvdSize2 ;
@@ -361,7 +361,7 @@ double k9CellCopyList::getfactor(bool _withMenus,bool _streams) {
 	uint64_t i=m_inbytes;
 	//if (o==0)
 	//	o=1;
-	//qDebug(TQString("factor : %1   realise : %2").tqarg(factor).tqarg((double)i/(double)o));
+	//qDebug("%s", (TQString("factor : %1   realise : %2").tqarg(factor).tqarg((double)i/(double)o).ascii()));
 	
 	if (factor<=1)
 		factor=1;
diff --git a/libk9copy/k9dvdauthor.cpp b/libk9copy/k9dvdauthor.cpp
index 3676109..256cf53 100755
--- a/libk9copy/k9dvdauthor.cpp
+++ b/libk9copy/k9dvdauthor.cpp
@@ -505,7 +505,7 @@ void k9DVDAuthor::DVDAuthorStderr(KProcess *proc, char *buffer, int buflen ) {
         //   m_copied+=totalBytes - m_lastPos;
         m_copied=totalBytes;
         m_lastPos=totalBytes;
-        //qDebug(TQString("copied : %1   totalSize : %2").tqarg(m_copied).tqarg(m_totalSize*512));
+        //qDebug("%s", (TQString("copied : %1   totalSize : %2").tqarg(m_copied).tqarg(m_totalSize*512).ascii());
         m_percent=(float)m_copied / (float)(m_totalSize/DVD_VIDEO_LB_LEN );
 
 
@@ -526,7 +526,7 @@ void k9DVDAuthor::DVDAuthorStderr(KProcess *proc, char *buffer, int buflen ) {
     }
     /*    else {
             if (!m_stderr.startsWith("libdvdread") && m_stderr.startsWith("libdvdnav"))
-    	   qDebug(m_stderr);
+    	   qDebug("%s", m_stderr.ascii());
         }
     */
     int end;
diff --git a/libk9copy/k9ifo2.cpp b/libk9copy/k9ifo2.cpp
index 57643c4..8c73213 100644
--- a/libk9copy/k9ifo2.cpp
+++ b/libk9copy/k9ifo2.cpp
@@ -870,7 +870,7 @@ void k9Ifo2::checkBuffer(TQString lib,uchar* _buffer) {
     if (memcmp(bufCopy,_buffer,bufCopySize)!=0) {
         TQString c;
         c="Buffer de sortie altéré: " +lib;
-        qDebug(c.latin1());
+        qDebug("%s", c.latin1());
     }
 }
 
diff --git a/libk9copy/k9mp4enc.cpp b/libk9copy/k9mp4enc.cpp
index e29f206..0f4aaca 100644
--- a/libk9copy/k9mp4enc.cpp
+++ b/libk9copy/k9mp4enc.cpp
@@ -392,7 +392,7 @@ void k9MP4Enc::execute ( k9DVDTitle *_title )
 				TQCString str=* ( m_process->args().at ( i ) );
 				s +=TQString ( str ) +" ";
 			}
-			qDebug ( s );
+			qDebug ( "%s", s.ascii() );
 			time->start();
 			m_timer3.start();
 			connect ( m_process, TQT_SIGNAL ( receivedStdout ( KProcess *, char *, int ) ),this, TQT_SLOT ( getStdout ( KProcess *, char *, int ) ) );
