diff -Nuar knetworkmanager9/knetworkmanager-0.9/src.ORI//knetworkmanager-connection_setting_wireless_security_widget.cpp knetworkmanager9/knetworkmanager-0.9/src/knetworkmanager-connection_setting_wireless_security_widget.cpp
--- knetworkmanager9/knetworkmanager-0.9/src.ORI//knetworkmanager-connection_setting_wireless_security_widget.cpp	2011-11-11 20:45:47.964680619 +0100
+++ knetworkmanager9/knetworkmanager-0.9/src/knetworkmanager-connection_setting_wireless_security_widget.cpp	2011-11-11 20:51:32.689072955 +0100
@@ -432,14 +432,15 @@
 	, _wireless_setting(wireless_setting)
 {
 	txtPSK->setText(_security_setting->getPSK());
-	connect(txtPSK, TQT_SIGNAL(textChanged(const TQString&)), this, TQT_SLOT(slotPSKChanged(const TQString&)));
+	connect(txtPSK, TQT_SIGNAL(lostFocus()), this, TQT_SLOT(slotPSKChanged()));
 }
 
 void
-WirelessSecurityWPAPSKImpl::slotPSKChanged(const TQString& psk)
+WirelessSecurityWPAPSKImpl::slotPSKChanged()
 {
 	if (_wireless_setting)
 	{
+		TQString psk = txtPSK->text();
 		_security_setting->setPSK(psk);
 	}
 }
diff -Nuar knetworkmanager9/knetworkmanager-0.9/src.ORI//knetworkmanager-connection_setting_wireless_security_widget.h knetworkmanager9/knetworkmanager-0.9/src/knetworkmanager-connection_setting_wireless_security_widget.h
--- knetworkmanager9/knetworkmanager-0.9/src.ORI//knetworkmanager-connection_setting_wireless_security_widget.h	2011-11-11 20:45:47.962680657 +0100
+++ knetworkmanager9/knetworkmanager-0.9/src/knetworkmanager-connection_setting_wireless_security_widget.h	2011-11-11 20:46:21.640030856 +0100
@@ -145,7 +145,7 @@
 			TQCString String2Hex(TQByteArray bytes, int final_len) const;
 
 		public slots:
-			void slotPSKChanged(const TQString&);
+			void slotPSKChanged();
 
 		private:
 			WirelessSecurity* _security_setting;
