diff -Nur kdelibs-3.5.8/kdeui/khelpmenu.cpp kdelibs-3.5.8.new/kdeui/khelpmenu.cpp
--- kdelibs-3.5.8/kdeui/khelpmenu.cpp	2007-10-08 10:52:12.000000000 +0100
+++ kdelibs-3.5.8.new/kdeui/khelpmenu.cpp	2007-10-09 09:42:02.000000000 +0100
@@ -41,6 +41,7 @@
 #include <kpopupmenu.h>
 #include <kstdaccel.h>
 #include <kstdaction.h>
+#include <kprocess.h>
 
 #include "kswitchlanguagedialog.h"
 
@@ -94,6 +95,8 @@
     KStdAction::reportBug(this, TQT_SLOT(reportBug()), actions);
     KStdAction::aboutApp(this, TQT_SLOT(aboutApplication()), actions);
     KStdAction::aboutKDE(this, TQT_SLOT(aboutKDE()), actions);
+    KStdAction::kubuntuTranslate(this, TQT_SLOT(kubuntuTranslate()), actions);
+    KStdAction::kubuntuGetHelpOnline(this, TQT_SLOT(kubuntuGetHelpOnline()), actions);
     KStdAction::switchApplicationLanguage(this, TQT_SLOT(switchApplicationLanguage()), actions);
   }
 }
@@ -164,6 +167,10 @@
     if (need_separator)
       mMenu->insertSeparator();
 
+    mMenu->insertItem( SmallIcon("launchpad"), i18n( "&Get Help Online" ), this, TQT_SLOT(kubuntuGetHelpOnline()) );
+    mMenu->insertItem( SmallIcon("locale"), i18n( "&Translate this Application" ), this, TQT_SLOT(kubuntuTranslate()) );
+    mMenu->insertSeparator();
+
     if (kapp->authorizeKAction("help_about_app"))
     {
       mMenu->insertItem( kapp->miniIcon(),
@@ -233,6 +240,24 @@
 }
 
 
+void KHelpMenu::kubuntuGetHelpOnline()
+{
+  const KAboutData *aboutData = KGlobal::instance()->aboutData();
+  TQString path = KStandardDirs::findExe( aboutData->appName() );
+  KProcess *proc = new KProcess;
+  *proc << "launchpad-integration" << "--file" << path << "--info";
+  proc->start(KProcess::DontCare);
+}
+
+void KHelpMenu::kubuntuTranslate()
+{
+  const KAboutData *aboutData = KGlobal::instance()->aboutData();
+  TQString path = KStandardDirs::findExe( aboutData->appName() );
+  KProcess *proc = new KProcess;
+  *proc << "launchpad-integration" << "--file" << path << "--translate";
+  proc->start(KProcess::DontCare);
+}
+
 void KHelpMenu::aboutKDE()
 {
   if( !mAboutKDE )
diff -Nur kdelibs-3.5.8/kdeui/khelpmenu.h kdelibs-3.5.8.new/kdeui/khelpmenu.h
--- kdelibs-3.5.8/kdeui/khelpmenu.h	2007-10-08 10:52:12.000000000 +0100
+++ kdelibs-3.5.8.new/kdeui/khelpmenu.h	2007-10-09 09:39:54.000000000 +0100
@@ -219,6 +219,9 @@
      */
     void aboutKDE();
 
+    void kubuntuTranslate();
+    void kubuntuGetHelpOnline();
+
     /**
      * Opens the standard "Report Bugs" dialog box.
      */
diff -Nur kdelibs-3.5.8/kdeui/kstdaction.cpp kdelibs-3.5.8.new/kdeui/kstdaction.cpp
--- kdelibs-3.5.8/kdeui/kstdaction.cpp	2007-10-08 10:52:12.000000000 +0100
+++ kdelibs-3.5.8.new/kdeui/kstdaction.cpp	2007-10-09 09:39:54.000000000 +0100
@@ -319,5 +319,9 @@
 	{ return KStdAction::create( AboutApp, name, recvr, slot, parent ); }
 KAction *aboutKDE( const TQObject *recvr, const char *slot, KActionCollection* parent, const char *name )
 	{ return KStdAction::create( AboutKDE, name, recvr, slot, parent ); }
+KAction *kubuntuTranslate( const TQObject *recvr, const char *slot, KActionCollection* parent, const char *name )
+	{ return KStdAction::create( KubuntuTranslate, name, recvr, slot, parent ); }
+KAction *kubuntuGetHelpOnline( const TQObject *recvr, const char *slot, KActionCollection* parent, const char *name )
+	{ return KStdAction::create( KubuntuGetHelpOnline, name, recvr, slot, parent ); }
 
 }
diff -Nur kdelibs-3.5.8/kdeui/kstdaction.h kdelibs-3.5.8.new/kdeui/kstdaction.h
--- kdelibs-3.5.8/kdeui/kstdaction.h	2007-10-08 10:52:12.000000000 +0100
+++ kdelibs-3.5.8.new/kdeui/kstdaction.h	2007-10-09 09:42:41.000000000 +0100
@@ -158,6 +158,7 @@
 		FullScreen, ///< @since 3.2
 		Clear, ///< @since 3.2
 		PasteText, ///< @since 3.2
+ 		KubuntuGetHelpOnline, KubuntuTranslate,
 		SwitchApplicationLanguage ///< @since 3.5.8
 	};
 
@@ -609,6 +610,12 @@
 	*/
 	KDEUI_EXPORT KAction *switchApplicationLanguage(const TQObject *recvr, const char *slot,
 				KActionCollection* parent, const char *name = 0 );
+
+	KDEUI_EXPORT KAction *kubuntuTranslate(const TQObject *recvr, const char *slot,
+				KActionCollection* parent, const char *name = 0 );
+
+	KDEUI_EXPORT KAction *kubuntuGetHelpOnline(const TQObject *recvr, const char *slot,
+				KActionCollection* parent, const char *name = 0 );
 }
 
 #endif // KSTDACTION_H
diff -Nur kdelibs-3.5.8/kdeui/kstdaction_p.h kdelibs-3.5.8.new/kdeui/kstdaction_p.h
--- kdelibs-3.5.8/kdeui/kstdaction_p.h	2007-10-08 10:52:12.000000000 +0100
+++ kdelibs-3.5.8.new/kdeui/kstdaction_p.h	2007-10-09 09:39:54.000000000 +0100
@@ -112,6 +112,8 @@
 	{ SwitchApplicationLanguage,     KStdAccel::AccelNone, "switch_application_language", I18N_NOOP("Switch application &language..."), 0, 0 },
 	{ AboutApp,      KStdAccel::AccelNone, "help_about_app", I18N_NOOP("&About %1"), 0, 0 },
 	{ AboutKDE,      KStdAccel::AccelNone, "help_about_kde", I18N_NOOP("About &Trinity"), 0,"about_kde" },
+	{ KubuntuGetHelpOnline,      KStdAccel::AccelNone, "kubuntu_get_help_online", I18N_NOOP("&Get Help Online"), 0, "launchpad" },
+	{ KubuntuTranslate,      KStdAccel::AccelNone, "kubuntu_translate", I18N_NOOP("&Translate This Application"), 0, "locale" },
 	{ ActionNone, KStdAccel::AccelNone, 0, 0, 0, 0 }
 };
 
diff -Nur kdelibs-3.5.8/kdeui/ui_standards.rc kdelibs-3.5.8.new/kdeui/ui_standards.rc
--- kdelibs-3.5.8/kdeui/ui_standards.rc	2007-10-08 10:52:12.000000000 +0100
+++ kdelibs-3.5.8.new/kdeui/ui_standards.rc	2007-10-09 09:39:54.000000000 +0100
@@ -178,6 +178,9 @@
     <Separator/>
     <Action name="switch_application_language"/>
     <Separator/>
+    <Action name="kubuntu_get_help_online"/>
+    <Action name="kubuntu_translate"/>
+    <Separator/>
     <MergeLocal name="about_merge"/>
     <Action name="help_about_app"/>
     <Action name="help_about_kde"/>
