--- kdeaddons-3.5.3/noatun-plugins/tippercanoe/sdlwrap.cpp.than	2006-06-07 18:52:09.000000000 +0200
+++ kdeaddons-3.5.3/noatun-plugins/tippercanoe/sdlwrap.cpp	2006-06-07 18:54:06.000000000 +0200
@@ -183,6 +183,7 @@
 	SDL_UpdateRect(surface, 0, 0, 0, 0);
 }
 
+#if 0
 int SdlScreen::winID(void)
 {
 	SDL_SysWMinfo info;
@@ -195,3 +196,4 @@
 	SDL_GetWMInfo(&info);
 	return info.info.x11.wmwindow;
 }
+#endif
--- kdeaddons-3.5.3/noatun-plugins/tippercanoe/main.cpp.than	2006-06-07 18:54:19.000000000 +0200
+++ kdeaddons-3.5.3/noatun-plugins/tippercanoe/main.cpp	2006-06-07 18:54:27.000000000 +0200
@@ -112,7 +112,7 @@
 {
 	time_t timer = time(NULL);
 
-	printf("%u\n", static_cast<SdlScreen *>(screen)->winID());
+	//printf("%u\n", static_cast<SdlScreen *>(screen)->winID());
 	fflush(stdout);
 
 	int frames = 0;
--- kdeaddons-3.5.3/noatun-plugins/tippercanoe/syna.h.than	2006-06-07 18:51:25.000000000 +0200
+++ kdeaddons-3.5.3/noatun-plugins/tippercanoe/syna.h	2006-06-07 18:51:56.000000000 +0200
@@ -108,7 +108,7 @@
 	int  sizeUpdate();
 	bool inputUpdate(int &mouseX,int &mouseY,int &mouseButtons,char &keyHit);
 	void show();
-	int winID();
+	//int winID();
 };
 
 // core
--- kdeaddons-3.5.3/noatun-plugins/tyler/main.cpp.than	2006-06-07 18:54:38.000000000 +0200
+++ kdeaddons-3.5.3/noatun-plugins/tyler/main.cpp	2006-06-07 18:55:00.000000000 +0200
@@ -52,6 +52,7 @@
 SDL_Surface *screen;
 }
 
+#if 0
 int winID(void)
 {
 	SDL_SysWMinfo info;
@@ -61,6 +62,7 @@
 	SDL_GetWMInfo(&info);
 	return info.info.x11.wmwindow;
 }
+#endif
 
 int main(int, char **)
 {
@@ -74,7 +76,7 @@
 	access_mutex = SDL_CreateMutex();
 
 	init_renderer1();
-	printf("%u\n", winID());
+	//printf("%u\n", winID());
 	fflush(stdout);
 	init_renderer2();
 	thread = SDL_CreateThread((int (*)(void *))renderer, NULL);
