--- trinity-tdepowersave-14.0.0~pre158+340d2058/po/CMakeLists.txt	2013-08-19 06:54:55.000000000 +0200
+++ trinity-tdepowersave-14.0.0~pre158+340d2058/po/CMakeLists.txt.new	2013-08-19 19:25:41.566465666 +0200
@@ -9,9 +9,4 @@
 #
 #################################################
 
-# file( GLOB_RECURSE po_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} tdepowersave.po )
-
-foreach( _po ${po_files} )
-  get_filename_component( _lang ${_po} PATH )
-  tde_create_translation( FILES ${_po} LANG ${_lang} )
-endforeach( )
+tde_create_translation( LANG auto OUTPUT_NAME tdepowersave )
diff --git a/ConfigureChecks.cmake b/ConfigureChecks.cmake
index c257983..68fca16 100644
--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -57,6 +57,14 @@ if( NOT DBUS_TQT_FOUND )
   tde_message_fatal( "dbus-tqt is required, but was not found on your system" )
 endif( )
 
+# check for msgfmt
+if( NOT DEFINED MSGFMT_EXECUTABLE )
+  find_program( MSGFMT_EXECUTABLE msgfmt )
+  if( NOT MSGFMT_EXECUTABLE )
+    tde_message_fatal( "msgfmt program is required, but was not found on your system" )
+  endif( )
+endif( )
+
 
 # required stuff
 find_package( TQt )
