--- kdetoys/kweather/Makefile.am.ORI	2012-08-01 10:42:11.015544034 +0200
+++ kdetoys/kweather/Makefile.am	2012-08-01 10:44:37.144904650 +0200
@@ -26,7 +26,8 @@
                                  weatherIface.skel weatherservice.stub \
                                  weatherbutton.cpp
 weather_panelapplet_la_LDFLAGS = $(KDE_PLUGIN) -module $(all_libraries)
-weather_panelapplet_la_LIBADD = $(LIB_KHTML)
+weather_panelapplet_la_LIBADD = $(LIB_KHTML) $(LIB_QT) -lDCOP
+
 
 appletdir = $(kde_datadir)/kicker/applets
 applet_DATA = kweather.desktop
@@ -67,7 +68,7 @@
 kde_services_DATA = kcmweather.desktop kcmweatherservice.desktop
 
 kweatherreport_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
-kweatherreport_la_LIBADD = $(LIB_KHTML) reportview.lo
+kweatherreport_la_LIBADD = $(LIB_KHTML) reportview.lo $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) -lDCOP 
 kweatherreport_la_SOURCES = reportmain.cpp weatherservice.stub
 
 KDE_ICON = kweather
