commit 298da70ab4f6da3780c617a85957a80f2e7ae71b
Author: Francois Andriot <francois.andriot@free.fr>
Date:   1398949579 +0200

    Fix FTBFS on Fedora 20
    Fix custom-installed Sword detection

diff --git a/bibletime/backend/clanguagemgr.cpp b/bibletime/backend/clanguagemgr.cpp
index 8f73ec3..ac94790 100644
--- a/bibletime/backend/clanguagemgr.cpp
+++ b/bibletime/backend/clanguagemgr.cpp
@@ -95,10 +95,10 @@ const CLanguageMgr::LangMap& CLanguageMgr::availableLanguages() {
 
 		for (ListCSwordModuleInfo::iterator it(mods.begin()); it != end_it; ++it) {
 			//   for (CSwordModuleInfo* m = mods.first(); m; m = mods.next()) {
-			abbrev = (*it)->module()->Lang();
+			const char* mod = (*it)->module()->Lang();
 
-			if (abbrev && !abbrevs.contains(abbrev)) {
-				abbrevs.append( abbrev );
+			if (mod && !abbrevs.contains(mod)) {
+				abbrevs.append( mod );
 			}
 		};
 
diff --git a/sword.m4 b/sword.m4
index 2f19b3f..8326ba3 100644
--- a/sword.m4
+++ b/sword.m4
@@ -28,7 +28,7 @@ AC_ARG_ENABLE(static-sword,
 dnl try to find Sword library files
 AC_MSG_CHECKING([for Sword library files])
 AC_REQUIRE([AC_FIND_ZLIB])
-ac_sword_library_dirs="$ac_sword_dir/lib $exec_prefix/lib $prefix/lib /usr/lib /usr/lib64 /usr/lib/sword /usr/local/lib /usr/local/lib/sword /usr/local/sword/lib"
+ac_sword_library_dirs="$ac_sword_dir/lib64 $ac_sword_dir/lib $exec_prefix/lib $prefix/lib64 $prefix/lib /usr/lib /usr/lib64 /usr/lib/sword /usr/local/lib /usr/local/lib/sword /usr/local/sword/lib"
 
 if test "x$ac_static_sword" = "xYES"; then
 	SEARCH_LIBS="libsword.a";
