--- piklab/src/common/common/range.h.ORI	2012-05-06 11:39:36.080834047 +0200
+++ piklab/src/common/common/range.h	2012-05-06 11:39:43.239684938 +0200
@@ -43,7 +43,7 @@
 {
 public:
   GenericRangeVector() {}
-  GenericRangeVector(const RangeType &range) { append(range); }
+  GenericRangeVector(const RangeType &range) { this->append(range); }
   bool isEmpty() const {
     uint nb = this->count();
     for (uint i=0; i<nb; i++) if ( !this->at(i).isEmpty() ) return false;
--- piklab/src/xml_to_data/device_xml_to_data.h.ORI	2012-05-06 11:45:56.824911257 +0200
+++ piklab/src/xml_to_data/device_xml_to_data.h	2012-05-06 11:46:04.135759246 +0200
@@ -76,7 +76,7 @@
     TQValueList<DataType *> list;
     for (it=_map.begin(); it!=_map.end(); ++it)
       list.append(const_cast<DataType *>(static_cast<const DataType *>(it.data())));
-    uint size = toCppString(list, ts);
+    uint size = this->toCppString(list, ts);
     ts << ";" << endl;
     ts << "const uint " << namespaceName() << "::DATA_SIZE = " << size << ";" << endl;
     file.close();
--- piklab/src/devices/base/device_group.h.ORI	2012-05-06 11:47:08.950411718 +0200
+++ piklab/src/devices/base/device_group.h	2012-05-06 11:47:17.607231757 +0200
@@ -52,7 +52,7 @@
 {
 protected:
   virtual void initSupported() {
-    TQValueList<DataType *> list = fromCppString(dataStream(), dataSize());
+    TQValueList<DataType *> list = this->fromCppString(dataStream(), dataSize());
     for (uint i=0; i<uint(list.count()); i++) addDevice(list[i]->name(), list[i], ::Group::Support::Tested);
   }
   virtual uint dataSize() const = 0;
