commit 80bc593eb31b3162fd870ee64eceb0ec90fac15c
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1327531088 -0600

    Fix linear alphabet string errors

diff --git a/NewsLog.txt b/NewsLog.txt
index bc32c09..cc5cdc4 100644
--- a/NewsLog.txt
+++ b/NewsLog.txt
@@ -1960,7 +1960,7 @@ Bugfixes
    deleted without ever having been initialised. Also init some other vars ASAP. (700035)
  - kolab/issue1712 Update revision when adding new attendees.(663228)
  - kolab/issue1672 Make sure to open the folder when saving a single message as well.(664661)
- - Fix lack of double-quotes around folder name, making GEQUOTAROOT fail on folders with a space in the
+ - Fix lack of double-quotes around folder name, making GETQUOTAROOT fail on folders with a space in the
    name.(665276)
  - Port the ability to add and remove subfolders from korganizer's resource view from proko2. (665274)
  - kolab/issue1721 Don't exit when the popup menu is canceled.(665679)
diff --git a/kioslaves/imap4/rfcdecoder.cc b/kioslaves/imap4/rfcdecoder.cc
index f91e9f8..08b641b 100644
--- a/kioslaves/imap4/rfcdecoder.cc
+++ b/kioslaves/imap4/rfcdecoder.cc
@@ -37,7 +37,7 @@
 // adapted to QT-Toolkit by Sven Carstens <s.carstens@gmx.de> 2000
 
 static unsigned char base64chars[] =
-  "ABCDEFGHIJKLMNOPTQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+,";
+  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+,";
 #define UNDEFINED 64
 #define MAXLINE  76
 
diff --git a/libkmime/kmime_util.cpp b/libkmime/kmime_util.cpp
index d4c8943..03a14e9 100644
--- a/libkmime/kmime_util.cpp
+++ b/libkmime/kmime_util.cpp
@@ -337,7 +337,7 @@ TQCString encodeRFC2047String(const TQString &src, const char *charset,
 
 TQCString uniqueString()
 {
-  static char chars[] = "0123456789abcdefghijklmnopqrstuvxyzABCDEFGHIJKLMNOPTQRSTUVWXYZ";
+  static char chars[] = "0123456789abcdefghijklmnopqrstuvxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
   time_t now;
   TQCString ret;
   char p[11];
diff --git a/mimelib/dw_cte.cpp b/mimelib/dw_cte.cpp
index 127a29a..4498597 100644
--- a/mimelib/dw_cte.cpp
+++ b/mimelib/dw_cte.cpp
@@ -353,7 +353,7 @@ static int to_cr(const char* srcBuf, size_t srcLen, char* destBuf,
 }
 
 
-static char base64tab[] = "ABCDEFGHIJKLMNOPTQRSTUVWXYZ"
+static char base64tab[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
     "abcdefghijklmnopqrstuvwxyz0123456789+/";
 
 static char base64idx[128] = {
diff --git a/mimelib/mediatyp.cpp b/mimelib/mediatyp.cpp
index 87ee345..7c766fe 100644
--- a/mimelib/mediatyp.cpp
+++ b/mimelib/mediatyp.cpp
@@ -231,7 +231,7 @@ void DwMediaType::SetBoundary(const DwString& aStr)
 void DwMediaType::CreateBoundary(unsigned aLevel)
 {
     // Create a random printable string and set it as the boundary parameter
-    static const char c[] = "ABCDEFGHIJKLMNOPTQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+    static const char c[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
     const int cLen = 64;
     char buf[80];
     strcpy(buf, "Boundary-");
