--- kdeutils/klaptopdaemon/acpi.cpp.ORI	2011-11-11 17:44:28.515033721 +0100
+++ kdeutils/klaptopdaemon/acpi.cpp	2011-11-11 17:50:52.703492308 +0100
@@ -178,7 +178,7 @@
 			proc << kdesu;
 			proc << "-u";
 			proc << "root";
-			proc <<  "dpkg-statoverride --update --add root root 6755 "+helper;
+			proc <<  "chown root "+helper+"; chmod +s "+helper;
 			proc.start(KProcess::Block);	// run it sync so has_acpi below sees the results
 		}
 	} else {
--- kdeutils/klaptopdaemon/apm.cpp.ORI	2011-11-11 17:45:11.515912373 +0100
+++ kdeutils/klaptopdaemon/apm.cpp	2011-11-11 17:51:07.095195808 +0100
@@ -166,7 +166,7 @@
 			proc << kdesu;
 			proc << "-u";
 			proc << "root";
-			proc <<  TQString("dpkg-statoverride --update --add root root 6755 ")+apm_name;
+			proc <<  TQString("chown root ")+apm_name+TQString("; chmod +s ")+apm_name;
 			proc.start(KProcess::Block);	// run it sync so has_apm below sees the results
 		}
 	} else {
@@ -200,7 +200,7 @@
 			proc << kdesu;
 			proc << "-u";
 			proc << "root";
-			proc <<  "dpkg-statoverride --update --add root root 6755 "+helper;
+			proc <<  "chown root "+helper+"; chmod +s "+helper;
 			proc.start(KProcess::Block);	// run it sync so has_acpi below sees the results
 		}
 	} else {
