--- kio-locate/src/SConscript.ORI	2012-05-02 00:23:36.637608539 +0200
+++ kio-locate/src/SConscript	2012-05-02 00:23:44.142459165 +0200
@@ -28,7 +28,7 @@
     pattern.cpp
 """
 obj.libs='tqt kio'
-obj.linkflags='-module' # add more link flags
+#obj.linkflags='-module' # add more link flags
 obj.execute()
 
 # Install the protocol files
--- kio-locate/admin/kde.py.install	2011-07-07 08:10:05.000000000 +0200
+++ kio-locate/admin/kde.py	2012-05-06 20:29:48.236024293 +0200
@@ -337,7 +337,7 @@
 		opts.Save(cachefile, env)
 
 	## set default variables, one can override them in sconscript files
-	env.Append(CXXFLAGS = ['-I'+env['KDEINCLUDEPATH'], '-I'+env['QTINCLUDEPATH'], '-I/usr/include/tqt' ],
+	env.Append(CXXFLAGS = ['-I'+env['KDEINCLUDEPATH'], '-I'+env['QTINCLUDEPATH'], '-I/opt/trinity/include/tqt' ],
 			LIBPATH = [env['KDELIBPATH'], env['QTLIBPATH'] ])
 	
 	env['QT_AUTOSCAN'] = 1
@@ -773,7 +773,7 @@
 		## Define this to 1 if you are writing documentation else to 0 :)
 		#if lenv.has_key('i_am_a_documentation_writer'):
 		for file in docbook_list:
-			lenv.Depends( folder+'index.cache.bz2', nodefile )
+			lenv.Depends( folder+'index.cache.bz2', file )
 
 		if lenv.has_key('_BUILDDIR_'): folder=lenv.join(lenv['_BUILDDIR_'], folder)
 
@@ -781,7 +782,7 @@
 		lenv.KDEinstall( 'KDEDOC', lenv.join(lang,destination), lenv.join(folder,'index.cache.bz2') )
 		
 		if env['_INSTALL']:
-			dir=lenv.join(lenv.getInstDirForResType('KDEDOC'), lang, destination)
+			dir=lenv.join(env['DESTDIR'], lenv.getInstDirForResType('KDEDOC'), lang, destination)
 			comp='mkdir -p %s && cd %s && rm -f common && ln -s ../common common' % (dir, dir)
 			lenv.Execute(comp)
 
