--- tdelibs/Makefile.am.in	2006-06-26 16:14:54.000000000 +0000
+++ tdelibs/Makefile.am.in	2006-06-26 16:16:33.000000000 +0000
@@ -54,10 +54,8 @@
 	find $$dirs -maxdepth 1 -name "*.cc" -print >> files ;\
 	find $$dirs -maxdepth 1 -name "*.h" -print >> files ;\
 	echo ./tdecore/kde-config.cpp.in >> files ;\
-	$(EXTRACTRC) `find $$dirs -maxdepth 1 \( -name "*.rc" -o -name "*.ui" \) ` ./dnssd/kcm_kdnssd.kcfg > rc.cpp; \
-	: > kde.pot ;\
+	$(EXTRACTRC) `find $$dirs -maxdepth 1 \( -name "*.rc" -o -name "*.ui" \) ` ./dnssd/kcm_kdnssd.kcfg > rc.cpp;
 	$(XGETTEXT) -ktranslate -x qt-messages.pot rc.cpp `cat files` && cat messages.po qt-messages.pot > $(podir)/tdelibs.pot
-	$(XGETTEXT) common_texts.cpp -o kde.pot
 	rm -f dirs
 
 EXTRA_DIST = admin kde.pot \
