--- trinity-python-trinity-14.0.0~pre168+4d311123/configure.py.ORI	2013-07-03 21:36:51.304975764 +0200
+++ trinity-python-trinity-14.0.0~pre168+4d311123/configure.py	2013-07-03 21:37:14.099193791 +0200
@@ -410,7 +410,7 @@
 
     if not opt_kdeincdir:
         incSearchPaths = [os.path.join (opt_tdebasedir, "include")]
-        incSearchPaths.append (os.path.join (opt_tdebasedir, "include", "kde")) # Red Hat
+        incSearchPaths.append (os.path.join (opt_tdebasedir, "include", "tde")) # Red Hat
         opt_kdeincdir = search ("tdeapplication.h", incSearchPaths)
 
 def check_kde_installation():
