--- ktechlab/src/Makefile.am.ORI	2012-08-06 11:04:54.798218350 +0200
+++ ktechlab/src/Makefile.am	2012-08-06 11:05:01.909069510 +0200
@@ -19,7 +19,7 @@
 	$(top_builddir)/src/mechanics/libmechanics.la $(top_builddir)/src/electronics/libelectronics.la \
 	$(top_builddir)/src/electronics/simulation/libelements.la $(top_builddir)/src/electronics/components/libcomponents.la \
 	$(top_builddir)/src/languages/liblanguages.la $(top_builddir)/src/drawparts/libdrawparts.la \
-	$(top_builddir)/src/core/libcore.la -lkutils $(LIB_GPSIM) -lktexteditor $(LIB_KFILE) $(LIB_KDEPRINT)
+	$(top_builddir)/src/core/libcore.la -lkutils $(LIB_GPSIM) -lktexteditor $(LIB_KFILE) $(LIB_KDEPRINT) $(LIB_QT)
 
 # which sources should be compiled for ktechlab
 ktechlab_SOURCES = ktechlab.cpp node.cpp connector.cpp itemlibrary.cpp \
