--- trinity-k9copy-3.5.13.2/k9author/k9avidecode.h.ORI	2013-07-29 21:52:57.859615464 +0200
+++ trinity-k9copy-3.5.13.2/k9author/k9avidecode.h	2013-07-29 21:57:19.206481299 +0200
@@ -43,7 +43,11 @@
 #else
 typedef int (*av_find_stream_info_t)(AVFormatContext *);
 #endif
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(54, 26, 0)
+typedef AVCodec* (*avcodec_find_decoder_t)(enum AVCodecID);
+#else
 typedef AVCodec* (*avcodec_find_decoder_t)(enum CodecID);
+#endif
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 8, 0)
 typedef int (*avcodec_open2_t)(AVCodecContext *, AVCodec *, AVDictionary **);
 #else
