diff -ur adept/adept/adept/dpkgpm.cpp adept/adept/adept/dpkgpm.cpp
--- adept/adept/adept/dpkgpm.cpp
+++ adept/adept/adept/dpkgpm.cpp
@@ -330,7 +330,7 @@
       char* list[5];
       char *line = strdup(m_statusBuffer.c_str());
       TokSplitString(':', line, list, sizeof(list)/sizeof(list[0]));
-      WriteApportReport(list[1], list[3]);
+//      WriteApportReport(list[1], list[3]);
       free(line);
       return;
    }
