--- configure.libpng	2011-09-15 07:20:32.000000000 +0200
+++ configure	2011-09-15 07:21:36.000000000 +0200
@@ -2971,8 +2971,8 @@
 if [ "$CFG_LIBPNG" = "system" ]; then
     QMAKE_CONFIG="$QMAKE_CONFIG system-png"
     if [ "$CFG_PNG" = "yes" ]; then
-	QMAKE_VARS="$QMAKE_VARS \"QMAKE_LIBS_QT+=-lpng\""
-	QMAKE_VARS="$QMAKE_VARS \"QMAKE_LIBS_QT_THREAD+=-lpng\""
+	QMAKE_VARS="$QMAKE_VARS \"QMAKE_LIBS_QT+=-lpng12\""
+	QMAKE_VARS="$QMAKE_VARS \"QMAKE_LIBS_QT_THREAD+=-lpng12\""
     fi
 fi
 if [ "$CFG_PNG" = "no" ]; then
--- plugins/src/imageformats/png/png.pro.libpng	2011-09-15 07:20:46.000000000 +0200
+++ plugins/src/imageformats/png/png.pro	2011-09-15 07:21:56.000000000 +0200
@@ -10,7 +10,7 @@
 SOURCES += main.cpp
 
 system-png {
-        unix:LIBS  += -lpng
+        unix:LIBS  += -lpng12
         win32:LIBS += libpng.lib
 }
 !system-png {
--- src/kernel/qt_gfx.pri.libpng	2011-09-15 07:21:19.000000000 +0200
+++ src/kernel/qt_gfx.pri	2011-09-15 07:22:13.000000000 +0200
@@ -102,7 +102,7 @@
 SOURCES+=$$KERNEL_CPP/qpngio.cpp
 png {
         system-png {
-	    unix:LIBS  += -lpng
+	    unix:LIBS  += -lpng12
 	    win32:LIBS += libpng.lib
 	} else {
 	    INCLUDEPATH  += 3rdparty/libpng
