--- gtk-qt-engine/kcm_gtk/kcmgtk.cpp.ORI	2013-07-23 23:35:16.614406882 +0200
+++ gtk-qt-engine/kcm_gtk/kcmgtk.cpp	2013-07-23 23:56:39.121594541 +0200
@@ -261,13 +261,21 @@
 		}
 	}
 	
+	bool gtk3installed = false;
 	widget->styleBox3->clear();
 	TQStringList otherGtk3Styles = gtk3Themes.keys();
-	otherGtk3Styles.remove(otherGtk3Styles.find("tdegtk"));
-	widget->styleBox3->insertStringList(otherGtk3Styles);
-	
-	bool gtk3installed = (gtk3Themes.find("tdegtk") != gtk3Themes.end());
+	if(!otherGtk3Styles.empty()) {
+		otherGtk3Styles.remove(otherGtk3Styles.find("tdegtk"));
+		if(!otherGtk3Styles.empty()) {
+			gtk3installed = (gtk3Themes.find("tdegtk") != gtk3Themes.end());
+		}
+	}
+	if(gtk3installed) {
+		widget->styleBox3->insertStringList(otherGtk3Styles);
+	}
 	widget->styleKde3->setEnabled(gtk3installed);
+	widget->styleBox3->setEnabled(gtk3installed);
+	widget->styleOther3->setEnabled(gtk3installed);
 	widget->warning4->setHidden(gtk3installed);
 	widget->warning5->setHidden(gtk3installed);
 	widget->warning6->setHidden(gtk3installed);
