diff -ru libr-0.6.0~maverick/src/libr-bfd.h libr-0.6.0/src/libr-bfd.h
--- libr-0.6.0~maverick/src/libr-bfd.h	2009-12-03 18:51:50.000000000 +0000
+++ libr-0.6.0/src/libr-bfd.h	2012-09-12 20:08:35.000000000 +0000
@@ -1,6 +1,8 @@
 #ifndef __LIBR_BFD_H
 #define __LIBR_BFD_H
 
+#include "config.h"
+
 #include <sys/types.h>
 #include <stdint.h>
 #include <bfd.h>
diff -ru libr-0.6.0~maverick/src/libr-internal.h libr-0.6.0/src/libr-internal.h
--- libr-0.6.0~maverick/src/libr-internal.h	2010-05-01 19:07:32.000000000 +0000
+++ libr-0.6.0/src/libr-internal.h	2012-09-12 20:08:35.000000000 +0000
@@ -7,7 +7,7 @@
 #define EXPORT_FN                  __attribute__((visibility ("protected")))
 #define INTERNAL_FN                __attribute__ ((visibility ("internal")))
 #define LIBR_TEMPFILE              "/tmp/libr-temp.XXXXXX"
-#define LIBR_TEMPFILE_LEN          21
+#define LIBR_TEMPFILE_LEN          22
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 
