diff -Nuar trinity-arts-3.5.13.2.ORI/flow/audioioesd.cc trinity-arts-3.5.13.2.P/flow/audioioesd.cc
--- trinity-arts-3.5.13.2.ORI/flow/audioioesd.cc	2013-04-08 19:05:22.665624664 +0200
+++ trinity-arts-3.5.13.2.P/flow/audioioesd.cc	2013-04-08 23:58:07.905872271 +0200
@@ -204,14 +204,10 @@
 		return write_fd;
 
 	case canRead:
-		return ESD_BUF_SIZE;
+		return param(fragmentSize);
 
 	case canWrite:
-		return ESD_BUF_SIZE;
-
-		// ESD handles are actually socket descriptors, and I know not
-		// of any portable way to peek at the socket's send or receive
-		// buffers.
+		return param(fragmentSize);
 
 	default:
 		return param(p);
