diff -Nuar arts.ORI/artsc/CMakeLists.txt arts.PATCHED/artsc/CMakeLists.txt
--- arts.ORI/artsc/CMakeLists.txt	2011-06-12 07:57:03.000000000 +0200
+++ arts.PATCHED/artsc/CMakeLists.txt	2011-11-14 21:53:12.000000000 +0100
@@ -43,6 +43,7 @@
 include_directories(
   ${CMAKE_CURRENT_BINARY_DIR}
   ${CMAKE_BINARY_DIR}
+  ${CMAKE_SOURCE_DIR}/artsc
   ${CMAKE_BINARY_DIR}/mcop
   ${CMAKE_SOURCE_DIR}/mcop
   ${CMAKE_BINARY_DIR}/flow
@@ -64,7 +65,7 @@
 tde_add_library( artsdsp SHARED
   SOURCES artsdsp.c
   VERSION 0.0.0
-  LINK artsc-shared
+  LINK artsc-shared dl
   DESTINATION ${LIB_INSTALL_DIR}
 )
 
diff -Nuar arts.ORI/CMakeLists.txt arts.PATCHED/CMakeLists.txt
--- arts.ORI/CMakeLists.txt	2011-03-28 04:09:15.000000000 +0200
+++ arts.PATCHED/CMakeLists.txt	2011-11-14 21:53:46.000000000 +0100
@@ -33,6 +33,7 @@
 option( WITH_AUDIOFILE "Enable audiofile (wav) support" ON )
 option( WITH_VORBIS "Enable Ogg/Vorbis support" ON )
 option( WITH_MAD "Enable MAD mp3 decoder support" ON )
+option( WITH_ESOUND "Enable ESOUND support" OFF )
 
 
 ##### paths setup ###############################
@@ -58,6 +59,8 @@
 check_include_file( "malloc.h" HAVE_MALLOC_H )
 check_include_file( "memory.h" HAVE_MEMORY_H )
 check_include_file( "dlfcn.h" HAVE_DLFCN_H )
+check_include_file( "sys/soundcard.h" HAVE_SYS_SOUNDCARD_H )
+check_include_file( "pthread.h" HAVE_LIBPTHREAD )
 
 
 ##### check for system libraries ################
@@ -145,6 +148,21 @@
 endif( WITH_ALSA )
 
 
+##### check for esound #######################
+
+set( HAVE_LIBESD 0 )
+if( WITH_ESOUND )
+
+  pkg_search_module( ESOUND esound )
+  if( ESOUND_FOUND )
+    set( HAVE_LIBESD 1 )
+  else( ESOUND_FOUND )
+    message(FATAL_ERROR "\nESOUND support is requested, but `libesd` not found" )
+  endif( ESOUND_FOUND )
+
+endif( WITH_ESOUND )
+
+
 ##### check for glib/gthread modules ############
 
 pkg_search_module( GLIB2 glib-2.0 )
diff -Nuar arts.ORI/config.h.cmake arts.PATCHED/config.h.cmake
--- arts.ORI/config.h.cmake	2010-10-12 03:53:06.000000000 +0200
+++ arts.PATCHED/config.h.cmake	2011-11-14 22:00:50.000000000 +0100
@@ -23,8 +23,14 @@
 
 #cmakedefine HAVE_LIBAUDIOFILE 1
 
+#cmakedefine HAVE_LIBESD 1
+
 #cmakedefine HAVE_LIBASOUND2 1
 #cmakedefine HAVE_ALSA_ASOUNDLIB_H 1
 #cmakedefine HAVE_SND_PCM_RESUME 1
 #cmakedefine ALSA_PCM_OLD_SW_PARAMS_API 1
 #cmakedefine ALSA_PCM_OLD_HW_PARAMS_API 1
+
+#cmakedefine HAVE_SYS_SOUNDCARD_H 1
+#cmakedefine HAVE_LIBPTHREAD 1
+#define HAVE_IOCTL_INT_ULONGINT_DOTS 3
diff -Nuar arts.ORI/flow/CMakeLists.txt arts.PATCHED/flow/CMakeLists.txt
--- arts.ORI/flow/CMakeLists.txt	2011-10-04 04:01:15.000000000 +0200
+++ arts.PATCHED/flow/CMakeLists.txt	2011-11-14 21:53:12.000000000 +0100
@@ -15,6 +15,7 @@
 
 include_directories(
   ${CMAKE_BINARY_DIR}
+  ${CMAKE_SOURCE_DIR}/artsc
   ${CMAKE_BINARY_DIR}/mcop
   ${CMAKE_SOURCE_DIR}/mcop
   ${CMAKE_CURRENT_BINARY_DIR}
diff -Nuar arts.ORI/mcop/CMakeLists.txt arts.PATCHED/mcop/CMakeLists.txt
--- arts.ORI/mcop/CMakeLists.txt	2010-10-12 03:53:05.000000000 +0200
+++ arts.PATCHED/mcop/CMakeLists.txt	2011-11-14 22:04:29.000000000 +0100
@@ -12,11 +12,11 @@
 configure_file( arts_export.h.in arts_export.h )
 
 if( NOT EXTENSION_DIR )
-  set( EXTENSION_DIR ${CMAKE_INSTALL_PREFIX}/lib )
+  set( EXTENSION_DIR ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX} )
 endif( NOT EXTENSION_DIR )
 
 if( NOT DTRADER_DIR )
-  set( DTRADER_DIR ${CMAKE_INSTALL_PREFIX}/lib/mcop )
+  set( DTRADER_DIR ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/mcop )
 endif( NOT DTRADER_DIR )
 
 add_definitions(
@@ -66,6 +66,6 @@
 tde_add_library( ${target} SHARED
   SOURCES ${${target}_SRCS}
   VERSION 1.0.0
-  LINK ltdlc-static
+  LINK ltdlc-static ${ESOUND_LIBRARIES}
   DESTINATION ${LIB_INSTALL_DIR}
 )
diff -Nuar arts.ORI/mcop_mt/CMakeLists.txt arts.PATCHED/mcop_mt/CMakeLists.txt
--- arts.ORI/mcop_mt/CMakeLists.txt	2010-10-12 03:53:05.000000000 +0200
+++ arts.PATCHED/mcop_mt/CMakeLists.txt	2011-11-14 21:53:12.000000000 +0100
@@ -11,6 +11,11 @@
 
 include_directories(
   ${CMAKE_BINARY_DIR}
+  ${CMAKE_BINARY_DIR}/flow
+  ${CMAKE_BINARY_DIR}/mcop
+  ${CMAKE_SOURCE_DIR}/flow
+  ${CMAKE_SOURCE_DIR}/mcop
+  ${GLIB2_INCLUDE_DIRS}
 )
 
 
@@ -19,6 +24,6 @@
 tde_add_library( mcop_mt SHARED
   SOURCES threads_posix.cc
   VERSION 1.0.0
-  LINK artsflow-shared
+  LINK artsflow-shared pthread
   DESTINATION ${LIB_INSTALL_DIR}
 )
