--- trinity-arts-3.5.13.2~pre15+85a8c7ad/flow/audioionull.cc.ORI	2013-05-26 13:35:39.081803512 +0200
+++ trinity-arts-3.5.13.2~pre15+85a8c7ad/flow/audioionull.cc	2013-05-26 13:36:02.342333674 +0200
@@ -133,6 +133,9 @@
 		if((_direction & directionWrite) && getParam(canWrite) >= _fragmentSize)
 			todo |= AudioSubSystem::ioWrite;
 
+		fprintf(stderr, "AudioIO: can write %d, fragment size %d, todo %d\n",
+			getParam(canWrite), _fragmentSize, todo);
+
 		if(!todo)
 			return;
 
