--- kpicosim/src/cassembler.cpp.ORI	2012-08-05 23:39:09.061007283 +0200
+++ kpicosim/src/cassembler.cpp	2012-08-05 23:42:53.639017712 +0200
@@ -575,7 +575,7 @@
 					for( j = 31 ; j >= 0 ; j-- )
 		 				fprintf( outfile, "%02X", INITP[ j ][ line ] ) ;
 			} else if ( strcmp( "name", varname ) == 0 ) {
-				fprintf( outfile, entityName.c_str() ) ;				
+				fprintf( outfile, "%s", entityName.c_str() ) ;				
 			} else if ( strcmp( "begin template", varname ) == 0 ) {
 				copy = true ;
 			}
