Index: b/cervisia/CMakeLists.txt
===================================================================
--- a/cervisia/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/cervisia/CMakeLists.txt	2012-11-05 23:41:26.000000000 +0100
@@ -58,6 +58,24 @@
   DESTINATION ${KCONF_UPDATE_INSTALL_DIR} )
 
 
+macro( _pod2man )
+  foreach( _pod ${ARGN} )
+    add_custom_command(
+      OUTPUT ${_pod}.1
+      COMMAND pod2man ${CMAKE_CURRENT_SOURCE_DIR}/${_pod}.pod > ${_pod}.1.in
+      COMMAND
+        sed -e 's%_KDEHTMLDIR_%'${HTML_INSTALL_DIR}'%g;'
+            -e 's%_KDECONFDIR_%'${CONFIG_INSTALL_DIR}'%g;'
+            < ${_pod}.1.in > ${_pod}.1
+      DEPENDS ${_pod} )
+    add_custom_target( "${_pod}-man" ALL DEPENDS ${_pod}.1 )
+    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/${_pod}.1 DESTINATION ${MAN_INSTALL_DIR}/man1 )
+  endforeach( )
+endmacro()
+
+_pod2man( cervisia )
+
+
 ##### cervisia (static) #########################
 
 tde_add_library( cervisia STATIC_PIC AUTOMOC
Index: b/cervisia/cvsservice/CMakeLists.txt
===================================================================
--- a/cervisia/cvsservice/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/cervisia/cvsservice/CMakeLists.txt	2012-11-05 23:41:26.000000000 +0100
@@ -41,7 +41,7 @@
 tde_add_library( cvsservice SHARED AUTOMOC
   SOURCES
     cvsservice.stub cvsjob.stub repository.stub
-  VERSION 0.1.0
+  VERSION 0.0.1
   LINK kdecore-shared
   DESTINATION ${LIB_INSTALL_DIR}
 )
Index: b/kapptemplate/CMakeLists.txt
===================================================================
--- a/kapptemplate/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/kapptemplate/CMakeLists.txt	2012-11-05 23:56:28.000000000 +0100
@@ -23,7 +23,11 @@
 
 install( PROGRAMS
     ${CMAKE_CURRENT_BINARY_DIR}/kapptemplate
+  DESTINATION ${BIN_INSTALL_DIR} )
+
+install( PROGRAMS
+    ${CMAKE_CURRENT_SOURCE_DIR}/mkinstalldirs
   DESTINATION ${DATA_INSTALL_DIR}/kapptemplate/bin )
 
 install( FILES
     kapptemplate.common kapptemplate.module

Index: b/kapptemplate/admin/CMakeLists.txt
===================================================================
--- a/kapptemplate/admin/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/kapptemplate/admin/CMakeLists.txt	2012-11-06 00:13:16.000000000 +0100
@@ -9,4 +9,9 @@
 #
 #################################################
 
-tde_install_empty_directory( ${DATA_INSTALL_DIR}/kapptemplate/admin )
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../../admin/
+  DESTINATION ${DATA_INSTALL_DIR}/kapptemplate/admin
+  USE_SOURCE_PERMISSIONS
+  REGEX "Makefile(|\\.in|\\.am)$|.*-orig$" EXCLUDE
+)
Index: b/kmtrace/CMakeLists.txt
===================================================================
--- a/kmtrace/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/kmtrace/CMakeLists.txt	2012-11-05 23:41:26.000000000 +0100
@@ -31,12 +31,16 @@
 install( PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/kminspector
   DESTINATION ${BIN_INSTALL_DIR} )
 
+install( FILES ktrace.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}
+)
+
 
 ##### ktrace_s (static) #########################
 
 tde_add_library( ktrace_s STATIC
   SOURCES ktrace.c
-  DESTINATION ${LIB_INSTALL_DIR}
+  DESTINATION ${LIB_INSTALL_DIR}/kmtrace
 )
 
 
Index: b/kompare/interfaces/CMakeLists.txt
===================================================================
--- a/kompare/interfaces/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/kompare/interfaces/CMakeLists.txt	2012-11-05 23:41:26.000000000 +0100
@@ -23,6 +23,7 @@
 
 tde_add_library( kompareinterface SHARED
   SOURCES kompareinterface.cpp
+  VERSION 0.0.0
   LINK ${TQT_LIBRARIES}
   DESTINATION ${LIB_INSTALL_DIR}
 )
Index: b/kunittest/CMakeLists.txt
===================================================================
--- a/kunittest/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/kunittest/CMakeLists.txt	2012-11-05 23:41:26.000000000 +0100
@@ -35,6 +35,7 @@
 
 tde_add_library( kunittestgui SHARED AUTOMOC
   SOURCES testerwidget.ui runnergui.cpp dcopinterface.skel
+  VERSION 0.0.0
   LINK kunittest-shared
   DESTINATION ${LIB_INSTALL_DIR}
 )
Index: b/scheck/CMakeLists.txt
===================================================================
--- a/scheck/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/scheck/CMakeLists.txt	2012-11-05 23:41:26.000000000 +0100
@@ -30,5 +30,5 @@
 tde_add_kpart( scheck AUTOMOC
   SOURCES scheck.cpp
   LINK kdeui-shared
-  DESTINATION ${PLUGIN_INSTALL_DIR}
+  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/styles
 )
Index: b/scripts/CMakeLists.txt
===================================================================
--- a/scripts/CMakeLists.txt	2012-11-05 23:40:05.000000000 +0100
+++ b/scripts/CMakeLists.txt	2012-11-05 23:41:26.000000000 +0100
@@ -36,7 +36,7 @@
       COMMAND pod2man ${CMAKE_CURRENT_SOURCE_DIR}/${_pod} > ${_pod}.1
       DEPENDS ${_pod} )
     add_custom_target( "${_pod}-man" ALL DEPENDS ${_pod}.1 )
-    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/${_pod}.1 DESTINATION ${MAN_INSTALL_DIR} )
+    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/${_pod}.1 DESTINATION ${MAN_INSTALL_DIR}/man1 )
   endforeach( )
 endmacro()
 
Index: b/kapptemplate/admin/Makefile.am
===================================================================
--- a/kapptemplate/admin/Makefile.am	2012-11-06 00:36:24.000000000 +0100
+++ b/kapptemplate/admin/Makefile.am	2012-11-06 00:48:24.000000000 +0100
@@ -1,8 +1,8 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(kde_datadir)/kapptemplate/admin
 	for file in $(srcdir)/*; do \
-		if [ -f $$file -a $$file != 'Makefile' -a $$file != 'Makefile.in' -a $$file != 'Makefile.am' ]; then \
-			destfile=`basename $$file` \
+		destfile=`basename $$file`; \
+		if [ -f $$file -a $$destfile != 'Makefile' -a $$destfile != 'Makefile.in' -a $$destfile != 'Makefile.am' ]; then \
 			$(INSTALL_DATA) $$file \
 				$(DESTDIR)$(kde_datadir)/kapptemplate/admin/$$destfile; \
 		fi \
