--- krusader/krusader/main.cpp.ORI	2012-08-06 00:19:36.475622376 +0200
+++ krusader/krusader/main.cpp	2012-08-06 00:19:48.867387323 +0200
@@ -199,7 +199,7 @@
        exit( 0 );
     TQCString regName = client->registerAs( KApplication::kApplication() ->name(), !singleInstanceMode );
     if( singleInstanceMode && regName != KApplication::kApplication()->name() ) {
-      fprintf( stderr, i18n( "Application already running!\n" ).ascii() );
+      fprintf( stderr, "%s", i18n( "Application already running!\n" ).ascii() );
 
       DCOPClient::mainClient()->send( KApplication::kApplication() ->name(), "Krusader-Interface",
                                     "moveToTop()", TQByteArray() );
