commit fdf6d34084255fa4e17833f8ef93626e5dd926e6
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1335632256 -0500

    GCC 4.7 fixes.

diff --git a/krusader/UserAction/tstring.h b/krusader/UserAction/tstring.h
index 94ece59..9356e5c 100644
--- a/krusader/UserAction/tstring.h
+++ b/krusader/UserAction/tstring.h
@@ -14,6 +14,9 @@
 #include <tqvaluelist.h>
 #include <utility>
 #include <cassert>
+
+using namespace std;
+
 template <class T>
 class TagString_t
 {
