commit 96f2a488786e370c355778e00c795e776e09d199
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1322168535 -0600

    Fix FTBFS

diff --git a/konversation/src/alias_preferences.h b/konversation/src/alias_preferences.h
index cf70b4f..aeb3062 100644
--- a/konversation/src/alias_preferences.h
+++ b/konversation/src/alias_preferences.h
@@ -13,6 +13,8 @@
 #ifndef EXALIASPREFERENCES_H
 #define EXALIASPREFERENCES_H
 
+#include <tqlistview.h>
+
 #include "alias_preferencesui.h"
 #include "konvisettingspage.h"
 
diff --git a/konversation/src/autoreplace_preferences.h b/konversation/src/autoreplace_preferences.h
index c73da29..9cab1f0 100644
--- a/konversation/src/autoreplace_preferences.h
+++ b/konversation/src/autoreplace_preferences.h
@@ -13,6 +13,8 @@
 #ifndef AUTOREPLACECONFIG_H
 #define AUTOREPLACECONFIG_H
 
+#include <tqlistview.h>
+
 #include "autoreplace_preferencesui.h"
 #include "konvisettingspage.h"
 
diff --git a/konversation/src/config/konversation.kcfg b/konversation/src/config/konversation.kcfg
index a2a2582..8cf2b82 100644
--- a/konversation/src/config/konversation.kcfg
+++ b/konversation/src/config/konversation.kcfg
@@ -294,7 +294,7 @@
       <whatsthis></whatsthis>
     </entry>
     <entry key="LogfileReaderGeometry" type="Size" name="LogfileReaderSize">
-      <default code="true">QSize(400, 200)</default>
+      <default code="true">TQSize(400, 200)</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
diff --git a/konversation/src/highlight_preferences.h b/konversation/src/highlight_preferences.h
index 67863a7..89974ca 100644
--- a/konversation/src/highlight_preferences.h
+++ b/konversation/src/highlight_preferences.h
@@ -17,7 +17,7 @@
 #include "konvisettingspage.h"
 
 #include <tqobject.h>
-
+#include <tqlistview.h>
 
 class Highlight_Config;
 class Highlight;
diff --git a/konversation/src/ignore_preferences.h b/konversation/src/ignore_preferences.h
index e910b78..a19ff03 100644
--- a/konversation/src/ignore_preferences.h
+++ b/konversation/src/ignore_preferences.h
@@ -17,7 +17,7 @@
 #include "konvisettingspage.h"
 
 #include <tqptrlist.h>
-
+#include <tqlistview.h>
 
 class Ignore;
 class Ignore_Config : public Ignore_ConfigUI, public KonviSettingsPage
diff --git a/konversation/src/quickbuttons_preferences.h b/konversation/src/quickbuttons_preferences.h
index c8cc0e1..ccf26af 100644
--- a/konversation/src/quickbuttons_preferences.h
+++ b/konversation/src/quickbuttons_preferences.h
@@ -13,6 +13,8 @@
 #ifndef TQUICKBUTTONSCONFIG_H
 #define TQUICKBUTTONSCONFIG_H
 
+#include <tqlistview.h>
+
 #include "quickbuttons_preferencesui.h"
 #include "konvisettingspage.h"
 
