commit ca3d6cef5d40d2c9faf1f0279f40b135a444d206
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1334302097 -0500

    Fix inadvertent "TQ" changes.

diff --git a/konversation/src/config/preferences.cpp b/konversation/src/config/preferences.cpp
index c28c844..5534ac8 100644
--- a/konversation/src/config/preferences.cpp
+++ b/konversation/src/config/preferences.cpp
@@ -105,7 +105,7 @@ const TQStringList Preferences::defaultQuickButtonList()
                          << "Kick,/KICK %u%n"
                          << "Ban,/BAN %u%n"
                          << "Part,/PART %c Leaving...%n"
-                         << "Quit,/TQUIT Leaving...%n";
+                         << "Quit,/QUIT Leaving...%n";
 }
 
 const TQStringList Preferences::quickButtonList()
diff --git a/konversation/src/dcctransfer.h b/konversation/src/dcctransfer.h
index af7bc97..f4b03a9 100644
--- a/konversation/src/dcctransfer.h
+++ b/konversation/src/dcctransfer.h
@@ -92,9 +92,9 @@ class DccTransfer : public TQObject
 
         // common settings for DccTransferRecv / DccTransferSend
 
-        // RETQUIRED
+        // REQUIRED
         void setConnectionId( int connectionId );
-        // RETQUIRED
+        // REQUIRED
         void setPartnerNick( const TQString& nick );
 
     signals:
diff --git a/konversation/src/dcctransferrecv.h b/konversation/src/dcctransferrecv.h
index aed4624..92f4201 100644
--- a/konversation/src/dcctransferrecv.h
+++ b/konversation/src/dcctransferrecv.h
@@ -50,11 +50,11 @@ class DccTransferRecv : public DccTransfer
         DccTransferRecv(TQObject* parent);
         virtual ~DccTransferRecv();
 
-        // RETQUIRED
+        // REQUIRED
         void setPartnerIp( const TQString& ip );
-        // RETQUIRED
+        // REQUIRED
         void setPartnerPort( const TQString& port );
-        // RETQUIRED
+        // REQUIRED
         void setFileSize( unsigned long fileSize );
         // OPTIONAL, if not specified, "unnamed_file"
         // TODO: "$sendername-$receiveddate" is better
diff --git a/konversation/src/dcctransfersend.h b/konversation/src/dcctransfersend.h
index 1a3e4ea..e4f50f0 100644
--- a/konversation/src/dcctransfersend.h
+++ b/konversation/src/dcctransfersend.h
@@ -39,11 +39,11 @@ class DccTransferSend : public DccTransfer
         DccTransferSend(TQObject* parent);
         virtual ~DccTransferSend();
 
-        // RETQUIRED
+        // REQUIRED
         void setFileURL( const KURL& url );
         // OPTIONAL
         void setFileName( const TQString& fileName );
-        // RETQUIED
+        // REQUIRED
         // FIXME: this setting should be an optional one or be removed: make DccTransferSend itself read the configuration
         void setOwnIp( const TQString& ownIp );
         // OPTIONAL
diff --git a/konversation/src/ircqueue.h b/konversation/src/ircqueue.h
index b1473d6..ec5d7d2 100644
--- a/konversation/src/ircqueue.h
+++ b/konversation/src/ircqueue.h
@@ -10,8 +10,8 @@
 */
 
 
-#ifndef IRCTQUEUE_H
-#define IRCTQUEUE_H
+#ifndef IRCQUEUE_H
+#define IRCQUEUE_H
 
 class Server;
 
diff --git a/konversation/src/outputfilter.cpp b/konversation/src/outputfilter.cpp
index 9ec34f6..f691a6d 100644
--- a/konversation/src/outputfilter.cpp
+++ b/konversation/src/outputfilter.cpp
@@ -1429,7 +1429,7 @@ namespace Konversation
 
         if(parameter.isEmpty())
         {
-            result = usage(i18n("Usage: %1TQUOTE command list").arg(commandChar));
+            result = usage(i18n("Usage: %1QUOTE command list").arg(commandChar));
         }
         else
         {
diff --git a/konversation/src/query.h b/konversation/src/query.h
index 513377e..c390ca7 100644
--- a/konversation/src/query.h
+++ b/konversation/src/query.h
@@ -10,8 +10,8 @@
   Copyright (C) 2005-2008 Eike Hein <hein@kde.org>
 */
 
-#ifndef TQUERY_H
-#define TQUERY_H
+#ifndef QUERY_H
+#define QUERY_H
 
 #include "chatwindow.h"
 #include "nickinfo.h"
diff --git a/konversation/src/queuetuner.h b/konversation/src/queuetuner.h
index c918171..be8fc55 100644
--- a/konversation/src/queuetuner.h
+++ b/konversation/src/queuetuner.h
@@ -10,8 +10,8 @@
 */
 
 
-#ifndef TQUEUETUNER_H
-#define TQUEUETUNER_H
+#ifndef QUEUETUNER_H
+#define QUEUETUNER_H
 
 #include "queuetunerbase.h"
 
diff --git a/konversation/src/quickbutton.h b/konversation/src/quickbutton.h
index 7bf9cf5..bb850ad 100644
--- a/konversation/src/quickbutton.h
+++ b/konversation/src/quickbutton.h
@@ -11,8 +11,8 @@
   email:     eisfuchs@tigress.com
 */
 
-#ifndef TQUICKBUTTON_H
-#define TQUICKBUTTON_H
+#ifndef QUICKBUTTON_H
+#define QUICKBUTTON_H
 
 #include <tqpushbutton.h>
 
diff --git a/konversation/src/quickbuttons_preferences.h b/konversation/src/quickbuttons_preferences.h
index ab04a55..8e2293c 100644
--- a/konversation/src/quickbuttons_preferences.h
+++ b/konversation/src/quickbuttons_preferences.h
@@ -10,8 +10,8 @@
   Copyright (C) 2006 John Tapsell <johnflux@gmail.com>
 */
 
-#ifndef TQUICKBUTTONSCONFIG_H
-#define TQUICKBUTTONSCONFIG_H
+#ifndef QUICKBUTTONSCONFIG_H
+#define QUICKBUTTONSCONFIG_H
 
 #include <tqlistview.h>
 
diff --git a/konversation/src/quickconnectdialog.h b/konversation/src/quickconnectdialog.h
index b41922d..3ba2ba8 100644
--- a/konversation/src/quickconnectdialog.h
+++ b/konversation/src/quickconnectdialog.h
@@ -12,8 +12,8 @@
   email:     mail@tuxipuxi.de
 */
 
-#ifndef TQUICKCONNECTDIALOG_H
-#define TQUICKCONNECTDIALOG_H
+#ifndef QUICKCONNECTDIALOG_H
+#define QUICKCONNECTDIALOG_H
 
 #include "common.h"
 
diff --git a/konversation/src/server.cpp b/konversation/src/server.cpp
index f04e75f..4c1cc6f 100644
--- a/konversation/src/server.cpp
+++ b/konversation/src/server.cpp
@@ -692,9 +692,9 @@ void Server::gotOwnResolvedHostByWelcome(KResolverResults res)
 
 void Server::quitServer()
 {
-    // Make clear this is deliberate even if the TQUIT never actually goes through the queue
+    // Make clear this is deliberate even if the QUIT never actually goes through the queue
     // (i.e. this is not redundant with _send_internal()'s updateConnectionState() call for
-    // a TQUIT).
+    // a QUIT).
     updateConnectionState(Konversation::SSDeliberatelyDisconnected);
 
     TQString command(Preferences::commandChar()+"QUIT");
@@ -1050,7 +1050,7 @@ int Server::getPreLength(const TQString& command, const TQString& dest)
 }
 
 //Commands greater than 1 have localizeable text:         0   1    2       3      4    5    6
-static TQStringList outcmds=TQStringList::split(TQChar(' '),"WHO TQUIT PRIVMSG NOTICE KICK PART TOPIC");
+static TQStringList outcmds=TQStringList::split(TQChar(' '),"WHO QUIT PRIVMSG NOTICE KICK PART TOPIC");
 
 int Server::_send_internal(TQString outputLine)
 {
