diff --git a/src/gstplayer.cpp b/src/gstplayer.cpp
index c4bf580..0585ac9 100644
--- a/src/gstplayer.cpp
+++ b/src/gstplayer.cpp
@@ -750,7 +750,7 @@ void KGStreamerPlayer::seek (int val /*offset_in_deciseconds*/) {
 void KGStreamerPlayer::volume (int val) {
     //fprintf (stderr, "position %d\n", val);
     if (gst_elm_play)
-        g_object_set (G_OBJECT (gst_elm_play), "volume", 1.0*val/100, 0L);
+        g_object_set (G_OBJECT (gst_elm_play), "volume", 1.0*val/100, NULL);
 }
 
 void KGStreamerPlayer::updatePosition () {
