diff -u -r kdenetwork/kopete/plugins/motionautoaway/motionawayplugin.cpp src/kdenetwork/kopete/plugins/motionautoaway/motionawayplugin.cpp
--- kdenetwork/kopete/plugins/motionautoaway/motionawayplugin.cpp	2011-08-12 22:11:02.000000000 +0200
+++ src/kdenetwork/kopete/plugins/motionautoaway/motionawayplugin.cpp	2011-12-15 23:25:02.000000000 +0100
@@ -75,7 +75,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,50)
 #define _LINUX_TIME_H
 #endif
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 #define DEF_WIDTH			352
 #define DEF_HEIGHT		288
diff -u -r kdenetwork/kopete/protocols/jabber/CMakeLists.txt src/kdenetwork/kopete/protocols/jabber/CMakeLists.txt
--- kdenetwork/kopete/protocols/jabber/CMakeLists.txt	2011-06-22 03:37:12.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/CMakeLists.txt	2011-12-15 22:56:58.000000000 +0100
@@ -80,6 +80,6 @@
     jabberclient-static kopetejabberui-static
     iris_jabber-static iris_xmpp_core-static iris_xmpp_im-static iris-static
     qca-static cutestuff_network-static cutestuff_util-static ${JINGLE_LIBRARIES}
-    kopete-shared
+    kopete-shared idn
   DESTINATION ${PLUGIN_INSTALL_DIR}
 )
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h	2011-06-22 03:37:10.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/sigslot.h	2011-12-15 22:56:58.000000000 +0100
@@ -80,6 +80,7 @@
 
 #include <set>
 #include <list>
+#include <stdlib.h>
 
 // On our copy of sigslot.h, we force single threading
 #define SIGSLOT_PURE_ISO
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/p2p/client/sessionclient.cc src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/p2p/client/sessionclient.cc
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/p2p/client/sessionclient.cc	2011-06-22 03:37:08.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/p2p/client/sessionclient.cc	2011-12-15 22:56:58.000000000 +0100
@@ -31,7 +31,7 @@
 #include "talk/p2p/client/sessionclient.h"
 #include "talk/p2p/base/helpers.h"
 #include "talk/base/logging.h"
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 #include "talk/xmpp/constants.h"
 #include "talk/xmllite/xmlprinter.h"
 #include <iostream>
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/qname.cc src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/qname.cc
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/qname.cc	2011-06-22 03:37:11.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/qname.cc	2011-12-15 22:56:58.000000000 +0100
@@ -28,7 +28,7 @@
 #include <string>
 #include "talk/base/common.h"
 #include "talk/xmllite/xmlelement.h"
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 #include "talk/xmllite/xmlconstants.h"
 
 //#define new TRACK_NEW
Only in src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite: tqname.cc
Only in src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite: tqname.h
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.cc src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.cc
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.cc	2011-06-22 03:37:11.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.cc	2011-12-15 22:56:58.000000000 +0100
@@ -32,7 +32,7 @@
 
 #include "talk/base/common.h"
 #include "talk/xmllite/xmlelement.h"
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 #include "talk/xmllite/xmlparser.h"
 #include "talk/xmllite/xmlbuilder.h"
 #include "talk/xmllite/xmlprinter.h"
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.h src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.h
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.h	2011-06-22 03:37:11.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlelement.h	2011-12-15 22:56:58.000000000 +0100
@@ -31,7 +31,7 @@
 #include <iosfwd>
 #include <string>
 #include "talk/base/scoped_ptr.h"
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 
 namespace buzz {
 
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlnsstack.h src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlnsstack.h
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlnsstack.h	2011-06-22 03:37:11.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmllite/xmlnsstack.h	2011-12-15 22:56:58.000000000 +0100
@@ -31,7 +31,7 @@
 #include <string>
 #include "talk/base/scoped_ptr.h"
 #include "talk/base/stl_decl.h"
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 
 namespace buzz {
 
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.cc src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.cc
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.cc	2011-06-22 03:37:09.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.cc	2011-12-15 22:56:58.000000000 +0100
@@ -29,7 +29,7 @@
 #include "talk/base/basicdefs.h"
 #include "talk/xmllite/xmlconstants.h"
 #include "talk/xmllite/xmlelement.h"
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 #include "talk/xmpp/jid.h"
 #include "talk/xmpp/constants.h"
 namespace buzz {
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.h src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.h
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.h	2011-06-22 03:37:09.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/constants.h	2011-12-15 22:56:58.000000000 +0100
@@ -29,7 +29,7 @@
 #define _CRICKET_XMPP_XMPPLIB_BUZZ_CONSTANTS_H_
 
 #include <string>
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 #include "talk/xmpp/jid.h"
 
 
diff -u -r kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmppengine.h src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmppengine.h
--- kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmppengine.h	2011-06-22 03:37:09.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmppengine.h	2011-12-15 22:56:58.000000000 +0100
@@ -30,7 +30,7 @@
 
 // also part of the API
 #include "talk/xmpp/jid.h"
-#include "talk/xmllite/tqname.h"
+#include "talk/xmllite/qname.h"
 #include "talk/xmllite/xmlelement.h"
 
 
diff -u -r kdenetwork/kopete/protocols/jabber/kioslave/CMakeLists.txt src/kdenetwork/kopete/protocols/jabber/kioslave/CMakeLists.txt
--- kdenetwork/kopete/protocols/jabber/kioslave/CMakeLists.txt	2011-06-22 03:37:08.000000000 +0200
+++ src/kdenetwork/kopete/protocols/jabber/kioslave/CMakeLists.txt	2011-12-15 22:56:58.000000000 +0100
@@ -35,7 +35,7 @@
 tde_add_kpart( kio_jabberdisco AUTOMOC
   SOURCES jabberdisco.cpp
   LINK
-    jabberclient-static
+    jabberclient-static idn
     iris_xmpp_core-static iris_xmpp_im-static iris_jabber-static iris-static
     qca-static cutestuff_network-static cutestuff_util-static kio-shared
   DESTINATION ${PLUGIN_INSTALL_DIR}
diff -u -r kdenetwork/kopete/protocols/msn/CMakeLists.txt src/kdenetwork/kopete/protocols/msn/CMakeLists.txt
--- kdenetwork/kopete/protocols/msn/CMakeLists.txt	2011-06-22 03:37:22.000000000 +0200
+++ src/kdenetwork/kopete/protocols/msn/CMakeLists.txt	2011-12-15 22:56:58.000000000 +0100
@@ -47,7 +47,7 @@
 
 tde_add_kpart( kopete_msn AUTOMOC
   SOURCES
-    dummy.cpp webcam.cpp
+    webcam.cpp
   LINK kopete_msn_shared-shared
   DESTINATION ${PLUGIN_INSTALL_DIR}
 )
diff -u -r kdenetwork/kopete/protocols/msn/msnswitchboardsocket.cpp src/kdenetwork/kopete/protocols/msn/msnswitchboardsocket.cpp
--- kdenetwork/kopete/protocols/msn/msnswitchboardsocket.cpp	2011-08-12 22:11:37.000000000 +0200
+++ src/kdenetwork/kopete/protocols/msn/msnswitchboardsocket.cpp	2011-12-15 22:56:58.000000000 +0100
@@ -54,7 +54,7 @@
 #include "kopetemessage.h"
 #include "kopetecontact.h"
 #include "kopeteuiglobal.h"
-#include "kopeteemoticons.h"
+#include "private/kopeteemoticons.h"
 //#include "kopeteaccountmanager.h"
 //#include "kopeteprotocol.h"
 
diff -u -r kdenetwork/kopete/protocols/sms/services/smsclient.cpp src/kdenetwork/kopete/protocols/sms/services/smsclient.cpp
--- kdenetwork/kopete/protocols/sms/services/smsclient.cpp	2011-08-12 22:11:13.000000000 +0200
+++ src/kdenetwork/kopete/protocols/sms/services/smsclient.cpp	2011-12-15 22:56:58.000000000 +0100
@@ -67,7 +67,7 @@
 		return;
 	}
 
-	TQString programName = c->readEntry(TQString("%1:%2").tqarg("SMSClient").tqarg("ProgramName"). TQString());
+	TQString programName = c->readEntry(TQString("%1:%2").tqarg("SMSClient").tqarg("ProgramName"));
 	if (programName.isNull())
 		programName = "/usr/bin/sms_client";
 
