diff -rcb abakus.ORI/src/SConscript abakus/src/SConscript
*** abakus.ORI/src/SConscript	2011-09-14 21:44:42.191599072 +0200
--- abakus/src/SConscript	2011-09-14 21:44:51.352602708 +0200
***************
*** 66,72 ****
  myenv.KDEaddpaths_includes('#/src/  #/')
  
  ## Necessary libraries to link against
! myenv.KDEaddlibs( 'qt-mt kio kdecore kdeprint kdeui' )
  
  #############################
  ## Data to install
--- 66,72 ----
  myenv.KDEaddpaths_includes('#/src/  #/')
  
  ## Necessary libraries to link against
! myenv.KDEaddlibs( 'qt-mt kio kdecore kdeprint kdeui DCOP' )
  
  #############################
  ## Data to install
