--- trinity-koffice-i18n-14.0.0~pre61+1ce1f0d2/koffice-i18n-pt_BR/docs/koffice/kword/storeprint.docbook.ORI	2014-03-23 18:45:24.486623559 +0100
+++ trinity-koffice-i18n-14.0.0~pre61+1ce1f0d2/koffice-i18n-pt_BR/docs/koffice/kword/storeprint.docbook	2014-03-23 18:46:01.804769286 +0100
@@ -664,7 +664,8 @@
 ><para
 >O botão de <guibutton
 >Opções do Sistema</guibutton
-> permite a um usuário alterar a configuração das impressoras e do sistema de impressão do &kde;.</note>
+> permite a um usuário alterar a configuração das impressoras e do sistema de impressão do &kde;.</para
+></note>
 
 <note
 ><para
