--- trinity-libtdeldap-14.0.0~pre91+a257ac4f/src/Makefile.am.ORI	2013-07-04 19:29:13.674536933 +0200
+++ trinity-libtdeldap-14.0.0~pre91+a257ac4f/src/Makefile.am	2013-07-04 19:29:33.216775942 +0200
@@ -7,7 +7,7 @@
 include_HEADERS = libtdeldap.h ldappasswddlg.h
 
 libtdeldap_la_SOURCES = libtdeldap.cpp ldaplogindlgbase.ui ldaplogindlg.cpp ldappasswddlg.cpp
-libtdeldap_la_LIBADD = -ltdeio $(LIB_TDEUI) -lldap
+libtdeldap_la_LIBADD = -ltdeio $(LIB_TDEUI) -lldap $(LIB_QT) $(LIB_TDECORE) -ltdesu -llber
 libtdeldap_la_LDFLAGS = -version-info $(lt_current):$(lt_revision):$(lt_age) -no-undefined \
 	$(all_libraries)
 
