diff -u -r applications/kgtk-qt3/CMakeLists.txt src/applications/kgtk-qt3/CMakeLists.txt
--- applications/kgtk-qt3/CMakeLists.txt	2011-05-08 21:37:48.000000000 +0200
+++ src/applications/kgtk-qt3/CMakeLists.txt	2011-12-16 16:42:01.000000000 +0100
@@ -16,9 +16,9 @@
 include(CheckFunctionExists)
 check_function_exists(getpeereid HAVE_GETPEEREID)
 
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I/usr/include/tqt -include tqt.h -DTQT_THREAD_SUPPORT")
-SET(QT_UIC_EXECUTABLE "/usr/bin/uic-tqt")
-#SET(QT_MOC_EXECUTABLE "/usr/bin/tmoc")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I/opt/trinity/include/tqt -include tqt.h -DTQT_THREAD_SUPPORT")
+SET(QT_UIC_EXECUTABLE "/opt/trinity/bin/uic-tqt")
+#SET(QT_MOC_EXECUTABLE "/opt/trinity/bin/tmoc")
 
 if (NOT "${KGTK_KDE3}" MATCHES "false")
     add_subdirectory(kdialogd3)
Only in src/applications/kgtk-qt3: .CMakeLists.txt.swp
