--- kdegraphics/kfile-plugins/dependencies/poppler-tqt/CMakeLists.txt.ORI	2011-11-22 16:56:38.608230401 -0500
+++ kdegraphics/kfile-plugins/dependencies/poppler-tqt/CMakeLists.txt	2011-11-22 16:56:57.829811773 -0500
@@ -21,6 +21,10 @@
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
   ${CMAKE_SOURCE_DIR}/kfile-plugins/dependencies/poppler-tqt
+  ${CMAKE_SOURCE_DIR}/kpdf/xpdf/xpdf
+  ${CMAKE_SOURCE_DIR}/kpdf/xpdf
+  ${CMAKE_SOURCE_DIR}/kpdf/xpdf/goo
+  ${CMAKE_SOURCE_DIR}/kpdf/xpdf/splash
   /usr/include/poppler
 )
 
