--- kdegraphics/kfile-plugins/dependencies/poppler-tqt/ConfigureChecks.cmake.ORI	2012-05-06 19:23:47.668667462 +0200
+++ kdegraphics/kfile-plugins/dependencies/poppler-tqt/ConfigureChecks.cmake	2012-05-06 19:21:43.311268611 +0200
@@ -1,3 +1,13 @@
+# Poppler
+if( BUILD_KPDF )
+  pkg_search_module( POPPLER poppler )
+  if( POPPLER_FOUND )
+    set( HAVE_POPPLER 1 )
+  else( )
+    tde_message_fatal( "poppler is required, but was not found on your system" )
+  endif( )
+endif( )
+
 tde_save( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
 set( CMAKE_REQUIRED_INCLUDES ${DBUS_TQT_INCLUDE_DIRS} ${TQT_INCLUDE_DIRS} ${QT_INCLUDE_DIRS})
 set( CMAKE_REQUIRED_LIBRARIES ${DBUS_TQT_LDFLAGS} ${TQT_LDFLAGS} ${QT_LDFLAGS} "-lpoppler" )
--- kdegraphics/kfile-plugins/dependencies/poppler-tqt/CMakeLists.txt.ORI	2012-05-06 19:24:31.405752482 +0200
+++ kdegraphics/kfile-plugins/dependencies/poppler-tqt/CMakeLists.txt	2012-05-06 19:22:12.544657204 +0200
@@ -21,7 +21,7 @@
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
   ${CMAKE_SOURCE_DIR}/kfile-plugins/dependencies/poppler-tqt
-  /usr/include/poppler
+  ${POPPLER_INCLUDE_DIRS}
 )
 
 link_directories(
