--- kdegraphics-3.5.13.1/kgamma/ConfigureChecks.cmake.ORI	2012-10-10 21:25:24.204928664 +0200
+++ kdegraphics-3.5.13.1/kgamma/ConfigureChecks.cmake	2012-10-10 21:26:51.812154610 +0200
@@ -11,10 +11,10 @@
 
 include( CheckIncludeFileCXX )
 
 pkg_search_module( XXF86VM xxf86vm )
 if( NOT XXF86VM_FOUND )
-  tde_message_fatal( "xxf86vm is required, but was not found on your system" )
+  set( XXF86VM_LIBRARIES "-lXxf86vm" )
 endif( )
 
 tde_save( CMAKE_REQUIRED_INCLUDES )
 set( CMAKE_REQUIRED_INCLUDES ${TQT_INCLUDE_DIRS} )
--- kdegraphics-3.5.13.1/kpovmodeler/ConfigureChecks.cmake.ORI	2012-10-10 21:33:43.491841294 +0200
+++ kdegraphics-3.5.13.1/kpovmodeler/ConfigureChecks.cmake	2012-10-10 21:34:11.245282172 +0200
@@ -23,5 +23,5 @@
 # xmu
 pkg_search_module( XMU xmu )
 if( NOT XMU_FOUND )
-  tde_message_fatal( "xmu is required, but was not found on your system" )
+  set( XMU_LIBRARIES "-lXmu" )
 endif( )
--- kdegraphics-3.5.13.1/ksvg/plugin/backends/libart/GlyphTracerLibart.cpp.ORI	2012-10-11 22:17:26.285831149 +0200
+++ kdegraphics-3.5.13.1/ksvg/plugin/backends/libart/GlyphTracerLibart.cpp	2012-10-11 22:17:31.815712301 +0200
@@ -31,11 +31,7 @@
 
 #include <config.h>
 
-#ifdef HAVE_FREETYPE_2_2_x
-  #define FT_VECTOR_PARAMETER const FT_Vector
-#else
-  #define FT_VECTOR_PARAMETER FT_Vector
-#endif
+#define FT_VECTOR_PARAMETER FT_Vector
 
 using namespace T2P;
 
