Index: b/siputils.py
===================================================================
--- a/siputils.py
+++ b/siputils.py
@@ -1547,9 +1547,12 @@
         mfile.write("\n$(OFILES): $(HFILES)\n")
 
         for mf in self._build["tqmoc_headers"].split():
-            root, discard = os.path.splitext(mf)
+            root, _ = os.path.splitext(mf)
             cpp = "tqmoc_" + root + ".cpp"
 
+            if self._src_dir != self.dir:
+                mf = os.path.join(self._src_dir, mf)
+
             mfile.write("\n%s: %s\n" % (cpp, mf))
             mfile.write("\t$(MOC) -o %s %s\n" % (cpp, mf))
 
