--- k3b-i18n-1.0.5/admin/cvs.sh.ORI	2012-05-09 19:45:56.106421990 +0200
+++ k3b-i18n-1.0.5/admin/cvs.sh	2012-05-09 19:46:16.264708192 +0200
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.1[0-9]*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
