--- kdewebdev/kommander/plugin/Makefile.am.ORI	2012-07-30 20:43:40.366904851 +0200
+++ kdewebdev/kommander/plugin/Makefile.am	2012-07-30 20:43:52.205813593 +0200
@@ -7,7 +7,7 @@
 libkommanderplugin_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -no-undefined
 
 # the libraries to link against.
-libkommanderplugin_la_LIBADD  =  $(LIB_KDEUI) $(LIB_QT)
+libkommanderplugin_la_LIBADD  =  $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT)
 
 libkommanderplugin_la_SOURCES = kommanderplugin.cpp specialinformation.cpp
 
--- kdewebdev/kommander/widget/Makefile.am.ORI	2012-07-30 20:49:46.120865640 +0200
+++ kdewebdev/kommander/widget/Makefile.am	2012-07-30 20:50:25.875198532 +0200
@@ -7,7 +7,7 @@
 libkommanderwidget_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -no-undefined
 
 # the libraries to link against.
-libkommanderwidget_la_LIBADD  = $(top_builddir)/kommander/factory/libkommanderfactory.la $(LIB_KIO) $(LIB_KDEUI) $(LIB_QT)
+libkommanderwidget_la_LIBADD  = $(top_builddir)/kommander/factory/libkommanderfactory.la $(LIB_KIO) $(LIB_KDEUI) $(LIB_QT) -lDCOP
 
 libkommanderwidget_la_SOURCES = expression.cpp function.cpp functionlib.cpp \
 	invokeclass.cpp kmdrmainwindow.cpp kommanderfunctions.cpp kommanderwidget.cpp \
