--- k3b/applications/k3b/libk3bdevice/configure.in.in.ORI	2012-08-04 14:47:03.081813892 +0200
+++ k3b/applications/k3b/libk3bdevice/configure.in.in	2012-08-04 14:47:25.421612485 +0200
@@ -165,7 +165,7 @@
   dbusqt_lib=NOTFOUND
   dbusqt=NOTFOUND
 
-  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
+  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0 /opt/trinity/include/dbus-1.0"
   AC_FIND_FILE(dbus/connection.h, $search_incs, dbusqt_incdir)
 
   if test -r $dbusqt_incdir/dbus/connection.h ; then
--- k3b/libk3bdevice/configure.in.in.ORI	2012-08-04 14:57:19.624255359 +0200
+++ k3b/libk3bdevice/configure.in.in	2012-08-04 14:57:33.719128285 +0200
@@ -165,7 +165,7 @@
   dbusqt_lib=NOTFOUND
   dbusqt=NOTFOUND
 
-  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
+  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0 /opt/trinity/include/dbus-1.0"
   AC_FIND_FILE(dbus/connection.h, $search_incs, dbusqt_incdir)
 
   if test -r $dbusqt_incdir/dbus/connection.h ; then
