--- kdebase/kdesu/kdesu/kdesu.cpp.orig	2011-10-26 21:06:48.000000000 +0200
+++ kdebase/kdesu/kdesu/kdesu.cpp	2012-01-01 16:54:11.468303046 +0100
@@ -274,7 +274,7 @@
 
     // Try to exec the command with kdesud.
     bool keep = !args->isSet("n") && have_daemon;
-    bool terminal = true;
+    bool terminal = args->isSet("t");
     bool new_dcop = args->isSet("newdcop");
     bool withIgnoreButton = args->isSet("ignorebutton");
     
