--- tdebase/startkde.ORI	2012-07-29 18:23:59.885538470 +0200
+++ tdebase/startkde	2012-07-29 18:26:19.022618054 +0200
@@ -152,6 +152,16 @@
   fi
 fi
 
+# Default Start Button icon for TDE
+if [ ! -e  $kdehome/share/config/kickerrc ]; then
+  if [ -r "%{tde_starticon}" ]; then
+    cat <<EOF >$kdehome/share/config/kickerrc
+[KMenu]
+CustomIcon=%{tde_starticon}
+EOF
+  fi
+fi
+
 # XCursor mouse theme needs to be applied here to work even for kded or ksmserver.
 if test -n "$kcminputrc_mouse_cursortheme" -o -n "$kcminputrc_mouse_cursorsize" ; then
     kapplymousetheme "$kcminputrc_mouse_cursortheme" "$kcminputrc_mouse_cursorsize"
