--- kdebase-3.5.12/kdesktop/lock/lockprocess.cc.ORI	2011-12-21 18:40:05.553301746 +0100
+++ kdebase-3.5.12/kdesktop/lock/lockprocess.cc	2011-12-21 18:43:01.643590270 +0100
@@ -1045,17 +1045,13 @@
             mHackProc << word;
         }
 
-	if (!mForbidden)
+	if (!mForbidden && mHackProc.start() == true)
 	{
-
-		if (mHackProc.start() == true)
-		{
 #ifdef HAVE_SETPRIORITY
 			setpriority(PRIO_PROCESS, mHackProc.pid(), mPriority);
 #endif
  		        //bitBlt(this, 0, 0, &mOriginal);
 	                return true;
-		}
 	}
 	else
 	// we aren't allowed to start the specified screensaver either because it didn't run for some reason

