--- kdebase/ConfigureChecks.cmake.afterlibdetect.ORI	2012-09-06 20:46:34.705339175 +0200
+++ kdebase/ConfigureChecks.cmake	2012-09-06 20:47:12.331548001 +0200
@@ -229,19 +229,21 @@
     tde_message_fatal( "dbus-1 is required, but was not found on your system" )
   endif( )
 
-  # check for dbus-tqt
-  pkg_check_modules( DBUS_TQT REQUIRED dbus-tqt )
-  tde_save( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
-  set( CMAKE_REQUIRED_INCLUDES ${DBUS_TQT_INCLUDE_DIRS} ${TQT_INCLUDE_DIRS} ${QT_INCLUDE_DIRS})
-  set( CMAKE_REQUIRED_LIBRARIES ${DBUS_TQT_LDFLAGS} ${TQT_LDFLAGS} ${QT_LDFLAGS} )
-  check_cxx_source_compiles("
-    #include <tqt.h>
-    #include <dbus/connection.h>
-    int main(int, char**) { return 0; } "
-    HAVE_DBUS_QT3_07 )
-  tde_restore( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
-  if( NOT HAVE_DBUS_QT3_07 )
-    tde_message_fatal( "dbus-tqt is required, but was not found on your system" )
+  if( BUILD_TSAK )
+    # check for dbus-tqt
+    pkg_check_modules( DBUS_TQT REQUIRED dbus-tqt )
+    tde_save( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
+    set( CMAKE_REQUIRED_INCLUDES ${DBUS_TQT_INCLUDE_DIRS} ${TQT_INCLUDE_DIRS} ${QT_INCLUDE_DIRS})
+    set( CMAKE_REQUIRED_LIBRARIES ${DBUS_TQT_LDFLAGS} ${TQT_LDFLAGS} ${QT_LDFLAGS} )
+    check_cxx_source_compiles("
+      #include <tqt.h>
+      #include <dbus/connection.h>
+      int main(int, char**) { return 0; } "
+      HAVE_DBUS_QT3_07 )
+    tde_restore( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
+    if( NOT HAVE_DBUS_QT3_07 )
+      tde_message_fatal( "dbus-tqt is required, but was not found on your system" )
+    endif( )
   endif( )
 
 endif( )
