--- kdebase/kdm/backend/CMakeLists.txt.ORI	2012-09-06 22:17:30.134745111 +0200
+++ kdebase/kdm/backend/CMakeLists.txt	2012-09-06 22:17:34.329664333 +0200
@@ -37,7 +37,7 @@
 
 tde_add_executable( kdm
   SOURCES
-    access.c auth.c bootman.c choose.c client.c consolekit.c
+    access.c auth.c bootman.c choose.c client.c
     ctrl.c daemon.c dm.c dpylist.c error.c genauth.c
     inifile.c krb5auth.c mitauth.c netaddr.c policy.c
     process.c protodpy.c reset.c resource.c rpcauth.c
--- kdebase/kdm/backend/dm.h.ORI	2012-09-06 17:33:31.284704186 -0400
+++ kdebase/kdm/backend/dm.h	2012-09-06 17:33:40.415522219 -0400
@@ -37,8 +37,6 @@
 #ifndef _DM_H_
 #define _DM_H_ 1
 
-#define WITH_CONSOLE_KIT
-
 #include "greet.h"
 #include <config.ci>
 
