--- kdebase/kdm/backend/CMakeLists.txt.ORI	2012-09-06 22:11:08.422091268 +0200
+++ kdebase/kdm/backend/CMakeLists.txt	2012-09-06 22:11:21.280843938 +0200
@@ -14,11 +14,11 @@
 include_directories(
   ${CMAKE_CURRENT_BINARY_DIR}
   ${CMAKE_BINARY_DIR}
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
 )
 
 link_directories(
-  ${DBUS_TQT_LIBRARY_DIRS}
+  ${DBUS_LIBRARY_DIRS}
 )
 
 ##### kdm (executable) ##########################
@@ -43,6 +43,6 @@
     process.c protodpy.c reset.c resource.c rpcauth.c
     server.c session.c sessreg.c socket.c streams.c
     util.c xdmauth.c xdmcp.c
-  LINK X11 ${XAU_LIBRARIES} ${DBUS_TQT_LIBRARIES} ${CRYPT_LIBRARY} ${PAM_LIBRARY} ${XDMCP_LIBRARIES}
+  LINK X11 ${XAU_LIBRARIES} ${DBUS_LIBRARIES} ${CRYPT_LIBRARY} ${PAM_LIBRARY} ${XDMCP_LIBRARIES}
   DESTINATION ${BIN_INSTALL_DIR}
 )
--- kdebase/ksmserver/CMakeLists.txt.ORI	2012-09-06 17:22:59.990317001 -0400
+++ kdebase/ksmserver/CMakeLists.txt	2012-09-06 17:23:16.340989494 -0400
@@ -19,13 +19,13 @@
   ${CMAKE_SOURCE_DIR}/kdmlib
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
   ${HAL_INCLUDE_DIRS}
 )
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
-  ${DBUS_TQT_LIBRARY_DIRS}
+  ${DBUS_LIBRARY_DIRS}
 )
 
 
@@ -43,5 +43,5 @@
     main.cpp server.cpp shutdowndlg.cpp startupdlg.cpp
     legacy.cpp startup.cpp shutdown.cpp client.cpp
     KSMServerInterface.skel server.skel timed.ui
-  LINK dmctl-static kdeui-shared krsync-shared ${HAL_LIBRARIES} ${DBUS_TQT_LIBRARIES}
+  LINK dmctl-static kdeui-shared krsync-shared ${HAL_LIBRARIES} ${DBUS_LIBRARIES}
 )
--- kdebase/kioslave/media/mediamanager/CMakeLists.txt.ORI	2012-09-07 23:57:56.858227336 +0200
+++ kdebase/kioslave/media/mediamanager/CMakeLists.txt	2012-09-07 23:58:54.948097583 +0200
@@ -20,12 +20,12 @@
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
   ${HAL_INCLUDE_DIRS}
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
 )
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
-  ${DBUS_TQT_LIBRARY_DIRS}
+  ${DBUS_LIBRARY_DIRS}
 )
 
 
@@ -48,7 +48,7 @@
 
   tde_add_kpart( ${target} AUTOMOC
     SOURCES ${${target}_SRCS}
-    LINK mediacommon-static kdeinit_kded-shared ${HAL_LIBRARIES} -lhal-storage ${DBUS_TQT_LIBRARIES}
+    LINK mediacommon-static kdeinit_kded-shared ${HAL_LIBRARIES} -lhal-storage ${DBUS_LIBRARIES} -ldbus-qt-1
     DESTINATION ${PLUGIN_INSTALL_DIR}
   )
 else( )
--- kdebase-3.5.13.1/ConfigureChecks.cmake.ORI	2012-10-08 22:57:40.751052066 +0200
+++ kdebase-3.5.13.1/ConfigureChecks.cmake	2012-10-08 22:57:49.226885124 +0200
@@ -254,19 +254,4 @@
     tde_message_fatal( "dbus-1 is required, but was not found on your system" )
   endif( )
 
-  # check for dbus-tqt
-  pkg_check_modules( DBUS_TQT REQUIRED dbus-tqt )
-  tde_save( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
-  set( CMAKE_REQUIRED_INCLUDES ${DBUS_TQT_INCLUDE_DIRS} ${TQT_INCLUDE_DIRS} ${QT_INCLUDE_DIRS})
-  set( CMAKE_REQUIRED_LIBRARIES ${DBUS_TQT_LDFLAGS} ${TQT_LDFLAGS} ${QT_LDFLAGS} )
-  check_cxx_source_compiles("
-    #include <tqt.h>
-    #include <dbus/connection.h>
-    int main(int, char**) { return 0; } "
-    HAVE_DBUS_QT3_07 )
-  tde_restore( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
-  if( NOT HAVE_DBUS_QT3_07 )
-    tde_message_fatal( "dbus-tqt is required, but was not found on your system" )
-  endif( )
-
 endif( )
