commit bd9c1479d7a965959147398d71f18be62b2aa8b6
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1333232418 -0500

    Ensure pagersettings.kcfg is installed.

diff --git a/kicker/applets/minipager/CMakeLists.txt b/kicker/applets/minipager/CMakeLists.txt
index b9ca9c7..a90269c 100644
--- a/kicker/applets/minipager/CMakeLists.txt
+++ b/kicker/applets/minipager/CMakeLists.txt
@@ -27,7 +27,7 @@ link_directories(
 ##### other data ################################
 
 install( FILES minipagerapplet.desktop DESTINATION ${DATA_INSTALL_DIR}/kicker/applets )
-
+install( FILES pagersettings.kcfg DESTINATION ${KCFG_INSTALL_DIR} )
 
 ##### minipager_panelapplet (module) ############
 
