--- kdebase/startkde.ORI	2011-12-12 20:09:21.984162840 +0100
+++ kdebase/startkde	2011-12-12 20:08:27.922834925 +0100
@@ -4,6 +4,7 @@
 #
 
 echo "[startkde] Starting startkde." 1>&2
+export KDEDIR="/opt/trinity"
 
 # When the X server dies we get a HUP signal from xinit. We must ignore it
 # because we still need to do some cleanup.
@@ -74,39 +75,10 @@
   echo "[startkde] KDEHOME is preset to $KDEHOME." 1>&2
   export KDEHOME=$KDEHOME
 else
-  # $KDEHOME is NOT already preset in the environment. Try to help.
-  # This might be overkill but does provide flexibility.
+  # $KDEHOME is NOT already preset in the environment.
+  # We always use ~/.trinity as default.
   echo "[startkde] KDEHOME is not set." 1>&2
-  if [ -d $HOME/.trinity ]; then
-    # OK, this one is obvious.
-    export KDEHOME=$HOME/.trinity
-  elif [ -d $HOME/.trinity ]; then
-    # Looks like the user had an old version of Trinity installed at last login
-    echo "[startkde] Migrating old ~/.trinity directory to new ~/.trinity name." 1>&2
-    mv $HOME/.trinity $HOME/.trinity
-    export KDEHOME=$HOME/.trinity
-  elif [ -f /usr/bin/kde4-config ]; then
-    # Looks like KDE4 is installed.
-    if [ -d $HOME/.kde ] && [ ! -d $HOME/.trinity ]; then
-      # Presume $HOME/.kde is being used for KDE4 as it already exists.
-      export KDEHOME=$HOME/.trinity
-    else
-      # Presume $HOME/.kde is being used for KDE4 to be on the safe side.
-      export KDEHOME=$HOME/.trinity
-    fi
-  elif [ -f /opt/trinity/bin/kde-config ]; then
-    # Looks like KDE3 or Trinity is installed and playing second fiddle to KDE4.
-    export KDEHOME=$HOME/.trinity
-  elif [ -f /opt/trinity/bin/kde-config ]; then
-    # Looks like KDE3 or Trinity is installed and playing second fiddle to KDE4.
-    export KDEHOME=$HOME/.trinity
-  elif [ -f /usr/bin/kde-config ] && [ -d $HOME/.kde ]; then
-    # Looks like KDE3 or Trinity is installed and not playing second fiddle to KDE4.
-    export KDEHOME=$HOME/.kde
-  else
-    # Resort to this and hope for the best!
-    export KDEHOME=$HOME/.trinity
-  fi
+  export KDEHOME=$HOME/.trinity
   echo "[startkde] Set KDEHOME to $KDEHOME." 1>&2
 fi
 
@@ -117,51 +89,25 @@
 fi
 
 # Modify the following environment variables only as necessary.
-if [ -d /opt/trinity/games ]; then
-  export PATH=/opt/trinity/games:$PATH
-fi
-if [ -d /opt/trinity/bin ]; then
-  export PATH=/opt/trinity/bin:$PATH
-fi
-if [ -d /opt/trinity/games ]; then
-  export PATH=/opt/trinity/games:$PATH
-fi
-if [ -d /opt/trinity/bin ]; then
-  export PATH=/opt/trinity/bin:$PATH
-fi
-if [ -d /opt/trinity/share ]; then
-  export XDG_DATA_DIRS=$XDG_DATA_DIRS:/opt/trinity/share/:/usr/share/
-fi
-if [ -d /opt/trinity/share ]; then
-  export XDG_DATA_DIRS=$XDG_DATA_DIRS:/opt/trinity/share/:/usr/share/
+if [ -d ${KDEDIR}/games ]; then
+  export PATH=${KDEDIR}/games:$PATH
 fi
-if [ -d /opt/trinity/etc/xdg ]; then
-  export XDG_CONFIG_DIRS=$XDG_CONFIG_DIRS:/opt/trinity/etc/xdg/:/etc/xdg/
+if [ -d ${KDEDIR}/bin ]; then
+  export PATH=${KDEDIR}/bin:$PATH
 fi
-if [ -d /opt/trinity/etc/xdg ]; then
-  export XDG_CONFIG_DIRS=$XDG_CONFIG_DIRS:/opt/trinity/etc/xdg/:/etc/xdg/
+if [ -d ${KDEDIR}/share ]; then
+  export XDG_DATA_DIRS=$XDG_DATA_DIRS:${KDEDIR}/share/:/usr/share/
 fi
-if [ -d /opt/trinity/share/man ]; then
-  export MANPATH=/opt/trinity/share/man:$MANPATH
+if [ -d ${KDEDIR}/etc/xdg ]; then
+  export XDG_CONFIG_DIRS=$XDG_CONFIG_DIRS:${KDEDIR}/etc/xdg/:/etc/xdg/
 fi
-if [ -d /opt/trinity/share/man ]; then
-  export MANPATH=/opt/trinity/share/man:$MANPATH
-fi
-if [ -d /opt/trinity ]; then
-  if [ -n "$KDEDIRS" ]; then
-    export KDEDIRS=$KDEDIRS:/opt/trinity/:/usr/
-  else
-    export KDEDIRS=/opt/trinity/:/usr/
-  fi
-fi
-if [ -d /opt/trinity ]; then
-  if [ -n "$KDEDIRS" ]; then
-    export KDEDIRS=$KDEDIRS:/opt/trinity/:/usr/
-  else
-    export KDEDIRS=/opt/trinity/:/usr/
-  fi
+if [ -d ${KDEDIR}/share/man ]; then
+  export MANPATH=${KDEDIR}/share/man:$MANPATH
 fi
 
+# The TDE directory must be first in $KDEDIRS
+export KDEDIRS=${KDEDIR}/:$KDEDIRS:/usr/
+
 test -n "$KDEHOME" && kdehome=`echo "$KDEHOME" | sed "s,^~/,$HOME/,"`
 echo "[startkde] kdehome: $kdehome" 1>&2
 
