--- trinity-tdebase-3.5.13.2/config.h.cmake.ORI	2013-02-11 13:57:40.955386048 +0100
+++ trinity-tdebase-3.5.13.2/config.h.cmake	2013-02-11 14:02:42.484413874 +0100
@@ -159,6 +159,9 @@
 
 #cmakedefine XDMCP 1
 
+#define TDM_DEFMINUID "@TDM_DEFMINUID@"
+#define TDM_DEFMAXUID "@TDM_DEFMAXUID@"
+
 
 // ksmserver
 #cmakedefine DBUS_SYSTEM_BUS "@DBUS_SYSTEM_BUS@"
--- trinity-tdebase-3.5.13.2/kdm/kfrontend/genkdmconf.c.ORI	2013-02-11 14:03:11.290843286 +0100
+++ trinity-tdebase-3.5.13.2/kdm/kfrontend/genkdmconf.c	2013-02-11 14:05:21.309267838 +0100
@@ -2761,6 +2761,10 @@
 			}
 		}
 	}
+#ifdef TDM_MINDEFUID
+	defminuid = TDM_MINDEFUID;
+	defmaxuid = TDM_MAXDEFUID;
+#else // TDM_MINDEFUID
 #ifdef __linux__
 	if (!stat( "/etc/debian_version", &st )) { /* debian */
 		defminuid = "1000";
@@ -2778,10 +2782,11 @@
 		defminuid = "500";
 		defmaxuid = "65000";
 	}
-#else
+#else // __linux__
 	defminuid = "1000";
 	defmaxuid = "65000";
-#endif
+#endif // __linux__
+#endif // TDM_MINDEFUID
 	for (i = 0; i < CONF_MAX_PRIO; i++)
 		for (cs = config; cs; cs = cs->next)
 			for (ce = cs->ents; ce; ce = ce->next)
