--- kdebase/kdm/backend/CMakeLists.txt.ORI	2012-09-06 22:11:08.422091268 +0200
+++ kdebase/kdm/backend/CMakeLists.txt	2012-09-06 22:11:21.280843938 +0200
@@ -14,11 +14,11 @@
 include_directories(
   ${CMAKE_CURRENT_BINARY_DIR}
   ${CMAKE_BINARY_DIR}
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
 )
 
 link_directories(
-  ${DBUS_TQT_LIBRARY_DIRS}
+  ${DBUS_LIBRARY_DIRS}
 )
 
 ##### kdm (executable) ##########################
@@ -43,6 +43,6 @@
     process.c protodpy.c reset.c resource.c rpcauth.c
     server.c session.c sessreg.c socket.c streams.c
     util.c xdmauth.c xdmcp.c
-  LINK X11 ${XAU_LIBRARIES} ${DBUS_TQT_LIBRARIES} ${CRYPT_LIBRARY} ${PAM_LIBRARY} ${XDMCP_LIBRARIES}
+  LINK X11 ${XAU_LIBRARIES} ${DBUS_LIBRARIES} ${CRYPT_LIBRARY} ${PAM_LIBRARY} ${XDMCP_LIBRARIES}
   DESTINATION ${BIN_INSTALL_DIR}
 )
--- kdebase/ksmserver/CMakeLists.txt.ORI	2012-09-06 17:22:59.990317001 -0400
+++ kdebase/ksmserver/CMakeLists.txt	2012-09-06 17:23:16.340989494 -0400
@@ -19,13 +19,13 @@
   ${CMAKE_SOURCE_DIR}/kdmlib
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
   ${HAL_INCLUDE_DIRS}
 )
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
-  ${DBUS_TQT_LIBRARY_DIRS}
+  ${DBUS_LIBRARY_DIRS}
 )
 
 
@@ -43,5 +43,5 @@
     main.cpp server.cpp shutdowndlg.cpp startupdlg.cpp
     legacy.cpp startup.cpp shutdown.cpp client.cpp
     KSMServerInterface.skel server.skel timed.ui
-  LINK dmctl-static kdeui-shared krsync-shared ${HAL_LIBRARIES} ${DBUS_TQT_LIBRARIES}
+  LINK dmctl-static kdeui-shared krsync-shared ${HAL_LIBRARIES} ${DBUS_LIBRARIES}
 )
--- kdebase/kioslave/media/mediamanager/CMakeLists.txt.ORI	2012-09-07 23:57:56.858227336 +0200
+++ kdebase/kioslave/media/mediamanager/CMakeLists.txt	2012-09-07 23:58:54.948097583 +0200
@@ -20,12 +20,12 @@
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
   ${HAL_INCLUDE_DIRS}
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
 )
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
-  ${DBUS_TQT_LIBRARY_DIRS}
+  ${DBUS_LIBRARY_DIRS}
 )
 
 
@@ -48,7 +48,7 @@
 
   tde_add_kpart( ${target} AUTOMOC
     SOURCES ${${target}_SRCS}
-    LINK mediacommon-static kdeinit_kded-shared ${HAL_LIBRARIES} -lhal-storage ${DBUS_TQT_LIBRARIES}
+    LINK mediacommon-static kdeinit_kded-shared ${HAL_LIBRARIES} -lhal-storage ${DBUS_LIBRARIES} -ldbus-qt-1
     DESTINATION ${PLUGIN_INSTALL_DIR}
   )
 else( )
--- trinity-tdebase-3.5.13.2/ConfigureChecks.cmake.ORI	2013-01-23 23:11:55.903373346 +0100
+++ trinity-tdebase-3.5.13.2/ConfigureChecks.cmake	2013-01-23 23:12:23.168860363 +0100
@@ -256,21 +256,3 @@
 endif( )
 
 
-# dbus-tqt (kdm, ksmserver, kioslaves(media))
-if( BUILD_KDM OR BUILD_KSMSERVER OR (BUILD_KIOSLAVES AND WITH_HAL) )
-  # check for dbus-tqt
-  pkg_check_modules( DBUS_TQT REQUIRED dbus-tqt )
-  tde_save( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
-  set( CMAKE_REQUIRED_INCLUDES ${DBUS_TQT_INCLUDE_DIRS} ${TQT_INCLUDE_DIRS} ${QT_INCLUDE_DIRS})
-  set( CMAKE_REQUIRED_LIBRARIES ${DBUS_TQT_LDFLAGS} ${TQT_LDFLAGS} ${QT_LDFLAGS} )
-  check_cxx_source_compiles("
-    #include <tqt.h>
-    #include <dbus/connection.h>
-    int main(int, char**) { return 0; } "
-    HAVE_DBUS_QT3_07 )
-  tde_restore( CMAKE_REQUIRED_INCLUDES CMAKE_REQUIRED_LIBRARIES )
-  if( NOT HAVE_DBUS_QT3_07 )
-    tde_message_fatal( "dbus-tqt is required, but was not found on your system" )
-  endif( )
-
-endif( )
--- trinity-tdebase-3.5.13.2/kicker/kicker/ui/CMakeLists.txt.ORI	2013-06-16 22:01:14.879046232 +0200
+++ trinity-tdebase-3.5.13.2/kicker/kicker/ui/CMakeLists.txt	2013-06-16 22:01:32.316688895 +0200
@@ -20,7 +20,7 @@
   ${CMAKE_SOURCE_DIR}/kicker/kicker/buttons
   ${CMAKE_SOURCE_DIR}/kdmlib
   ${CMAKE_SOURCE_DIR}/libkonq
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
 )
@@ -56,5 +56,5 @@
 tde_add_library( ${target} STATIC_PIC AUTOMOC
   SOURCES ${${target}_SRCS}
   DEPENDENCIES kicker_core-static kickermain-static
-  LINK dmctl-static kickoffsearch_interfaces-shared ${DBUS_TQT_LIBRARIES} ${HAL_LIBRARIES}
+  LINK dmctl-static kickoffsearch_interfaces-shared ${DBUS_LIBRARIES} ${HAL_LIBRARIES}
 )
--- trinity-tdebase-3.5.13.2/kicker/kicker/core/CMakeLists.txt.ORI	2013-06-16 22:05:07.891261997 +0200
+++ trinity-tdebase-3.5.13.2/kicker/kicker/core/CMakeLists.txt	2013-06-16 22:05:17.673060725 +0200
@@ -17,7 +17,7 @@
   ${CMAKE_SOURCE_DIR}/kicker/libkicker
   ${CMAKE_SOURCE_DIR}/kicker/kicker/ui
   ${CMAKE_SOURCE_DIR}/kicker/kicker/buttons
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
 )
--- trinity-tdebase-3.5.13.2/kicker/kicker/buttons/CMakeLists.txt.ORI	2013-06-16 22:09:44.143564983 +0200
+++ trinity-tdebase-3.5.13.2/kicker/kicker/buttons/CMakeLists.txt	2013-06-16 22:09:48.450475959 +0200
@@ -17,7 +17,7 @@
   ${CMAKE_SOURCE_DIR}/kicker/kicker/core
   ${CMAKE_SOURCE_DIR}/kicker/kicker/ui
   ${CMAKE_SOURCE_DIR}/libkonq
-  ${DBUS_TQT_INCLUDE_DIRS}
+  ${DBUS_INCLUDE_DIRS}
   ${TDE_INCLUDE_DIR}
   ${TQT_INCLUDE_DIRS}
 )
