commit 03e19305ec704b8749bb7564e16ed8d78a80d516
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1340675585 -0500

    Update default konqueror maximum image preview size to 10MB.

diff --git a/kcontrol/konq/previews.cpp b/kcontrol/konq/previews.cpp
index cbe4d40..53c16f1 100644
--- a/kcontrol/konq/previews.cpp
+++ b/kcontrol/konq/previews.cpp
@@ -67,14 +67,14 @@ KPreviewOptions::KPreviewOptions( TQWidget *parent, const char */*name*/ )
 
     setQuickHelp( i18n("<h1>Preview Options</h1> Here you can modify the behavior "
                 "of Konqueror when it shows the files in a folder."
-                "<h2>The list of protocols:</h2> check the protocols over which "
+                "<h2>The list of protocols:</h2> Check the protocols over which "
                 "previews should be shown; uncheck those over which they should not. "
                 "For instance, you might want to show previews over SMB if the local "
                 "network is fast enough, but you might disable it for FTP if you often "
                 "visit very slow FTP sites with large images."
-                "<h2>Maximum File Size:</h2> select the maximum file size for which "
-                "previews should be generated. For instance, if set to 1 MB (the default), "
-                "no preview will be generated for files bigger than 1 MB, for speed reasons."));
+                "<h2>Maximum File Size:</h2> Select the maximum file size for which "
+                "previews should be generated. For instance, if set to 10 MB (the default), "
+                "no preview will be generated for files bigger than 10 MB, for speed reasons."));
 
     // Listview containing checkboxes for all protocols that support listing
     KListView *listView = new KListView( this, "listView" );
@@ -152,8 +152,8 @@ KPreviewOptions::KPreviewOptions( TQWidget *parent, const char */*name*/ )
     load();
 }
 
-// Default: 1 MB
-#define DEFAULT_MAXSIZE (1024*1024)
+// Default: 10 MB
+#define DEFAULT_MAXSIZE (1024*1024*10)
 
 void KPreviewOptions::load(bool useDefaults)
 {
@@ -171,7 +171,7 @@ void KPreviewOptions::load(bool useDefaults)
         else
           it.current()->setOn( group.readBoolEntry( protocol, false ) );
     }
-    // config key is in bytes (default value 1MB), numinput is in MB
+    // config key is in bytes (default value 10MB), numinput is in MB
     m_maxSize->setValue( ((double)group.readNumEntry( "MaximumSize", DEFAULT_MAXSIZE )) / (1024*1024) );
 
     m_boostSize->setChecked( group.readBoolEntry( "BoostSize", false /*default*/ ) );
