diff -u -r kdebase.orig/config.h.cmake kdebase/config.h.cmake
--- kdebase.orig/config.h.cmake	2011-10-30 00:02:15.000000000 +0300
+++ kdebase/config.h.cmake	2011-11-19 00:59:26.088061733 +0200
@@ -114,6 +114,7 @@
 #cmakedefine HAVE_SYSINFO 1
 #cmakedefine HAVE_STRNLEN 1
 #cmakedefine HAVE_GETIFADDRS 1
+#cmakedefine HAVE_CRYPT 1
 
 #cmakedefine HAVE_SETUSERCONTEXT 1
 #cmakedefine HAVE_GETUSERSHELL 1
diff -u -r kdebase.orig/kdm/backend/CMakeLists.txt kdebase/kdm/backend/CMakeLists.txt
--- kdebase.orig/kdm/backend/CMakeLists.txt	2011-09-03 22:32:00.000000000 +0300
+++ kdebase/kdm/backend/CMakeLists.txt	2011-11-19 01:05:03.820839491 +0200
@@ -43,6 +43,6 @@
     process.c protodpy.c reset.c resource.c rpcauth.c
     server.c session.c sessreg.c socket.c streams.c
     util.c xdmauth.c xdmcp.c
-  LINK X11 ${XAU_LIBRARIES} ${DBUS_TQT_LIBRARIES} ${PAM_LIBRARY} ${XDMCP_LIBRARIES}
+  LINK X11 ${XAU_LIBRARIES} ${DBUS_TQT_LIBRARIES} ${CRYPT_LIBRARY} ${PAM_LIBRARY} ${XDMCP_LIBRARIES}
   DESTINATION ${BIN_INSTALL_DIR}
 )
diff -u -r kdebase.orig/kdm/ConfigureChecks.cmake kdebase/kdm/ConfigureChecks.cmake
--- kdebase.orig/kdm/ConfigureChecks.cmake	2011-03-16 22:49:38.000000000 +0200
+++ kdebase/kdm/ConfigureChecks.cmake	2011-11-19 01:05:40.426381860 +0200
@@ -27,6 +27,16 @@
 check_function_exists( auth_timeok HAVE_AUTH_TIMEOK )
 tde_restore( CMAKE_REQUIRED_LIBRARIES )
 
+check_function_exists( crypt LIBC_HAVE_CRYPT )
+if( LIBC_HAVE_CRYPT )
+  set( HAVE_CRYPT 1 CACHE INTERNAL "" FORCE )
+else( )
+  check_library_exists( crypt crypt "" HAVE_CRYPT )
+  if( HAVE_CRYPT )
+    set( CRYPT_LIBRARY crypt )
+  endif( )
+endif( )
+
 check_include_file( lastlog.h HAVE_LASTLOG_H )
 check_include_file( termio.h HAVE_TERMIO_H )
 

