--- kdebase/config.h.cmake.ORI	2012-08-20 11:19:03.411461759 +0200
+++ kdebase/config.h.cmake	2012-08-20 11:19:40.587461736 +0200
@@ -174,3 +174,6 @@
 
 // kdm, kcontrol
 #cmakedefine WITH_XRANDR "@WITH_XRANDR@"
+
+// tsak
+#cmakedefine BUILD_TSAK "@BUILD_TSAK@"
--- kdebase/kdesktop/lock/main.cc.tsak	2012-08-20 13:55:47.265335687 +0200
+++ kdebase/kdesktop/lock/main.cc	2012-08-20 13:56:31.594856008 +0200
@@ -207,8 +207,11 @@
         struct stat st;
         KSimpleConfig* tdmconfig;
         OPEN_TDMCONFIG_AND_SET_GROUP
+#ifdef BUILD_TSAK
         trinity_desktop_lock_use_sak = tdmconfig->readBoolEntry("UseSAK", true);
-
+#else
+        trinity_desktop_lock_use_sak = false;
+#endif
         LockProcess process;
 
         // Start loading core functions, such as the desktop wallpaper interface
@@ -272,7 +275,11 @@
         trinity_desktop_lock_use_system_modal_dialogs = !KDesktopSettings::useUnmanagedLockWindows();
         trinity_desktop_lock_delay_screensaver_start = KDesktopSettings::delaySaverStart();
         if (trinity_desktop_lock_use_system_modal_dialogs) {
+#ifdef BUILD_TSAK
             trinity_desktop_lock_use_sak = tdmconfig->readBoolEntry("UseSAK", true);
+#else
+            trinity_desktop_lock_use_sak = false;
+#endif
         }
         else {
             trinity_desktop_lock_use_sak = false;			// If SAK is enabled with unmanaged windows, the SAK dialog will never close and will "burn in" the screen
--- kdebase/kdm/kfrontend/kgapp.cpp.tsak	2012-08-20 14:00:44.387336567 +0200
+++ kdebase/kdm/kfrontend/kgapp.cpp	2012-08-20 14:01:26.280461134 +0200
@@ -195,7 +195,11 @@
 	KProcess *dcop = 0;
 	KProcess *kwin = 0;
 
+#ifdef BUILD_TSAK
 	trinity_desktop_lock_use_sak = _useSAK;
+#else
+	trinity_desktop_lock_use_sak = false;
+#endif
 	if (trinity_desktop_lock_use_sak) {
 		tsak = new KProcess;
 		*tsak << TQCString( argv0, strrchr( argv0, '/' ) - argv0 + 2 ) + "tsak";
