--- tdebase/starttde.ORI	2014-04-20 18:05:28.820084805 +0200
+++ tdebase/starttde	2014-04-20 18:06:08.059237113 +0200
@@ -81,7 +81,7 @@
 
 # When the X server dies we get a HUP signal from xinit. We must ignore it
 # because we still need to do some cleanup.
-trap '[starttde] echo GOT SIGHUP' HUP
+trap 'echo "[starttde] GOT SIGHUP"' HUP
 
 # Check if a TDE session is already running.
 if kcheckrunning >/dev/null 2>&1; then
