--- trinity-tdebase/twin/data/update_default_rules.cpp.ORI	2013-07-27 11:49:19.337804333 +0200
+++ trinity-tdebase/twin/data/update_default_rules.cpp	2013-07-27 19:12:32.144222960 +0200
@@ -15,12 +15,18 @@
 #include <kdebug.h>
 #include <kinstance.h>
 #include <kstandarddirs.h>
+#include <tdeaboutdata.h>
+#include <tdecmdlineargs.h>
+#include <tdeglobal.h>
 
 int main( int argc, char* argv[] )
     {
     if( argc != 2 )
         return 1;
-    TDEInstance inst( "twin_update_default_rules" );
+    TDEAboutData about( "twin_update_default_rules", I18N_NOOP("TWin"), 0, 0, TDEAboutData::License_GPL, I18N_NOOP( "(c) 2004 KWin and KControl Authors" ) );
+    TDECmdLineArgs::init( argc, argv, &about );
+    TDEInstance inst( &about );
+    Q_UNUSED( TDEGlobal::locale() ); // jump-start locales to get to translated desriptions
     TQString file = locate( "data", TQString( "twin/default_rules/" ) + argv[ 1 ] );
     if( file.isEmpty())
         {
@@ -43,8 +49,10 @@
         dest_cfg.setGroup( TQString::number( pos ));
         for( TQMap< TQString, TQString >::ConstIterator it = entries.begin();
              it != entries.end();
-             ++it )
+             ++it ) {
+				 printf("DEBUG: %s\n", it.key().latin1());
             dest_cfg.writeEntry( it.key(), *it );
+			}
         }
     dest_cfg.setGroup( "General" );
     dest_cfg.writeEntry( "count", pos );
