--- trinity-tdebase-3.5.13.2/kioslave/media/mediamanager/halbackend.cpp.slash	2013-08-15 18:24:06.092882626 +0200
+++ trinity-tdebase-3.5.13.2/kioslave/media/mediamanager/halbackend.cpp	2013-08-15 18:32:20.535855050 +0200
@@ -554,6 +554,10 @@
         media_name.replace(pattern, i18n(pattern.utf8()));
         medium->setLabel(media_name);
     } else {
+        // If media_name begins with '%', it is likely to be a directory path.
+        if( media_name.at(0) == '%') {
+            media_name.replace(QChar('%'), "/");
+        }
         medium->setLabel(i18n(media_name.utf8()));
     }
 
