--- ./kcontrol/displayconfig/displayconfig.desktop.displayconfigtranslation	2014-05-30 12:16:09.448919875 +0200
+++ ./kcontrol/displayconfig/displayconfig.desktop	2014-05-30 12:19:18.104578759 +0200
@@ -13,6 +13,7 @@
 Categories=Qt;TDE;X-TDE-settings-system;
 Comment=Configure display
 Comment[en_US]=Configure display
+Comment[fr]=Configuration de l'affichage
 Comment[ru]=Конфигурация экрана
 Comment[uk]=Конфігурація екрану
 GenericName=
@@ -21,6 +22,7 @@
 MimeType=
 Name=Monitor & Display
 Name[en_US]=Monitor & Display
+Name[fr]=Moniteurs & Affichage
 Name[ru]=Монитор & Экран
 Name[uk]=Монітор & Екран
 NoDisplay=false
--- ./kcontrol/iccconfig/iccconfig.desktop.displayconfigtranslation	2014-05-30 12:16:09.448919875 +0200
+++ ./kcontrol/iccconfig/iccconfig.desktop	2014-05-30 12:20:36.526770672 +0200
@@ -12,6 +12,7 @@
 Categories=Qt;TDE;X-TDE-settings-hardware;
 Comment=Configure display ICC profile
 Comment[en_US]=Configure display ICC profile
+Comment[fr]=Configurer les profils de couleurs ICC
 Comment[ru]=Конфигурация профиля ICC экрана
 Comment[uk]=Конфігурація профілю ICC екрану
 GenericName=
@@ -20,6 +21,7 @@
 MimeType=
 Name=ICC Color Profile
 Name[en_US]=ICC Color Profile
+Name[fr]=Profils de couleurs ICC
 Name[ru]=ICC Профиль цвета
 Name[uk]=ICC Профіль кольору
 
