--- ./libkonq/konqbookmarkmanager.h.ori	2013-05-08 23:26:03.558587419 +0200
+++ ./libkonq/konqbookmarkmanager.h	2013-05-08 23:29:25.317517880 +0200
@@ -3,6 +3,8 @@
 
 #include <kbookmarkmanager.h>
 #include <kstandarddirs.h>
+#include <kurl.h>
+#include <tdeio/job.h>
 #include <libkonq_export.h>
 
 class LIBKONQ_EXPORT KonqBookmarkManager
@@ -11,7 +13,14 @@
     static KBookmarkManager * self() {
         if ( !s_bookmarkManager )
         {
-            TQString bookmarksFile = locateLocal("data", TQString::fromLatin1("konqueror/bookmarks.xml"));
+            TQString globalBookmarkFile = locate( "data",  TQString::fromLatin1( "konqueror/bookmarks.xml" ) );
+            TQString bookmarksFile = locateLocal( "data", TQString::fromLatin1("konqueror/bookmarks.xml" ), true);
+            if ( globalBookmarkFile != TQString::null && bookmarksFile == TQString::null ) {
+                TDEIO::file_copy( 
+                   KURL::fromPathOrURL( globalBookmarkFile ), 
+                   KURL::fromPathOrURL( bookmarksFile ), -1, false, false
+                );
+            }
             s_bookmarkManager = KBookmarkManager::managerForFile( bookmarksFile );
         }
         return s_bookmarkManager;
