--- trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/background/CMakeLists.txt.ORI	2013-07-17 15:07:53.620475336 +0200
+++ trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/background/CMakeLists.txt	2013-07-17 15:08:37.773235444 +0200
@@ -23,6 +23,7 @@
 
   link_directories(
     ${TQT_LIBRARY_DIRS}
+    ${LIBART_LIBRARY_DIRS}
   )
 
   ##### bgnd (static) #############################
--- trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/kicker/CMakeLists.txt.ftbfs	2013-07-17 15:10:49.001523716 +0200
+++ trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/kicker/CMakeLists.txt	2013-07-17 15:11:58.278148735 +0200
@@ -22,6 +22,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${LIBART_LIBRARY_DIRS}
 )
 
 ##### other data ################################
--- trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/tdm/CMakeLists.txt.ftbfs	2013-07-17 15:14:35.389300184 +0200
+++ trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/tdm/CMakeLists.txt	2013-07-17 15:14:41.479267342 +0200
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${LIBART_LIBRARY_DIRS}
 )
 
 ##### other data ################################
--- trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/icons/CMakeLists.txt.ftbfs	2013-07-17 15:16:26.158703430 +0200
+++ trinity-tdebase-14.0.0~pre738+d41051b3/kcontrol/icons/CMakeLists.txt	2013-07-17 15:16:31.722673486 +0200
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${LIBART_LIBRARY_DIRS}
 )
 
 ##### other data ################################
--- trinity-tdebase-14.0.0~pre738+d41051b3/tdm/kfrontend/CMakeLists.txt.ftbfs	2013-07-17 15:22:20.506802399 +0200
+++ trinity-tdebase-14.0.0~pre738+d41051b3/tdm/kfrontend/CMakeLists.txt	2013-07-17 15:22:27.893762894 +0200
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${LIBART_LIBRARY_DIRS}
 )
 
 
--- trinity-tdebase-14.0.0~pre738+d41051b3/kdesktop/CMakeLists.txt.ftbfs	2013-07-17 15:26:02.711616155 +0200
+++ trinity-tdebase-14.0.0~pre738+d41051b3/kdesktop/CMakeLists.txt	2013-07-17 15:26:08.085587520 +0200
@@ -36,6 +36,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${LIBART_LIBRARY_DIRS}
 )
 
 
