--- ./kicker/applets/clock/clock.cpp.orig	2014-09-13 20:24:15.429011133 +0200
+++ ./kicker/applets/clock/clock.cpp	2014-09-13 20:25:02.281084675 +0200
@@ -1515,11 +1515,8 @@
             preferences();
             break;
         case 103:
-            proc << locate("exe", "tdesu");
-            proc << "--nonewdcop";
-            proc << TQString("%1 tde-clock.desktop --lang %2")
-                .arg(locate("exe", "tdecmshell"))
-                .arg(TDEGlobal::locale()->language());
+            // Mageia/Mandriva only
+            proc << locate("exe", "drakclock");
             proc.start(TDEProcess::DontCare);
             break;
         case 104:
