commit bfb0bc00e959ed3cac6cf4e517a24de50446be50
Author: Serghei Amelian <serghei.amelian@gmail.com>
Date:   1329176749 +0200

    [tdesdk] cmake port

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 806890a..c6f7d3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,7 @@ include( CheckIncludeFile )
 include( CheckTypeSize )
 include( CheckCSourceCompiles )
 include( CheckCXXSourceCompiles )
+include( CheckFunctionExists )
 include( CheckLibraryExists )
 
 
@@ -55,6 +56,21 @@ option( BUILD_DOC "Build documentation" ${BUILD_ALL} )
 option( BUILD_KAPPTEMPLATE "Build kapptemplate" ${BUILD_ALL} )
 option( BUILD_KBABEL "Build kbabel" ${BUILD_ALL} )
 option( BUILD_KBUGBUSTER "Build kbugbuster" ${BUILD_ALL} )
+option( BUILD_KDEACCOUNTS_PLUGIN "Build kdeaccounts-plugin" ${BUILD_ALL} )
+option( BUILD_KFILE_PLUGINS "Build kfile-plugins" ${BUILD_ALL} )
+option( BUILD_KIOSLAVE "Build kioslave" ${BUILD_ALL} )
+option( BUILD_KMTRACE "Build kmtrace" ${BUILD_ALL} )
+option( BUILD_KOMPARE "Build kompare" ${BUILD_ALL} )
+option( BUILD_KPROFILEMETHOD "Build kprofilemethod" ${BUILD_ALL} )
+option( BUILD_KSPY "Build kspy" ${BUILD_ALL} )
+option( BUILD_KSTARTPERF "Build kstartperf" ${BUILD_ALL} )
+option( BUILD_KUIVIEWER "Build kuiviewer" ${BUILD_ALL} )
+option( BUILD_KUNITTEST "Build kunittest" ${BUILD_ALL} )
+option( BUILD_POXML "Build poxml" ${BUILD_ALL} )
+option( BUILD_SCHECK "Build scheck" ${BUILD_ALL} )
+option( BUILD_SCRIPTS "Build scripts" ${BUILD_ALL} )
+option( BUILD_TDECACHEGRIND "Build tdecachegrind" ${BUILD_ALL} )
+option( BUILD_UMBRELLO "Build umbrello" ${BUILD_ALL} )
 
 
 ##### configure checks ##########################
@@ -83,6 +96,21 @@ tde_conditional_add_subdirectory( BUILD_DOC doc )
 tde_conditional_add_subdirectory( BUILD_KAPPTEMPLATE kapptemplate )
 tde_conditional_add_subdirectory( BUILD_KBABEL kbabel )
 tde_conditional_add_subdirectory( BUILD_KBUGBUSTER kbugbuster )
+tde_conditional_add_subdirectory( BUILD_KDEACCOUNTS_PLUGIN kdeaccounts-plugin )
+tde_conditional_add_subdirectory( BUILD_KFILE_PLUGINS kfile-plugins )
+tde_conditional_add_subdirectory( BUILD_KIOSLAVE kioslave )
+tde_conditional_add_subdirectory( BUILD_KMTRACE kmtrace )
+tde_conditional_add_subdirectory( BUILD_KOMPARE kompare )
+tde_conditional_add_subdirectory( BUILD_KPROFILEMETHOD kprofilemethod )
+tde_conditional_add_subdirectory( BUILD_KSPY kspy )
+tde_conditional_add_subdirectory( BUILD_KSTARTPERF kstartperf )
+tde_conditional_add_subdirectory( BUILD_KUIVIEWER kuiviewer )
+tde_conditional_add_subdirectory( BUILD_KUNITTEST kunittest )
+tde_conditional_add_subdirectory( BUILD_POXML poxml )
+tde_conditional_add_subdirectory( BUILD_SCHECK scheck )
+tde_conditional_add_subdirectory( BUILD_SCRIPTS scripts )
+tde_conditional_add_subdirectory( BUILD_TDECACHEGRIND kcachegrind )
+tde_conditional_add_subdirectory( BUILD_UMBRELLO umbrello )
 
 
 ##### write configure files #####################
diff --git a/config.h.cmake b/config.h.cmake
index 139f3c7..7ab8b07 100644
--- a/config.h.cmake
+++ b/config.h.cmake
@@ -1 +1,4 @@
 #cmakedefine VERSION "@version@"
+
+// kcachegrind
+#cmakedefine HAVE_MMAP 1
diff --git a/kbabel/kbabel/CMakeLists.txt b/kbabel/kbabel/CMakeLists.txt
index b5164b4..6b263b5 100644
--- a/kbabel/kbabel/CMakeLists.txt
+++ b/kbabel/kbabel/CMakeLists.txt
@@ -62,4 +62,5 @@ tde_add_library( kbabel STATIC AUTOMOC
     fontpreferences.ui editordiffpreferences.ui editorpreferences.ui
     searchpreferences.ui colorpreferences.ui kbcataloglistview.cpp
     kbcataloglistviewitem.cpp errorlistview.cpp
+  DEPENDENCIES kbabelcommon-shared
 )
diff --git a/kmtrace/ktrace.c b/kmtrace/ktrace.c
index 21fca37..8b9eb2f 100644
--- a/kmtrace/ktrace.c
+++ b/kmtrace/ktrace.c
@@ -778,12 +778,12 @@ kuntrace()
 			"Mallocs:        %8ld   Frees:        %8ld   Leaks:        %8ld\n"
 			"Logged Mallocs: %8ld   Logged Frees: %8ld   Logged Leaks: %8ld\n"
 			"Avg. Free lookups: %ld  Malloc collisions: %ld  Max offset: %ld\n",
-			tr_max_mallocs, TR_CACHE_SIZE, tr_flashes,
+			tr_max_mallocs, (long int)TR_CACHE_SIZE, tr_flashes,
 			tr_mallocs, tr_frees, tr_current_mallocs,
 			tr_logged_mallocs, tr_logged_frees,
 			tr_logged_mallocs - tr_logged_frees,
 			tr_frees > 0 ? ( tr_failed_free_lookups / tr_frees ) : 0,
-			tr_malloc_collisions, tr_max_offset);
+			tr_malloc_collisions, (long int)tr_max_offset);
 #endif
 	fclose (mallstream);
 	mallstream = NULL;
