Index: b/src/tdedistutils.py
===================================================================
--- a/src/tdedistutils.py
+++ b/src/tdedistutils.py
@@ -2272,7 +2272,7 @@
 
 ###########################################################################
 def get_libdir_name():
-    if os.uname()[4] in ['x86_64','mips64','ppc64','sparc64','s390x']:
-        return 'lib64'
-    else:
+    #if os.uname()[4] in ['x86_64','mips64','ppc64','sparc64','s390x']:
+    #    return 'lib64'
+    #else:
         return 'lib'
