--- tqt3/src/sql/drivers/odbc/qsql_odbc.cpp.odbc	2012-02-13 19:10:01.000000000 +0100
+++ tqt3/src/sql/drivers/odbc/qsql_odbc.cpp	2012-02-13 21:47:25.232367854 +0100
@@ -61,13 +61,13 @@
 #endif
 
 // newer platform SDKs use SQLLEN instead of SQLINTEGER
-#ifdef SQLLEN
+#if defined(SQLLEN) || defined(Q_OS_WIN64) || defined(Q_OS_UNIX)
 # define TQSQLLEN SQLLEN
 #else
 # define TQSQLLEN SQLINTEGER
 #endif
 
-#ifdef SQLULEN
+#if defined(SQLULEN) || defined(Q_OS_WIN64) || defined(Q_OS_UNIX)
 # define TQSQLULEN SQLULEN
 #else
 # define TQSQLULEN SQLUINTEGER
