--- tqt3/src/kernel/qpixmap_x11.cpp.quiet	2012-02-13 19:10:01.000000000 +0100
+++ tqt3/src/kernel/qpixmap_x11.cpp	2012-02-13 21:24:08.190653338 +0100
@@ -394,8 +394,8 @@
 {
 #if defined(QT_CHECK_STATE)
     if ( tqApp->type() == TQApplication::Tty ) {
-	tqWarning( "TQPixmap: Cannot create a TQPixmap when no GUI "
-		  "is being used" );
+//	tqWarning( "TQPixmap: Cannot create a TQPixmap when no GUI "
+//		  "is being used" );
     }
 #endif
 
