--- ./tools/assistant/lib/lib.pro.ori	2013-06-25 20:20:21.576374929 +0200
+++ ./tools/assistant/lib/lib.pro	2013-06-25 20:20:36.008078008 +0200
@@ -3,8 +3,7 @@
 VERSION		= 1.0
 
 CONFIG		+= qt warn_on release
-CONFIG		+= staticlib
-CONFIG		-= dll
+CONFIG		+= dll
 
 SOURCES		= qassistantclient.cpp
 HEADERS         += $$QT_SOURCE_TREE/include/ntqassistantclient.h
--- ./tools/designer/editor/editor.pro.ori	2013-06-25 20:21:49.469566420 +0200
+++ ./tools/designer/editor/editor.pro	2013-06-25 20:21:59.535359278 +0200
@@ -1,6 +1,6 @@
 TEMPLATE	= lib
-CONFIG		+= qt warn_on staticlib
-CONFIG		-= dll
+CONFIG		+= qt warn_on
+CONFIG		+= dll
 HEADERS		= editor.h  \
 		  parenmatcher.h  \
 		  completion.h \
--- ./tools/designer/designer/designer.pro.ori	2013-06-25 20:22:39.279541337 +0200
+++ ./tools/designer/designer/designer.pro	2013-06-25 20:23:08.892943751 +0200
@@ -1,7 +1,7 @@
 TEMPLATE	= lib
 
-CONFIG		+= qt warn_on staticlib qmake_cache
-CONFIG 		-= dll
+CONFIG		+= qt warn_on qmake_cache
+CONFIG 		+= dll
 !force_static:!win32:contains(QT_PRODUCT,qt-internal) {
    CONFIG          -= staticlib
    CONFIG          += dll
