--- trinity-tqt3-14.0.0~pre193+a9ec6a15/src/kernel/ntqstyle.h.ORI	2014-05-04 12:49:48.579786273 -0400
+++ trinity-tqt3-14.0.0~pre193+a9ec6a15/src/kernel/ntqstyle.h	2014-05-04 12:50:00.948216121 -0400
@@ -287,6 +287,7 @@
 		TQ_UINT32 topLevelWidgetFlags;
 		TQPixmap paletteBgPixmap;
 
+/*
 #ifdef ENABLE_TQSTYLECONTROLELEMENTDATA_SLOW_COPY
 	public:
 		TQStyleControlElementData(const TQStyleControlElementData&);
@@ -295,6 +296,7 @@
 		// Disable copy constructor
 		TQStyleControlElementData(const TQStyleControlElementData&);
 #endif // ENABLE_TQSTYLECONTROLELEMENTDATA_SLOW_COPY
+*/
 
 };
 
--- trinity-tqt3-14.0.0~pre193+a9ec6a15/src/kernel/qstyle.cpp.ORI	2014-05-04 12:50:38.499448860 -0400
+++ trinity-tqt3-14.0.0~pre193+a9ec6a15/src/kernel/qstyle.cpp	2014-05-04 12:50:49.766018474 -0400
@@ -2836,6 +2836,7 @@
 	//
 }
 
+/*
 TQStyleControlElementData::TQStyleControlElementData(const TQStyleControlElementData& orig) {
 	isNull = orig.isNull;
 	widgetObjectTypes = orig.widgetObjectTypes;
@@ -2898,5 +2899,6 @@
 	topLevelWidgetFlags = orig.topLevelWidgetFlags;
 	paletteBgPixmap = orig.paletteBgPixmap;
 }
+*/
 
 #endif // QT_NO_STYLE
