--- tqt3/src/tools/qgpluginmanager.cpp.ndebug	2012-02-13 19:10:01.000000000 +0100
+++ tqt3/src/tools/qgpluginmanager.cpp	2012-02-13 19:25:05.653477747 +0100
@@ -506,10 +506,12 @@
 			     (*f).latin1(),
 			     (const char*) TQFile::encodeName( old->library() ) );
 		} else {
+/*
 		    qWarning("%s: Feature %s already defined in %s!",
 			     (const char*) TQFile::encodeName( old->library() ),
 			     (*f).latin1(),
 			     (const char*) TQFile::encodeName( plugin->library() ) );
+*/
 		}
 	    }
 	}
