--- python-tqt/pyqtconfig.py.in.ORI	2013-12-28 15:49:34.901906208 +0100
+++ python-tqt/pyqtconfig.py.in	2013-12-28 15:49:42.849851496 +0100
@@ -21,7 +21,7 @@
 # modules that %Import PyTQt modules.
 
 
-import sipconfig
+from sip4_tqt import sipconfig
 
 
 # These are installation specific values created when PyTQt was configured.
--- python-tqt/build.py.ORI	2013-12-28 15:52:10.923832969 +0100
+++ python-tqt/build.py	2013-12-28 15:52:18.638779926 +0100
@@ -35,7 +35,7 @@
 
 # Get the SIP configuration.
 try:
-    import sipconfig
+    from sip4_tqt import sipconfig
 except:
     print "Unable to import the sipconfig module.  Please make sure you have"
     print "SIP v3.9 or later installed."
--- python-tqt/configure.py.ORI	2013-12-28 15:52:31.294692944 +0100
+++ python-tqt/configure.py	2013-12-28 15:52:37.494650327 +0100
@@ -28,7 +28,7 @@
 import shutil
 import py_compile
 
-import sipconfig
+from sip4_tqt import sipconfig
 
 src_dir = os.path.dirname(os.path.abspath(__file__))
 
