--- tqt3/src/tools/qsettings.cpp.umask	2012-02-13 19:10:01.000000000 +0100
+++ tqt3/src/tools/qsettings.cpp	2012-02-13 19:28:00.802555882 +0100
@@ -997,7 +997,10 @@
 #endif
 
 	TQFile file( filename + ".tmp" );
-	if (! file.open(IO_WriteOnly)) {
+	mode_t old_umask = umask(0022);
+	bool f = file.open(IO_WriteOnly);
+	umask(old_umask);
+	if (! f) {
 
 #ifdef QT_CHECK_STATE
 	    qWarning("TQSettings::sync: failed to open '%s' for writing",
