--- trinity-libtqt-perl-14.0.0~pre127+d5696aa6/configure.in.ORI	2013-07-04 19:50:09.707137994 +0200
+++ trinity-libtqt-perl-14.0.0~pre127+d5696aa6/configure.in	2013-07-04 19:50:18.592226468 +0200
@@ -43,7 +43,7 @@
 KDE_SET_PREFIX
 
 dnl generate the config header
-AM_CONFIG_HEADER(config.h) dnl at the distribution this done
+AC_CONFIG_HEADER(config.h) dnl at the distribution this done
 
 dnl Checks for programs.
 AC_PROG_CC
