commit 7d0d82b7d93af5822d4c84c4ebd4ad82b9e463d0
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1334801652 -0500

    Fix usage of obsolete libpng jmpbuf member

diff --git a/digikam/kioslave/digikamthumbnail.cpp b/digikam/kioslave/digikamthumbnail.cpp
index 96cba8e..6dba518 100644
--- a/digikam/kioslave/digikamthumbnail.cpp
+++ b/digikam/kioslave/digikamthumbnail.cpp
@@ -392,7 +392,7 @@ TQImage kio_digikamthumbnailProtocol::loadPNG(const TQString& path)
         return qimage;
     }
 
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
         png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
         fclose(f);
diff --git a/digikam/libs/dimg/loaders/pngloader.cpp b/digikam/libs/dimg/loaders/pngloader.cpp
index 3367561..fd5d771 100644
--- a/digikam/libs/dimg/loaders/pngloader.cpp
+++ b/digikam/libs/dimg/loaders/pngloader.cpp
@@ -120,7 +120,7 @@ bool PNGLoader::load(const TQString& filePath, DImgLoaderObserver *observer)
     // PNG error handling. If an error occurs during reading, libpng
     // will jump here
 
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
         DDebug() << k_funcinfo << "Internal libPNG error during reading file. Process aborted!" << endl;
         png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
@@ -526,7 +526,7 @@ bool PNGLoader::save(const TQString& filePath, DImgLoaderObserver *observer)
     // PNG error handling. If an error occurs during writing, libpng
     // will jump here
 
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
         DDebug() << k_funcinfo << "Internal libPNG error during writing file. Process aborted!" << endl;
         fclose(f);
