--- digikam/digikam/digikam/Makefile.am.ORI	2012-08-03 19:41:42.791088483 +0200
+++ digikam/digikam/digikam/Makefile.am	2012-08-03 19:41:56.661794764 +0200
@@ -111,6 +111,8 @@
 #       if shared libsqlite3 is used LIB_SQLITE3_LOCAL is null.
 libdigikam_la_LIBADD = $(LIB_SQLITE3) \
 	               $(LIB_SQLITE3_LOCAL) \
+		       $(LIB_QT) \
+		       $(LIB_KPARTS) \
 	               $(LIB_KIO) \
 	               $(LIB_KABC) \
 	               $(LIB_KHTML) \
