commit 1f71905047d874cdc6c63049e566a4d7b976b0fc
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1327530712 -0600

    Fix linear alphabet string errors

diff --git a/kcron/ctcron.cpp b/kcron/ctcron.cpp
index e470a72..f5cb04c 100644
--- a/kcron/ctcron.cpp
+++ b/kcron/ctcron.cpp
@@ -178,7 +178,7 @@ istream& operator >> (istream& inputStream, CTCron& cron)
     if ((line.find("#") == 0) && (line.find("\\") != 1))
     {
       // If the first 10 characters don't contain a character, it's probably a disabled entry.
-      int first_text = line.find_first_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPTQRSTUVWXYZ");
+      int first_text = line.find_first_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
       if (first_text < 0)
          continue;
 
diff --git a/kuser/misc.cpp b/kuser/misc.cpp
index 6b26a72..c16f563 100644
--- a/kuser/misc.cpp
+++ b/kuser/misc.cpp
@@ -135,7 +135,7 @@ void addShell(const TQString &shell)
 TQCString genSalt( int len )
 {
   TQCString salt( len + 1 );
-  const char * set = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPTQRSTUVWXYZ0123456789./";
+  const char * set = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789./";
     
   salt[0] = set[getpid() % strlen(set)];
   for( int i = 1; i < len; i++ ) {
