commit 14ba7d8af6ef7248dcd3eb85ae977950d1758600
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1336066403 -0500

    Fix 'format not a string literal' error

diff --git a/kbarcode/csvimportdlg.cpp b/kbarcode/csvimportdlg.cpp
index 26e59da..6b07a7b 100644
--- a/kbarcode/csvimportdlg.cpp
+++ b/kbarcode/csvimportdlg.cpp
@@ -334,7 +334,6 @@ void CSVImportDlg::accept()
     TQValueList<int> headers;
     TQStringList list;
     TQString name = getDatabaseName();
-    int i = 0;
 
     TQString q = "INSERT INTO " + name + " (";
     for( int c = 0; c < table->horizontalHeader()->count(); c++ ) {
@@ -375,7 +374,7 @@ void CSVImportDlg::accept()
 
         TQSqlQuery query;
         if( !query.exec( line ) )
-            qDebug( i18n("Could not import the following line:") + line );
+            qDebug( "%s", i18n("Could not import the following line:") + line );
             //KMessageBox::error( this, i18n("Could not import the following line:") + line );
     }
 
