commit b83d045598947bb355ed6a2996b24225da6a9ffa
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1326310679 -0600

    Fix kdeedu makefile glitch
    This closes Bug 785

diff --git a/kalzium/src/solver/Makefile b/kalzium/src/solver/Makefile
index 4b96fdb..6012982 100644
--- a/kalzium/src/solver/Makefile
+++ b/kalzium/src/solver/Makefile
@@ -89,4 +89,4 @@ install :
 .depend: parser.ml lexer.ml
 	$(OCAMLDEP) $(INCLUDES) *.mli *.ml > $@
 
-include .depend
+-include .depend
