--- kdeedu/libkdeedu/kdeeducore/Makefile.am.ORI	2012-08-01 16:03:57.158018060 +0200
+++ kdeedu/libkdeedu/kdeeducore/Makefile.am	2012-08-01 16:04:06.206657197 +0200
@@ -11,7 +11,7 @@
 libkdeeducode_includedir = $(includedir)/libkdeedu
 libkdeeducode_include_HEADERS = keduvocdata.h
 
-libkdeeducore_la_LIBADD = $(LIB_KDECORE)
+libkdeeducore_la_LIBADD = $(LIB_KDECORE) $(LIB_QT)
 
 METASOURCES = AUTO
 
--- kdeedu/libkdeedu/kdeeduui/Makefile.am.ORI	2012-08-01 16:08:47.875060426 +0200
+++ kdeedu/libkdeedu/kdeeduui/Makefile.am	2012-08-01 16:08:53.275867176 +0200
@@ -10,7 +10,7 @@
 libkdeeduuiinclude_HEADERS = kdeeduglossary.h
 
 libkdeeduui_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 3:5:0
-libkdeeduui_la_LIBADD = $(LIB_KDECORE) $(LIB_KIO) $(LIB_KDEUI) $(LIB_KHTML)
+libkdeeduui_la_LIBADD = $(LIB_KDECORE) $(LIB_KIO) $(LIB_KDEUI) $(LIB_KHTML) $(LIB_QT)
 
 METASOURCES = AUTO
 
--- kdeedu/libkdeedu/extdate/Makefile.am.ORI	2012-08-01 16:10:28.272515295 +0200
+++ kdeedu/libkdeedu/extdate/Makefile.am	2012-08-01 16:10:37.999176797 +0200
@@ -7,7 +7,7 @@
 libextdate_la_SOURCES = extdatetime.cpp extcalendarsystem.cpp extcalendarsystemgregorian.cpp extdatetbl.cpp extdatepicker.cpp extdatetimeedit.cpp extdatewidget.cpp
 
 libextdate_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 3:0:2
-libextdate_la_LIBADD = $(LIB_KDEUI)
+libextdate_la_LIBADD = $(LIB_KDEUI) $(LIB_QT) $(LIB_KDECORE)
 
 test_extdate_SOURCES = test_extdate.cc
 test_extdate_LDADD = libextdate.la
--- kdeedu/kiten/Makefile.am.ORI	2012-08-01 16:15:45.679552203 +0200
+++ kdeedu/kiten/Makefile.am	2012-08-01 16:15:53.606341454 +0200
@@ -13,7 +13,7 @@
 lib_LTLIBRARIES = libkiten.la
 
 libkiten_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined -version-info 1:0:0
-libkiten_la_LIBADD = $(LIB_KDEUI) $(LIB_KDEPRINT)
+libkiten_la_LIBADD = $(LIB_KDEUI) $(LIB_KDEPRINT) $(LIB_QT) $(LIB_KDECORE)
 
 libkiten_la_SOURCES = asyndeta.cpp dict.cpp widgets.cpp kromajiedit.cpp rad.cpp \
 	deinf.cpp kitenconfig.kcfgc
--- kdeedu/kvoctrain/kvoctrain/kvt-core/Makefile.am.ORI	2012-08-01 16:24:48.865399440 +0200
+++ kdeedu/kvoctrain/kvoctrain/kvt-core/Makefile.am	2012-08-01 16:24:54.832227336 +0200
@@ -12,6 +12,7 @@
 
 SUBDIRS = kvt-xml
 
-libkvoctraincore_la_LIBADD= kvt-xml/libkvtxml.la
+libkvoctraincore_la_LIBADD= kvt-xml/libkvtxml.la ../common-dialogs/libcommondlg.la $(LIB_QT) $(LIB_KDECORE) $(LIB_KIO) $(LIB_KDEUI) -lDCOP
+libkvoctraincore_la_LDFLAGS = $(QT_LDFLAGS)
 
 noinst_HEADERS = grammarmanager.h
