--- tdeedu/kiten/dict.h
+++ tdeedu/kiten/dict.h	2012-04-11 01:14:04.481645011 -0500
@@ -65,12 +65,12 @@
 	, data(d)
 	, dataSize(s)
 {
-	setRawData(data, dataSize / sizeof(T));
+	this->setRawData(data, dataSize / sizeof(T));
 }
 
 template<class T> Array<T>::~Array()
 {
-	resetRawData(data, dataSize / sizeof(T));
+	this->resetRawData(data, dataSize / sizeof(T));
 }
 
 // File manages all the files, pointers, and memory management associated
