diff -urN tdelibs/kio/kio/kdirwatch.cpp tdelibs.new/kio/kio/kdirwatch.cpp
--- tdelibs/kio/kio/kdirwatch.cpp	2012-04-05 13:26:08.000000000 -0500
+++ tdelibs.new/kio/kio/kdirwatch.cpp	2012-04-08 17:51:15.000000000 -0500
@@ -1155,6 +1155,7 @@
     if(!e->dirty) return NoChange;
     e->dirty = false;
   }
+  if (e->isDir) return Changed;
 #endif
 
   // Shouldn't happen: Ignore "unknown" notification method
@@ -1184,6 +1185,8 @@
   if (exists) {
 
     if (e->m_status == NonExistent) {
+      // ctime is the 'creation time' on windows, but with qMax
+      // we get the latest change of any kind, on any platform.
       e->m_ctime = stat_buf.st_ctime;
       e->m_status = Normal;
       e->m_nlink = stat_buf.st_nlink;
