commit 0536f0b707bf86636096c962e9bb32adf3bb8112
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1341041394 -0500

    Export kdemain symbol in iso kioslave
    This closes Bug 465

diff --git a/kioslave/iso/iso.cpp b/kioslave/iso/iso.cpp
index 5865139..4bd128e 100644
--- a/kioslave/iso/iso.cpp
+++ b/kioslave/iso/iso.cpp
@@ -56,7 +56,7 @@ static const unsigned char zisofs_magic[8] = {
 
 using namespace KIO;
 
-extern "C" { int kdemain(int argc, char **argv); }
+extern "C" { KDE_EXPORT int kdemain(int argc, char **argv); }
 
 int kdemain( int argc, char **argv )
 {
