--- ./kdecore/kconfigbackend.cpp.orig	2011-08-11 04:30:15.000000000 +0200
+++ ./kdecore/kconfigbackend.cpp	2012-06-18 23:27:20.034671093 +0200
@@ -255,8 +255,7 @@
       mLocalFileName = KGlobal::dirs()->saveLocation(resType) + mfileName;
 
    if (useKDEGlobals)
-      mGlobalFileName = KGlobal::dirs()->saveLocation("config") +
-	      TQString::tqfromLatin1("kdeglobals");
+      mGlobalFileName = KGlobal::dirs()->saveLocation("config", TQString(), false) + TQString::fromLatin1("kdeglobals");
    else
       mGlobalFileName = TQString::null;
 
