commit 3e284fadb9110cdd2a2dcf9c2e960c74f15e0360
Author: Slávek Banko <slavek.banko@axis.cz>
Date:   1339570958 +0200

    Fix FTBFS - incomplete build kspell2
    This closes Bug 657

diff --git a/kspell2/ui/CMakeLists.txt b/kspell2/ui/CMakeLists.txt
index 65ea158..a83774d 100644
--- a/kspell2/ui/CMakeLists.txt
+++ b/kspell2/ui/CMakeLists.txt
@@ -54,12 +54,14 @@ tde_add_library( ${target} STATIC_PIC AUTOMOC
 set( target kspell2 )
 
 set( ${target}_SRCS
+  configwidget.cpp highlighter.cpp configdialog.cpp
+  kspell2ui.ui configui.ui
   dialog.cpp
 )
 
 tde_add_library( ${target} SHARED AUTOMOC
   SOURCES ${${target}_SRCS}
   VERSION 1.0.0
-  LINK kspell2-static kspell2base-static kutils-shared
+  LINK kspell2base-static kutils-shared
   DESTINATION ${LIB_INSTALL_DIR}
 )
