commit 2b035349c31fe64c31d2c050892b117a3a807179
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1326668093 -0600

    Add fam/gamin support to tdelibs CMake

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ce186b..a1a6429 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,6 +77,7 @@ OPTION( WITH_ELFICON "Enable ELF embedded icon support" ${WITH_ALL_OPTIONS} )
 OPTION( WITH_AVAHI "Enable AVAHI support" OFF )
 OPTION( WITH_PCRE "Enable pcre regex support for kjs" ON )
 OPTION( WITH_INOTIFY "Enable inotify support for kio" ON )
+OPTION( WITH_GAMIN "Enable FAM/GAMIN support" OFF )
 
 OPTION( WITH_ASPELL "Enable aspell support" OFF )
 OPTION( WITH_HSPELL "Enable hspell support" OFF )
@@ -786,6 +787,23 @@ if( WITH_INOTIFY )
 endif( )
 
 
+##### check for FAM/GAMIN ##########################
+
+if( WITH_GAMIN )
+  check_include_file( "fam.h" HAVE_FAM_H )
+  if( HAVE_FAM_H )
+    pkg_search_module( GAMIN gamin )
+    if( GAMIN_FOUND )
+      set( HAVE_FAM 1 )
+    else( GAMIN_FOUND )
+      message(FATAL_ERROR "\nfam/gamin support was requested, but gamin was not found on your system" )
+    endif( GAMIN_FOUND )
+  else( HAVE_FAM_H )
+    message(FATAL_ERROR "\nfam/gamin support was requested, but fam was not found on your system" )
+  endif( HAVE_FAM_H )
+endif( WITH_GAMIN )
+
+
 ##### check for aspell ##########################
 # we need ASPELL_DATADIR too
 
diff --git a/config.h.cmake b/config.h.cmake
index d9b0ef5..4e4bd18 100644
--- a/config.h.cmake
+++ b/config.h.cmake
@@ -120,7 +120,7 @@
 #undef HAVE_FADVISE
 
 /* Define if your system has libfam */
-#undef HAVE_FAM
+#cmakedefine HAVE_FAM 1
 
 /* Define to 1 if you have the <float.h> header file. */
 #cmakedefine HAVE_FLOAT_H 1
diff --git a/kio/kio/CMakeLists.txt b/kio/kio/CMakeLists.txt
index 9d8790f..25d4fad 100644
--- a/kio/kio/CMakeLists.txt
+++ b/kio/kio/CMakeLists.txt
@@ -25,8 +25,12 @@ include_directories(
   ${CMAKE_SOURCE_DIR}/kio
   ${CMAKE_SOURCE_DIR}/kio/kssl
   ${CMAKE_SOURCE_DIR}/interfaces
+  ${GAMIN_INCLUDEDIR}
 )
 
+link_directories(
+  ${GAMIN_LIBDIR}
+)
 
 ##### headers ###################################
 
@@ -112,4 +116,5 @@
 
 tde_add_library( ${target} STATIC_PIC AUTOMOC
   SOURCES ${${target}_SRCS}
+  LINK ${GAMIN_LIBRARIES}
 )
 
