--- kdelibs/kdefx/CMakeLists.txt.ORI	2012-09-05 14:49:17.332664590 -0400
+++ kdelibs/kdefx/CMakeLists.txt	2012-09-05 14:50:29.605179741 -0400
@@ -20,6 +20,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/keramik/CMakeLists.txt.ORI	2012-09-05 14:52:39.488511648 -0400
+++ kdelibs/kstyles/keramik/CMakeLists.txt	2012-09-05 14:52:46.135375114 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/plastik/CMakeLists.txt.ORI	2012-09-05 14:55:52.184554103 -0400
+++ kdelibs/kstyles/plastik/CMakeLists.txt	2012-09-05 14:55:59.389406152 -0400
@@ -25,6 +25,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/asteroid/CMakeLists.txt.ORI	2012-09-05 14:57:14.714859423 -0400
+++ kdelibs/kstyles/asteroid/CMakeLists.txt	2012-09-05 14:57:25.833631121 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/highcolor/CMakeLists.txt.ORI	2012-09-05 14:59:43.880796883 -0400
+++ kdelibs/kstyles/highcolor/CMakeLists.txt	2012-09-05 14:59:52.769614406 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/highcontrast/CMakeLists.txt.ORI	2012-09-05 15:00:52.531387606 -0400
+++ kdelibs/kstyles/highcontrast/CMakeLists.txt	2012-09-05 15:00:59.908236179 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/light/CMakeLists.txt.ORI	2012-09-05 15:02:37.873225343 -0400
+++ kdelibs/kstyles/light/CMakeLists.txt	2012-09-05 15:02:46.535047560 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdecore/CMakeLists.txt.ORI	2012-09-05 15:04:44.597624532 -0400
+++ kdelibs/kdecore/CMakeLists.txt	2012-09-05 15:04:58.009349300 -0400
@@ -37,6 +37,7 @@
 link_directories(
   ${TQT_LIBRARY_DIRS}
   ${LIBIDN_LIBRARY_DIRS}
+  ${XCOMPOSITE_LIBRARY_DIRS}
 )
 
 
--- kdelibs/dcop/client/CMakeLists.txt.ORI	2012-09-05 15:09:40.211558975 -0400
+++ kdelibs/dcop/client/CMakeLists.txt	2012-09-05 15:09:46.973420255 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kjs/CMakeLists.txt.ORI	2012-09-05 15:10:44.871247689 -0400
+++ kdelibs/kjs/CMakeLists.txt	2012-09-05 15:10:55.546035695 -0400
@@ -21,6 +21,7 @@
 link_directories(
   ${TQT_LIBRARY_DIRS}
   ${LIBPCRE_LIBDIR}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/CMakeLists.txt.ORI	2012-09-05 15:11:50.982934593 -0400
+++ kdelibs/kio/misc/CMakeLists.txt	2012-09-05 15:11:59.380767769 -0400
@@ -40,6 +40,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdecore/kconfig_compiler/CMakeLists.txt.ORI	2012-09-05 15:12:54.956663587 -0400
+++ kdelibs/kdecore/kconfig_compiler/CMakeLists.txt	2012-09-05 15:13:01.689529800 -0400
@@ -17,6 +17,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kwallet/client/CMakeLists.txt.ORI	2012-09-05 15:13:46.102647178 -0400
+++ kdelibs/kwallet/client/CMakeLists.txt	2012-09-05 15:13:54.441481441 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kwallet/backend/CMakeLists.txt.ORI	2012-09-05 15:14:39.946576908 -0400
+++ kdelibs/kwallet/backend/CMakeLists.txt	2012-09-05 15:14:46.189452800 -0400
@@ -20,6 +20,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kunittest/CMakeLists.txt.ORI	2012-09-05 15:15:30.724567350 -0400
+++ kdelibs/kunittest/CMakeLists.txt	2012-09-05 15:15:37.191438762 -0400
@@ -18,6 +18,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/kdesasl/CMakeLists.txt.ORI	2012-09-05 15:16:23.686514140 -0400
+++ kdelibs/kio/misc/kdesasl/CMakeLists.txt	2012-09-05 15:16:30.107386437 -0400
@@ -18,6 +18,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdesu/CMakeLists.txt.ORI	2012-09-05 15:17:18.220429423 -0400
+++ kdelibs/kdesu/CMakeLists.txt	2012-09-05 15:17:25.132291924 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/kntlm/CMakeLists.txt.ORI	2012-09-05 15:18:10.203395218 -0400
+++ kdelibs/kio/misc/kntlm/CMakeLists.txt	2012-09-05 15:18:16.090278085 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdewidgets/CMakeLists.txt.ORI	2012-09-05 15:19:01.043383536 -0400
+++ kdelibs/kdewidgets/CMakeLists.txt	2012-09-05 15:19:08.404237043 -0400
@@ -29,6 +29,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kded/CMakeLists.txt.ORI	2012-09-05 15:19:53.290343643 -0400
+++ kdelibs/kded/CMakeLists.txt	2012-09-05 15:19:59.050228986 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kinit/CMakeLists.txt.ORI	2012-09-05 15:20:56.013094954 -0400
+++ kdelibs/kinit/CMakeLists.txt	2012-09-05 15:21:01.877978180 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kconf_update/CMakeLists.txt.ORI	2012-09-05 15:21:46.470090230 -0400
+++ kdelibs/kconf_update/CMakeLists.txt	2012-09-05 15:21:52.129977515 -0400
@@ -17,6 +17,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kabc/vcard/CMakeLists.txt.ORI	2012-09-05 15:22:33.653150519 -0400
+++ kdelibs/kabc/vcard/CMakeLists.txt	2012-09-05 15:22:39.350037047 -0400
@@ -21,6 +21,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeui/CMakeLists.txt.ORI	2012-09-05 15:23:43.705755018 -0400
+++ kdelibs/kdeui/CMakeLists.txt	2012-09-05 15:23:52.102587723 -0400
@@ -33,6 +33,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kimgio/CMakeLists.txt.ORI	2012-09-05 15:24:52.729379655 -0400
+++ kdelibs/kimgio/CMakeLists.txt	2012-09-05 15:24:59.825238243 -0400
@@ -20,6 +20,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/kthemestyle/CMakeLists.txt.ORI	2012-09-05 15:25:41.501407611 -0400
+++ kdelibs/kstyles/kthemestyle/CMakeLists.txt	2012-09-05 15:25:48.051277056 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/libkmid/CMakeLists.txt.ORI	2012-09-05 15:26:30.215436548 -0400
+++ kdelibs/libkmid/CMakeLists.txt	2012-09-05 15:26:36.349314265 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/http/kcookiejar/CMakeLists.txt.ORI	2012-09-05 15:27:21.689410284 -0400
+++ kdelibs/kioslave/http/kcookiejar/CMakeLists.txt	2012-09-05 15:27:28.281278845 -0400
@@ -22,6 +22,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/utils/installtheme/CMakeLists.txt.ORI	2012-09-05 15:28:14.233362491 -0400
+++ kdelibs/kstyles/utils/installtheme/CMakeLists.txt	2012-09-05 15:28:21.719213198 -0400
@@ -17,6 +17,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 ##### kinstalltheme #############################
--- kdelibs/dnssd/CMakeLists.txt.ORI	2012-09-05 15:29:09.718255841 -0400
+++ kdelibs/dnssd/CMakeLists.txt	2012-09-05 15:29:15.350143500 -0400
@@ -20,6 +20,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 ##### headers ###################################
--- kdelibs/kstyles/highcontrast/config/CMakeLists.txt.ORI	2012-09-05 15:30:26.325727548 -0400
+++ kdelibs/kstyles/highcontrast/config/CMakeLists.txt	2012-09-05 15:30:31.885616614 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/libkscreensaver/CMakeLists.txt.ORI	2012-09-05 15:31:25.686543044 -0400
+++ kdelibs/libkscreensaver/CMakeLists.txt	2012-09-05 15:31:31.287431270 -0400
@@ -21,6 +21,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/pics/CMakeLists.txt.ORI	2012-09-05 15:33:15.116358829 -0400
+++ kdelibs/pics/CMakeLists.txt	2012-09-05 15:33:22.821205010 -0400
@@ -28,6 +28,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/tools/escputil/CMakeLists.txt.ORI	2012-09-05 15:34:18.809087161 -0400
+++ kdelibs/kdeprint/tools/escputil/CMakeLists.txt	2012-09-05 15:34:24.838966757 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/arts/message/CMakeLists.txt.ORI	2012-09-05 15:35:54.035185423 -0400
+++ kdelibs/arts/message/CMakeLists.txt	2012-09-05 15:35:59.572074829 -0400
@@ -18,6 +18,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kstyles/plastik/config/CMakeLists.txt.ORI	2012-09-05 15:37:59.047687962 -0400
+++ kdelibs/kstyles/plastik/config/CMakeLists.txt	2012-09-05 15:38:04.596577086 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/networkstatus/CMakeLists.txt.ORI	2012-09-05 15:39:15.031169518 -0400
+++ kdelibs/networkstatus/CMakeLists.txt	2012-09-05 15:39:22.330023641 -0400
@@ -22,6 +22,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/CMakeLists.txt.ORI	2012-09-05 15:40:44.810374939 -0400
+++ kdelibs/kio/CMakeLists.txt	2012-09-05 15:40:54.052190177 -0400
@@ -30,6 +30,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/tests/CMakeLists.txt.ORI	2012-09-05 15:41:48.532100927 -0400
+++ kdelibs/kio/tests/CMakeLists.txt	2012-09-05 15:41:54.080989975 -0400
@@ -18,6 +18,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/kioexec/CMakeLists.txt.ORI	2012-09-05 15:42:48.434903055 -0400
+++ kdelibs/kio/kioexec/CMakeLists.txt	2012-09-05 15:42:54.144788864 -0400
@@ -22,6 +22,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/kfile/CMakeLists.txt.ORI	2012-09-05 15:43:44.232787087 -0400
+++ kdelibs/kio/misc/kfile/CMakeLists.txt	2012-09-05 15:43:51.120649315 -0400
@@ -21,6 +21,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
@@ -36,4 +37,4 @@
   SOURCES ${${target}_SRCS}
   LINK kio-shared
   DESTINATION ${BIN_INSTALL_DIR}
-)
\ No newline at end of file
+)
--- kdelibs/kdoctools/CMakeLists.txt.ORI	2012-09-05 15:44:33.205807471 -0400
+++ kdelibs/kdoctools/CMakeLists.txt	2012-09-05 15:44:39.466682225 -0400
@@ -25,6 +25,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/ksendbugmail/CMakeLists.txt.ORI	2012-09-05 15:45:24.401783247 -0400
+++ kdelibs/kio/misc/ksendbugmail/CMakeLists.txt	2012-09-05 15:45:31.286645499 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/interfaces/kscript/CMakeLists.txt.ORI	2012-09-05 15:46:39.496306960 -0400
+++ kdelibs/interfaces/kscript/CMakeLists.txt	2012-09-05 15:46:45.331193604 -0400
@@ -25,6 +25,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kparts/CMakeLists.txt.ORI	2012-09-05 15:47:41.529101602 -0400
+++ kdelibs/kparts/CMakeLists.txt	2012-09-05 15:47:47.294989542 -0400
@@ -25,6 +25,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/CMakeLists.txt.ORI	2012-09-05 15:48:34.508071783 -0400
+++ kdelibs/kdeprint/CMakeLists.txt	2012-09-05 15:48:41.700931937 -0400
@@ -40,6 +40,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/cups/cupsdconf2/CMakeLists.txt.ORI	2012-09-05 15:49:26.236065937 -0400
+++ kdelibs/kdeprint/cups/cupsdconf2/CMakeLists.txt	2012-09-05 15:49:31.824957243 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/http/CMakeLists.txt.ORI	2012-09-05 15:50:45.284528209 -0400
+++ kdelibs/kioslave/http/CMakeLists.txt	2012-09-05 15:50:52.468388422 -0400
@@ -27,6 +27,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/file/CMakeLists.txt.ORI	2012-09-05 15:51:43.357398032 -0400
+++ kdelibs/kioslave/file/CMakeLists.txt	2012-09-05 15:51:50.506258876 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/ftp/CMakeLists.txt.ORI	2012-09-05 15:52:39.615302790 -0400
+++ kdelibs/kioslave/ftp/CMakeLists.txt	2012-09-05 15:52:45.935179729 -0400
@@ -21,6 +21,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/arts/kde/CMakeLists.txt.ORI	2012-09-05 15:53:32.040281826 -0400
+++ kdelibs/arts/kde/CMakeLists.txt	2012-09-05 15:53:38.541155204 -0400
@@ -25,6 +25,7 @@
 link_directories(
   ${TQT_LIBRARY_DIRS}
   ${ARTS_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 # pkg-config stripping -I/usr/include
--- kdelibs/interfaces/kimproxy/library/CMakeLists.txt.ORI	2012-09-05 15:54:52.117721726 -0400
+++ kdelibs/interfaces/kimproxy/library/CMakeLists.txt	2012-09-05 15:55:02.693515629 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kresources/CMakeLists.txt.ORI	2012-09-05 15:55:55.219491830 -0400
+++ kdelibs/kresources/CMakeLists.txt	2012-09-05 15:56:01.964360344 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/iso/CMakeLists.txt.ORI	2012-09-05 15:57:00.178225273 -0400
+++ kdelibs/kioslave/iso/CMakeLists.txt	2012-09-05 15:57:06.535101301 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/gzip/CMakeLists.txt.ORI	2012-09-05 15:58:24.363583139 -0400
+++ kdelibs/kioslave/gzip/CMakeLists.txt	2012-09-05 15:58:29.777477507 -0400
@@ -18,6 +18,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/bzip2/CMakeLists.txt.ORI	2012-09-05 15:59:29.076320316 -0400
+++ kdelibs/kioslave/bzip2/CMakeLists.txt	2012-09-05 15:59:34.117221927 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kioslave/metainfo/CMakeLists.txt.ORI	2012-09-05 16:00:23.606255852 -0400
+++ kdelibs/kioslave/metainfo/CMakeLists.txt	2012-09-05 16:00:29.200146638 -0400
@@ -20,6 +20,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/krsync/CMakeLists.txt.ORI	2012-09-05 16:01:17.567202186 -0400
+++ kdelibs/krsync/CMakeLists.txt	2012-09-05 16:01:23.763081184 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/kssld/CMakeLists.txt.ORI	2012-09-05 16:02:12.759124170 -0400
+++ kdelibs/kio/misc/kssld/CMakeLists.txt	2012-09-05 16:02:19.426993911 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeui/kdetrayproxy/CMakeLists.txt.ORI	2012-09-05 16:03:48.650271606 -0400
+++ kdelibs/kdeui/kdetrayproxy/CMakeLists.txt	2012-09-05 16:03:54.752156274 -0400
@@ -20,6 +20,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/kwalletd/CMakeLists.txt.ORI	2012-09-05 16:04:55.631005088 -0400
+++ kdelibs/kio/misc/kwalletd/CMakeLists.txt	2012-09-05 16:05:03.666853065 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/knewstuff/CMakeLists.txt.ORI	2012-09-05 16:05:51.805942029 -0400
+++ kdelibs/knewstuff/CMakeLists.txt	2012-09-05 16:05:57.713830182 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/kpasswdserver/CMakeLists.txt.ORI	2012-09-05 16:06:46.686902706 -0400
+++ kdelibs/kio/kpasswdserver/CMakeLists.txt	2012-09-05 16:06:53.345776552 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kio/misc/kpac/CMakeLists.txt.ORI	2012-09-05 16:07:43.907818289 -0400
+++ kdelibs/kio/misc/kpac/CMakeLists.txt	2012-09-05 16:07:49.444713316 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 ##### other data ################################
--- kdelibs/kabc/CMakeLists.txt.ORI	2012-09-05 16:08:49.154580815 -0400
+++ kdelibs/kabc/CMakeLists.txt	2012-09-05 16:08:57.154429020 -0400
@@ -36,6 +36,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/khtml/java/CMakeLists.txt.ORI	2012-09-05 16:09:55.224326707 -0400
+++ kdelibs/khtml/java/CMakeLists.txt	2012-09-05 16:10:02.194194348 -0400
@@ -28,6 +28,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kutils/CMakeLists.txt.ORI	2012-09-05 16:10:45.848365097 -0400
+++ kdelibs/kutils/CMakeLists.txt	2012-09-05 16:10:51.715253615 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/interfaces/kscript/sample/CMakeLists.txt.ORI	2012-09-05 16:11:52.986088900 -0400
+++ kdelibs/interfaces/kscript/sample/CMakeLists.txt	2012-09-05 16:12:00.529945437 -0400
@@ -20,6 +20,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/khtml/kmultipart/CMakeLists.txt.ORI	2012-09-05 16:12:49.105021375 -0400
+++ kdelibs/khtml/kmultipart/CMakeLists.txt	2012-09-05 16:12:55.035908513 -0400
@@ -25,6 +25,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kcert/CMakeLists.txt.ORI	2012-09-05 16:13:43.686982413 -0400
+++ kdelibs/kcert/CMakeLists.txt	2012-09-05 16:13:51.788828139 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/arts/knotify/CMakeLists.txt.ORI	2012-09-05 16:14:42.802856412 -0400
+++ kdelibs/arts/knotify/CMakeLists.txt	2012-09-05 16:14:50.217715126 -0400
@@ -24,6 +24,7 @@
 link_directories(
   ${TQT_LIBRARY_DIRS}
   ${ARTS_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/interfaces/kmediaplayer/CMakeLists.txt.ORI	2012-09-05 16:15:47.850616551 -0400
+++ kdelibs/interfaces/kmediaplayer/CMakeLists.txt	2012-09-05 16:15:53.805503001 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/management/CMakeLists.txt.ORI	2012-09-05 16:16:51.704398570 -0400
+++ kdelibs/kdeprint/management/CMakeLists.txt	2012-09-05 16:16:58.402270762 -0400
@@ -32,6 +32,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/lpdunix/CMakeLists.txt.ORI	2012-09-05 16:18:05.348992771 -0400
+++ kdelibs/kdeprint/lpdunix/CMakeLists.txt	2012-09-05 16:18:13.274841407 -0400
@@ -28,6 +28,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/ext/CMakeLists.txt.ORI	2012-09-05 16:19:06.696820851 -0400
+++ kdelibs/kdeprint/ext/CMakeLists.txt	2012-09-05 16:19:13.589689130 -0400
@@ -26,6 +26,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/interfaces/kmediaplayer/kfileaudiopreview/CMakeLists.txt.ORI	2012-09-05 16:20:09.255625011 -0400
+++ kdelibs/interfaces/kmediaplayer/kfileaudiopreview/CMakeLists.txt	2012-09-05 16:20:16.883479148 -0400
@@ -28,6 +28,7 @@
 link_directories(
   ${TQT_LIBRARY_DIRS}
   ${ARTS_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/khtml/CMakeLists.txt.ORI	2012-09-05 16:21:17.056328088 -0400
+++ kdelibs/khtml/CMakeLists.txt	2012-09-05 16:21:23.855197987 -0400
@@ -46,6 +46,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kspell2/ui/CMakeLists.txt.ORI	2012-09-05 16:22:21.155101145 -0400
+++ kdelibs/kspell2/ui/CMakeLists.txt	2012-09-05 16:22:28.929952270 -0400
@@ -25,6 +25,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kcmshell/CMakeLists.txt.ORI	2012-09-05 16:23:26.764844457 -0400
+++ kdelibs/kcmshell/CMakeLists.txt	2012-09-05 16:23:32.943726066 -0400
@@ -23,6 +23,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kmdi/kmdi/CMakeLists.txt.ORI	2012-09-05 16:24:30.703618985 -0400
+++ kdelibs/kmdi/kmdi/CMakeLists.txt	2012-09-05 16:24:36.946499290 -0400
@@ -25,6 +25,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kmdi/CMakeLists.txt.ORI	2012-09-05 16:25:27.631527239 -0400
+++ kdelibs/kmdi/CMakeLists.txt	2012-09-05 16:25:33.639411986 -0400
@@ -27,6 +27,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kspell2/plugins/ispell/CMakeLists.txt.ORI	2012-09-05 16:26:26.062406048 -0400
+++ kdelibs/kspell2/plugins/ispell/CMakeLists.txt	2012-09-05 16:26:31.159308217 -0400
@@ -19,6 +19,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kspell2/plugins/aspell/CMakeLists.txt.ORI	2012-09-05 16:27:38.670011962 -0400
+++ kdelibs/kspell2/plugins/aspell/CMakeLists.txt	2012-09-05 16:27:44.931891687 -0400
@@ -21,6 +21,7 @@
 link_directories(
   ${TQT_LIBRARY_DIRS}
   ${ASPELL_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kabc/formats/CMakeLists.txt.ORI	2012-09-05 16:28:38.501862464 -0400
+++ kdelibs/kabc/formats/CMakeLists.txt	2012-09-05 16:28:44.382749443 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kabc/plugins/ldapkio/CMakeLists.txt.ORI	2012-09-05 16:29:38.283713293 -0400
+++ kdelibs/kabc/plugins/ldapkio/CMakeLists.txt	2012-09-05 16:29:44.129600887 -0400
@@ -27,6 +27,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kabc/plugins/file/CMakeLists.txt.ORI	2012-09-05 16:30:42.311481817 -0400
+++ kdelibs/kabc/plugins/file/CMakeLists.txt	2012-09-05 16:30:47.634379406 -0400
@@ -27,6 +27,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/interfaces/ktexteditor/CMakeLists.txt.ORI	2012-09-05 16:31:40.221367404 -0400
+++ kdelibs/interfaces/ktexteditor/CMakeLists.txt	2012-09-05 16:31:47.174233563 -0400
@@ -28,6 +28,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kabc/plugins/dir/CMakeLists.txt.ORI	2012-09-05 16:32:39.930217776 -0400
+++ kdelibs/kabc/plugins/dir/CMakeLists.txt	2012-09-05 16:32:45.775105207 -0400
@@ -27,6 +27,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kabc/plugins/net/CMakeLists.txt.ORI	2012-09-05 16:33:52.745814992 -0400
+++ kdelibs/kabc/plugins/net/CMakeLists.txt	2012-09-05 16:34:01.271650687 -0400
@@ -27,6 +27,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/cups/CMakeLists.txt.ORI	2012-09-05 16:35:01.843483032 -0400
+++ kdelibs/kdeprint/cups/CMakeLists.txt	2012-09-05 16:35:10.784310629 -0400
@@ -35,6 +35,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/rlpr/CMakeLists.txt.ORI	2012-09-05 16:36:01.783326981 -0400
+++ kdelibs/kdeprint/rlpr/CMakeLists.txt	2012-09-05 16:36:08.502197359 -0400
@@ -29,6 +29,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kate/plugins/wordcompletion/CMakeLists.txt.ORI	2012-09-05 16:37:02.425156810 -0400
+++ kdelibs/kate/plugins/wordcompletion/CMakeLists.txt	2012-09-05 16:37:08.514039285 -0400
@@ -21,6 +21,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 ##### other data ################################
--- kdelibs/kate/plugins/isearch/CMakeLists.txt.ORI	2012-09-05 16:38:01.583014723 -0400
+++ kdelibs/kate/plugins/isearch/CMakeLists.txt	2012-09-05 16:38:07.610898319 -0400
@@ -22,6 +22,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kate/interfaces/CMakeLists.txt.ORI	2012-09-05 16:39:02.822865446 -0400
+++ kdelibs/kate/interfaces/CMakeLists.txt	2012-09-05 16:39:11.673700838 -0400
@@ -21,6 +21,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kdeprint/lpr/CMakeLists.txt.ORI	2012-09-05 16:40:04.179723987 -0400
+++ kdelibs/kdeprint/lpr/CMakeLists.txt	2012-09-05 16:40:15.471513833 -0400
@@ -29,6 +29,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kate/plugins/insertfile/CMakeLists.txt.ORI	2012-09-05 16:41:07.687541674 -0400
+++ kdelibs/kate/plugins/insertfile/CMakeLists.txt	2012-09-05 16:41:13.326436653 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kate/plugins/kdatatool/CMakeLists.txt.ORI	2012-09-05 16:42:09.313393591 -0400
+++ kdelibs/kate/plugins/kdatatool/CMakeLists.txt	2012-09-05 16:42:15.058286525 -0400
@@ -24,6 +24,7 @@
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 
--- kdelibs/kate/part/CMakeLists.txt.ORI	2012-09-05 16:43:06.975318648 -0400
+++ kdelibs/kate/part/CMakeLists.txt	2012-09-05 16:43:13.228202040 -0400
@@ -35,6 +35,7 @@
 link_directories(
   ${TQT_LIBRARY_DIRS}
   ${LUA_LIBRARY_DIRS}
+  ${XRENDER_LIBRARY_DIRS}
 )
 
 ##### kate (static library) #####################
