--- trinity-tdelibs-14.0.0~pre599+bae79307/tdeui/kpassivepopup.cpp.ORI	2013-07-19 22:31:16.453073343 +0200
+++ trinity-tdelibs-14.0.0~pre599+bae79307/tdeui/kpassivepopup.cpp	2013-07-19 22:31:55.288484612 +0200
@@ -120,7 +120,9 @@
                              const TQPixmap &icon )
 {
     // kdDebug() << "KPassivePopup::setView " << caption << ", " << text << endl;
-    setView( standardView( caption, text, icon, this ) );
+    const TQString captionISO = TQString::fromLocal8Bit(caption.latin1());
+    const TQString textISO = TQString::fromLocal8Bit(text.latin1());
+    setView( standardView( captionISO, textISO, icon, this ) );
 }
 
 TQVBox * KPassivePopup::standardView( const TQString& caption,
