diff -urN tdemultimedia/configure.in.in tdemultimedia.new/configure.in.in
--- tdemultimedia/configure.in.in	2010-08-11 22:36:30.000000000 -0500
+++ tdemultimedia.new/configure.in.in	2012-01-12 22:54:04.000000000 -0600
@@ -35,7 +35,7 @@
 	then
 		MMX_SUPPORT="no"
 	else
-		as_ver=`echo $as_ver |sed -e "s/.*version //;s/ .*//"`
+		as_ver=`echo|as -v 2>&1 | grep ".*version.*" | awk '{print $NF}'`
 		as_major=`echo $as_ver |cut -d. -f1`
 		if test $as_major -gt 2
 		then
