--- pykdeextensions/src/kdedistutils.py.libtool	2011-08-21 23:01:12.000000000 +0200
+++ pykdeextensions/src/kdedistutils.py	2012-05-11 23:41:07.326232457 +0200
@@ -790,6 +791,8 @@
             cmdlist.append(sysconfig.get_config_var("LDSHARED").split()[0])
             # link_flags
             cmdlist.append("-module")
+            cmdlist.append("-avoid-version")
+            cmdlist.append("-shared")
             cmdlist.append("-export-dynamic")
             # object
             cmdlist.append(outputfile)
@@ -1703,6 +1705,8 @@
             cmdlist.append(sysconfig.get_config_var("LDSHARED").split()[0])
             # link_flags
             cmdlist.append("-module")
+            cmdlist.append("-avoid-version")
+            cmdlist.append("-shared")
             cmdlist.append("-export-dynamic")
             # object
             cmdlist.append(outputfile)
