--- pykdeextensions-3.5.13.1/src/kdedistutils.py.extramodule	2012-09-08 17:33:05.000000000 +0000
+++ pykdeextensions-3.5.13.1/src/kdedistutils.py	2012-10-02 19:19:40.185384207 +0000
@@ -712,6 +712,8 @@
             
             # Build the 'stub' code.
             cppcode = self.cpptemplate % {"moduledir": self.data_dir,
+                                            "extramodule": os.getenv("EXTRA_MODULE_DIR"),
+                                            "pyqt_dir": self.pyqt_dir,
                                             "modulename": modulename,
                                             "factoryfunction": factoryfunction,
                                             "python_version": python_version}
@@ -854,6 +856,8 @@
 #include <sip.h>
 
 #define MODULE_DIR "%(moduledir)s"
+#define EXTRA_MODULE_DIR "%(extramodule)s"
+#define PYQT_DIR "%(pyqt_dir)s"
 #define MODULE_NAME "%(modulename)s"
 #define FACTORY "%(factoryfunction)s"
 #define CPP_FACTORY %(factoryfunction)s
--- pykdeextensions-3.5.13.1/src/kdedistutils.py.ORI	2013-04-07 17:54:28.531813437 +0200
+++ pykdeextensions-3.5.13.1/src/kdedistutils.py	2013-04-07 17:56:15.510529614 +0200
@@ -903,6 +906,18 @@
     if(!pyize->appendToSysPath (path.latin1 ())) {
         return report_error ("***Failed to set sys.path\n");
     }
+
+    // Add the extra path to the python script to the interpreter search path.
+    TQString extrapath = TQString(EXTRA_MODULE_DIR);
+    if(!pyize->appendToSysPath (extrapath.latin1 ())) {
+      return report_error ("***Failed to set extra sys.path\n");
+    }
+
+    // Add the PYQT path to the python script to the interpreter search path.
+    TQString pyqt_dir = TQString(PYQT_DIR);
+    if(!pyize->appendToSysPath (pyqt_dir.latin1 ())) {
+      return report_error ("***Failed to set PYQT sys.path\n");
+    }
     
     // Load the Python script.
     PyObject *pyModule = pyize->importModule ((char *)script.latin1 ());
