--- akode-2.0.2/akode/lib/Makefile.am.ORI	2012-01-16 19:17:09.849211160 +0100
+++ akode-2.0.2/akode/lib/Makefile.am	2012-01-16 19:17:24.496410002 +0100
@@ -12,6 +12,7 @@
                       player.cpp magic.cpp
 
 AM_CPPFLAGS = -DAKODE_SEARCHDIR=\"$(libdir)\"
+AM_CXXFLAGS = -fPIC
 
 libakode_la_LDFLAGS = -no-undefined -version-info 2:0:0 $(all_libraries)
 libakode_la_LIBADD =  $(LIBPTHREAD) $(AKODE_LIBDL) $(LIBSEM)
--- akode-2.0.2/configure.in.in.ORI	2013-05-19 23:25:31.450512262 +0200
+++ akode-2.0.2/configure.in.in	2013-05-19 23:25:37.156390421 +0200
@@ -55,7 +55,7 @@
 
 dnl without this order in this file, automake will be confused!
 dnl
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 
 dnl checks for programs.
 dnl first check for c/c++ compilers
--- akode-2.0.2/admin/configure.in.min~	2013-05-19 23:35:21.532965131 +0200
+++ akode-2.0.2/admin/configure.in.min	2013-05-19 23:34:19.011289846 +0200
@@ -41,7 +41,7 @@
 KDE_SET_PREFIX
 
 dnl generate the config header
-AM_CONFIG_HEADER(config.h) dnl at the distribution this done
+AC_CONFIG_HEADER(config.h) dnl at the distribution this done
 
 dnl Checks for programs.
 AC_CHECK_COMPILERS
--- akode-2.0.2/admin/cvs.sh.autotools	2007-01-15 10:06:45.000000000 +0100
+++ akode-2.0.2/admin/cvs.sh	2013-05-19 23:37:14.145581582 +0200
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.1[0-9]*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
@@ -128,7 +128,7 @@
 echo "*** Creating configure"
 call_and_fix_autoconf
 
-if egrep "^AM_CONFIG_HEADER" configure.in >/dev/null 2>&1; then
+if egrep "^AC_CONFIG_HEADER" configure.in >/dev/null 2>&1; then
   echo "*** Creating config.h template"
   $AUTOHEADER || exit 1
   touch config.h.in
@@ -184,7 +184,7 @@
   fi
 fi
 $ACLOCAL $ACLOCALFLAGS
-if egrep "^AM_CONFIG_HEADER" configure.in >/dev/null 2>&1; then
+if egrep "^AC_CONFIG_HEADER" configure.in >/dev/null 2>&1; then
   echo "*** Creating config.h template"
   $AUTOHEADER || exit 1
   touch config.h.in
--- akode-2.0.2/akode/configure.in.in.ORI	2013-05-19 23:39:02.874283156 +0200
+++ akode-2.0.2/akode/configure.in.in	2013-05-19 23:39:13.381061201 +0200
@@ -2,7 +2,7 @@
 dnl AC_OUTPUT(akode/akode-config)
 dnl AC_OUTPUT(akode/akode.pc)
 
-AM_CONFIG_HEADER(akode/lib/akode_export.h)
+AC_CONFIG_HEADER(akode/lib/akode_export.h)
 
 AC_DEFUN([AC_CHECK_LIBFLAC],
 [
