--- akode-2.0.2/akode/lib/Makefile.am.ORI	2012-01-16 19:17:09.849211160 +0100
+++ akode-2.0.2/akode/lib/Makefile.am	2012-01-16 19:17:24.496410002 +0100
@@ -12,6 +12,7 @@
                       player.cpp magic.cpp
 
 AM_CPPFLAGS = -DAKODE_SEARCHDIR=\"$(libdir)\"
+AM_CXXFLAGS = -fPIC
 
 libakode_la_LDFLAGS = -no-undefined -version-info 2:0:0 $(all_libraries)
 libakode_la_LIBADD =  $(LIBPTHREAD) $(AKODE_LIBDL) $(LIBSEM)
