*** kdenetwork/kopete/protocols/testbed/testbedaccount.h.ORI	2011-08-24 00:30:58.403965000 +0200
--- kdenetwork/kopete/protocols/testbed/testbedaccount.h	2011-08-24 00:31:29.706965000 +0200
***************
*** 59,65 ****
  	/**
  	 * 'Connect' to the testbed server.  Only sets myself() online.
  	 */
! 	virtual void connect( const Kopete::OnlineStatus& initialStatus = Kopete::OnlineStatus::OnlineStatus() );
  	/**
  	 * Disconnect from the server.  Only sets myself() offline.
  	 */
--- 59,65 ----
  	/**
  	 * 'Connect' to the testbed server.  Only sets myself() online.
  	 */
! 	virtual void connect( const Kopete::OnlineStatus& initialStatus = Kopete::OnlineStatus() );
  	/**
  	 * Disconnect from the server.  Only sets myself() offline.
  	 */
*** kdenetwork/kopete/kopete/contactlist/kopetegroupviewitem.cpp.ORI	2011-08-24 00:28:22.080965001 +0200
--- kdenetwork/kopete/kopete/contactlist/kopetegroupviewitem.cpp	2011-08-24 00:28:36.149965002 +0200
***************
*** 175,181 ****
  	// the sorting itself is rather slow. Therefore we call delayedSort, which tries
  	// to group multiple sort requests into one.
  	using namespace Kopete::UI::ListView;
! 	if ( ListView::ListView *lv = dynamic_cast<ListView::ListView *>( listView() ) )
  		lv->delayedSort();
  	else
  		listView()->sort();
--- 175,181 ----
  	// the sorting itself is rather slow. Therefore we call delayedSort, which tries
  	// to group multiple sort requests into one.
  	using namespace Kopete::UI::ListView;
! 	if ( ListView *lv = dynamic_cast<ListView *>( listView() ) )
  		lv->delayedSort();
  	else
  		listView()->sort();
*** kdenetwork/kopete/kopete/chatwindow/krichtexteditpart.cpp.ORI	2011-08-24 00:29:00.953965001 +0200
--- kdenetwork/kopete/kopete/chatwindow/krichtexteditpart.cpp	2011-08-24 00:29:21.080965001 +0200
***************
*** 38,44 ****
  KopeteRichTextEditPart::KopeteRichTextEditPart( TQWidget *wparent, const char *wname, TQObject*, const char*, const TQStringList& )
  	: KParts::ReadOnlyPart( wparent, wname ? wname : "rich_text_part" )
  {
! 	KopeteRichTextEditPart::KopeteRichTextEditPart( wparent, wname, false );
  }
  
  KopeteRichTextEditPart::KopeteRichTextEditPart( TQWidget *parent, const char *name, int capabilities )
--- 38,44 ----
  KopeteRichTextEditPart::KopeteRichTextEditPart( TQWidget *wparent, const char *wname, TQObject*, const char*, const TQStringList& )
  	: KParts::ReadOnlyPart( wparent, wname ? wname : "rich_text_part" )
  {
! 	KopeteRichTextEditPart( wparent, wname, false );
  }
  
  KopeteRichTextEditPart::KopeteRichTextEditPart( TQWidget *parent, const char *name, int capabilities )
