--- kdenetwork/kopete/protocols/sms/services/smsclient.cpp.ORI	2011-11-27 05:22:47.001212554 -0500
+++ kdenetwork/kopete/protocols/sms/services/smsclient.cpp	2011-11-27 05:22:58.253085350 -0500
@@ -67,7 +67,7 @@
 		return;
 	}
 
-	TQString programName = c->readEntry(TQString("%1:%2").tqarg("SMSClient").tqarg("ProgramName"). TQString());
+	TQString programName = c->readEntry(TQString("%1:%2").tqarg("SMSClient").tqarg("ProgramName"), TQString());
 	if (programName.isNull())
 		programName = "/usr/bin/sms_client";
 
