From caedf8323c45b376c824de343f31de78664b2511 Mon Sep 17 00:00:00 2001
From: Serghei Amelian <serghei.amelian@gmail.com>
Date: Sun, 12 Feb 2012 21:08:53 +0000
Subject: [TDEMacros.cmake] added tde_save_and_set() macro

---
diff --git a/modules/TDEMacros.cmake b/modules/TDEMacros.cmake
index 075b7eb..1091755 100644
--- a/modules/TDEMacros.cmake
+++ b/modules/TDEMacros.cmake
@@ -1283,6 +1283,11 @@ macro( tde_save )
   endforeach()
 endmacro()
 
+macro( tde_save_and_set _var )
+  set( __bak_${_var} ${${_var}} )
+  set( ${_var} ${ARGN} )
+endmacro( )
+
 macro( tde_restore )
   foreach( _var ${ARGN} )
     set( ${_var} ${__bak_${_var}} )
--
cgit v0.9.0.2-39-g756e
