Index: kdenetwork/kopete/libkopete/avdevice/configure.in.in
===================================================================
--- kdenetwork/kopete/libkopete/avdevice/configure.in.in	(revision 0)
+++ kdenetwork/kopete/libkopete/avdevice/configure.in.in	(revision 1243951)
@@ -0,0 +1,21 @@
+# -- Check for libv4l1-videodev.h presence --------------------
+
+AC_MSG_CHECKING([for libv4l1-videodev.h])
+
+AC_TRY_COMPILE([
+#include <libv4l1-videodev.h>
+],
+[
+],
+        ac_have_libv4l1_videodev_h=yes
+,
+        ac_have_libv4l1_videodev_h=no
+)
+
+if test $ac_have_libv4l1_videodev_h = yes; then
+        AC_DEFINE(HAVE_LIBV4L1_VIDEODEV_H, 1, [Define if you have libv4l1-videodev.h installed])
+fi
+
+AC_MSG_RESULT($ac_have_libv4l1_videodev_h)
+
+# -- End libv4l1-videodev.h ----------------------------------------------
Index: kdenetwork/kopete/libkopete/avdevice/videodevice.h
===================================================================
--- kdenetwork/kopete/libkopete/avdevice/videodevice.h	(revision 1243950)
+++ kdenetwork/kopete/libkopete/avdevice/videodevice.h	(revision 1243951)
@@ -52,7 +52,11 @@
 
 #include <linux/fs.h>
 #include <linux/kernel.h>
+#if defined(HAVE_LIBV4L1_VIDEODEV_H)
+#include <libv4l1-videodev.h>
+#else
 #include <linux/videodev.h>
+#endif
 #define VIDEO_MODE_PAL_Nc  3
 #define VIDEO_MODE_PAL_M   4
 #define VIDEO_MODE_PAL_N   5
