diff -Nuar kdegraphics.ORI/ksnapshot/CMakeLists.txt kdegraphics/ksnapshot/CMakeLists.txt
--- kdegraphics.ORI/ksnapshot/CMakeLists.txt	2011-05-11 23:29:40.000000000 +0200
+++ kdegraphics/ksnapshot/CMakeLists.txt	2011-10-31 15:45:47.107958154 +0100
@@ -40,6 +40,6 @@
   SOURCES
     ksnapshotiface.skel main.cpp ksnapshot.cpp regiongrabber.cpp
     windowgrabber.cpp ksnapshotwidget.ui
-  LINK kdeprint-shared
+  LINK kdeprint-shared Xext
   DESTINATION ${BIN_INSTALL_DIR}
 )
