Index: kexi/kexiutils/utils.cpp
===================================================================
--- kexi/kexiutils/utils.cpp	(revision 716273)
+++ kexi/kexiutils/utils.cpp	(working copy)
@@ -57,7 +57,7 @@
 	TQApplication::restoreOverrideCursor();
 }
 void DelayedCursorHandler::show() {
-	TQApplication::setOverrideCursor( KCursor::waitCursor() );
+	TQApplication::setOverrideCursor( KCursor::waitCursor(), true/*replace*/ );
 }
 
 DelayedCursorHandler _delayedCursorHandler;
@@ -89,7 +89,8 @@
 
 WaitCursorRemover::~WaitCursorRemover()
 {
-	_delayedCursorHandler.start(true);
+	if (m_reactivateCursor)
+		_delayedCursorHandler.start(true);
 }
 
 //--------------------------------------------------------------------------------
