--- koffice/chalk/plugins/viewplugins/scripting/chalkcore/krs_iterator.h.ORI	2012-04-26 20:22:29.322777779 +0200
+++ koffice/chalk/plugins/viewplugins/scripting/chalkcore/krs_iterator.h	2012-04-26 20:23:21.216311169 +0200
@@ -137,12 +137,12 @@
             }
             initiales = initiales.upper();
             // set/get general
-            addFunction("set" + initiales, &Iterator::setPixel);
-            addFunction("get" + initiales, &Iterator::getPixel);
+            this->addFunction("set" + initiales, &Iterator::setPixel);
+            this->addFunction("get" + initiales, &Iterator::getPixel);
             kdDebug(41011) << ( "get" + initiales ) << endl;
             // Various colorSpace
-            addFunction("invertColor", &Iterator::invertColor);
-            addFunction("darken", &Iterator::darken);
+            this->addFunction("invertColor", &Iterator::invertColor);
+            this->addFunction("darken", &Iterator::darken);
         }
     
         ~Iterator()
