--- koffice/lib/store/Makefile.am.ORI	2012-08-05 21:05:16.181010835 +0200
+++ koffice/lib/store/Makefile.am	2012-08-05 21:05:25.027837330 +0200
@@ -7,7 +7,7 @@
 
 ####### Files
 
-libkstore_la_LIBADD = $(LIB_KIO)
+libkstore_la_LIBADD = $(LIB_KIO) $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI)
 libkstore_la_SOURCES = KoStore.cpp KoTarStore.cpp KoDirectoryStore.cpp KoZipStore.cpp KoStoreDrag.cpp KoStoreBase.cpp KoXmlWriter.cpp
 libkstore_la_LDFLAGS = $(all_libraries) -version-info 3:0:0 $(KDE_LDFLAGS) -no-undefined
 include_HEADERS = KoStore.h KoStoreDevice.h KoXmlWriter.h
--- koffice/lib/kwmf/Makefile.am.ORI	2012-08-05 21:09:16.183303850 +0200
+++ koffice/lib/kwmf/Makefile.am	2012-08-05 21:09:35.742920240 +0200
@@ -3,7 +3,7 @@
 INCLUDES= $(KOFFICECORE_INCLUDES) -I$(srcdir) $(all_includes)
 libkwmf_la_LDFLAGS = $(all_libraries) -version-info 3:0:0 -no-undefined
 # We use kdecore for kdDebug :)
-libkwmf_la_LIBADD = $(LIB_KDECORE)
+libkwmf_la_LIBADD = $(LIB_KDECORE) $(LIB_QT)
 
 ####### Files
 
@@ -16,4 +16,4 @@
 
 libkowmf_la_SOURCES  = kowmfreadprivate.cc kowmfstack.cc kowmfread.cc kowmfwrite.cc kowmfpaint.cc
 libkowmf_la_LDFLAGS = $(all_libraries) -version-info 2:0:0  -no-undefined
-libkowmf_la_LIBADD = $(LIB_KDECORE)
+libkowmf_la_LIBADD = $(LIB_KDECORE) $(LIB_QT)
--- koffice/lib/kofficecore/Makefile.am.ORI	2012-08-05 21:12:34.924406087 +0200
+++ koffice/lib/kofficecore/Makefile.am	2012-08-05 21:12:47.700155525 +0200
@@ -4,7 +4,7 @@
 KDE_CXXFLAGS = $(USE_RTTI) $(WOVERLOADED_VIRTUAL)
 INCLUDES= $(KSTORE_INCLUDES) $(KWMF_INCLUDES) $(all_includes)
 libkofficecore_la_LIBADD  = $(LIB_KSTORE) $(LIB_KOWMF) $(LIB_KPARTS) \
-	$(LIB_KDEPRINT) $(LIB_KABC) $(LIB_KWMF)
+	$(LIB_KDEPRINT) $(LIB_KABC) $(LIB_KWMF) -lDCOP
 
 ####### Files
 
--- koffice/lib/koproperty/Makefile.am.ORI	2012-08-05 21:16:10.320181691 +0200
+++ koffice/lib/koproperty/Makefile.am	2012-08-05 21:16:17.135048036 +0200
@@ -2,7 +2,7 @@
 
 lib_LTLIBRARIES =  libkoproperty.la
 
-libkoproperty_la_LIBADD = $(LIB_KDEUI) ./editors/libkopropertyeditors.la
+libkoproperty_la_LIBADD = $(LIB_KDEUI) ./editors/libkopropertyeditors.la $(LIB_QT) $(LIB_KDECORE)
 libkoproperty_la_LDFLAGS = -no-undefined $(all_libraries) -version-info 2:0:0
 libkoproperty_la_SOURCES = property.cpp customproperty.cpp set.cpp editor.cpp \
 	editoritem.cpp factory.cpp widget.cpp
--- koffice/kpresenter/Makefile.am.ORI	2012-08-05 21:41:53.120199863 +0200
+++ koffice/kpresenter/Makefile.am	2012-08-05 21:42:03.358832558 +0200
@@ -72,7 +72,7 @@
 if include_ARTS
 ARTSKDE = -lartskde
 endif
-libkpresenterprivate_la_LIBADD = autoformEdit/libaf.la $(LIB_KOFFICEUI) $(LIB_KOTEXT) $(LIB_KOPAINTER) $(ARTSKDE)
+libkpresenterprivate_la_LIBADD = autoformEdit/libaf.la $(LIB_KOFFICEUI) $(LIB_KOTEXT) $(LIB_KOPAINTER) $(ARTSKDE) -lsoundserver_idl -lmcop
 
 ## The part
 kde_module_LTLIBRARIES = libkpresenterpart.la
