diff -u -r qt3/src/sql/drivers/odbc/qsql_odbc.cpp src/qt3/src/sql/drivers/odbc/qsql_odbc.cpp
--- qt3/src/sql/drivers/odbc/qsql_odbc.cpp	2011-10-30 19:55:58.000000000 +0100
+++ src/qt3/src/sql/drivers/odbc/qsql_odbc.cpp	2012-01-13 01:24:59.000000000 +0100
@@ -61,19 +61,18 @@
 #endif
 
 // newer platform SDKs use SQLLEN instead of SQLINTEGER
-#ifdef SQLLEN
+#if defined(SQLLEN) || defined(Q_OS_WIN64) || defined(Q_OS_UNIX)
 # define QSQLLEN SQLLEN
 #else
 # define QSQLLEN SQLINTEGER
 #endif
 
-#ifdef SQLULEN
+#if defined(SQLULEN) || defined(Q_OS_WIN64) || defined(Q_OS_UNIX)
 # define QSQLULEN SQLULEN
 #else
 # define QSQLULEN SQLUINTEGER
 #endif
 
-
 static const QSQLLEN COLNAMESIZE = 256;
 //Map Qt parameter types to ODBC types
 static const SQLSMALLINT qParamType[ 4 ] = { SQL_PARAM_INPUT, SQL_PARAM_INPUT, SQL_PARAM_OUTPUT, SQL_PARAM_INPUT_OUTPUT };

diff -u -r qt3/plugins/src/sqldrivers/odbc/odbc.pro src/qt3/plugins/src/sqldrivers/odbc/odbc.pro
--- qt3/plugins/src/sqldrivers/odbc/odbc.pro	2011-10-30 19:55:57.000000000 +0100
+++ src/qt3/plugins/src/sqldrivers/odbc/odbc.pro	2012-01-13 01:39:11.000000000 +0100
@@ -10,14 +10,14 @@
 
 mac {
         !contains( LIBS, .*odbc.* ) {
-            LIBS        *= -liodbc
+            LIBS        *= -lodbc
         }
 }
 
 unix {
 	OBJECTS_DIR	= .obj
 	!contains( LIBS, .*odbc.* ) {
-	    LIBS 	*= -liodbc
+	    LIBS 	*= -lodbc
 	}
 }
 
diff -u -r qt3/src/sql/qt_sql.pri src/qt3/src/sql/qt_sql.pri
--- qt3/src/sql/qt_sql.pri	2011-10-30 19:55:58.000000000 +0100
+++ src/qt3/src/sql/qt_sql.pri	2012-01-13 01:39:31.000000000 +0100
@@ -110,13 +110,13 @@
 
 		mac {
 			!contains( LIBS, .*odbc.* ) {
-				LIBS        *= -liodbc
+				LIBS        *= -lodbc
 			}
 		}
 
 		unix {
 			!contains( LIBS, .*odbc.* ) {
-				LIBS        *= -liodbc
+				LIBS        *= -lodbc
 			}
 		}
