commit c1faffb8ed0ebae85f21eb3e3ca81a2692a81ae9
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1347382645 -0500

    Use cflags from ruby pc file

diff --git a/korundum/rubylib/korundum/Makefile.am b/korundum/rubylib/korundum/Makefile.am
index 5642728..541e3f5 100644
--- a/korundum/rubylib/korundum/Makefile.am
+++ b/korundum/rubylib/korundum/Makefile.am
@@ -1,4 +1,4 @@
-INCLUDES = -I$(top_srcdir)/smoke -I$(top_srcdir)/qtruby/rubylib/qtruby $(all_includes) -I$(RUBY_ARCHDIR)
+INCLUDES = -I$(top_srcdir)/smoke -I$(top_srcdir)/qtruby/rubylib/qtruby $(all_includes) -I$(RUBY_ARCHDIR) $(RUBY_CFLAGS)
 
 rubylibdir = $(RUBY_ARCHDIR)
 rubylib_LTLIBRARIES = korundum.la
diff --git a/korundum/rubylib/korundum/configure.in.in b/korundum/rubylib/korundum/configure.in.in
index 5782123..e0d581b 100644
--- a/korundum/rubylib/korundum/configure.in.in
+++ b/korundum/rubylib/korundum/configure.in.in
@@ -15,6 +15,7 @@ else
       RUBY_SITEARCHDIR=`ruby -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"sitearchdir"@:>@)'`
       RUBY_SITEDIR=`ruby -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"sitelibdir"@:>@)'`
       RUBY_RUBYLIBDIR=`ruby -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"rubylibdir"@:>@)'`
+      RUBY_CFLAGS=`$PKGCONFIG ruby --cflags`
     fi
   fi
   if test -z "$RUBY_VERSION"; then
@@ -23,10 +24,11 @@ else
     RUBY_SITEDIR=`ruby -r rbconfig -e 'printf("%s",Config::CONFIG@<:@"sitelibdir"@:>@)'`
     RUBY_RUBYLIBDIR=`ruby -r rbconfig -e 'printf("%s",Config::CONFIG@<:@"rubylibdir"@:>@)'`
   fi
-  AC_MSG_RESULT([archdir $RUBY_ARCHDIR, sitearchdir $RUBY_SITEARCHDIR, sitedir $RUBY_SITEDIR])
+  AC_MSG_RESULT([archdir $RUBY_ARCHDIR, sitearchdir $RUBY_SITEARCHDIR, sitedir $RUBY_SITEDIR, cflags $RUBY_CFLAGS])
   AC_SUBST(RUBY_ARCHDIR)
   AC_SUBST(RUBY_SITEARCHDIR)
   AC_SUBST(RUBY_RUBYLIBDIR)
   AC_SUBST(RUBY_SITEDIR)
+  AC_SUBST(RUBY_CFLAGS)
 fi
 
diff --git a/qtruby/rubylib/qtruby/Makefile.am b/qtruby/rubylib/qtruby/Makefile.am
index 156f80a..56b320f 100644
--- a/qtruby/rubylib/qtruby/Makefile.am
+++ b/qtruby/rubylib/qtruby/Makefile.am
@@ -1,4 +1,4 @@
-INCLUDES = -I$(top_srcdir)/smoke $(all_includes) -I$(RUBY_ARCHDIR)
+INCLUDES = -I$(top_srcdir)/smoke $(all_includes) -I$(RUBY_ARCHDIR) $(RUBY_CFLAGS)
 
 noinst_HEADERS = qtruby.h marshall.h smokeruby.h extconf.rb 
 
diff --git a/qtruby/rubylib/qtruby/configure.in.in b/qtruby/rubylib/qtruby/configure.in.in
index 3cfa630..a875b2f 100644
--- a/qtruby/rubylib/qtruby/configure.in.in
+++ b/qtruby/rubylib/qtruby/configure.in.in
@@ -17,6 +17,7 @@ else
       RUBY_RUBYLIBDIR=`$RUBY -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"rubylibdir"@:>@)'`
       RUBY_LIBDIR=`$RUBY -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"libdir"@:>@)'`
       RUBY_LIBRUBYARG=`$RUBY -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"LIBRUBYARG_SHARED"@:>@)'`
+      RUBY_CFLAGS=`$PKGCONFIG ruby --cflags`
     fi
   fi
   if test -z "$RUBY_VERSION"; then
@@ -27,12 +28,13 @@ else
     RUBY_LIBDIR=`$RUBY -r rbconfig -e 'printf("%s",Config::CONFIG@<:@"libdir"@:>@)'`
     RUBY_LIBRUBYARG=`$RUBY -r rbconfig -e 'printf("%s",Config::CONFIG@<:@"LIBRUBYARG_SHARED"@:>@)'`
   fi
-  AC_MSG_RESULT([archdir $RUBY_ARCHDIR, sitearchdir $RUBY_SITEARCHDIR, sitedir $RUBY_SITEDIR, libdir $RUBY_LIBDIR, librubyarg $RUBY_LIBRUBYARG])
+  AC_MSG_RESULT([archdir $RUBY_ARCHDIR, sitearchdir $RUBY_SITEARCHDIR, sitedir $RUBY_SITEDIR, libdir $RUBY_LIBDIR, librubyarg $RUBY_LIBRUBYARG, cflags $RUBY_CFLAGS])
   AC_SUBST(RUBY_ARCHDIR)
   AC_SUBST(RUBY_SITEARCHDIR)
   AC_SUBST(RUBY_SITEDIR)
   AC_SUBST(RUBY_RUBYLIBDIR)
   AC_SUBST(RUBY_LIBDIR)
   AC_SUBST(RUBY_LIBRUBYARG)
+  AC_SUBST(RUBY_CFLAGS)
 fi
 
