commit a901a1232ab8f8432c63dc97e0efd9fed051d2c7
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1348264715 -0500

    Fix FTBFS on x86 systems

diff --git a/korundum/bin/krubyinit.cpp b/korundum/bin/krubyinit.cpp
index 12b0f6e..2a7f6c5 100644
--- a/korundum/bin/krubyinit.cpp
+++ b/korundum/bin/krubyinit.cpp
@@ -1,5 +1,7 @@
 #include <ruby.h>
 
+#define HAVE_STRLCAT_PROTO 1
+#define HAVE_STRLCPY_PROTO 1
 #include "config.h"
 
 #ifndef HAVE_RUBY_1_9
diff --git a/korundum/rubylib/korundum/Korundum.cpp b/korundum/rubylib/korundum/Korundum.cpp
index 124e2a1..da9f099 100644
--- a/korundum/rubylib/korundum/Korundum.cpp
+++ b/korundum/rubylib/korundum/Korundum.cpp
@@ -40,6 +40,8 @@
 #include <smokeruby.h>
 #include <smoke.h>
 
+#define HAVE_STRLCAT_PROTO 1
+#define HAVE_STRLCPY_PROTO 1
 #include "config.h"
 
 #ifndef HAVE_RUBY_1_9
diff --git a/korundum/rubylib/korundum/kdehandlers.cpp b/korundum/rubylib/korundum/kdehandlers.cpp
index 6961301..11b97b5 100644
--- a/korundum/rubylib/korundum/kdehandlers.cpp
+++ b/korundum/rubylib/korundum/kdehandlers.cpp
@@ -52,6 +52,8 @@
 #include <dom/dom_string.h>
 #include <dom/html_element.h>
 
+#define HAVE_STRLCAT_PROTO 1
+#define HAVE_STRLCPY_PROTO 1
 #include "config.h"
 
 #ifndef HAVE_RUBY_1_9
diff --git a/qtruby/bin/qtrubyinit.cpp b/qtruby/bin/qtrubyinit.cpp
index c9569d9..dd732b1 100644
--- a/qtruby/bin/qtrubyinit.cpp
+++ b/qtruby/bin/qtrubyinit.cpp
@@ -1,5 +1,7 @@
 #include <ruby.h>
 
+#define HAVE_STRLCAT_PROTO 1
+#define HAVE_STRLCPY_PROTO 1
 #include "config.h"
 
 #ifndef HAVE_RUBY_1_9
diff --git a/qtruby/rubylib/qtruby/Qt.cpp b/qtruby/rubylib/qtruby/Qt.cpp
index 370ad6d..b264d44 100644
--- a/qtruby/rubylib/qtruby/Qt.cpp
+++ b/qtruby/rubylib/qtruby/Qt.cpp
@@ -60,6 +60,8 @@
 #include "smokeruby.h"
 #include "smoke.h"
 
+#define HAVE_STRLCAT_PROTO 1
+#define HAVE_STRLCPY_PROTO 1
 #include "config.h"
 
 #ifndef HAVE_RUBY_1_9
diff --git a/qtruby/rubylib/qtruby/handlers.cpp b/qtruby/rubylib/qtruby/handlers.cpp
index f3d0dda..2f6b5e7 100644
--- a/qtruby/rubylib/qtruby/handlers.cpp
+++ b/qtruby/rubylib/qtruby/handlers.cpp
@@ -47,6 +47,8 @@
 #define HINT_BYTES HINT_BYTE
 #endif
 
+#define HAVE_STRLCAT_PROTO 1
+#define HAVE_STRLCPY_PROTO 1
 #include "config.h"
 
 #ifndef HAVE_RUBY_1_9
