--- trinity-tdebindings-3.5.13.2/qtruby/rubylib/qtruby/configure.in.in.ORI	2013-08-17 10:49:32.741192663 +0200
+++ trinity-tdebindings-3.5.13.2/qtruby/rubylib/qtruby/configure.in.in	2013-08-17 10:52:27.904466644 +0200
@@ -54,6 +54,13 @@
     RUBY_INCLUDEDIR=$RUBY_ARCHDIR
     RUBY_LIBRUBYARG=`$RUBY -r rbconfig -e 'printf("%s",Config::CONFIG@<:@"LIBRUBYARG_SHARED"@:>@)'`
   fi
+
+  # Removes trailing slashes, if any, to avoid fail to install with recent libtool.
+  RUBY_ARCHDIR=${RUBY_ARCHDIR%/}
+  RUBY_SITEDIR=${RUBY_SITEDIR%/}
+  RUBY_LIBDIR=${RUBY_LIBDIR%/}
+  RUBY_RUBYLIBDIR=${RUBY_RUBYLIBDIR%/}
+
   AC_MSG_RESULT([
                   archdir $RUBY_ARCHDIR,
                   sitearchdir $RUBY_SITEARCHDIR,
--- trinity-tdebindings-3.5.13.2/korundum/rubylib/korundum/configure.in.in.ORI	2013-08-17 10:58:23.873872686 +0200
+++ trinity-tdebindings-3.5.13.2/korundum/rubylib/korundum/configure.in.in	2013-08-17 10:58:44.416433828 +0200
@@ -54,6 +54,13 @@
     RUBY_INCLUDEDIR=$RUBY_ARCHDIR
     RUBY_LIBRUBYARG=`$RUBY -r rbconfig -e 'printf("%s",Config::CONFIG@<:@"LIBRUBYARG_SHARED"@:>@)'`
   fi
+
+  # Removes trailing slashes, if any, to avoid fail to install with recent libtool.
+  RUBY_ARCHDIR=${RUBY_ARCHDIR%/}
+  RUBY_SITEDIR=${RUBY_SITEDIR%/}
+  RUBY_LIBDIR=${RUBY_LIBDIR%/}
+  RUBY_RUBYLIBDIR=${RUBY_RUBYLIBDIR%/}
+
   AC_MSG_RESULT([
                   archdir $RUBY_ARCHDIR,
                   sitearchdir $RUBY_SITEARCHDIR,
