--- rosegarden/src/gui/editors/segment/segmentcanvas/AudioPreviewUpdater.cpp.ORI	2012-05-06 12:31:01.464997883 +0200
+++ rosegarden/src/gui/editors/segment/segmentcanvas/AudioPreviewUpdater.cpp	2012-05-06 12:31:19.271625694 +0200
@@ -22,6 +22,7 @@
     COPYING included with this distribution for more information.
 */
 
+#include <unistd.h>
 
 #include "AudioPreviewUpdater.h"
 
--- rosegarden/src/gui/editors/segment/segmentcanvas/AudioPreviewUpdater.h.ORI	2012-05-06 12:32:47.222788008 +0200
+++ rosegarden/src/gui/editors/segment/segmentcanvas/AudioPreviewUpdater.h	2012-05-06 12:33:01.046499259 +0200
@@ -26,6 +26,7 @@
 #ifndef _RG_AUDIOPREVIEWUPDATER_H_
 #define _RG_AUDIOPREVIEWUPDATER_H_
 
+#include <unistd.h>
 #include <tqobject.h>
 #include <tqrect.h>
 #include <vector>
