diff -rcb kdebase/ksmserver/shutdowndlg.h kdebase.patch/ksmserver/shutdowndlg.h
*** kdebase/ksmserver/shutdowndlg.h	Sat Sep 25 23:33:51 2010
--- kdebase.patch/ksmserver/shutdowndlg.h	Thu Dec  9 14:58:43 2010
***************
*** 32,38 ****
  
  /* We acknowledge the the dbus API is unstable */
  #define DBUS_API_SUBJECT_TO_CHANGE
! #include <dbus/connection.h>
  #include <libhal.h>
  
  // The (singleton) widget that makes/fades the desktop gray.
--- 32,38 ----
  
  /* We acknowledge the the dbus API is unstable */
  #define DBUS_API_SUBJECT_TO_CHANGE
! #include <dbus/dbus.h>
  #include <libhal.h>
  
  // The (singleton) widget that makes/fades the desktop gray.
