Index: kicker/applets/systemtray/systemtrayapplet.cpp
===================================================================
--- kicker/applets/systemtray/systemtrayapplet.cpp	(revision 1262481)
+++ kicker/applets/systemtray/systemtrayapplet.cpp	(working copy)
@@ -34,6 +34,7 @@
 #include <tqstyle.h>
 #include <tqgrid.h>
 #include <tqpainter.h>
+#include <tqimage.h>
 
 #include <dcopclient.h>
 #include <kapplication.h>
Index: kicker/applets/clock/clock.h
===================================================================
--- kicker/applets/clock/clock.h	(revision 1262481)
+++ kicker/applets/clock/clock.h	(working copy)
@@ -41,7 +41,6 @@
 
 #include <kickertip.h>
 #include "settings.h"
-#include "kshadowengine.h"
 
 class TQTimer;
 class TQBoxLayout;
@@ -283,8 +282,6 @@
         TQDate clockGetDate();
 
         virtual void updateKickerTip(KickerTip::Data&);
-    
-        KTextShadowEngine *shadowEngine();
 
     k_dcop:
         void reconfigure();
@@ -344,7 +341,6 @@
         TQStringList _remotezonelist;
         KPopupMenu* menu;
         ClockAppletToolTip m_tooltip;
-        KTextShadowEngine *m_shadowEngine;
 };
 
 
Index: kicker/applets/clock/clock.cpp
===================================================================
--- kicker/applets/clock/clock.cpp	(revision 1262481)
+++ kicker/applets/clock/clock.cpp	(working copy)
@@ -277,10 +277,7 @@
 {
     TQRect tr(0, 0, width(), height());
     
-    if (!KickerSettings::transparent())
-        p->drawText(tr, AlignCenter, _timeStr);
-    else
-        _applet->shadowEngine()->drawText(*p, tr, AlignCenter, _timeStr, size());
+    p->drawText(tr, AlignCenter, _timeStr);
 }
 
 //************************************************************
@@ -862,10 +859,7 @@
     else
         tr = TQRect(4, 2, width() - 8, height() - 4);
         
-    if (!KickerSettings::transparent())
-        p->drawText(tr, AlignCenter, _timeStr);
-    else
-        _applet->shadowEngine()->drawText(*p, tr, AlignCenter, _timeStr, size());
+    p->drawText(tr, AlignCenter, _timeStr);
     
     alreadyDrawing = false;
 }
@@ -899,8 +893,7 @@
       _prefs(new Prefs(sharedConfig())),
       zone(new Zone(config())),
       menu(0),
-      m_tooltip(this),
-      m_shadowEngine(0)
+      m_tooltip(this)
 {
     DCOPObject::setObjId("ClockApplet");
     _prefs->readConfig();
@@ -938,7 +931,6 @@
 
 ClockApplet::~ClockApplet()
 {
-    delete m_shadowEngine;
     //reverse for the moment
     KGlobal::locale()->removeCatalogue("clockapplet");
     KGlobal::locale()->removeCatalogue("timezones"); // For time zone translations
@@ -959,15 +951,7 @@
 }
 
 
-KTextShadowEngine *ClockApplet::shadowEngine()
-{
-    if (!m_shadowEngine)
-        m_shadowEngine = new KTextShadowEngine();
 
-    return m_shadowEngine;
-}
-
-
 int ClockApplet::widthForHeight(int h) const
 {
     if (orientation() == Qt::Vertical)
 
