Index: kdebase/kcontrol/iccconfig/iccconfig.cpp
===================================================================
--- kdebase/kcontrol/iccconfig/iccconfig.cpp	(revision 1220926)
+++ kdebase/kcontrol/iccconfig/iccconfig.cpp	(revision 1220927)
@@ -286,7 +286,7 @@
   base->deleteProfileButton->setFixedWidth(90);
 
   XRROutputInfo *output_info;
-  KRandrSimpleAPI *randrsimple = new KRandrSimpleAPI::KRandrSimpleAPI();
+  KRandrSimpleAPI *randrsimple = new KRandrSimpleAPI();
 
   config->setReadDefaults( useDefaults );
 
@@ -356,7 +356,7 @@
 {
 	int i;
 	int j;
-	KRandrSimpleAPI *randrsimple = new KRandrSimpleAPI::KRandrSimpleAPI();
+	KRandrSimpleAPI *randrsimple = new KRandrSimpleAPI();
 
 	// Write system configuration
 	systemconfig->setGroup(NULL);
@@ -408,4 +408,4 @@
      " for a more lifelike and vibrant image.");
 }
 
-#include "iccconfig.moc"
\ No newline at end of file
+#include "iccconfig.moc"
