--- kdebase/ConfigureChecks.cmake.libdetect	2012-09-06 20:55:39.679897201 +0200
+++ kdebase/ConfigureChecks.cmake	2012-09-06 21:06:41.454048511 +0200
@@ -100,22 +100,26 @@
 # xext (kwin/kompmgr)
 if( WITH_XEXT )
   pkg_search_module( XEXT xext )
-  if( XEXT_FOUND )
-    set( HAVE_XEXT 1 )
-  else( XEXT_FOUND )
-    tde_message_fatal( "xext is requested, but was not found on your system" )
+  if( NOT XEXT_FOUND )
+    find_path( XEXT_INCLUDE_DIR  "X11/extensions/Xext.h" )
+    if ( NOT XEXT_INCLUDE_DIR )
+      tde_message_fatal( "xext is requested, but was not found on your system" )
+    endif( )
   endif( )
+  set( HAVE_XEXT 1 )
 endif( )
 
 
 # xtest (kxkb)
 if( WITH_XTEST )
   pkg_search_module( XTEST xtst )
-  if( XTEST_FOUND )
-    set( HAVE_XTEST 1 )
-  else( XTEST_FOUND )
-    tde_message_fatal( "xtest is requested, but was not found on your system" )
+  if( NOT XTEST_FOUND )
+    find_path ( XTEST_INCLUDE_DIR "X11/extensions/XTest.h" )
+    if( NOT XTEST_INCLUDE_DIR )
+      tde_message_fatal( "xtest is requested, but was not found on your system" )
+    endif( )
   endif( )
+  set( HAVE_XTEST 1 )
 endif( )
 
 
@@ -125,12 +129,20 @@
   if( HAVE_XSSLIB )
     pkg_search_module( XSS xscrnsaver )
   else( )
-    check_library_exists( Xext XScreenSaverQueryInfo "" HAVE_XSSLIB )
-    if( HAVE_XSSLIB )
-      pkg_search_module( XSS xext )
+    find_path( XSS_LIBRARY_DIR "libXss.so" )
+    if( XSS_LIBRARY_DIR )
+      set( HAVE_XSSLIB 1 )
+    else( )
+      check_library_exists( Xext XScreenSaverQueryInfo "" HAVE_XSSLIB )
+      if( HAVE_XSSLIB )
+        pkg_search_module( XSS xext )
+      endif( )
     endif( )
   endif( )
-  check_include_file( X11/extensions/scrnsaver.h HAVE_XSCREENSAVER_H )
+  find_path( XSCREENSAVER_INCLUDE_DIR "X11/extensions/scrnsaver.h" )
+  if( XSCREENSAVER_INCLUDE_DIR )
+    set( HAVE_XSCREENSAVER_H 1 )
+  endif( )
   if( HAVE_XSSLIB AND HAVE_XSCREENSAVER_H )
     set( HAVE_XSCREENSAVER 1 )
   else( )
--- kdebase/kcontrol/ConfigureChecks.cmake.ORI	2012-09-06 21:14:07.070747861 +0200
+++ kdebase/kcontrol/ConfigureChecks.cmake	2012-09-06 21:18:33.159501490 +0200
@@ -59,10 +59,13 @@
 
 if( WITH_LIBUSB )
   pkg_search_module( LIBUSB libusb )
-  if( LIBUSB_FOUND )
-    set( HAVE_LIBUSB 1 CACHE INTERNAL "" FORCE )
-  else( )
-    tde_message_fatal( "libusb are requested, but not found on your system" )
+  if( NOT LIBUSB_FOUND )
+    find_path( LIBUSB_LIBRARY_DIR libusb.so )
+    if( LIBUSB_LIBRARY_DIR )
+      set( HAVE_LIBUSB 1 CACHE INTERNAL "" FORCE )
+    else( )
+      #tde_message_fatal( "libusb are requested, but not found on your system" )
+    endif( )
   endif( )
 endif( )
 
--- kdebase/kdm/ConfigureChecks.cmake.ORI	2012-09-06 21:43:53.814013510 +0200
+++ kdebase/kdm/ConfigureChecks.cmake	2012-09-06 21:44:10.768708628 +0200
@@ -112,7 +112,7 @@
 # Xau
 pkg_search_module( XAU xau )
 if( NOT XAU_FOUND )
-  tde_message_fatal( "Xau are required, but not found on your system" )
+  set( XAU_LIBRARIES "-lXau" )
 endif()
 
 
@@ -122,7 +122,7 @@
   if( XDMCP_FOUND )
     set( XDMCP 1 CACHE INTERNAL "" FORCE )
   else()
-    tde_message_fatal( "xdmcp is requested, but was not found on your system" )
+    set( XDMCP_LIBRARIES "-lXdmcp" )
   endif()
 endif()
 
