commit 82d4a938ce57d8611e1ef2e26924bb4367483b0c
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1341027705 -0500

    Disable unneccesary fsync() in cached IMAP handler

diff --git a/kmail/kmfoldercachedimap.cpp b/kmail/kmfoldercachedimap.cpp
index 72f1113..e7009a2 100644
--- a/kmail/kmfoldercachedimap.cpp
+++ b/kmail/kmfoldercachedimap.cpp
@@ -478,7 +478,7 @@ int KMFolderCachedImap::writeUidCache()
     str << lastUid() << endl;
     uidcache.flush();
     if ( uidcache.status() == IO_Ok ) {
-      fsync( uidcache.handle() ); /* this is probably overkill */
+      // fsync( uidcache.handle() ); /* this is probably overkill */
       uidcache.close();
       if ( uidcache.status() == IO_Ok )
         return 0;
