--- kaffeine/kaffeine/src/player-parts/kaffeine-part/Makefile.am.ORI	2012-08-03 12:13:32.063876101 +0200
+++ kaffeine/kaffeine/src/player-parts/kaffeine-part/Makefile.am	2012-08-03 12:13:38.423736088 +0200
@@ -8,5 +8,5 @@
 METASOURCES = AUTO
 
 libkaffeinepart_la_SOURCES = kaffeinepart.cpp mrl.cpp playlistimport.cpp
-libkaffeinepart_la_LIBADD  = $(LIB_KPARTS) $(all_libraries)
+libkaffeinepart_la_LIBADD  = $(LIB_KPARTS) $(all_libraries) $(LIB_QT) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KIO)
 libkaffeinepart_la_LDFLAGS = -no-undefined -avoid-version
--- kaffeine/kaffeine/src/input/disc/plugins/Makefile.am.ORI	2012-08-03 12:17:08.799687224 +0200
+++ kaffeine/kaffeine/src/input/disc/plugins/Makefile.am	2012-08-03 12:17:35.613454250 +0200
@@ -18,7 +18,7 @@
 kaffeineinclude_HEADERS = kaffeineaudioencoder.h
 
 libkaffeineaudioencoder_la_SOURCES = kaffeineaudioencoder.cpp
-libkaffeineaudioencoder_la_LIBADD  = $(LIB_KPARTS)
+libkaffeineaudioencoder_la_LIBADD  = $(LIB_KPARTS) $(LIB_KDEUI) $(LIB_QT) $(LIB_OGGVORBIS)
 libkaffeineaudioencoder_la_LDFLAGS = $(all_libraries) -version-info 0:1:0 -no-undefined
 
 kde_servicetypes_DATA = kaffeineaudioencoder.desktop
--- kaffeine/kaffeine/src/input/dvb/plugins/epg/Makefile.am.ORI	2012-08-03 12:19:42.699882457 +0200
+++ kaffeine/kaffeine/src/input/dvb/plugins/epg/Makefile.am	2012-08-03 12:19:47.451553785 +0200
@@ -10,7 +10,7 @@
 kaffeineinclude_HEADERS = kaffeinedvbevents.h kaffeinedvbsection.h
 
 libkaffeineepgplugin_la_SOURCES = kaffeinedvbevents.cpp kaffeinedvbsection.cpp
-libkaffeineepgplugin_la_LIBADD  = $(LIB_KPARTS)
+libkaffeineepgplugin_la_LIBADD  = $(LIB_KPARTS) $(LIB_QT) $(LIB_KDEUI) $(LIB_KDECORE)
 libkaffeineepgplugin_la_LDFLAGS = $(all_libraries) -version-info 0:1:0 -no-undefined
 
 kde_servicetypes_DATA = kaffeineepgplugin.desktop
--- kaffeine/kaffeine/src/input/dvb/plugins/stream/Makefile.am.ORI	2012-08-03 12:20:56.687973949 +0200
+++ kaffeine/kaffeine/src/input/dvb/plugins/stream/Makefile.am	2012-08-03 12:21:01.669658770 +0200
@@ -10,7 +10,7 @@
 kaffeineinclude_HEADERS = kaffeinedvbplugin.h
 
 libkaffeinedvbplugin_la_SOURCES = kaffeinedvbplugin.cpp
-libkaffeinedvbplugin_la_LIBADD  = $(LIB_KPARTS)
+libkaffeinedvbplugin_la_LIBADD  = $(LIB_KPARTS) $(LIB_KDEUI) $(LIB_QT)
 libkaffeinedvbplugin_la_LDFLAGS = $(all_libraries) -version-info 0:1:0 -no-undefined
 
 kde_servicetypes_DATA = kaffeinedvbplugin.desktop
