--- trinity-kaffeine-14.0.0~pre143+aa821e98/kaffeine/src/input/disc/paranoia.h.ORI	2014-02-06 19:14:04.868957613 +0100
+++ trinity-kaffeine-14.0.0~pre143+aa821e98/kaffeine/src/input/disc/paranoia.h	2014-02-06 19:14:17.705677449 +0100
@@ -33,7 +33,7 @@
 extern "C"
 {
 #ifdef HAVE_LIBCDIO_090
-#include <cdio/paranoiacdda.h>
+#include <cdio/paranoia/cdda.h>
 #include <cdio/paranoia/paranoia.h>
 #else // HAVE_LIBCDIO_090
 #include <cdio/cdda.h>
