commit 701a124908ec0072b12cd3884589a86431f36080
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1324316159 -0600

    Remove additional unneeded tq method conversions

diff --git a/kaffeine/src/input/audiobrowser/googlefetcher.cpp b/kaffeine/src/input/audiobrowser/googlefetcher.cpp
index c74317d..0e76b88 100644
--- a/kaffeine/src/input/audiobrowser/googlefetcher.cpp
+++ b/kaffeine/src/input/audiobrowser/googlefetcher.cpp
@@ -108,7 +108,7 @@ void GoogleFetcher::slotLoadImageURLs(GoogleFetcher::ImageSize size)
 
     kdDebug() << "Performing Google Search: " << url << endl;
 
-    search.load(url.url()+TQString("&start=%1&sa=N").tqarg(galleryPage*20) );
+    search.load(url.url()+TQString("&start=%1&sa=N").arg(galleryPage*20) );
 
     TQString text = search.toString().string();
 
diff --git a/kaffeine/src/input/audiobrowser/googlefetcherdialog.cpp b/kaffeine/src/input/audiobrowser/googlefetcherdialog.cpp
index 5bb169e..ccfea35 100644
--- a/kaffeine/src/input/audiobrowser/googlefetcherdialog.cpp
+++ b/kaffeine/src/input/audiobrowser/googlefetcherdialog.cpp
@@ -87,9 +87,9 @@ GoogleFetcherDialog::~GoogleFetcherDialog()
 void GoogleFetcherDialog::setLayout()
 {
     setCaption(TQString("%1 - %2 (%3)")
-              .tqarg(m_artist)
-              .tqarg(m_album)
-              .tqarg(m_imageList.size()));
+              .arg(m_artist)
+              .arg(m_album)
+              .arg(m_imageList.size()));
 
     m_iconWidget->clear();
     for(uint i = 0; i < m_imageList.size(); i++)
@@ -237,7 +237,7 @@ CoverIconViewItem::~CoverIconViewItem()
 void CoverIconViewItem::imageData(KIO::Job *, const TQByteArray &data)
 {
     int currentSize = m_buffer.size();
-    m_buffer.tqresize(currentSize + data.size(), TQGArray::SpeedOptim);
+    m_buffer.resize(currentSize + data.size(), TQGArray::SpeedOptim);
     memcpy(&(m_buffer.data()[currentSize]), data.data(), data.size());
 }
 
diff --git a/kaffeine/src/input/audiobrowser/playlist.cpp b/kaffeine/src/input/audiobrowser/playlist.cpp
index 3735b61..e67603e 100644
--- a/kaffeine/src/input/audiobrowser/playlist.cpp
+++ b/kaffeine/src/input/audiobrowser/playlist.cpp
@@ -1774,14 +1774,14 @@ void PlayList::updateStatus()
 		TQValueList<MRL>::ConstIterator end(m_queue.end());
 		for (TQValueList<MRL>::ConstIterator it = m_queue.begin(); it != end; ++it)
 			total = total.addSecs(TQTime().secsTo((*it).length()));
-		status = i18n("Queue: %1 Entries, Playtime: %2").tqarg(m_queue.count()).tqarg(total.toString("h:mm:ss"));
+		status = i18n("Queue: %1 Entries, Playtime: %2").arg(m_queue.count()).arg(total.toString("h:mm:ss"));
 	}
 	else
 	{
-		//status = i18n("Entries: %1, Playtime: %2  (Total: %3, %4)").tqarg(TQString::number(m_countVisible)).tqarg(msToTimeString(m_playTimeVisible)).tqarg(TQString::number(m_list->childCount())).tqarg(msToTimeString(m_playTime));
+		//status = i18n("Entries: %1, Playtime: %2  (Total: %3, %4)").arg(TQString::number(m_countVisible)).arg(msToTimeString(m_playTimeVisible)).arg(TQString::number(m_list->childCount())).arg(msToTimeString(m_playTime));
 
 
-		status = i18n("Entries: %1, Playtime: %2").tqarg(TQString::number(m_countVisible)).tqarg(msToTimeString(m_playTimeVisible));
+		status = i18n("Entries: %1, Playtime: %2").arg(TQString::number(m_countVisible)).arg(msToTimeString(m_playTimeVisible));
 	}
 	emit statusBarMessage(status);
 }
@@ -1886,7 +1886,7 @@ void PlayList::saveCurrentPlaylist()
 
 void PlayList::removeCurrentPlaylist()
 {
-	int code = KMessageBox::warningContinueCancel(0, i18n("Remove '%1' from list and from disk?").tqarg(m_playlistSelector->text(m_currentPlaylist)),TQString(),KStdGuiItem::del());
+	int code = KMessageBox::warningContinueCancel(0, i18n("Remove '%1' from list and from disk?").arg(m_playlistSelector->text(m_currentPlaylist)),TQString(),KStdGuiItem::del());
 	if (code == KMessageBox::Continue)
 	{
 		TQString pl = m_playlistDirectory + m_playlistSelector->text(m_currentPlaylist) + ".kaffeine";
diff --git a/kaffeine/src/input/disc/cddb.cpp b/kaffeine/src/input/disc/cddb.cpp
index 9581651..231e008 100644
--- a/kaffeine/src/input/disc/cddb.cpp
+++ b/kaffeine/src/input/disc/cddb.cpp
@@ -302,7 +302,7 @@ bool CDDB::parse_read_resp(TQTextStream *stream, TQTextStream *write_stream)
   for (int i = 0; i < m_tracks; i++)
     {
       if (m_names[i].isEmpty())
-        m_names[i] += i18n("Track %1").tqarg(i);
+        m_names[i] += i18n("Track %1").arg(i);
       m_names[i].replace(TQRegExp("/"), "%2f");
       kdDebug(7101) << "CDDB: found Track " << i+1 << ": `" << m_names[i]
         << "'" << endl;
@@ -328,7 +328,7 @@ bool CDDB::searchLocal(unsigned int id, TQFile *ret_file)
 {
   TQDir dir;
   TQString filename;
-  filename = TQString("%1").tqarg(id, 0, 16).rightJustify(8, '0');
+  filename = TQString("%1").arg(id, 0, 16).rightJustify(8, '0');
   TQStringList::ConstIterator it;
   for (it = cddb_dirs.begin(); it != cddb_dirs.end(); ++it)
     {
diff --git a/kaffeine/src/input/disc/disc.cpp b/kaffeine/src/input/disc/disc.cpp
index e392dca..31bba0c 100644
--- a/kaffeine/src/input/disc/disc.cpp
+++ b/kaffeine/src/input/disc/disc.cpp
@@ -370,7 +370,7 @@ void Disc::trackSelected( TQListViewItem *it )
 	if ( !it )
 		return;
 
-	MRL mrl( TQString("cdda://%1/%2").tqarg( currentDevice ).tqarg( it->text(0).toInt() ) );
+	MRL mrl( TQString("cdda://%1/%2").arg( currentDevice ).arg( it->text(0).toInt() ) );
 	mrl.setTitle( it->text(1) );
 	mrl.setArtist( artistLab->text() );
 	mrl.setAlbum( albumLab->text() );
@@ -465,7 +465,7 @@ void Disc::startCD( const TQString &device, bool rip )
 			for ( i=0; i<para->getTracks(); i++ ) {
 				it = new KListViewItem( list, TQString().sprintf("%02d", i+1), cddb->track( i ), para->trackTime(i) );
 				if ( i==0 ) {
-					mrl.setURL( TQString("cdda://%1/1").tqarg( currentDevice ) );
+					mrl.setURL( TQString("cdda://%1/1").arg( currentDevice ) );
 					mrl.setTitle( cddb->track(i) );
 					mrl.setArtist( artistLab->text() );
 					mrl.setAlbum( albumLab->text() );
@@ -481,7 +481,7 @@ void Disc::startCD( const TQString &device, bool rip )
 				for ( i=0; i<para->getTracks(); i++ ) {
 					it = new KListViewItem( list, TQString().sprintf("%02d", i+1), cddb->track( i ), para->trackTime(i) );
 					if ( i==0 ) {
-						mrl.setURL( TQString("cdda://%1/1").tqarg( currentDevice ) );
+						mrl.setURL( TQString("cdda://%1/1").arg( currentDevice ) );
 						mrl.setTitle( cddb->track(i) );
 						mrl.setArtist( artistLab->text() );
 						mrl.setAlbum( albumLab->text() );
@@ -495,7 +495,7 @@ void Disc::startCD( const TQString &device, bool rip )
 				for ( i=0; i<para->getTracks(); i++ ) {
 					it = new KListViewItem( list, TQString().sprintf("%02d", i+1), i18n("Track")+TQString().sprintf("%02d", i+1), para->trackTime(i) );
 					if ( i==0 ) {
-						mrl.setURL( TQString("cdda://%1/1").tqarg( currentDevice ) );
+						mrl.setURL( TQString("cdda://%1/1").arg( currentDevice ) );
 						mrl.setTitle( i18n("Track")+TQString().sprintf("%02d", i+1) );
 						mrl.setArtist( artistLab->text() );
 						mrl.setAlbum( albumLab->text() );
@@ -531,7 +531,7 @@ void Disc::startDVD( const TQString &device )
 	bool ok=false;
 
 	if ( !device.isEmpty() )
-		mrl.setURL( TQString("dvd://%1").tqarg(device) );
+		mrl.setURL( TQString("dvd://%1").arg(device) );
 	else {
 		DCOPRef mediamanager("kded","mediamanager");
 		DCOPReply reply = mediamanager.call("fullList()");
@@ -555,12 +555,12 @@ void Disc::startDVD( const TQString &device )
 				TQString choice = KInputDialog::getItem( i18n("DVD Video"), i18n("Several DVD Video found. Choose one:"),
 					devList, 0, false, &ok );
 				if ( ok )
-					mrl.setURL( TQString("dvd://%1").tqarg(choice) );
+					mrl.setURL( TQString("dvd://%1").arg(choice) );
 				else
 					return;
 			}
 			else if ( devList.count()==1 )
-				mrl.setURL( TQString("dvd://%1").tqarg(devList[0]) );
+				mrl.setURL( TQString("dvd://%1").arg(devList[0]) );
 			else {
 				//KMessageBox::information( 0, i18n("No DVD Video found."), i18n("Warning") );
 				//return;
@@ -591,7 +591,7 @@ void Disc::startVCD( const TQString &device )
 	bool ok=false;
 
 	if ( !device.isEmpty() )
-		mrl.setURL( TQString("vcd://%1").tqarg(device) );
+		mrl.setURL( TQString("vcd://%1").arg(device) );
 	else {
 		DCOPRef mediamanager("kded","mediamanager");
 		DCOPReply reply = mediamanager.call("fullList()");
@@ -609,12 +609,12 @@ void Disc::startVCD( const TQString &device )
 				TQString choice = KInputDialog::getItem( i18n("VCD-SVCD"), i18n("Several (S)VCD found. Choose one:"),
 					devList, 0, false, &ok );
 				if ( ok )
-					mrl.setURL( TQString("vcd://%1").tqarg(choice) );
+					mrl.setURL( TQString("vcd://%1").arg(choice) );
 				else
 					return;
 			}
 			else if ( devList.count()==1 )
-				mrl.setURL( TQString("vcd://%1").tqarg(devList[0]) );
+				mrl.setURL( TQString("vcd://%1").arg(devList[0]) );
 			else {
 				//KMessageBox::information( 0, i18n("No (S)VCD found."), i18n("Warning") );
 				//return;
@@ -649,7 +649,7 @@ bool Disc::currentTrack( MRL &mrl )
 		return false;
 	while ( it!=0 ) {
 		if ( it->text(0).toInt()==trackCurrent ) {
-			mrl.setURL( TQString("cdda://%1/%2").tqarg( currentDevice ).tqarg( it->text(0).toInt() ) );
+			mrl.setURL( TQString("cdda://%1/%2").arg( currentDevice ).arg( it->text(0).toInt() ) );
 			mrl.setTitle( it->text(1) );
 			mrl.setArtist( artistLab->text() );
 			mrl.setAlbum( albumLab->text() );
@@ -681,7 +681,7 @@ bool Disc::nextTrack( MRL &mrl )
 		return false;
 	while ( it!=0 ) {
 		if ( it->text(0).toInt()==trackCurrent+1 ) {
-			mrl.setURL( TQString("cdda://%1/%2").tqarg( currentDevice ).tqarg( it->text(0).toInt() ) );
+			mrl.setURL( TQString("cdda://%1/%2").arg( currentDevice ).arg( it->text(0).toInt() ) );
 			mrl.setTitle( it->text(1) );
 			mrl.setArtist( artistLab->text() );
 			mrl.setAlbum( albumLab->text() );
@@ -695,7 +695,7 @@ bool Disc::nextTrack( MRL &mrl )
 	}
 
 	it = list->firstChild();
-	mrl.setURL( TQString("cdda://%1/%2").tqarg( currentDevice ).tqarg( it->text(0).toInt() ) );
+	mrl.setURL( TQString("cdda://%1/%2").arg( currentDevice ).arg( it->text(0).toInt() ) );
 	mrl.setTitle( it->text(1) );
 	mrl.setArtist( artistLab->text() );
 	mrl.setAlbum( albumLab->text() );
@@ -718,7 +718,7 @@ bool Disc::previousTrack( MRL &mrl )
 		return false;
 	while ( it!=0 ) {
 		if ( it->text(0).toInt()==trackCurrent-1 ) {
-			mrl.setURL( TQString("cdda://%1/%2").tqarg( currentDevice ).tqarg( it->text(0).toInt() ) );
+			mrl.setURL( TQString("cdda://%1/%2").arg( currentDevice ).arg( it->text(0).toInt() ) );
 			mrl.setTitle( it->text(1) );
 			mrl.setArtist( artistLab->text() );
 			mrl.setAlbum( albumLab->text() );
@@ -746,7 +746,7 @@ bool Disc::trackNumber( int number, MRL &mrl )
 		return false;
 	while ( it!=0 ) {
 		if ( it->text(0).toInt()==number ) {
-			mrl.setURL( TQString("cdda://%1/%2").tqarg( currentDevice ).tqarg( it->text(0).toInt() ) );
+			mrl.setURL( TQString("cdda://%1/%2").arg( currentDevice ).arg( it->text(0).toInt() ) );
 			mrl.setTitle( it->text(1) );
 			mrl.setArtist( artistLab->text() );
 			mrl.setAlbum( albumLab->text() );
diff --git a/kaffeine/src/input/disc/paranoia.cpp b/kaffeine/src/input/disc/paranoia.cpp
index 03f39fa..36b052a 100644
--- a/kaffeine/src/input/disc/paranoia.cpp
+++ b/kaffeine/src/input/disc/paranoia.cpp
@@ -288,14 +288,14 @@ bool Paranoia::loadEncoder( TQWidget *parent )
 
 	KService::Ptr service = KService::serviceByDesktopName( encoderDesktop );
 	if (!service) {
-		KMessageBox::error( parent, i18n("Loading of encoder '%1' failed.").tqarg(encoderDesktop) );
+		KMessageBox::error( parent, i18n("Loading of encoder '%1' failed.").arg(encoderDesktop) );
 		return false;
 	}
 
 	if ( service->serviceTypes().contains("KaffeineAudioEncoder") ) {
 		currentEncoder = KParts::ComponentFactory::createPartInstanceFromService<KaffeineAudioEncoder>(service, 0, service->name().ascii(), 0, 0, 0, &error);
 		if (error > 0) {
-			KMessageBox::error( parent, i18n("Loading of encoder '%1' failed.").tqarg(encoderDesktop) );
+			KMessageBox::error( parent, i18n("Loading of encoder '%1' failed.").arg(encoderDesktop) );
 			return false;
 		}
 		else
diff --git a/kaffeine/src/input/dvb/dvbcam.cpp b/kaffeine/src/input/dvb/dvbcam.cpp
index b430b38..41e9895 100644
--- a/kaffeine/src/input/dvb/dvbcam.cpp
+++ b/kaffeine/src/input/dvb/dvbcam.cpp
@@ -79,7 +79,7 @@ int DvbCam::probe( int adapter, int ca_device )
 	int fdCa;
 	ca_caps_t cap;
 	ca_slot_info_t info;
-	TQString s = TQString("/dev/dvb/adapter%1/ca%2").tqarg( adapter ).tqarg( ca_device );
+	TQString s = TQString("/dev/dvb/adapter%1/ca%2").arg( adapter ).arg( ca_device );
 
 	if ( (fdCa = open( s.ascii(), O_RDWR )) < 0) {
 		fprintf(stderr, "DvbCam::probe(): %s:", s.ascii());
@@ -413,11 +413,11 @@ int DvbCam::aiCallback(void *arg, uint8_t slot_id, uint16_t session_number, uint
 	(void)session_number;
 
 	fprintf(stderr, "CAM Application type: %02x\n", application_type);
-	std->dvbcam->setAppType( TQString("0x%1").tqarg(application_type, 0, 16 ) );
+	std->dvbcam->setAppType( TQString("0x%1").arg(application_type, 0, 16 ) );
 	fprintf(stderr, "CAM Application manufacturer: %04x\n", application_manufacturer);
-	std->dvbcam->setAppManu( TQString("0x%1").tqarg(application_manufacturer, 0, 16 ) );
+	std->dvbcam->setAppManu( TQString("0x%1").arg(application_manufacturer, 0, 16 ) );
 	fprintf(stderr, "CAM Manufacturer code: %04x\n", manufacturer_code);
-	std->dvbcam->setManuCode( TQString("0x%1").tqarg(manufacturer_code, 0, 16 ) );
+	std->dvbcam->setManuCode( TQString("0x%1").arg(manufacturer_code, 0, 16 ) );
 	fprintf(stderr, "CAM Menu string: %.*s\n", menu_string_length, menu_string);
 	TQString s = (const char*)menu_string;
 	s.truncate( menu_string_length );
@@ -531,7 +531,7 @@ int DvbCam::mmi_menu_callback(void *arg, uint8_t slot_id, uint16_t session_numbe
 		fprintf(stderr, "%i. %.*s\n", i+1, items[i].text_length, items[i].text);
 		s = (const char*)items[i].text;
 		s.truncate( items[i].text_length );
-		std->menuList.append( TQString("%1. %2").tqarg(i+1).tqarg(s) );
+		std->menuList.append( TQString("%1. %2").arg(i+1).arg(s) );
 
 	}
 
diff --git a/kaffeine/src/input/dvb/dvbconfig.cpp b/kaffeine/src/input/dvb/dvbconfig.cpp
index c920090..5f9e228 100644
--- a/kaffeine/src/input/dvb/dvbconfig.cpp
+++ b/kaffeine/src/input/dvb/dvbconfig.cpp
@@ -187,10 +187,10 @@ bool DVBconfig::haveDvbDevice()
 		d.setPath( "/dev/dvb/"+list[i]+"/" );
 		flist = d.entryList( "frontend*", TQDir::System, TQDir::Name );
 		for ( j=0; j<(int)flist.count(); j++ ) {
-			fdFrontend = open( TQString("/dev/dvb/%1/%2").tqarg( list[i] ).tqarg( flist[j] ).ascii(), O_RDWR);
+			fdFrontend = open( TQString("/dev/dvb/%1/%2").arg( list[i] ).arg( flist[j] ).ascii(), O_RDWR);
 			if ( fdFrontend>0 ) {
 				if ( (res = ioctl( fdFrontend, FE_GET_INFO, &info ) < 0) )
-					perror( TQString("/dev/dvb/%1/%2 FE_GET_INFO: ").tqarg( list[i] ).tqarg( flist[j] ).ascii() );
+					perror( TQString("/dev/dvb/%1/%2 FE_GET_INFO: ").arg( list[i] ).arg( flist[j] ).ascii() );
 				else {
 					close( fdFrontend );
 					return true;
@@ -229,7 +229,7 @@ void DVBconfig::startup()
 			if ( !probeMfe && t!="frontend0" )
 				continue;
 			t1 = TQTime::currentTime();
-			fdFrontend = open( TQString("/dev/dvb/%1/%2").tqarg( s ).tqarg( t ).ascii(), O_RDWR);
+			fdFrontend = open( TQString("/dev/dvb/%1/%2").arg( s ).arg( t ).ascii(), O_RDWR);
 			if ( fdFrontend>0 ) {
 				if ( !(res = ioctl( fdFrontend, FE_GET_INFO, &info ) < 0) ) {
 					if ( (info.type==FE_OFDM)
@@ -247,7 +247,7 @@ void DVBconfig::startup()
 				close( fdFrontend );
 			}
 			else {
-				perror( TQString("/dev/dvb/%1/%2  %3/%4").tqarg( s ).tqarg( t ).tqarg( errno ).tqarg( -EBUSY ).ascii() );
+				perror( TQString("/dev/dvb/%1/%2  %3/%4").arg( s ).arg( t ).arg( errno ).arg( -EBUSY ).ascii() );
 			}
 		}
 	}
@@ -450,30 +450,30 @@ void DVBconfig::readConfig()
 		shiftDir+= "/";
 	filenameFormat = config->readEntry( "filenameFormat", "%name" );
 	for ( i=0; i<(int)devList.count(); i++ ) {
-		devList.at(i)->source = config->readEntry( TQString("DVB%1").tqarg(i), "" );
-		devList.at(i)->tuningTimeout = config->readNumEntry( TQString("DVB%1_TIMEOUT").tqarg(i), 1500 );
-		devList.at(i)->camMaxService = config->readNumEntry( TQString("DVB%1_CAM_MAX").tqarg(i), 1 );
-		devList.at(i)->priority = config->readNumEntry( TQString("DVB%1_PRIORITY").tqarg(i), 10 );
+		devList.at(i)->source = config->readEntry( TQString("DVB%1").arg(i), "" );
+		devList.at(i)->tuningTimeout = config->readNumEntry( TQString("DVB%1_TIMEOUT").arg(i), 1500 );
+		devList.at(i)->camMaxService = config->readNumEntry( TQString("DVB%1_CAM_MAX").arg(i), 1 );
+		devList.at(i)->priority = config->readNumEntry( TQString("DVB%1_PRIORITY").arg(i), 10 );
 		if ( devList.at(i)->type!=FE_QPSK )
 			continue;
-		devList.at(i)->numLnb = config->readNumEntry( TQString("DVB%1_NLNB").tqarg(i), 1 );
+		devList.at(i)->numLnb = config->readNumEntry( TQString("DVB%1_NLNB").arg(i), 1 );
 		for ( j=0; j<devList.at(i)->numLnb; j++ ) {
-			devList.at(i)->lnb[j].switchFreq = config->readNumEntry( TQString("DVB%1_LNB%2_switch").tqarg(i).tqarg(j), 11700 );
-			devList.at(i)->lnb[j].loFreq = config->readNumEntry( TQString("DVB%1_LNB%2_lo").tqarg(i).tqarg(j), 9750 );
-			devList.at(i)->lnb[j].hiFreq = config->readNumEntry( TQString("DVB%1_LNB%2_hi").tqarg(i).tqarg(j), 10600 );
-			devList.at(i)->lnb[j].rotorType = config->readNumEntry( TQString("DVB%1_LNB%2_rotor").tqarg(i).tqarg(j), 0 );
-			devList.at(i)->lnb[j].source = config->readListEntry( TQString("DVB%1_LNB%2_source").tqarg(i).tqarg(j) );
-			devList.at(i)->lnb[j].position = config->readIntListEntry( TQString("DVB%1_LNB%2_position").tqarg(i).tqarg(j) );
-			devList.at(i)->lnb[j].speed13v = config->readDoubleNumEntry( TQString("DVB%1_LNB%2_speed13v").tqarg(i).tqarg(j), 2.5 );
-			devList.at(i)->lnb[j].speed18v = config->readDoubleNumEntry( TQString("DVB%1_LNB%2_speed18v").tqarg(i).tqarg(j), 1.5 );
+			devList.at(i)->lnb[j].switchFreq = config->readNumEntry( TQString("DVB%1_LNB%2_switch").arg(i).arg(j), 11700 );
+			devList.at(i)->lnb[j].loFreq = config->readNumEntry( TQString("DVB%1_LNB%2_lo").arg(i).arg(j), 9750 );
+			devList.at(i)->lnb[j].hiFreq = config->readNumEntry( TQString("DVB%1_LNB%2_hi").arg(i).arg(j), 10600 );
+			devList.at(i)->lnb[j].rotorType = config->readNumEntry( TQString("DVB%1_LNB%2_rotor").arg(i).arg(j), 0 );
+			devList.at(i)->lnb[j].source = config->readListEntry( TQString("DVB%1_LNB%2_source").arg(i).arg(j) );
+			devList.at(i)->lnb[j].position = config->readIntListEntry( TQString("DVB%1_LNB%2_position").arg(i).arg(j) );
+			devList.at(i)->lnb[j].speed13v = config->readDoubleNumEntry( TQString("DVB%1_LNB%2_speed13v").arg(i).arg(j), 2.5 );
+			devList.at(i)->lnb[j].speed18v = config->readDoubleNumEntry( TQString("DVB%1_LNB%2_speed18v").arg(i).arg(j), 1.5 );
 		}
-		devList.at(i)->secMini = config->readNumEntry( TQString("DVB%1_SEC_MINI").tqarg(i), 0 );
-		devList.at(i)->secTwice = config->readNumEntry( TQString("DVB%1_SEC_TWICE").tqarg(i), 0 );
-		devList.at(i)->doS2 = config->readNumEntry( TQString("DVB%1_DOS2").tqarg(i), 0 );
+		devList.at(i)->secMini = config->readNumEntry( TQString("DVB%1_SEC_MINI").arg(i), 0 );
+		devList.at(i)->secTwice = config->readNumEntry( TQString("DVB%1_SEC_TWICE").arg(i), 0 );
+		devList.at(i)->doS2 = config->readNumEntry( TQString("DVB%1_DOS2").arg(i), 0 );
 	}
 	j = config->readNumEntry( "NumCategories", 0 );
 	for ( i=0; i<j; i++ )
-		categories.append( new Category( config->readEntry( TQString("CategoryName%1").tqarg(i), "" ), config->readEntry( TQString("CategoryIcon%1").tqarg(i), "kaffeine" ) ) );
+		categories.append( new Category( config->readEntry( TQString("CategoryName%1").arg(i), "" ), config->readEntry( TQString("CategoryIcon%1").arg(i), "kaffeine" ) ) );
 	allIcon = config->readEntry( "AllIcon", "kaffeine" );
 	tvIcon = config->readEntry( "TvIcon", "kdvbtv" );
 	radioIcon = config->readEntry( "RadioIcon", "kdvbra" );
@@ -512,31 +512,31 @@ void DVBconfig::saveConfig()
 	config->writeEntry( "SenderPort", senderPort );
 	config->writeEntry( "ProbeMFE", probeMfe );
 	for ( i=0; i<(int)devList.count(); i++ ) {
-		config->writeEntry( TQString("DVB%1").tqarg(i), devList.at(i)->source );
-		config->writeEntry( TQString("DVB%1_TIMEOUT").tqarg(i), devList.at(i)->tuningTimeout );
-		config->writeEntry( TQString("DVB%1_PRIORITY").tqarg(i), devList.at(i)->priority );
-		config->writeEntry( TQString("DVB%1_CAM_MAX").tqarg(i), devList.at(i)->camMaxService );
+		config->writeEntry( TQString("DVB%1").arg(i), devList.at(i)->source );
+		config->writeEntry( TQString("DVB%1_TIMEOUT").arg(i), devList.at(i)->tuningTimeout );
+		config->writeEntry( TQString("DVB%1_PRIORITY").arg(i), devList.at(i)->priority );
+		config->writeEntry( TQString("DVB%1_CAM_MAX").arg(i), devList.at(i)->camMaxService );
 		if ( devList.at(i)->type!=FE_QPSK )
 			continue;
-		config->writeEntry( TQString("DVB%1_NLNB").tqarg(i), devList.at(i)->numLnb );
+		config->writeEntry( TQString("DVB%1_NLNB").arg(i), devList.at(i)->numLnb );
 		for ( j=0; j<devList.at(i)->numLnb; j++ ) {
-			config->writeEntry( TQString("DVB%1_LNB%2_switch").tqarg(i).tqarg(j), devList.at(i)->lnb[j].switchFreq );
-			config->writeEntry( TQString("DVB%1_LNB%2_lo").tqarg(i).tqarg(j), devList.at(i)->lnb[j].loFreq );
-			config->writeEntry( TQString("DVB%1_LNB%2_hi").tqarg(i).tqarg(j), devList.at(i)->lnb[j].hiFreq );
-			config->writeEntry( TQString("DVB%1_LNB%2_rotor").tqarg(i).tqarg(j), devList.at(i)->lnb[j].rotorType );
-			config->writeEntry( TQString("DVB%1_LNB%2_source").tqarg(i).tqarg(j), devList.at(i)->lnb[j].source );
-			config->writeEntry( TQString("DVB%1_LNB%2_position").tqarg(i).tqarg(j), devList.at(i)->lnb[j].position );
-			config->writeEntry( TQString("DVB%1_LNB%2_speed13v").tqarg(i).tqarg(j), devList.at(i)->lnb[j].speed13v );
-			config->writeEntry( TQString("DVB%1_LNB%2_speed18v").tqarg(i).tqarg(j), devList.at(i)->lnb[j].speed18v );
+			config->writeEntry( TQString("DVB%1_LNB%2_switch").arg(i).arg(j), devList.at(i)->lnb[j].switchFreq );
+			config->writeEntry( TQString("DVB%1_LNB%2_lo").arg(i).arg(j), devList.at(i)->lnb[j].loFreq );
+			config->writeEntry( TQString("DVB%1_LNB%2_hi").arg(i).arg(j), devList.at(i)->lnb[j].hiFreq );
+			config->writeEntry( TQString("DVB%1_LNB%2_rotor").arg(i).arg(j), devList.at(i)->lnb[j].rotorType );
+			config->writeEntry( TQString("DVB%1_LNB%2_source").arg(i).arg(j), devList.at(i)->lnb[j].source );
+			config->writeEntry( TQString("DVB%1_LNB%2_position").arg(i).arg(j), devList.at(i)->lnb[j].position );
+			config->writeEntry( TQString("DVB%1_LNB%2_speed13v").arg(i).arg(j), devList.at(i)->lnb[j].speed13v );
+			config->writeEntry( TQString("DVB%1_LNB%2_speed18v").arg(i).arg(j), devList.at(i)->lnb[j].speed18v );
 		}
-		config->writeEntry( TQString("DVB%1_SEC_MINI").tqarg(i), devList.at(i)->secMini );
-		config->writeEntry( TQString("DVB%1_SEC_TWICE").tqarg(i), devList.at(i)->secTwice );
-		config->writeEntry( TQString("DVB%1_DOS2").tqarg(i), devList.at(i)->doS2 );
+		config->writeEntry( TQString("DVB%1_SEC_MINI").arg(i), devList.at(i)->secMini );
+		config->writeEntry( TQString("DVB%1_SEC_TWICE").arg(i), devList.at(i)->secTwice );
+		config->writeEntry( TQString("DVB%1_DOS2").arg(i), devList.at(i)->doS2 );
 	}
 	config->writeEntry( "NumCategories", categories.count() );
 	for ( i=0; i<(int)categories.count(); i++ ) {
-		config->writeEntry( TQString("CategoryName%1").tqarg(i), categories.at(i)->name );
-		config->writeEntry( TQString("CategoryIcon%1").tqarg(i), categories.at(i)->icon );
+		config->writeEntry( TQString("CategoryName%1").arg(i), categories.at(i)->name );
+		config->writeEntry( TQString("CategoryIcon%1").arg(i), categories.at(i)->icon );
 	}
 	config->writeEntry( "AllIcon", allIcon );
 	config->writeEntry( "TvIcon", tvIcon );
@@ -589,7 +589,7 @@ DvbConfigDialog::DvbConfigDialog( DvbPanel *pan, DVBconfig *dc, TQWidget *parent
 	timeoutSpin.setAutoDelete( true );
 
 	for ( i=0; i<(int)dvbConfig->devList.count(); i++ ) {
-		page = addPage( i18n("DVB Device")+" "+TQString("%1:%2").tqarg(dvbConfig->devList.at(i)->adapter).tqarg(dvbConfig->devList.at(i)->tuner), i18n("Device Settings"),
+		page = addPage( i18n("DVB Device")+" "+TQString("%1:%2").arg(dvbConfig->devList.at(i)->adapter).arg(dvbConfig->devList.at(i)->tuner), i18n("Device Settings"),
 			KGlobal::instance()->iconLoader()->loadIcon( "hwinfo", KIcon::NoGroup, KIcon::SizeMedium ) );
 		vb = new TQVBoxLayout( page, 6, 6 );
 		gb = new TQGroupBox( "", page );
diff --git a/kaffeine/src/input/dvb/dvbpanel.cpp b/kaffeine/src/input/dvb/dvbpanel.cpp
index 1b54229..365b7ac 100644
--- a/kaffeine/src/input/dvb/dvbpanel.cpp
+++ b/kaffeine/src/input/dvb/dvbpanel.cpp
@@ -307,7 +307,7 @@ void DvbPanel::diskStatus()
 
 	freemb = (double)(((double)(buf.f_bavail)*(double)(buf.f_bsize))/(1024.0*1024.0));
 	if ( freemb<300 )
-		emit showOSD( i18n("Warning: low disc space")+TQString(" (%1").tqarg((int)freemb)+i18n("MB")+")", 5000, 3 );
+		emit showOSD( i18n("Warning: low disc space")+TQString(" (%1").arg((int)freemb)+i18n("MB")+")", 5000, 3 );
 }
 
 
@@ -649,15 +649,15 @@ void DvbPanel::dumpEvents()
 		for( i=0; i<events.getNSource(); i++ ) {
 			if ( !(esrc=events.getNEventSource( i )) )
 				continue;
-			tt << esrc->getSource()+TQString(" : %1 TS with events.\n").tqarg( esrc->getNTsid() );
+			tt << esrc->getSource()+TQString(" : %1 TS with events.\n").arg( esrc->getNTsid() );
 			for ( m=0; m<esrc->getNTsid(); m++ ) {
 				if ( !(et=esrc->getNEventTsid( m )) )
 					continue;
-				tt << TQString("TSID %1 : %2 services with events\n").tqarg( et->getTsid() ).tqarg( et->getNSid() );
+				tt << TQString("TSID %1 : %2 services with events\n").arg( et->getTsid() ).arg( et->getNSid() );
 				for ( n=0; n<et->getNSid(); n++ ) {
 					if ( !(es=et->getNEventSid( n )) )
 						continue;
-					tt << TQString("     SID %1 : %2 events\n").tqarg( es->getSid() ).tqarg( es->getNDesc() );
+					tt << TQString("     SID %1 : %2 events\n").arg( es->getSid() ).arg( es->getNDesc() );
 					k+= es->getNDesc();
 				}
 			}
@@ -681,10 +681,10 @@ void DvbPanel::dumpEvents()
 						if ( !(desc=esEvents->at( j )) )
 							continue;
 						tt << "Source: "+desc->source+"\n";
-						tt << TQString("Network ID: %1\n").tqarg(desc->nid);
-						tt << TQString("Transport Stream ID: %1\n").tqarg(desc->tsid);
-						tt << TQString("Service ID: %1\n").tqarg(desc->sid);
-						tt << TQString("Event ID: %1\n").tqarg(desc->eid);
+						tt << TQString("Network ID: %1\n").arg(desc->nid);
+						tt << TQString("Transport Stream ID: %1\n").arg(desc->tsid);
+						tt << TQString("Service ID: %1\n").arg(desc->sid);
+						tt << TQString("Event ID: %1\n").arg(desc->eid);
 						tt << "Title: "+desc->title+"\n";
 						tt << "Subtitle: "+desc->subtitle+"\n";
 						tt << desc->startDateTime.toString( "dd-MM-yyyy hh:mm:ss" )+"\n";
diff --git a/kaffeine/src/input/dvb/dvbsi.cpp b/kaffeine/src/input/dvb/dvbsi.cpp
index cce9d0c..7fd8669 100644
--- a/kaffeine/src/input/dvb/dvbsi.cpp
+++ b/kaffeine/src/input/dvb/dvbsi.cpp
@@ -620,7 +620,7 @@ bool DVBsi::tablePAT( unsigned char *buf )
 			desc->tp.tsid = tsid;
 			desc->sid = sid;
 			desc->pmtpid = pmt;
-			desc->name = TQString("TSID:%1-SID:%2").tqarg(tsid).tqarg(sid);
+			desc->name = TQString("TSID:%1-SID:%2").arg(tsid).arg(sid);
 		}
 	}
 
@@ -854,7 +854,7 @@ bool DVBsi::parseVCT( int pid, int tid, int timeout, int sid )
 
 		ChannelDesc *desc = new ChannelDesc();
 		desc->tp.tsid = channel_tsid;
-		desc->name = TQString("%1-%2 %3").tqarg(major_channel_num).tqarg(minor_channel_num).tqarg(short_name);
+		desc->name = TQString("%1-%2 %3").arg(major_channel_num).arg(minor_channel_num).arg(short_name);
 		desc->sid = program_number;
 		if (access_controlled == 1)
 			desc->fta = 1;
diff --git a/kaffeine/src/input/dvb/dvbstream.cpp b/kaffeine/src/input/dvb/dvbstream.cpp
index 4ef290b..be8ae7d 100644
--- a/kaffeine/src/input/dvb/dvbstream.cpp
+++ b/kaffeine/src/input/dvb/dvbstream.cpp
@@ -65,9 +65,9 @@ DvbStream::DvbStream( Device *d, const TQString &charset, EventTable *et )
 	fdFrontend = fdDvr = 0;
 	ndmx = 0;
 	currentTransponder = Transponder();
-	frontendName = TQString("/dev/dvb/adapter%1/frontend%2").tqarg( dvbDevice->adapter ).tqarg( dvbDevice->tuner );
-	dvrName = TQString("/dev/dvb/adapter%1/dvr%2").tqarg( dvbDevice->adapter ).tqarg( dvbDevice->tuner );
-	demuxName = TQString("/dev/dvb/adapter%1/demux%2").tqarg( dvbDevice->adapter ).tqarg( dvbDevice->tuner );
+	frontendName = TQString("/dev/dvb/adapter%1/frontend%2").arg( dvbDevice->adapter ).arg( dvbDevice->tuner );
+	dvrName = TQString("/dev/dvb/adapter%1/dvr%2").arg( dvbDevice->adapter ).arg( dvbDevice->tuner );
+	demuxName = TQString("/dev/dvb/adapter%1/demux%2").arg( dvbDevice->adapter ).arg( dvbDevice->tuner );
 
 	out.setAutoDelete( true );
 
diff --git a/kaffeine/src/input/dvb/kgradprogress.cpp b/kaffeine/src/input/dvb/kgradprogress.cpp
index 7898f84..07030e4 100644
--- a/kaffeine/src/input/dvb/kgradprogress.cpp
+++ b/kaffeine/src/input/dvb/kgradprogress.cpp
@@ -43,7 +43,7 @@ void KGradProgress::setProgress( int progress )
 {
 	current = progress;
 	draw();
-	tqrepaint(false);
+	repaint(false);
 }
 
 
diff --git a/kaffeine/src/input/dvb/plugins/epg/kaffeinedvbsection.cpp b/kaffeine/src/input/dvb/plugins/epg/kaffeinedvbsection.cpp
index 44619b7..3c49848 100644
--- a/kaffeine/src/input/dvb/plugins/epg/kaffeinedvbsection.cpp
+++ b/kaffeine/src/input/dvb/plugins/epg/kaffeinedvbsection.cpp
@@ -70,7 +70,7 @@ KaffeineDVBsection::~KaffeineDVBsection()
 bool KaffeineDVBsection::setFilter( int pid, int tid, int timeout, bool checkcrc )
 {
 	struct dmx_sct_filter_params sctfilter;
-	TQString demuxer = TQString("/dev/dvb/adapter%1/demux%2").tqarg( adapter ).tqarg( tuner );
+	TQString demuxer = TQString("/dev/dvb/adapter%1/demux%2").arg( adapter ).arg( tuner );
 
 	if ((fdDemux = open( demuxer.ascii(), O_RDWR | O_NONBLOCK )) < 0) {
 		perror ("open failed");
diff --git a/kaffeine/src/input/dvb/scandialog.cpp b/kaffeine/src/input/dvb/scandialog.cpp
index 7f760e6..9b68d93 100644
--- a/kaffeine/src/input/dvb/scandialog.cpp
+++ b/kaffeine/src/input/dvb/scandialog.cpp
@@ -71,7 +71,7 @@ ScanDialog::ScanDialog( TQPtrList<DvbStream> *d, TQPtrList<ChannelDesc> *ch, TQS
 		tmp = dvb->at(i)->getSources( true );
 		for ( j=0; j<(int)tmp.count(); j++ ) {
 			if ( tmp[j].startsWith( "AUTO" ) )
-				list.append( TQString("AUTO(%1:%2)").tqarg(dvb->at(i)->getAdapter()).tqarg(dvb->at(i)->getTuner()) );
+				list.append( TQString("AUTO(%1:%2)").arg(dvb->at(i)->getAdapter()).arg(dvb->at(i)->getTuner()) );
 			else
 				list.append( tmp[j] );
 		}
@@ -614,25 +614,25 @@ bool ScanDialog::getTransData()
 		int i;
 		for ( i=177; i<227; i+=7 ) {
 			if ( offset07->isChecked() ) {
-				s = TQString("T %1 7MHz AUTO AUTO AUTO AUTO AUTO AUTO").tqarg( (i*1000000)+500000 );
+				s = TQString("T %1 7MHz AUTO AUTO AUTO AUTO AUTO AUTO").arg( (i*1000000)+500000 );
 				parseTp( s, ds->getType(), "" );
 			}
 			if ( offset125p->isChecked() ) {
-				s = TQString("T %1 7MHz AUTO AUTO AUTO AUTO AUTO AUTO").tqarg( (i*1000000)+500000+125000 );
+				s = TQString("T %1 7MHz AUTO AUTO AUTO AUTO AUTO AUTO").arg( (i*1000000)+500000+125000 );
 				parseTp( s, ds->getType(), "" );
 			}
 		}
 		for ( i=474; i<859; i+=8 ) {
 			if ( offset167m->isChecked() ) {
-				s = TQString("T %1 8MHz AUTO AUTO AUTO AUTO AUTO AUTO").tqarg( (i*1000000)-167000 );
+				s = TQString("T %1 8MHz AUTO AUTO AUTO AUTO AUTO AUTO").arg( (i*1000000)-167000 );
 				parseTp( s, ds->getType(), "" );
 			}
 			if ( offset0->isChecked() ) {
-				s = TQString("T %1 8MHz AUTO AUTO AUTO AUTO AUTO AUTO").tqarg( i*1000000 );
+				s = TQString("T %1 8MHz AUTO AUTO AUTO AUTO AUTO AUTO").arg( i*1000000 );
 				parseTp( s, ds->getType(), "" );
 			}
 			if ( offset167p->isChecked() ) {
-				s = TQString("T %1 8MHz AUTO AUTO AUTO AUTO AUTO AUTO").tqarg( (i*1000000)+167000 );
+				s = TQString("T %1 8MHz AUTO AUTO AUTO AUTO AUTO AUTO").arg( (i*1000000)+167000 );
 				parseTp( s, ds->getType(), "" );
 			}
 		}
@@ -1010,7 +1010,7 @@ void ScanDialog::addFound( ChannelDesc *chan, bool scan )
 	int i;
 	DListViewItem *it;
 
-	it = new DListViewItem( foundList, chan, TQString( "%1").tqarg( chan->tp.snr ), chan->name );
+	it = new DListViewItem( foundList, chan, TQString( "%1").arg( chan->tp.snr ), chan->name );
 
 	if ( chan->type==1 ) {
 		if ( scan )
@@ -1030,7 +1030,7 @@ void ScanDialog::addFound( ChannelDesc *chan, bool scan )
 	}
 
 	if ( scan )
-		progressLab->setText( TQString( i18n("Found: %1 TV - %2 radio") ).tqarg(ntv).tqarg(nradio) );
+		progressLab->setText( TQString( i18n("Found: %1 TV - %2 radio") ).arg(ntv).arg(nradio) );
 	if ( chan->provider.isEmpty() )
 		return;
 	for ( i=0; i<providerComb->count(); i++ ) {
diff --git a/kaffeine/src/instwizard.cpp b/kaffeine/src/instwizard.cpp
index 8e2f412..0bc4a31 100644
--- a/kaffeine/src/instwizard.cpp
+++ b/kaffeine/src/instwizard.cpp
@@ -49,7 +49,7 @@
 
 void InstWizard::internalWizard()
 {
-	setCaption(i18n("Kaffeine %1 Installation Wizard").tqarg(KAFFEINE_VERSION));
+	setCaption(i18n("Kaffeine %1 Installation Wizard").arg(KAFFEINE_VERSION));
 	resize(400,400);
 
 	/************** installation check ************************/
@@ -79,7 +79,7 @@ void InstWizard::internalWizard()
 	if (KDE::version() >= KDE_MAKE_VERSION(3,2,0))
 		info << "<font color=\"DarkGreen\">" << i18n("Ok.") << "</font>";
 	else
-		info << "<font color=\"DarkRed\">" << i18n("Kaffeine requires KDE >= %1.").tqarg("3.2") << "</font>";
+		info << "<font color=\"DarkRed\">" << i18n("Kaffeine requires KDE >= %1.").arg("3.2") << "</font>";
 
 	//XINE-LIB
 	//Should engine depending stuff be here?
@@ -96,13 +96,13 @@ void InstWizard::internalWizard()
 
 		if ((major == 1) && (minor == 0) && (sub == 0) && (xineVersion.contains("rc", false)))
 		{
-			info << "<font color=\"DarkRed\">" << i18n("Kaffeine requires xine-lib >= %1. Download the latest version here:").tqarg("1.0") << " <a href=\"http://www.xinehq.de\">http://www.xinehq.de</a>.</font>";
+			info << "<font color=\"DarkRed\">" << i18n("Kaffeine requires xine-lib >= %1. Download the latest version here:").arg("1.0") << " <a href=\"http://www.xinehq.de\">http://www.xinehq.de</a>.</font>";
 		}
 		else
 			info << "<font color=\"DarkGreen\">" << i18n("Ok.") << "</font>";
 	}
 	else
-		info << "<font color=\"DarkRed\">" << i18n("Kaffeine requires xine-lib >= %1. Download the latest version here:").tqarg("1.0")
+		info << "<font color=\"DarkRed\">" << i18n("Kaffeine requires xine-lib >= %1. Download the latest version here:").arg("1.0")
 		<< " <a href=\"http://www.xinehq.de\">http://www.xinehq.de</a>.</font>";
 	*/
 	//WIN32-CODECS
diff --git a/kaffeine/src/kaffeine.cpp b/kaffeine/src/kaffeine.cpp
index 5b188da..9efebc9 100644
--- a/kaffeine/src/kaffeine.cpp
+++ b/kaffeine/src/kaffeine.cpp
@@ -154,8 +154,8 @@ Kaffeine::Kaffeine() : DCOPObject("KaffeineIface"),
 	//createStandardStatusBarAction();
 	createGUI("kaffeineui.rc");
 
-	//statusBar()->insertItem(i18n("Entries: %1, Playtime: %2  (Total: %3, %4)").tqarg("0").tqarg("00:00:00").tqarg("0").tqarg("00:00:00"), 9, 0, true);
-	//statusBar()->insertItem(i18n("Entries: %1, Playtime: %2").tqarg("0").tqarg("0:00:00"), 9, 0, true);
+	//statusBar()->insertItem(i18n("Entries: %1, Playtime: %2  (Total: %3, %4)").arg("0").arg("00:00:00").arg("0").arg("00:00:00"), 9, 0, true);
+	//statusBar()->insertItem(i18n("Entries: %1, Playtime: %2").arg("0").arg("0:00:00"), 9, 0, true);
 	//statusBar()->insertItem(i18n("No player"), 10, 0, true);
 
 	TQString stamp =  locateLocal("appdata", "wizard_stamp_v0.7.1");
@@ -367,7 +367,7 @@ void Kaffeine::slotLoadPart(const TQString& desktopName)
 	KService::Ptr service = KService::serviceByDesktopName(desktopName);
 	if (!service)
 	{
-		KMessageBox::detailedError(this, i18n("Loading of player part '%1' failed.").tqarg(desktopName), i18n("%1 not found in search path.").tqarg(TQString(desktopName)+ ".desktop"));
+		KMessageBox::detailedError(this, i18n("Loading of player part '%1' failed.").arg(desktopName), i18n("%1 not found in search path.").arg(TQString(desktopName)+ ".desktop"));
 		return;
 	}
 
@@ -387,7 +387,7 @@ void Kaffeine::slotLoadPart(const TQString& desktopName)
 		int error = 0;
 		m_mediaPart = KParts::ComponentFactory::createPartInstanceFromService<KaffeinePart>(service, m_playerContainer, service->name().ascii(), TQT_TQOBJECT(this), 0, m_engineParameters, &error);
 		if (error > 0) {
-			KMessageBox::detailedError(this, i18n("Loading of player part '%1' failed.").tqarg(service->name()), KLibLoader::self()->lastErrorMessage());
+			KMessageBox::detailedError(this, i18n("Loading of player part '%1' failed.").arg(service->name()), KLibLoader::self()->lastErrorMessage());
 		}
 		else {
 			connect(m_mediaPart, TQT_SIGNAL(setWindowCaption(const TQString&)), TQT_TQOBJECT(this), TQT_SLOT(slotChangeCaption(const TQString&)));
@@ -455,7 +455,7 @@ void Kaffeine::slotLoadingCanceled(const TQString& message)
 		delete m_mediaPart;
 		m_mediaPart = NULL;
 	}
-	KMessageBox::detailedError(this, i18n("Loading of player part '%1' failed.").tqarg(name), message);
+	KMessageBox::detailedError(this, i18n("Loading of player part '%1' failed.").arg(name), message);
 }
 
 void Kaffeine::slotPlaybackFailed()
@@ -1439,7 +1439,7 @@ TQString Kaffeine::askForOtherDevice(const TQString& type)
 	TQVBox* page = dialog->makeVBoxMainWidget();
 	page->setSpacing(5);
 	page->setMargin(5);
-	new TQLabel(i18n("No %1 in drive, or wrong path to device.").tqarg(type), page);
+	new TQLabel(i18n("No %1 in drive, or wrong path to device.").arg(type), page);
 	new TQLabel(TQString("\n") + i18n("Please select correct drive:"), page);
 	DrivesCombo* drives = new DrivesCombo(page);
 
diff --git a/kaffeine/src/kmultitabbar.cpp b/kaffeine/src/kmultitabbar.cpp
index 308bfa5..ecfa2c5 100644
--- a/kaffeine/src/kmultitabbar.cpp
+++ b/kaffeine/src/kmultitabbar.cpp
@@ -111,7 +111,7 @@ void KMultiTabBarInternal::setStyle(enum KMultiTabBar::KMultiTabBarStyle style)
 		mainLayout->setAutoAdd(true);
 
 	}
-        viewport()->tqrepaint();
+        viewport()->repaint();
 }
 
 void KMultiTabBarInternal::drawContents ( TQPainter * paint, int clipx, int clipy, int clipw, int cliph )
@@ -189,8 +189,8 @@ void KMultiTabBarInternal::mousePressEvent(TQMouseEvent *ev)
 
 void KMultiTabBarInternal::resizeEvent(TQResizeEvent *ev) {
 /*	kdDebug()<<"KMultiTabBarInternal::resizeEvent"<<endl;
-	kdDebug()<<"KMultiTabBarInternal::resizeEvent - box tqgeometry"<<box->tqgeometry()<<endl;
-	kdDebug()<<"KMultiTabBarInternal::resizeEvent - tqgeometry"<<tqgeometry()<<endl;*/
+	kdDebug()<<"KMultiTabBarInternal::resizeEvent - box geometry"<<box->geometry()<<endl;
+	kdDebug()<<"KMultiTabBarInternal::resizeEvent - geometry"<<geometry()<<endl;*/
 	if (ev) TQScrollView::resizeEvent(ev);
 
 	if ( (m_style==KMultiTabBar::KDEV3) ||
@@ -379,7 +379,7 @@ void KMultiTabBarInternal::setPosition(enum KMultiTabBar::KMultiTabBarPosition p
 	m_position=pos;
 	for (uint i=0;i<m_tabs.count();i++)
 		m_tabs.at(i)->setTabsPosition(m_position);
-	viewport()->tqrepaint();
+	viewport()->repaint();
 }
 
 
@@ -436,13 +436,13 @@ void KMultiTabBarButton::slotClicked()
 void KMultiTabBarButton::setPosition(KMultiTabBar::KMultiTabBarPosition pos)
 {
 	m_position=pos;
-	tqrepaint();
+	repaint();
 }
 
 void KMultiTabBarButton::setStyle(KMultiTabBar::KMultiTabBarStyle style)
 {
 	m_style=style;
-	tqrepaint();
+	repaint();
 }
 
 void KMultiTabBarButton::hideEvent( TQHideEvent* he) {
@@ -528,7 +528,7 @@ void KMultiTabBarTab::setTabsPosition(KMultiTabBar::KMultiTabBarPosition pos)
 	}
 
 	setPosition(pos);
-//	tqrepaint();
+//	repaint();
 }
 
 void KMultiTabBarTab::setIcon(const TQString& icon)
@@ -992,7 +992,7 @@ void KMultiTabBar::fontChange(const TQFont& /* oldFont */)
 {
 	for (uint i=0;i<tabs()->count();i++)
 		tabs()->at(i)->resize();
-	tqrepaint();
+	repaint();
 }
 
 TQPtrList<KMultiTabBarTab>* KMultiTabBar::tabs() {return m_internal->tabs();}
diff --git a/kaffeine/src/player-parts/gstreamer-part/gstreamer_part.cpp b/kaffeine/src/player-parts/gstreamer-part/gstreamer_part.cpp
index 2255e67..fb6e7a3 100644
--- a/kaffeine/src/player-parts/gstreamer-part/gstreamer_part.cpp
+++ b/kaffeine/src/player-parts/gstreamer-part/gstreamer_part.cpp
@@ -737,7 +737,7 @@ void GStreamerPart::setAudioSink( TQString sinkName )
 {
 	GstElement* sink = gst_element_factory_make(sinkName.ascii(), "audiosink");
 	if ( !sink ) {
-		KMessageBox::error(0, i18n("Error: Can't init new Audio Driver %1 - using %2!").tqarg(sinkName).tqarg(m_audioSinkName));
+		KMessageBox::error(0, i18n("Error: Can't init new Audio Driver %1 - using %2!").arg(sinkName).arg(m_audioSinkName));
 		return;
 	}
 
@@ -785,7 +785,7 @@ bool GStreamerPart::initGStreamer()
 	 */
 	m_audiosink = gst_element_factory_make( m_audioSinkName.ascii(), "audiosink" );
 	if ( !m_audiosink ) {
-		KMessageBox::error(0, i18n("Can't init Audio Driver '%1' - trying another one...").tqarg(m_audioSinkName));
+		KMessageBox::error(0, i18n("Can't init Audio Driver '%1' - trying another one...").arg(m_audioSinkName));
 		if ( (m_audiosink = gst_element_factory_make("alsasink", "audiosink")) != NULL )
 			kdDebug() << "GStreamerPart: Using audio-driver: alsasink" << endl;
 		else {
@@ -809,7 +809,7 @@ bool GStreamerPart::initGStreamer()
 	 */
 	m_videosink = gst_element_factory_make( m_videoSinkName.ascii(), "videosink" );
 	if ( !m_videosink ) {
-		KMessageBox::error(0, i18n("Can't init Video Driver '%1' - trying another one...").tqarg(m_videoSinkName));
+		KMessageBox::error(0, i18n("Can't init Video Driver '%1' - trying another one...").arg(m_videoSinkName));
 		if ((m_videosink = gst_element_factory_make("xvimagesink", "videosink")) != NULL)
 			kdDebug() << "GStreamerPart: Using video-driver: xvimagesink" << endl;
 		else {
diff --git a/kaffeine/src/player-parts/gstreamer-part/video.cpp b/kaffeine/src/player-parts/gstreamer-part/video.cpp
index eb6203f..96752aa 100644
--- a/kaffeine/src/player-parts/gstreamer-part/video.cpp
+++ b/kaffeine/src/player-parts/gstreamer-part/video.cpp
@@ -134,7 +134,7 @@ void VideoWindow::newCapsset( const GstCaps *caps )
 	TQSize frame = getFrameSize();
 	correctByAspectRatio( frame );
 	emit signalNewFrameSize( frame );
-	//set correct tqgeometry
+	//set correct geometry
 	setGeometry();
 }
 
@@ -177,7 +177,7 @@ void VideoWindow::newState()
 	TQSize frame = getFrameSize();
 	correctByAspectRatio( frame );
 	emit signalNewFrameSize( frame );
-	//set correct tqgeometry
+	//set correct geometry
 	setGeometry();
 }
 
diff --git a/kaffeine/src/player-parts/xine-part/kxinewidget.cpp b/kaffeine/src/player-parts/xine-part/kxinewidget.cpp
index 58a9818..acc9937 100644
--- a/kaffeine/src/player-parts/xine-part/kxinewidget.cpp
+++ b/kaffeine/src/player-parts/xine-part/kxinewidget.cpp
@@ -153,14 +153,14 @@ KXineWidget::~KXineWidget()
 	{
 		debugOut("Unwire audio filters");
 		unwireAudioFilters();
-		debugOut(TQString("Dispose visual plugin: %1").tqarg(m_visualPluginName ));
+		debugOut(TQString("Dispose visual plugin: %1").arg(m_visualPluginName ));
 		delete m_visualPlugin;
 		m_visualPlugin = NULL;
 	}
 #else
 	if (m_xinePost)
 	{
-		debugOut(TQString("Dispose visual plugin: %1").tqarg(m_visualPluginName));
+		debugOut(TQString("Dispose visual plugin: %1").arg(m_visualPluginName));
 		m_postAudioSource = xine_get_audio_source(m_xineStream);
 		xine_post_wire_audio_port(m_postAudioSource, m_audioDriver);
 		xine_post_dispose(m_xineEngine, m_xinePost);
@@ -279,7 +279,7 @@ void KXineWidget::saveXineConfig()
 		xine_config_update_entry (m_xineEngine, &config);
 	}
 
-	debugOut(TQString("Save xine config to: %1").tqarg(m_configFilePath));
+	debugOut(TQString("Save xine config to: %1").arg(m_configFilePath));
 	xine_config_save(m_xineEngine, m_configFilePath.ascii());
 }
 
@@ -331,7 +331,7 @@ void KXineWidget::frameOutputCallback(void* p, int video_width, int video_height
 	/* frame size changed */
 	if ( (video_width != vw->m_videoFrameWidth) || (video_height != vw->m_videoFrameHeight) )
 	{
-		debugOut(TQString("New video frame size: %1x%2 - aspect ratio: %3").tqarg(video_width).tqarg(video_height).tqarg(video_aspect));
+		debugOut(TQString("New video frame size: %1x%2 - aspect ratio: %3").arg(video_width).arg(video_height).arg(video_aspect));
 		vw->m_videoFrameWidth = video_width;
 		vw->m_videoFrameHeight = video_height;
 		vw->m_videoAspect = video_aspect;
@@ -343,7 +343,7 @@ void KXineWidget::frameOutputCallback(void* p, int video_width, int video_height
 		{
 			vw->m_newParentSize = vw->parentWidget()->size() - TQSize((vw->width() - video_width), vw->height() - video_height);
 
-			debugOut(TQString("Resize video window to: %1x%2").tqarg(vw->m_newParentSize.width()).tqarg(vw->m_newParentSize.height()));
+			debugOut(TQString("Resize video window to: %1x%2").arg(vw->m_newParentSize.width()).arg(vw->m_newParentSize.height()));
 
 			/* we should not do a resize() inside a xine thread,
 			   but post an event to the main thread */
@@ -391,7 +391,7 @@ void KXineWidget::xineEventListener(void *p, const xine_event_t* xineEvent)
 			channels = xine_get_stream_info(vw->m_xineStream, XINE_STREAM_INFO_MAX_AUDIO_CHANNEL);
 			for(i = 0; i < channels; i++)
 			{
-				slang = TQString("%1.").tqarg(i+1);
+				slang = TQString("%1.").arg(i+1);
 				if (xine_get_audio_lang(vw->m_xineStream, i, lang))
 					slang += lang;
 				tmp << slang;
@@ -415,7 +415,7 @@ void KXineWidget::xineEventListener(void *p, const xine_event_t* xineEvent)
 			channels = xine_get_stream_info(vw->m_xineStream, XINE_STREAM_INFO_MAX_SPU_CHANNEL);
 			for(i = 0; i<channels; i++)
 			{
-				slang = TQString("%1.").tqarg(i+1);
+				slang = TQString("%1.").arg(i+1);
 				if (xine_get_spu_lang(vw->m_xineStream, i, lang))
 					slang += lang;
 				tmp << slang;
@@ -472,7 +472,7 @@ void KXineWidget::xineEventListener(void *p, const xine_event_t* xineEvent)
 			debugOut("xine event: dropped frames");
 			xine_dropped_frames_t* dropped = (xine_dropped_frames_t*)xineEvent->data;
 
-			warningOut(TQString("Skipped frames: %1 - discarded frames: %2").tqarg(dropped->skipped_frames/10).tqarg(dropped->discarded_frames/10));
+			warningOut(TQString("Skipped frames: %1 - discarded frames: %2").arg(dropped->skipped_frames/10).arg(dropped->discarded_frames/10));
 
 			break;
 		}
@@ -752,7 +752,7 @@ void KXineWidget::timerEvent( TQTimerEvent* tevent )
 			else
 			{
 				//restart
-				warningOut(TQString("Message: '%1' was blocked!").tqarg(m_xineMessage));
+				warningOut(TQString("Message: '%1' was blocked!").arg(m_xineMessage));
 				m_recentMessagesTimer.start(1500);
 			}
 			break;
@@ -795,7 +795,7 @@ void KXineWidget::videoDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 
 	int pos, time, length;
 
-	debugOut(TQString("New video driver: %1").tqarg(entry->enum_values[entry->num_value]));
+	debugOut(TQString("New video driver: %1").arg(entry->enum_values[entry->num_value]));
 
 	if (vw->m_osd)
 	{
@@ -835,7 +835,7 @@ void KXineWidget::videoDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 	vw->unwireAudioFilters();
 	if (vw->m_visualPlugin)
 	{
-		debugOut(TQString("Dispose visual plugin: %1").tqarg(vw->m_visualPluginName));
+		debugOut(TQString("Dispose visual plugin: %1").arg(vw->m_visualPluginName));
 		delete vw->m_visualPlugin;
 		vw->m_visualPlugin = NULL;
 	}
@@ -855,7 +855,7 @@ void KXineWidget::videoDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 
 	if (!vw->m_videoDriver)
 	{
-		vw->m_xineError = i18n("Error: Can't init new Video Driver %1 - using %2!").tqarg(entry->enum_values[entry->num_value]).tqarg(vw->m_videoDriverName);
+		vw->m_xineError = i18n("Error: Can't init new Video Driver %1 - using %2!").arg(entry->enum_values[entry->num_value]).arg(vw->m_videoDriverName);
 		TQApplication::postEvent(vw, new TQTimerEvent( TIMER_EVENT_NEW_XINE_ERROR));
 		playing = false;
 		vw->m_videoDriver = xine_open_video_driver(vw->m_xineEngine,
@@ -869,7 +869,7 @@ void KXineWidget::videoDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 	else
 	{
 		vw->m_videoDriverName = entry->enum_values[entry->num_value];
-		vw->m_statusString = i18n("Using Video Driver: %1").tqarg(vw->m_videoDriverName);
+		vw->m_statusString = i18n("Using Video Driver: %1").arg(vw->m_videoDriverName);
 		TQApplication::postEvent(vw, new TQTimerEvent(TIMER_EVENT_NEW_STATUS));
 	}
 
@@ -902,7 +902,7 @@ void KXineWidget::audioDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 
 	int pos, time, length;
 
-	debugOut(TQString("New audio driver: %1").tqarg(entry->enum_values[entry->num_value]));
+	debugOut(TQString("New audio driver: %1").arg(entry->enum_values[entry->num_value]));
 
 	if (vw->m_osd)
 	{
@@ -931,7 +931,7 @@ void KXineWidget::audioDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 	vw->unwireAudioFilters();
 	if (vw->m_visualPlugin)
 	{
-		debugOut(TQString("Dispose visual plugin: %1").tqarg(vw->m_visualPluginName));
+		debugOut(TQString("Dispose visual plugin: %1").arg(vw->m_visualPluginName));
 		delete vw->m_visualPlugin;
 		vw->m_visualPlugin = NULL;
 	}
@@ -945,7 +945,7 @@ void KXineWidget::audioDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 
 	if (!vw->m_audioDriver)
 	{
-		vw->m_xineError = i18n("Error: Can't init new Audio Driver %1 - using %2!").tqarg(entry->enum_values[entry->num_value]).tqarg(vw->m_audioDriverName);
+		vw->m_xineError = i18n("Error: Can't init new Audio Driver %1 - using %2!").arg(entry->enum_values[entry->num_value]).arg(vw->m_audioDriverName);
 		TQApplication::postEvent(vw, new TQTimerEvent( TIMER_EVENT_NEW_XINE_ERROR));
 		playing = false;
 		vw->m_audioDriver = xine_open_audio_driver(vw->m_xineEngine, vw->m_audioDriverName.ascii(), NULL);
@@ -953,7 +953,7 @@ void KXineWidget::audioDriverChangedCallback(void* p, xine_cfg_entry_t* entry)
 	else
 	{
 		vw->m_audioDriverName = entry->enum_values[entry->num_value];
-		vw->m_statusString = i18n("Using Audio Driver: %1").tqarg(vw->m_audioDriverName);
+		vw->m_statusString = i18n("Using Audio Driver: %1").arg(vw->m_audioDriverName);
 		TQApplication::postEvent(vw, new TQTimerEvent(TIMER_EVENT_NEW_STATUS));
 	}
 
@@ -1148,18 +1148,18 @@ bool KXineWidget::initXine()
 		int count = 1;
 		debugOut("Xinerama extension present");
 		XineramaQueryScreens(x11Display(), &count);
-		debugOut(TQString("%1 screens detected").tqarg(count));
+		debugOut(TQString("%1 screens detected").arg(count));
 		if (count > 1)
 			// multihead -> assuming square pixels
 			m_displayRatio = 1.0;
 	}
 #endif
 
-	debugOut(TQString("Display aspect ratio (v/h): %1").tqarg(m_displayRatio));*/
+	debugOut(TQString("Display aspect ratio (v/h): %1").arg(m_displayRatio));*/
 
 	/**** INIT XINE ENGINE ****/
 
-	debugOut(TQString("Using xine version %1").tqarg(xine_get_version_string()));
+	debugOut(TQString("Using xine version %1").arg(xine_get_version_string()));
 
 	m_xineEngine = xine_new();
 	if (!m_xineEngine)
@@ -1209,7 +1209,7 @@ bool KXineWidget::initXine()
 	else
 		m_audioDriverName = m_audioChoices[i];
 
-	debugOut(TQString("Use audio driver %1").tqarg(m_audioDriverName));
+	debugOut(TQString("Use audio driver %1").arg(m_audioDriverName));
 
 	drivers = xine_list_video_output_plugins(m_xineEngine);
 	i = 0;
@@ -1235,7 +1235,7 @@ bool KXineWidget::initXine()
 	else
 		m_videoDriverName = m_videoChoices[i];
 
-	debugOut(TQString("Use video driver %1").tqarg(m_videoDriverName));
+	debugOut(TQString("Use video driver %1").arg(m_videoDriverName));
 
 	m_mixerInfo = strdup(i18n("Use software audio mixer").local8Bit());
 	m_softwareMixer = (bool)xine_config_register_bool(m_xineEngine, "audio.mixer_software", 1, m_mixerInfo,
@@ -1307,7 +1307,7 @@ bool KXineWidget::initXine()
 
 	if (!m_videoDriver && m_videoDriverName != "auto")
 	{
-		emit signalXineError(i18n("Can't init Video Driver '%1' - trying 'auto'...").tqarg(m_videoDriverName));
+		emit signalXineError(i18n("Can't init Video Driver '%1' - trying 'auto'...").arg(m_videoDriverName));
 		m_videoDriverName = "auto";
 		m_videoDriver = xine_open_video_driver(m_xineEngine,
 #ifndef HAVE_XCB
@@ -1331,7 +1331,7 @@ bool KXineWidget::initXine()
 
 	if (!m_audioDriver && m_audioDriverName != "auto")
 	{
-		emit signalXineError(i18n("Can't init Audio Driver '%1' - trying 'auto'...").tqarg(m_audioDriverName));
+		emit signalXineError(i18n("Can't init Audio Driver '%1' - trying 'auto'...").arg(m_audioDriverName));
 		m_audioDriverName = "auto";
 		m_audioDriver = xine_open_audio_driver (m_xineEngine, m_audioDriverName.ascii(), NULL);
 	}
@@ -1395,12 +1395,12 @@ void KXineWidget::initOSD()
 	{
 		if (!xine_osd_set_font(m_osd, m_osdFont, fontsizetable[m_osdSize]))
 		{
-			debugOut(TQString("Font ->%1<- specified for OSD doesn't exists.").tqarg(m_osdFont));
+			debugOut(TQString("Font ->%1<- specified for OSD doesn't exists.").arg(m_osdFont));
 			free(m_osdFont);
 			m_osdFont = strdup("sans");
 			xine_osd_set_font(m_osd, m_osdFont, fontsizetable[m_osdSize]);
 		}
-		debugOut(TQString("Font for OSD: %1").tqarg(m_osdFont));
+		debugOut(TQString("Font for OSD: %1").arg(m_osdFont));
 		xine_osd_set_text_palette(m_osd, XINE_TEXTPALETTE_WHITE_BLACK_TRANSPARENT, XINE_OSD_TEXT1);
 		m_osdUnscaled = (xine_osd_get_capabilities(m_osd) & XINE_OSD_CAP_UNSCALED);
 		if (m_osdUnscaled)
@@ -1428,7 +1428,7 @@ bool KXineWidget::playDvb()
 	{
 		if (!m_visualPlugin)
 		{
-			debugOut(TQString("Init visual plugin: %1").tqarg(m_visualPluginName));
+			debugOut(TQString("Init visual plugin: %1").arg(m_visualPluginName));
 			m_visualPlugin = new PostFilter(m_visualPluginName, m_xineEngine, m_audioDriver, m_videoDriver, NULL);
 		}
 
@@ -1438,7 +1438,7 @@ bool KXineWidget::playDvb()
 	{
 		if (m_visualPlugin)
 		{
-			debugOut(TQString("Dispose visual plugin: %1").tqarg(m_visualPluginName));
+			debugOut(TQString("Dispose visual plugin: %1").arg(m_visualPluginName));
 			delete m_visualPlugin;
 			m_visualPlugin = NULL;
 		}
@@ -1978,7 +1978,7 @@ void KXineWidget::slotPlay()
 	if ( turl.startsWith("/") )
 		turl.prepend("file://");
 
-	debugOut(TQString("Playing: %1").tqarg(turl.local8Bit().data()));
+	debugOut(TQString("Playing: %1").arg(turl.local8Bit().data()));
 
 	xine_set_param( m_xineStream, XINE_PARAM_METRONOM_PREBUFFER, 12000 );
 	if (!xine_open(m_xineStream, TQFile::encodeName(turl))) {
@@ -2002,7 +2002,7 @@ void KXineWidget::slotPlay()
 	{
 		if (m_visualPluginName && (!m_xinePost))
 		{
-			debugOut(TQString("Init visual plugin: %1").tqarg(m_visualPluginName));
+			debugOut(TQString("Init visual plugin: %1").arg(m_visualPluginName));
 			m_xinePost = xine_post_init(m_xineEngine, m_visualPluginName, 0,
 			                            &m_audioDriver,
 			                            &m_videoDriver);
@@ -2016,7 +2016,7 @@ void KXineWidget::slotPlay()
 	{
 		if (m_xinePost)
 		{
-			debugOut(TQString("Dispose visual plugin: %1").tqarg(m_visualPluginName));
+			debugOut(TQString("Dispose visual plugin: %1").arg(m_visualPluginName));
 			m_postAudioSource = xine_get_audio_source(m_xineStream);
 			xine_post_wire_audio_port(m_postAudioSource, m_audioDriver);
 			xine_post_dispose(m_xineEngine, m_xinePost);
@@ -2248,7 +2248,7 @@ void KXineWidget::sendXineError()
 	else
 	{
 		errorOut(error);
-		errorOut(TQString("Can't play: %1 - trying next").tqarg(m_trackURL));
+		errorOut(TQString("Can't play: %1 - trying next").arg(m_trackURL));
 		TQTimer::singleShot(0, this, TQT_SLOT(slotPlay()));
 	}
 }
@@ -2290,7 +2290,7 @@ void KXineWidget::showOSDMessage(const TQString& message, uint duration, int pri
 		return;
 	prevOsdPriority = priority;
 
-	//debugOut(TQString("OSD: draw text: %1").tqarg(message));
+	//debugOut(TQString("OSD: draw text: %1").arg(message));
 	xine_osd_clear(m_osd);
 	xine_osd_draw_text(m_osd, 0, 0, message.local8Bit(), XINE_OSD_TEXT1);
 
@@ -2348,7 +2348,7 @@ void KXineWidget::slotCreateVideoFilter(const TQString& name, TQWidget* parent)
 	wireVideoFilters();
 #else
 	parent = parent;
-	warningOut(TQString("Not implemented [CreateVideoFilter %1]").tqarg(name));
+	warningOut(TQString("Not implemented [CreateVideoFilter %1]").arg(name));
 #endif
 }
 
@@ -2364,7 +2364,7 @@ void KXineWidget::slotCreateAudioFilter(const TQString& name, TQWidget* parent)
 	wireAudioFilters();
 #else
 	parent = parent;
-	warningOut(TQString("Not implemented [CreateAudioFilter %1]").tqarg(name));
+	warningOut(TQString("Not implemented [CreateAudioFilter %1]").arg(name));
 #endif
 }
 
@@ -2478,7 +2478,7 @@ void KXineWidget::wireAudioFilters()
 	{
 		if (!m_visualPlugin)
 		{
-			debugOut(TQString("Init visual plugin: %1").tqarg(m_visualPluginName));
+			debugOut(TQString("Init visual plugin: %1").arg(m_visualPluginName));
 			m_visualPlugin = new PostFilter(m_visualPluginName, m_xineEngine, m_audioDriver, m_videoDriver, NULL);
 		}
 
@@ -2488,7 +2488,7 @@ void KXineWidget::wireAudioFilters()
 	{
 		if (m_visualPlugin)
 		{
-			debugOut(TQString("Dispose visual plugin: %1").tqarg(m_visualPluginName));
+			debugOut(TQString("Dispose visual plugin: %1").arg(m_visualPluginName));
 			delete m_visualPlugin;
 			m_visualPlugin = NULL;
 		}
@@ -2571,7 +2571,7 @@ TQStringList KXineWidget::getVisualPlugins() const
 void KXineWidget::slotSetVisualPlugin(const TQString& visual)
 {
 	if (m_visualPluginName == visual) return;
-	debugOut(TQString("New visualization plugin: %1").tqarg(visual));
+	debugOut(TQString("New visualization plugin: %1").arg(visual));
 
 #ifndef USE_TQT_ONLY
 	unwireAudioFilters();
@@ -2660,7 +2660,7 @@ void KXineWidget::slotSetVolume(int vol)
 	if (!isXineReady()) return;
 	if (m_softwareMixer)
 	{
-		//debugOut(TQString("Set software amplification level: %1").tqarg(vol));
+		//debugOut(TQString("Set software amplification level: %1").arg(vol));
 		if (m_volumeGain)
 			xine_set_param(m_xineStream, XINE_PARAM_AUDIO_AMP_LEVEL, vol*2);
 		else
@@ -2668,7 +2668,7 @@ void KXineWidget::slotSetVolume(int vol)
 	}
 	else
 	{
-		//debugOut(TQString("Set audio mixer volume: %1").tqarg(vol));
+		//debugOut(TQString("Set audio mixer volume: %1").arg(vol));
 		xine_set_param(m_xineStream, XINE_PARAM_AUDIO_VOLUME, vol);
 	}
 	emit signalXineStatus(i18n("Volume") + ": " + TQString::number(vol) +"%");
@@ -2872,7 +2872,7 @@ void KXineWidget::slotStop()
 
 void KXineWidget::slotSetAudiocdDevice(const TQString& device)
 {
-	debugOut(TQString("Set AudioCD device to %1").tqarg(device));
+	debugOut(TQString("Set AudioCD device to %1").arg(device));
 
 	xine_cfg_entry_t config;
 	xine_config_lookup_entry(m_xineEngine, "input.cdda_device", &config);
@@ -2884,7 +2884,7 @@ void KXineWidget::slotSetAudiocdDevice(const TQString& device)
 
 void KXineWidget::slotSetVcdDevice(const TQString& device)
 {
-	debugOut(TQString("Set VCD device to %1").tqarg(device));
+	debugOut(TQString("Set VCD device to %1").arg(device));
 
 	xine_cfg_entry_t config;
 	xine_config_lookup_entry(m_xineEngine, "input.vcd_device", &config);
@@ -2896,7 +2896,7 @@ void KXineWidget::slotSetVcdDevice(const TQString& device)
 
 void KXineWidget::slotSetDvdDevice(const TQString& device)
 {
-	debugOut(TQString("Set DVD device to %1").tqarg(device));
+	debugOut(TQString("Set DVD device to %1").arg(device));
 
 	xine_cfg_entry_t config;
 	xine_config_lookup_entry(m_xineEngine, "input.dvd_device", &config);
@@ -2966,7 +2966,7 @@ void KXineWidget::setStreamSaveDir(const TQString& dir)
 
 	if (!xine_config_lookup_entry(m_xineEngine, "misc.save_dir", &config)) return; /* older xine-lib */
 
-	debugOut(TQString("Set misc.save_dir to: %1").tqarg(dir));
+	debugOut(TQString("Set misc.save_dir to: %1").arg(dir));
 	config.str_value = TQFile::encodeName(dir).data();
 	xine_config_update_entry (m_xineEngine, &config);
 }
@@ -2982,7 +2982,7 @@ const TQString KXineWidget::getStreamSaveDir()
 
 void KXineWidget::setBroadcasterPort(const uint port)
 {
-	debugOut(TQString("Set broadcaster port to %1").tqarg(port));
+	debugOut(TQString("Set broadcaster port to %1").arg(port));
 	xine_set_param(m_xineStream, XINE_PARAM_BROADCASTER_PORT, port);
 }
 
@@ -3018,7 +3018,7 @@ void KXineWidget::slotSpeedFaster()
 		{
 			xine_set_param(m_xineStream, XINE_PARAM_SPEED, XINE_SPEED_FAST_4);
 			m_currentSpeed = Fast2;
-			emit signalXineStatus(i18n("Fast Forward %1").tqarg("x2"));
+			emit signalXineStatus(i18n("Fast Forward %1").arg("x2"));
 			break;
 		}
 		case Fast2:
@@ -3035,14 +3035,14 @@ void KXineWidget::slotSpeedFaster()
 		{
 			xine_set_param(m_xineStream, XINE_PARAM_SPEED, XINE_SPEED_SLOW_2);
 			m_currentSpeed = Slow1;
-			emit signalXineStatus(i18n("Slow Motion %1").tqarg("x1"));
+			emit signalXineStatus(i18n("Slow Motion %1").arg("x1"));
 			break;
 		}
 		default:
 		{
 			xine_set_param(m_xineStream, XINE_PARAM_SPEED, XINE_SPEED_FAST_2);
 			m_currentSpeed = Fast1;
-			emit signalXineStatus(i18n("Fast Forward %1").tqarg("x1"));
+			emit signalXineStatus(i18n("Fast Forward %1").arg("x1"));
 			break;
 		}
 	}
@@ -3056,7 +3056,7 @@ void KXineWidget::slotSpeedSlower()
 		{
 			xine_set_param(m_xineStream, XINE_PARAM_SPEED, XINE_SPEED_SLOW_4);
 			m_currentSpeed = Slow2;
-			emit signalXineStatus(i18n("Slow Motion %1").tqarg("x2"));
+			emit signalXineStatus(i18n("Slow Motion %1").arg("x2"));
 			break;
 		}
 		case Slow2:
@@ -3073,14 +3073,14 @@ void KXineWidget::slotSpeedSlower()
 		{
 			xine_set_param(m_xineStream, XINE_PARAM_SPEED, XINE_SPEED_FAST_2);
 			m_currentSpeed = Fast1;
-			emit signalXineStatus(i18n("Fast Forward %1").tqarg("x1"));
+			emit signalXineStatus(i18n("Fast Forward %1").arg("x1"));
 			break;
 		}
 		default:
 		{
 			xine_set_param(m_xineStream, XINE_PARAM_SPEED, XINE_SPEED_SLOW_2);
 			m_currentSpeed = Slow1;
-			emit signalXineStatus(i18n("Slow Motion %1").tqarg("x1"));
+			emit signalXineStatus(i18n("Slow Motion %1").arg("x1"));
 			break;
 		}
 	}
@@ -3093,14 +3093,14 @@ TQString KXineWidget::getSupportedExtensions() const
 
 void KXineWidget::slotSetAudioChannel(int ch)
 {
-	debugOut(TQString("Switch to audio channel %1").tqarg(ch-1));
+	debugOut(TQString("Switch to audio channel %1").arg(ch-1));
 	xine_set_param(m_xineStream, XINE_PARAM_AUDIO_CHANNEL_LOGICAL, ch-1);
 }
 
 
 void KXineWidget::slotSetSubtitleChannel(int ch)
 {
-	debugOut(TQString("Switch to subtitle channel %1").tqarg(ch-1));
+	debugOut(TQString("Switch to subtitle channel %1").arg(ch-1));
 	xine_set_param(m_xineStream, XINE_PARAM_SPU_CHANNEL, ch-1);
 }
 
@@ -3306,7 +3306,7 @@ void KXineWidget::slotSetDeinterlaceConfig(const TQString& config)
 	if (m_deinterlaceFilter)
 		m_deinterlaceFilter->setConfig(config);
 #else
-	warningOut(TQString ("Not implemented [SetDeinterlaceConfig %1]").tqarg(config));
+	warningOut(TQString ("Not implemented [SetDeinterlaceConfig %1]").arg(config));
 #endif
 }
 
@@ -3318,7 +3318,7 @@ void KXineWidget::slotToggleDeinterlace()
 	if (m_deinterlaceFilter)
 	{
 		m_deinterlaceEnabled = !m_deinterlaceEnabled;
-		debugOut(TQString("Deinterlace enabled: %1").tqarg(m_deinterlaceEnabled));
+		debugOut(TQString("Deinterlace enabled: %1").arg(m_deinterlaceEnabled));
 		if ( m_deinterlaceEnabled ) s = i18n("Deinterlace: on");
 		else s = i18n("Deinterlace: off");
 		showOSDMessage( s, 2000 );
@@ -3786,7 +3786,7 @@ TQImage KXineWidget::getScreenshot() const
 	else
 		height = (int)((double) height / scaleFactor + 0.5);
 
-	debugOut(TQString("Screenshot: scale picture from %1x%2 to %3x%4").tqarg(screenShot.width()).tqarg(screenShot.height()).tqarg(width).tqarg(height));
+	debugOut(TQString("Screenshot: scale picture from %1x%2 to %3x%4").arg(screenShot.width()).arg(screenShot.height()).arg(width).arg(height));
 	screenShot = screenShot.smoothScale(width, height);
 
 	delete []rgbPile;
@@ -3849,7 +3849,7 @@ void KXineWidget::getScreenshot(uchar*& rgb32BitData, int& videoWidth, int& vide
 		break;
 		default:
 		{
-			warningOut(TQString("Screenshot: Format %1 not supportet!").tqarg((char*)&format));
+			warningOut(TQString("Screenshot: Format %1 not supportet!").arg((char*)&format));
 			delete [] yuv;
 			return;
 		}
@@ -3880,7 +3880,7 @@ void KXineWidget::getScreenshot(uchar*& rgb32BitData, int& videoWidth, int& vide
 	      desired_ratio = 4.0 / 3.0;
 	      break;
 	    default:
-	      warningOut(TQString("Screenshot: Unknown aspect ratio: %1 - using 4:3").tqarg(ratio));
+	      warningOut(TQString("Screenshot: Unknown aspect ratio: %1 - using 4:3").arg(ratio));
 	    case XINE_VO_ASPECT_STQUARE:
 	      debugOut("Screenshot: got video aspect: 1:1");
 	      desired_ratio = image_ratio;
@@ -3888,7 +3888,7 @@ void KXineWidget::getScreenshot(uchar*& rgb32BitData, int& videoWidth, int& vide
 	  }
 	*/
 
-	debugOut(TQString("Screenshot: using scale factor: %1").tqarg(m_videoAspect));
+	debugOut(TQString("Screenshot: using scale factor: %1").arg(m_videoAspect));
 	scaleFactor = m_videoAspect;
 
 	delete [] yuv;
diff --git a/kaffeine/src/player-parts/xine-part/xine_part.cpp b/kaffeine/src/player-parts/xine-part/xine_part.cpp
index f9b3fb1..1b9c123 100644
--- a/kaffeine/src/player-parts/xine-part/xine_part.cpp
+++ b/kaffeine/src/player-parts/xine-part/xine_part.cpp
@@ -304,7 +304,7 @@ void XinePart::slotPlay(bool forcePlay)
 	{
 		TQString sub;
 		if ((!mrl.subtitleFiles().isEmpty()) && (mrl.currentSubtitle() > -1))
-			sub = TQString("#subtitle:%1").tqarg(mrl.subtitleFiles()[mrl.currentSubtitle()]);
+			sub = TQString("#subtitle:%1").arg(mrl.subtitleFiles()[mrl.currentSubtitle()]);
 
 		m_xine->clearQueue();
 		m_xine->appendToQueue(mrl.url() + sub );
@@ -786,7 +786,7 @@ void XinePart::slotNewPosition(int pos, const TQTime& playtime)
 		}
 		else
 		{
-			timeMessage = i18n("%1 of %2").tqarg(calcLength.toString("h:mm:ss")).tqarg(length.toString("h:mm:ss"));
+			timeMessage = i18n("%1 of %2").arg(calcLength.toString("h:mm:ss")).arg(length.toString("h:mm:ss"));
 			m_xine->showOSDMessage(timeMessage, 600, OSD_MESSAGE_LOW_PRIORITY);
 		}
 	}
@@ -1623,7 +1623,7 @@ void XinePart::audiocdMRLS(MRL::List& mrls, bool& ok, bool& supported, const TQS
 	for (uint i = 0; i < list.count(); i++)
 	{
 		mrl = MRL(list[i]);
-		mrl.setTitle(i18n("AudioCD Track %1").tqarg(i+1));
+		mrl.setTitle(i18n("AudioCD Track %1").arg(i+1));
 		mrl.setTrack(TQString::number(i+1));
 		if (xine_open(xineStreamForMeta, TQFile::encodeName(mrl.url())))
 		{
@@ -1698,7 +1698,7 @@ void XinePart::vcdMRLS(MRL::List& mrls, bool& ok, bool& supported, const TQStrin
 		mrl = MRL(list[i]);
 		mrl.setMime("video/vcd");
 		mrl.setTrack(TQString::number(i+1));
-		mrl.setTitle(i18n("VCD Track %1").tqarg(i+1));
+		mrl.setTitle(i18n("VCD Track %1").arg(i+1));
 		mrls.append(mrl);
 	}
 	if (mrls.count())
diff --git a/kaffeine/src/player-parts/xine-part/xineconfig.cpp b/kaffeine/src/player-parts/xine-part/xineconfig.cpp
index 74076e4..6d6a2d4 100644
--- a/kaffeine/src/player-parts/xine-part/xineconfig.cpp
+++ b/kaffeine/src/player-parts/xine-part/xineconfig.cpp
@@ -272,7 +272,7 @@ XineConfig::XineConfig(const xine_t* const xine) :
 		else
 			icon = "edit";
 
-		xineFrame = addPage(*it, i18n("%1 Options").tqarg(*it), KGlobal::iconLoader()->loadIcon(icon, KIcon::Panel,
+		xineFrame = addPage(*it, i18n("%1 Options").arg(*it), KGlobal::iconLoader()->loadIcon(icon, KIcon::Panel,
 		                    KIcon::SizeMedium));
 		xineLayout = new TQVBoxLayout(xineFrame, marginHint(), spacingHint());
 		tabWidget = new TQTabWidget(xineFrame);
diff --git a/kaffeine/src/startwindow.cpp b/kaffeine/src/startwindow.cpp
index a86696e..67948a9 100644
--- a/kaffeine/src/startwindow.cpp
+++ b/kaffeine/src/startwindow.cpp
@@ -113,7 +113,7 @@ void StartWindow::registerTarget( const TQString& uiName, const TQString& pixNam
 	//f.setPointSize( f.pointSize()+2 );
 	//f.setBold(true);
 	//btn->setFont( f );
-	btn->setTextLabel( TQString("&%1 %2").tqarg(buttons.count() + 1).tqarg(uiName));
+	btn->setTextLabel( TQString("&%1 %2").arg(buttons.count() + 1).arg(uiName));
 	TQToolTip::add(btn, uiName);
 	btn->setTextPosition( TQToolButton::Under );
 	btn->setUsesTextLabel( true );
