--- ksensors-0.7.3/admin/cvs.sh.autotools	2013-06-04 21:59:18.325691011 +0200
+++ ksensors-0.7.3/admin/cvs.sh	2013-06-04 21:58:49.715264098 +0200
@@ -64,7 +64,7 @@
     echo "*** KDE requires automake 1.6.1 or newer"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* ) : ;;
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.1[0-9]* ) : ;;
   "" )
     echo "*** AUTOMAKE NOT FOUND!."
     echo "*** KDE requires automake 1.6.1 or newer"
@@ -123,7 +123,7 @@
 echo "*** Creating configure"
 call_and_fix_autoconf
 
-if egrep "^AM_CONFIG_HEADER" configure.in >/dev/null 2>&1; then
+if egrep "^A[CM]_CONFIG_HEADER" configure.in >/dev/null 2>&1; then
   echo "*** Creating config.h template"
   $AUTOHEADER || exit 1
 fi
--- ksensors-0.7.3/admin/configure.in.min.ORI	2013-06-04 21:59:24.783561656 +0200
+++ ksensors-0.7.3/admin/configure.in.min	2013-06-04 21:59:33.526386508 +0200
@@ -41,7 +41,7 @@
 KDE_SET_PREFIX
 
 dnl generate the config header
-AM_CONFIG_HEADER(config.h) dnl at the distribution this done
+AC_CONFIG_HEADER(config.h) dnl at the distribution this done
 
 dnl Checks for programs.
 AC_CHECK_COMPILERS
