--- kbfx/plugins/applications/CMakeLists.txt.ORI	2012-08-04 18:24:43.221636976 +0200
+++ kbfx/plugins/applications/CMakeLists.txt	2012-08-04 18:26:32.602463608 +0200
@@ -21,6 +21,7 @@
 	    ${KDE3_UI_LIBRARY}
 	    ${KDE3_KHTML_LIBRARY}
 	    ${M_LIBRARY}
+	    ${KDE3_DCOP_LIBRARY}
 	    )
 
 	KDE3_INSTALL_LIBTOOL_FILE(kbfxplasmadatastub ${LIB_INSTALL_DIR}/kbfx/plugins)
--- kbfx/plugins/settings/CMakeLists.txt.ORI	2012-08-04 18:28:43.968853387 +0200
+++ kbfx/plugins/settings/CMakeLists.txt	2012-08-04 18:29:10.408328041 +0200
@@ -21,6 +21,7 @@
 	    ${KDE3_UI_LIBRARY}
 	    ${KDE3_KHTML_LIBRARY}
 	    ${M_LIBRARY}
+	    ${KDE3_DCOP_LIBRARY}
 	    )
 
 	KDE3_INSTALL_LIBTOOL_FILE(kbfxplasmadatasettings ${LIB_INSTALL_DIR}/kbfx/plugins)
