--- kio/bookmarks/kbookmarkmanager.cc~	2007-10-06 16:11:19.000000000 +0200
+++ kio/bookmarks/kbookmarkmanager.cc	2007-10-06 16:11:19.000000000 +0200
@@ -706,7 +706,7 @@
    config.setGroup("Bookmarks");
 
    // add bookmark dialog usage - no reparse
-   s_self->m_advancedaddbookmark = config.readBoolEntry("AdvancedAddBookmarkDialog", false);
+   s_self->m_advancedaddbookmark = config.readBoolEntry("AdvancedAddBookmarkDialog", true);
 
    // these three alter the menu, therefore all need a reparse
    s_self->m_contextmenu = config.readBoolEntry("ContextMenuActions", true);
