--- branches/trinity/kdelibs/kdecore/fixx11h.h	2011/01/03 05:50:03	1211108
+++ branches/trinity/kdelibs/kdecore/fixx11h.h	2011/01/12 22:19:14	1214091
@@ -57,15 +57,18 @@
 
 // template --->
 // Affects: Should be without side effects.
+#ifndef _XTYPEDEF_BOOL
 #ifdef Bool
 #ifndef FIXX11H_Bool
 #define FIXX11H_Bool
 typedef Bool XBool;
 #undef Bool
+#define _XTYPEDEF_BOOL
 typedef XBool Bool;
 #endif
 #undef Bool
 #endif
+#endif // _XTYPEDEF_BOOL
 // template <---
 
 // Affects: Should be without side effects.
