commit 24f144faf98249012e7b1657a5dfe93750f0dfde
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1326095653 -0600

    Add inotify CMake support to tdelibs

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e26c252..ede0a3a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,6 +76,7 @@
 OPTION( WITH_UTEMPTER "Use utempter for utmp management" OFF )
 OPTION( WITH_AVAHI "Enable AVAHI support" OFF )
 OPTION( WITH_PCRE "Enable pcre regex support for kjs" ON )
+OPTION( WITH_INOTIFY "Enable inotify support for kio" ON )
 
 OPTION( WITH_ASPELL "Enable aspell support" OFF )
 OPTION( WITH_HSPELL "Enable hspell support" OFF )
@@ -726,6 +727,26 @@ if( WITH_GCC_VISIBILITY )
   set( HAVE_PCREPOSIX 1 )
 endif( )

+##### check for inotify ###########################
+
+if( WITH_INOTIFY )
+  check_include_file( "inotify.h" INOTIFY_FOUND )
+  check_include_file( "sys/inotify.h" INOTIFY_SYS_FOUND )
+  if( NOT INOTIFY_FOUND )
+    if( NOT INOTIFY_SYS_FOUND )
+      message(FATAL_ERROR "\ninotify support was requested, but inotify was not found on your system" )
+    endif( NOT INOTIFY_SYS_FOUND )
+  endif( NOT INOTIFY_FOUND )
+  if( INOTIFY_FOUND )
+      set( HAVE_INOTIFY 1 )
+  endif( INOTIFY_FOUND )
+  if( INOTIFY_SYS_FOUND )
+      set( HAVE_INOTIFY 1 )
+      set( HAVE_SYS_INOTIFY 1 )
+  endif( INOTIFY_SYS_FOUND )
+endif( )
+
+
 ##### check for aspell ##########################
 # we need ASPELL_DATADIR too
 
diff --git a/config.h.cmake b/config.h.cmake
index 1893270..d9b0ef5 100644
--- a/config.h.cmake
+++ b/config.h.cmake
@@ -234,7 +234,7 @@
 #cmakedefine HAVE_INITGROUPS_PROTO 1
 
 /* Define if your system has Linux Inode Notification */
-#undef HAVE_INOTIFY
+#cmakedefine HAVE_INOTIFY 1
 
 /* Define to 1 if you have the <inttypes.h> header file. */
 #cmakedefine HAVE_INTTYPES_H 1
@@ -555,7 +555,7 @@
 #cmakedefine HAVE_SYS_FILIO_H 1
 
 /* Define if your system has glibc support for inotify */
-#undef HAVE_SYS_INOTIFY
+#cmakedefine HAVE_SYS_INOTIFY 1
 
 /* Define to 1 if you have the <sys/mman.h> header file. */
 #cmakedefine HAVE_SYS_MMAN_H 1
