--- kmymoney/doc/en/Makefile.am.ORI	2011-11-26 17:57:22.473298551 +0100
+++ kmymoney/doc/en/Makefile.am	2011-11-26 17:58:18.611423284 +0100
@@ -111,7 +111,7 @@
 kmymoney-user.pdf: $(DOCBOOK_FILES) $(SCREENSHOT_FILES)
 	$(MAKE) get-files
 	$(MEINPROC) --check --stylesheet `dirname $(KDE_XSL_STYLESHEET)`/kde-nochunk.xsl  $(srcdir)/index.docbook -o index.html;
-	recode utf8..latin1 index.html
+	recode utf8..latin1 index.html || :
 	html2ps -o kmymoney-user.ps -n index.html
 	ps2pdf kmymoney-user.ps kmymoney-user.pdf
 	rm index.html kmymoney-user.ps
--- kmymoney/developer-doc/phb/Makefile.am.ORI	2011-11-26 12:00:54.708855347 -0500
+++ kmymoney/developer-doc/phb/Makefile.am	2011-11-26 12:01:01.683692300 -0500
@@ -106,7 +106,7 @@
 kmymoney-phb.pdf: $(DOCBOOK_FILES)
 	$(MAKE) get-files
 	$(MEINPROC) --stylesheet `dirname $(KDE_XSL_STYLESHEET)`/kde-nochunk.xsl $(srcdir)/phb.docbook -o index.html
-	recode utf8..latin1 index.html
+	recode utf8..latin1 index.html || :
 	html2ps -o kmymoney-phb.ps -n index.html
 	ps2pdf kmymoney-phb.ps kmymoney-phb.pdf
 	rm index.html kmymoney-phb.ps
