--- libktorrent/interfaces/functions.cpp.orig	2006-05-21 00:06:28.000000000 +0200
+++ libktorrent/interfaces/functions.cpp	2006-05-21 00:06:54.000000000 +0200
@@ -46,7 +46,11 @@
 	QString KBytesPerSecToString(double speed,int precision)
 	{
 		KLocale* loc = KGlobal::locale();
-		return i18n("%1 KB/s").arg(loc->formatNumber(speed,precision));
+		QString tmp = loc->formatNumber(speed,precision);
+		if (tmp.startsWith("inf"))
+			return i18n("%1 KB/sec").arg("> 0.0");
+		else
+			return i18n("%1 KB/sec").arg(tmp);
 	}
 
 	QString DurationToString(Uint32 nsecs)
