commit 04fccf73370ad95a70b8e107240e50f9fdbf98fc
Author: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date:   1324497836 -0600

    Rename obsolete tq methods to standard names

diff --git a/src/app/configdialog.cpp b/src/app/configdialog.cpp
index a429095..3dbdb53 100644
--- a/src/app/configdialog.cpp
+++ b/src/app/configdialog.cpp
@@ -93,9 +93,9 @@ template<class T>
 void addConfigPage(KDialogBase* dialog, T* content, const TQString& header, const TQString& name, const char* iconName) {
 	TQFrame* page=dialog->addPage(name, header, BarIcon(iconName, 32));
 	content->reparent(page, TQPoint(0,0));
-	TQVBoxLayout* tqlayout=new TQVBoxLayout(page, 0, KDialog::spacingHint());
-	tqlayout->addWidget(content);
-	tqlayout->addStretch();
+	TQVBoxLayout* layout=new TQVBoxLayout(page, 0, KDialog::spacingHint());
+	layout->addWidget(content);
+	layout->addStretch();
 }
 
 template<class T>
diff --git a/src/app/configimagelistpage.ui b/src/app/configimagelistpage.ui
index e2fb15b..01cfe95 100644
--- a/src/app/configimagelistpage.ui
+++ b/src/app/configimagelistpage.ui
@@ -150,7 +150,7 @@
         </widget>
         <widget class="TQLayoutWidget">
             <property name="name">
-                <cstring>tqlayout6</cstring>
+                <cstring>layout6</cstring>
             </property>
             <grid>
                 <property name="name">
@@ -258,7 +258,7 @@
         </widget>
         <widget class="TQLayoutWidget">
             <property name="name">
-                <cstring>tqlayout7</cstring>
+                <cstring>layout7</cstring>
             </property>
             <hbox>
                 <property name="name">
diff --git a/src/app/configimageviewpage.ui b/src/app/configimageviewpage.ui
index 2f816c0..963f13d 100644
--- a/src/app/configimageviewpage.ui
+++ b/src/app/configimageviewpage.ui
@@ -73,7 +73,7 @@
         </widget>
         <widget class="TQLayoutWidget" row="4" column="0" rowspan="1" colspan="6">
             <property name="name">
-                <cstring>tqlayout4</cstring>
+                <cstring>layout4</cstring>
             </property>
             <hbox>
                 <property name="name">
diff --git a/src/app/configmiscpage.ui b/src/app/configmiscpage.ui
index 9df8e96..7e5874e 100644
--- a/src/app/configmiscpage.ui
+++ b/src/app/configmiscpage.ui
@@ -137,7 +137,7 @@
         </widget>
         <widget class="TQLayoutWidget">
             <property name="name">
-                <cstring>tqlayout1</cstring>
+                <cstring>layout1</cstring>
             </property>
             <grid>
                 <property name="name">
diff --git a/src/app/main.cpp b/src/app/main.cpp
index bef08ae..989bffd 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -117,7 +117,7 @@ KDE_EXPORT int kdemain (int argc, char *argv[]) {
 	aboutData.addCredit("Marco Gazzetta", I18N_NOOP("Fixed crash when trying to generate a thumbnail for a broken JPEG file (v0.16.0)"), "mililani@pobox.com");
 	aboutData.addCredit("GeniusR13", I18N_NOOP("Fixed compilation on KDE 3.0 (v0.16.1)"), "geniusr13@gmx.net");
 	aboutData.addCredit("Ian Koenig", I18N_NOOP("First RPM spec file"), "iguy@ionsphere.org");
-	aboutData.addCredit("Meni Livne", I18N_NOOP("Toolbar tqlayout patch for RTL languages (v0.16.0)"), "livne@kde.org");
+	aboutData.addCredit("Meni Livne", I18N_NOOP("Toolbar layout patch for RTL languages (v0.16.0)"), "livne@kde.org");
 	aboutData.addCredit("Angelo Naselli", I18N_NOOP("Printing support (v1.0.0)"), "anaselli@linux.it");
 	aboutData.addCredit("Jos van den Oever", I18N_NOOP("File info view (v1.0.0)\nPatch to toggle auto-zoom on click (v1.0.0)"), "jos@vandenoever.info");
 	aboutData.addCredit("Jeroen Peters", I18N_NOOP("Configurable mouse wheel behavior (v1.1.1)"), "jpeters@coldmail.nl");
diff --git a/src/app/mainwindow.cpp b/src/app/mainwindow.cpp
index 2f2275b..70b42d7 100644
--- a/src/app/mainwindow.cpp
+++ b/src/app/mainwindow.cpp
@@ -199,8 +199,8 @@ bool MainWindow::queryClose() {
 
 	KConfig* config=KGlobal::config();
 
-	// Don't store dock tqlayout if only the image dock is visible. This avoid
-	// saving tqlayout when in "fullscreen" or "image only" mode.
+	// Don't store dock layout if only the image dock is visible. This avoid
+	// saving layout when in "fullscreen" or "image only" mode.
 	if (mFileViewController->isVisible() || mDirViewController->widget()->isVisible()) {
 		mDockArea->writeDockConfig(config,CONFIG_DOCK_GROUP);
 	}
@@ -867,8 +867,8 @@ void MainWindow::createWidgets() {
 	mDockArea->manager()->setSplitterOpaqueResize(true);
 	
 	mViewModeWidget=new TQWidget(mCentralStack);
-	TQVBoxLayout* tqlayout=new TQVBoxLayout(mViewModeWidget);
-	tqlayout->setAutoAdd(true);
+	TQVBoxLayout* layout=new TQVBoxLayout(mViewModeWidget);
+	layout->setAutoAdd(true);
 	mCentralStack->addWidget(mViewModeWidget);
 
 	// Status bar
@@ -923,7 +923,7 @@ void MainWindow::createWidgets() {
 	setGeometry(20,20,720,520);
 
 	// Default dock config
-	// (The "magic numbers" were found by adjusting the tqlayout from within the
+	// (The "magic numbers" were found by adjusting the layout from within the
 	// app and looking at the result in the configuration file)
 	mFolderDock->manualDock(mFileDock, KDockWidget::DockLeft, 4000);
 	mImageDock->manualDock(mFolderDock, KDockWidget::DockBottom, 3734);
diff --git a/src/app/vtabwidget.cpp b/src/app/vtabwidget.cpp
index 66e9b9e..40a1a0a 100644
--- a/src/app/vtabwidget.cpp
+++ b/src/app/vtabwidget.cpp
@@ -47,9 +47,9 @@ VTabWidget::VTabWidget(TQWidget* parent)
 	d->mTabBar->setPosition(KMultiTabBar::Left);
 	d->mTabBar->setStyle(KMultiTabBar::KDEV3ICON);
 	d->mStack=new TQWidgetStack(this);
-	TQHBoxLayout* tqlayout=new TQHBoxLayout(this);
-	tqlayout->add(d->mTabBar);
-	tqlayout->add(d->mStack);
+	TQHBoxLayout* layout=new TQHBoxLayout(this);
+	layout->add(d->mTabBar);
+	layout->add(d->mStack);
 }
 
 
diff --git a/src/gvcore/deletedialogbase.ui b/src/gvcore/deletedialogbase.ui
index f7d6f90..7a8ffd4 100644
--- a/src/gvcore/deletedialogbase.ui
+++ b/src/gvcore/deletedialogbase.ui
@@ -21,7 +21,7 @@
         </property>
         <widget class="TQLayoutWidget">
             <property name="name">
-                <cstring>tqlayout3</cstring>
+                <cstring>layout3</cstring>
             </property>
             <hbox>
                 <property name="name">
diff --git a/src/gvcore/externaltooldialogbase.ui b/src/gvcore/externaltooldialogbase.ui
index 2c25a92..e35f222 100644
--- a/src/gvcore/externaltooldialogbase.ui
+++ b/src/gvcore/externaltooldialogbase.ui
@@ -153,7 +153,7 @@
                 </widget>
                 <widget class="TQLayoutWidget" row="0" column="0">
                     <property name="name">
-                        <cstring>tqlayout3</cstring>
+                        <cstring>layout3</cstring>
                     </property>
                     <grid>
                         <property name="name">
diff --git a/src/gvcore/filethumbnailview.cpp b/src/gvcore/filethumbnailview.cpp
index c6760fe..4867781 100644
--- a/src/gvcore/filethumbnailview.cpp
+++ b/src/gvcore/filethumbnailview.cpp
@@ -77,8 +77,8 @@ public:
 	ProgressWidget(FileThumbnailView* view, int count)
 	: TQFrame(view)
 	{
-		TQHBoxLayout* tqlayout=new TQHBoxLayout(this, 3, 3);
-		tqlayout->setAutoAdd(true);
+		TQHBoxLayout* layout=new TQHBoxLayout(this, 3, 3);
+		layout->setAutoAdd(true);
 		setFrameStyle( TQFrame::StyledPanel | TQFrame::Raised );
 		
 		mStop=new TQPushButton(this);
@@ -93,7 +93,7 @@ public:
 	
 	void polish() {
 		TQFrame::polish();
-		setMinimumWidth(tqlayout()->minimumSize().width());
+		setMinimumWidth(layout()->minimumSize().width());
 		//setFixedHeight( mProgressBar->height() );
 		setFixedHeight( mStop->height() );
 	}
diff --git a/src/gvcore/fileviewcontroller.cpp b/src/gvcore/fileviewcontroller.cpp
index 1cd795e..86a0282 100644
--- a/src/gvcore/fileviewcontroller.cpp
+++ b/src/gvcore/fileviewcontroller.cpp
@@ -268,10 +268,10 @@ FileViewController::FileViewController(TQWidget* parent,KActionCollection* actio
 	d->initFilterCombo();
 	d->mStack=new TQWidgetStack(this);
 	
-	TQVBoxLayout *tqlayout=new TQVBoxLayout(this);
-	tqlayout->addWidget(d->mToolBar);
-	tqlayout->addWidget(d->mFilterBar);
-	tqlayout->addWidget(d->mStack);
+	TQVBoxLayout *layout=new TQVBoxLayout(this);
+	layout->addWidget(d->mToolBar);
+	layout->addWidget(d->mFilterBar);
+	layout->addWidget(d->mStack);
 
 	// Actions
 	mSelectFirst=new KAction(i18n("&First"),
diff --git a/src/gvcore/fullscreenbar.cpp b/src/gvcore/fullscreenbar.cpp
index be2b517..bc8375f 100644
--- a/src/gvcore/fullscreenbar.cpp
+++ b/src/gvcore/fullscreenbar.cpp
@@ -123,7 +123,7 @@ void FullScreenBar::showEvent(TQShowEvent* event) {
 	if (!d->mFirstShow) return;
 	d->mFirstShow=false;
 	move(0, -height());
-	tqlayout()->setResizeMode(TQLayout::Fixed);
+	layout()->setResizeMode(TQLayout::Fixed);
 }
 
 
diff --git a/src/gvcore/imageviewconfig.kcfg b/src/gvcore/imageviewconfig.kcfg
index 6424202..e5c1e04 100644
--- a/src/gvcore/imageviewconfig.kcfg
+++ b/src/gvcore/imageviewconfig.kcfg
@@ -18,7 +18,7 @@
       <default>false</default>
     </entry>
     <entry name="backgroundColor" key="background color" type="Color">
-      <default code="true">TQApplication::tqpalette().active().dark()</default>
+      <default code="true">TQApplication::palette().active().dark()</default>
     </entry>
     <entry name="enlargeSmallImages" key="enlarge small images" type="Bool">
       <default>false</default>
diff --git a/src/gvcore/imageviewcontroller.cpp b/src/gvcore/imageviewcontroller.cpp
index f24a2e8..7387cb8 100644
--- a/src/gvcore/imageviewcontroller.cpp
+++ b/src/gvcore/imageviewcontroller.cpp
@@ -275,12 +275,12 @@ ImageViewController::ImageViewController(TQWidget* parent, Document* document, K
 
 	d->mContainer=new TQWidget(parent);
 	d->mContainer->setMinimumWidth(1); // Make sure we can resize the toolbar smaller than its minimum size
-	TQVBoxLayout* tqlayout=new TQVBoxLayout(d->mContainer);
+	TQVBoxLayout* layout=new TQVBoxLayout(d->mContainer);
 	d->mToolBar=new KToolBar(d->mContainer, "", true);
 
-	tqlayout->add(d->mToolBar);
+	layout->add(d->mToolBar);
 	d->mStack=new TQWidgetStack(d->mContainer);
-	tqlayout->add(d->mStack);
+	layout->add(d->mStack);
 	
 	d->mImageView=new ImageView(d->mStack, document, actionCollection);
 	d->mStack->addWidget(d->mImageView);
diff --git a/src/gvcore/printdialog.cpp b/src/gvcore/printdialog.cpp
index 003fb3a..3b013b3 100644
--- a/src/gvcore/printdialog.cpp
+++ b/src/gvcore/printdialog.cpp
@@ -82,8 +82,8 @@ PrintDialogPage::PrintDialogPage( Document* document, TQWidget *parent, const ch
 	mContent = new PrintDialogPageBase(this);
 	setTitle( mContent->caption() );
 
-	TQVBoxLayout *tqlayout = new TQVBoxLayout( this );
-	tqlayout->addWidget( mContent );
+	TQVBoxLayout *layout = new TQVBoxLayout( this );
+	layout->addWidget( mContent );
 
 	connect(mContent->mWidth, TQT_SIGNAL( valueChanged( double )), TQT_SLOT( slotWidthChanged( double )));
 	connect(mContent->mHeight, TQT_SIGNAL( valueChanged( double )), TQT_SLOT( slotHeightChanged( double )));
diff --git a/src/gvcore/printdialogpagebase.ui b/src/gvcore/printdialogpagebase.ui
index 543877c..6ae7696 100644
--- a/src/gvcore/printdialogpagebase.ui
+++ b/src/gvcore/printdialogpagebase.ui
@@ -24,7 +24,7 @@
         </property>
         <widget class="TQLayoutWidget">
             <property name="name">
-                <cstring>tqlayout2</cstring>
+                <cstring>layout2</cstring>
             </property>
             <hbox>
                 <property name="name">
@@ -170,7 +170,7 @@
                 </widget>
                 <widget class="TQLayoutWidget">
                     <property name="name">
-                        <cstring>tqlayout4</cstring>
+                        <cstring>layout4</cstring>
                     </property>
                     <hbox>
                         <property name="name">
@@ -236,7 +236,7 @@
                 </widget>
                 <widget class="TQLayoutWidget">
                     <property name="name">
-                        <cstring>tqlayout4</cstring>
+                        <cstring>layout4</cstring>
                     </property>
                     <hbox>
                         <property name="name">
diff --git a/src/gvcore/thumbnaildetailsdialogbase.ui b/src/gvcore/thumbnaildetailsdialogbase.ui
index e9ef011..ae03c98 100644
--- a/src/gvcore/thumbnaildetailsdialogbase.ui
+++ b/src/gvcore/thumbnaildetailsdialogbase.ui
@@ -29,7 +29,7 @@
         </widget>
         <widget class="TQLayoutWidget">
             <property name="name">
-                <cstring>tqlayout6</cstring>
+                <cstring>layout6</cstring>
             </property>
             <grid>
                 <property name="name">
diff --git a/src/spec/gwenview-mdk.spec b/src/spec/gwenview-mdk.spec
index 67a3cdf..c524954 100644
--- a/src/spec/gwenview-mdk.spec
+++ b/src/spec/gwenview-mdk.spec
@@ -317,7 +317,7 @@ rm -rf $RPM_BUILD_ROOT
  - Avoid data loss if the JPEG images are saved while being rotated by
    JPEGTran.
  - The back button in Konqueror now works correctly with gvimagepart.
- - The default tqlayout is more user-friendly.
+ - The default layout is more user-friendly.
  - Non-trivial URLs (e.g. http query URL) are correctly handled.
  - You can now drop images on the image view.
 
