--- trinity-kkbswitch-14.0.0~pre9+c7bbe542/kkbswitch/kbconfig.cpp.ORI	2013-09-08 11:56:58.314462456 +0200
+++ trinity-kkbswitch-14.0.0~pre9+c7bbe542/kkbswitch/kbconfig.cpp	2013-09-08 11:57:19.464008724 +0200
@@ -15,6 +15,8 @@
  *                                                                         *
  ***************************************************************************/
 
+#include <unistd.h>
+
 #include <config.h> 
 #include <tdeversion.h>
 #include <tdelocale.h>
