commit b4c7fd481649eaa362dc426e6b5f8d23b0c9e836
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1335631847 -0500

    GCC 4.7 fixes.

diff --git a/smb4k/configdlg/smb4kshareoptions.cpp b/smb4k/configdlg/smb4kshareoptions.cpp
index 24e309a..6627b51 100644
--- a/smb4k/configdlg/smb4kshareoptions.cpp
+++ b/smb4k/configdlg/smb4kshareoptions.cpp
@@ -40,6 +40,7 @@
 // system includes
 #include <sys/types.h>
 #include <pwd.h>
+#include <unistd.h>
 
 // applications specific includes
 #include "smb4kshareoptions.h"
